// react-aria-components/dist/exports/Autocomplete.js
import {Autocomplete as $8f09b710ef85b337$export$2f2b9559550c7bbc, AutocompleteContext as $8f09b710ef85b337$export$36e687af51cd0967, AutocompleteStateContext as $8f09b710ef85b337$export$68ee3368b6d68148, FieldInputContext as $8f09b710ef85b337$export$698f465ec27e93df, SelectableCollectionContext as $8f09b710ef85b337$export$b0d3ecf7112093a7} from "../private/Autocomplete.js";
import "client-only";
import {useFilter as $32dbe569a1cce7f3$re_export$useFilter} from "react-aria/useFilter";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.





export {$8f09b710ef85b337$export$2f2b9559550c7bbc as Autocomplete, $8f09b710ef85b337$export$36e687af51cd0967 as AutocompleteContext, $8f09b710ef85b337$export$68ee3368b6d68148 as AutocompleteStateContext, $8f09b710ef85b337$export$b0d3ecf7112093a7 as SelectableCollectionContext, $8f09b710ef85b337$export$698f465ec27e93df as FieldInputContext, $32dbe569a1cce7f3$re_export$useFilter as useFilter};
