// react-aria-components/dist/exports/CheckboxGroup.mjs
import {Checkbox as $ed8ccb2e23e76301$export$48513f6b9f8ce62d, CheckboxContext as $ed8ccb2e23e76301$export$b085522c77523c51, CheckboxGroup as $ed8ccb2e23e76301$export$4aa08d5625cb8ead, CheckboxGroupContext as $ed8ccb2e23e76301$export$baf37c4be89255b8, CheckboxGroupStateContext as $ed8ccb2e23e76301$export$139c5b8563afc1fc} from "../private/Checkbox.mjs";
import {Label as $43a3b93638fe5db9$export$b04be29aa201d4f5} from "../private/Label.mjs";
import {Text as $efe09c6d1c304b50$export$5f1af8db9871e1d6} from "../private/Text.mjs";
import {FieldError as $1f3c3b1a70cec653$export$f551688fc98f2e09} from "../private/FieldError.mjs";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.







export {$ed8ccb2e23e76301$export$48513f6b9f8ce62d as Checkbox, $ed8ccb2e23e76301$export$4aa08d5625cb8ead as CheckboxGroup, $ed8ccb2e23e76301$export$b085522c77523c51 as CheckboxContext, $ed8ccb2e23e76301$export$baf37c4be89255b8 as CheckboxGroupContext, $ed8ccb2e23e76301$export$139c5b8563afc1fc as CheckboxGroupStateContext, $43a3b93638fe5db9$export$b04be29aa201d4f5 as Label, $efe09c6d1c304b50$export$5f1af8db9871e1d6 as Text, $1f3c3b1a70cec653$export$f551688fc98f2e09 as FieldError};
