// react-aria-components/dist/exports/Checkbox.mjs
import {Checkbox as $ed8ccb2e23e76301$export$48513f6b9f8ce62d, CheckboxButton as $ed8ccb2e23e76301$export$6e7a18c0548f3129, CheckboxContext as $ed8ccb2e23e76301$export$b085522c77523c51, CheckboxField as $ed8ccb2e23e76301$export$94195a47b94ed396} from "../private/Checkbox.mjs";
import {Text as $efe09c6d1c304b50$export$5f1af8db9871e1d6} from "../private/Text.mjs";
import {FieldError as $1f3c3b1a70cec653$export$f551688fc98f2e09} from "../private/FieldError.mjs";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.






export {$ed8ccb2e23e76301$export$48513f6b9f8ce62d as Checkbox, $ed8ccb2e23e76301$export$94195a47b94ed396 as CheckboxField, $ed8ccb2e23e76301$export$6e7a18c0548f3129 as CheckboxButton, $ed8ccb2e23e76301$export$b085522c77523c51 as CheckboxContext, $efe09c6d1c304b50$export$5f1af8db9871e1d6 as Text, $1f3c3b1a70cec653$export$f551688fc98f2e09 as FieldError};
