// react-aria-components/dist/exports/CheckboxGroup.js
import {Checkbox as $4bd9daf9bf54cf04$export$48513f6b9f8ce62d, CheckboxContext as $4bd9daf9bf54cf04$export$b085522c77523c51, CheckboxGroup as $4bd9daf9bf54cf04$export$4aa08d5625cb8ead, CheckboxGroupContext as $4bd9daf9bf54cf04$export$baf37c4be89255b8, CheckboxGroupStateContext as $4bd9daf9bf54cf04$export$139c5b8563afc1fc} from "../private/Checkbox.js";
import {Label as $3e4839e5b30e7b17$export$b04be29aa201d4f5} from "../private/Label.js";
import {Text as $20d769b1e2b13352$export$5f1af8db9871e1d6} from "../private/Text.js";
import {FieldError as $6567560e1d9cc847$export$f551688fc98f2e09} from "../private/FieldError.js";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.







export {$4bd9daf9bf54cf04$export$48513f6b9f8ce62d as Checkbox, $4bd9daf9bf54cf04$export$4aa08d5625cb8ead as CheckboxGroup, $4bd9daf9bf54cf04$export$b085522c77523c51 as CheckboxContext, $4bd9daf9bf54cf04$export$baf37c4be89255b8 as CheckboxGroupContext, $4bd9daf9bf54cf04$export$139c5b8563afc1fc as CheckboxGroupStateContext, $3e4839e5b30e7b17$export$b04be29aa201d4f5 as Label, $20d769b1e2b13352$export$5f1af8db9871e1d6 as Text, $6567560e1d9cc847$export$f551688fc98f2e09 as FieldError};
