// react-aria-components/dist/exports/ColorArea.mjs
import {ColorArea as $e3bcd4910eec2b11$export$b2103f68a961418e, ColorAreaContext as $e3bcd4910eec2b11$export$ebe63fadcdce34ed, ColorAreaStateContext as $e3bcd4910eec2b11$export$6b32221de49982e} from "../private/ColorArea.mjs";
import {ColorThumb as $ceafedee624ffe11$export$a3cc47cee1c1ccc} from "../private/ColorThumb.mjs";
import "client-only";
import {parseColor as $58109fe2e91c0085$re_export$parseColor, getColorChannels as $58109fe2e91c0085$re_export$getColorChannels} from "react-stately/Color";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.






export {$e3bcd4910eec2b11$export$b2103f68a961418e as ColorArea, $e3bcd4910eec2b11$export$ebe63fadcdce34ed as ColorAreaContext, $e3bcd4910eec2b11$export$6b32221de49982e as ColorAreaStateContext, $58109fe2e91c0085$re_export$parseColor as parseColor, $58109fe2e91c0085$re_export$getColorChannels as getColorChannels, $ceafedee624ffe11$export$a3cc47cee1c1ccc as ColorThumb};
