// react-aria-components/dist/exports/ColorArea.js
import {ColorArea as $a78e35c6fd3df8be$export$b2103f68a961418e, ColorAreaContext as $a78e35c6fd3df8be$export$ebe63fadcdce34ed, ColorAreaStateContext as $a78e35c6fd3df8be$export$6b32221de49982e} from "../private/ColorArea.js";
import {ColorThumb as $80f3336a74d25baa$export$a3cc47cee1c1ccc} from "../private/ColorThumb.js";
import "client-only";
import {parseColor as $10b510ccd9efa802$re_export$parseColor, getColorChannels as $10b510ccd9efa802$re_export$getColorChannels} from "react-stately/Color";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.






export {$a78e35c6fd3df8be$export$b2103f68a961418e as ColorArea, $a78e35c6fd3df8be$export$ebe63fadcdce34ed as ColorAreaContext, $a78e35c6fd3df8be$export$6b32221de49982e as ColorAreaStateContext, $10b510ccd9efa802$re_export$parseColor as parseColor, $10b510ccd9efa802$re_export$getColorChannels as getColorChannels, $80f3336a74d25baa$export$a3cc47cee1c1ccc as ColorThumb};
