// react-aria-components/dist/exports/ColorField.mjs
import {ColorField as $baf01eb6a6ce4d91$export$b865d4358897bb17, ColorFieldContext as $baf01eb6a6ce4d91$export$44644b8a16031b5b, ColorFieldStateContext as $baf01eb6a6ce4d91$export$96b6d32b05a1a8ed} from "../private/ColorField.mjs";
import {Input as $41fb335299a4a39e$export$f5b8910cec6cf069} from "../private/Input.mjs";
import {Label as $43a3b93638fe5db9$export$b04be29aa201d4f5} from "../private/Label.mjs";
import {Text as $efe09c6d1c304b50$export$5f1af8db9871e1d6} from "../private/Text.mjs";
import {FieldError as $1f3c3b1a70cec653$export$f551688fc98f2e09} from "../private/FieldError.mjs";
import "client-only";
import {parseColor as $bceb0a0e9ce9b056$re_export$parseColor, getColorChannels as $bceb0a0e9ce9b056$re_export$getColorChannels} from "react-stately/Color";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.









export {$baf01eb6a6ce4d91$export$b865d4358897bb17 as ColorField, $baf01eb6a6ce4d91$export$44644b8a16031b5b as ColorFieldContext, $baf01eb6a6ce4d91$export$96b6d32b05a1a8ed as ColorFieldStateContext, $bceb0a0e9ce9b056$re_export$parseColor as parseColor, $bceb0a0e9ce9b056$re_export$getColorChannels as getColorChannels, $41fb335299a4a39e$export$f5b8910cec6cf069 as Input, $43a3b93638fe5db9$export$b04be29aa201d4f5 as Label, $efe09c6d1c304b50$export$5f1af8db9871e1d6 as Text, $1f3c3b1a70cec653$export$f551688fc98f2e09 as FieldError};
