// react-aria-components/dist/exports/DateField.js
import {DateField as $ebd46e18226db4de$export$d9781c7894a82487, DateFieldContext as $ebd46e18226db4de$export$7b3e670c86da5fe8, DateFieldStateContext as $ebd46e18226db4de$export$3b08bebcf796eea0, DateInput as $ebd46e18226db4de$export$7edc06cf1783b30f, DateSegment as $ebd46e18226db4de$export$336ab7fa954c4b5f} from "../private/DateField.js";
import {Label as $3e4839e5b30e7b17$export$b04be29aa201d4f5} from "../private/Label.js";
import {Text as $20d769b1e2b13352$export$5f1af8db9871e1d6} from "../private/Text.js";
import {FieldError as $6567560e1d9cc847$export$f551688fc98f2e09} from "../private/FieldError.js";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.







export {$ebd46e18226db4de$export$d9781c7894a82487 as DateField, $ebd46e18226db4de$export$7edc06cf1783b30f as DateInput, $ebd46e18226db4de$export$336ab7fa954c4b5f as DateSegment, $ebd46e18226db4de$export$7b3e670c86da5fe8 as DateFieldContext, $ebd46e18226db4de$export$3b08bebcf796eea0 as DateFieldStateContext, $3e4839e5b30e7b17$export$b04be29aa201d4f5 as Label, $20d769b1e2b13352$export$5f1af8db9871e1d6 as Text, $6567560e1d9cc847$export$f551688fc98f2e09 as FieldError};
