// react-aria-components/dist/exports/DisclosureGroup.mjs
import {Disclosure as $a3e16eab0ea4036c$export$74a362b31437ec83, DisclosureContext as $a3e16eab0ea4036c$export$d665dd135a51b28a, DisclosureGroup as $a3e16eab0ea4036c$export$944aceb4f8c89f10, DisclosureGroupStateContext as $a3e16eab0ea4036c$export$1d40e3e0cc4d5de, DisclosurePanel as $a3e16eab0ea4036c$export$feabaa331e1d464c, DisclosureStateContext as $a3e16eab0ea4036c$export$dab3ea4a6ef094da} from "../private/Disclosure.mjs";
import {Heading as $2ec61d1d4f780267$export$a8a3e93435678ff9} from "../private/Heading.mjs";
import {Button as $7705c033048f6da7$export$353f5b6fc5456de1} from "../private/Button.mjs";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.






export {$a3e16eab0ea4036c$export$74a362b31437ec83 as Disclosure, $a3e16eab0ea4036c$export$944aceb4f8c89f10 as DisclosureGroup, $a3e16eab0ea4036c$export$1d40e3e0cc4d5de as DisclosureGroupStateContext, $a3e16eab0ea4036c$export$feabaa331e1d464c as DisclosurePanel, $a3e16eab0ea4036c$export$dab3ea4a6ef094da as DisclosureStateContext, $a3e16eab0ea4036c$export$d665dd135a51b28a as DisclosureContext, $2ec61d1d4f780267$export$a8a3e93435678ff9 as Heading, $7705c033048f6da7$export$353f5b6fc5456de1 as Button};
