// react-aria-components/dist/exports/DisclosureGroup.js
import {Disclosure as $9322d6c8018c24e4$export$74a362b31437ec83, DisclosureContext as $9322d6c8018c24e4$export$d665dd135a51b28a, DisclosureGroup as $9322d6c8018c24e4$export$944aceb4f8c89f10, DisclosureGroupStateContext as $9322d6c8018c24e4$export$1d40e3e0cc4d5de, DisclosurePanel as $9322d6c8018c24e4$export$feabaa331e1d464c, DisclosureStateContext as $9322d6c8018c24e4$export$dab3ea4a6ef094da} from "../private/Disclosure.js";
import {Heading as $cd487658cd83358d$export$a8a3e93435678ff9} from "../private/Heading.js";
import {Button as $fc203795b9b363cd$export$353f5b6fc5456de1} from "../private/Button.js";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.






export {$9322d6c8018c24e4$export$74a362b31437ec83 as Disclosure, $9322d6c8018c24e4$export$944aceb4f8c89f10 as DisclosureGroup, $9322d6c8018c24e4$export$1d40e3e0cc4d5de as DisclosureGroupStateContext, $9322d6c8018c24e4$export$feabaa331e1d464c as DisclosurePanel, $9322d6c8018c24e4$export$dab3ea4a6ef094da as DisclosureStateContext, $9322d6c8018c24e4$export$d665dd135a51b28a as DisclosureContext, $cd487658cd83358d$export$a8a3e93435678ff9 as Heading, $fc203795b9b363cd$export$353f5b6fc5456de1 as Button};
