// react-aria-components/dist/exports/Popover.mjs
import {Popover as $542a13ca2fa5b484$export$5b6b19405a83ff9d, PopoverContext as $542a13ca2fa5b484$export$9b9a0cd73afb7ca4} from "../private/Popover.mjs";
import {Dialog as $f2ff30fde7b014be$export$3ddf2d174ce01153, DialogTrigger as $f2ff30fde7b014be$export$2e1e1122cf0cba88} from "../private/Dialog.mjs";
import {OverlayArrow as $4fcfe18fac72dabd$export$746d02f47f4d381} from "../private/OverlayArrow.mjs";
import "client-only";
import {Pressable as $a9f6acaa1d41c5c2$re_export$Pressable} from "react-aria/Pressable";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.







export {$542a13ca2fa5b484$export$5b6b19405a83ff9d as Popover, $542a13ca2fa5b484$export$9b9a0cd73afb7ca4 as PopoverContext, $f2ff30fde7b014be$export$2e1e1122cf0cba88 as DialogTrigger, $f2ff30fde7b014be$export$3ddf2d174ce01153 as Dialog, $4fcfe18fac72dabd$export$746d02f47f4d381 as OverlayArrow, $a9f6acaa1d41c5c2$re_export$Pressable as Pressable};
