// react-aria-components/dist/exports/Popover.js
import {Popover as $03df2f2d3cffb62f$export$5b6b19405a83ff9d, PopoverContext as $03df2f2d3cffb62f$export$9b9a0cd73afb7ca4} from "../private/Popover.js";
import {Dialog as $acf8e70c2f419f18$export$3ddf2d174ce01153, DialogTrigger as $acf8e70c2f419f18$export$2e1e1122cf0cba88} from "../private/Dialog.js";
import {OverlayArrow as $5cdd5922a5c055ad$export$746d02f47f4d381} from "../private/OverlayArrow.js";
import "client-only";
import {Pressable as $56b8f26c3ff7cc5a$re_export$Pressable} from "react-aria/Pressable";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.







export {$03df2f2d3cffb62f$export$5b6b19405a83ff9d as Popover, $03df2f2d3cffb62f$export$9b9a0cd73afb7ca4 as PopoverContext, $acf8e70c2f419f18$export$2e1e1122cf0cba88 as DialogTrigger, $acf8e70c2f419f18$export$3ddf2d174ce01153 as Dialog, $5cdd5922a5c055ad$export$746d02f47f4d381 as OverlayArrow, $56b8f26c3ff7cc5a$re_export$Pressable as Pressable};
