// react-aria-components/dist/exports/Switch.mjs
import {Switch as $24585aa46d6ffdff$export$b5d5cf8927ab7262, SwitchButton as $24585aa46d6ffdff$export$72111f742dee7cb8, SwitchContext as $24585aa46d6ffdff$export$8699e3b644d5a28a, SwitchField as $24585aa46d6ffdff$export$208c2e617baf9fc3} from "../private/Switch.mjs";
import {Text as $efe09c6d1c304b50$export$5f1af8db9871e1d6} from "../private/Text.mjs";
import {FieldError as $1f3c3b1a70cec653$export$f551688fc98f2e09} from "../private/FieldError.mjs";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.






export {$24585aa46d6ffdff$export$b5d5cf8927ab7262 as Switch, $24585aa46d6ffdff$export$208c2e617baf9fc3 as SwitchField, $24585aa46d6ffdff$export$72111f742dee7cb8 as SwitchButton, $24585aa46d6ffdff$export$8699e3b644d5a28a as SwitchContext, $efe09c6d1c304b50$export$5f1af8db9871e1d6 as Text, $1f3c3b1a70cec653$export$f551688fc98f2e09 as FieldError};
