// react-aria-components/dist/exports/Tabs.mjs
import {Tab as $b4f18e3395fe64d7$export$3e41faf802a29e71, TabList as $b4f18e3395fe64d7$export$e51a686c67fdaa2d, TabListStateContext as $b4f18e3395fe64d7$export$364712098d2aa57c, TabPanel as $b4f18e3395fe64d7$export$3d96ec278d3efce4, TabPanels as $b4f18e3395fe64d7$export$5dae8d435677f210, Tabs as $b4f18e3395fe64d7$export$b2539bed5023c21c, TabsContext as $b4f18e3395fe64d7$export$cfa7aa87c26e7d1f} from "../private/Tabs.mjs";
import {SelectionIndicator as $91fe5e721c7f36c1$export$17f80983afe4e444} from "../private/SelectionIndicator.mjs";
import "client-only";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.





export {$b4f18e3395fe64d7$export$b2539bed5023c21c as Tabs, $b4f18e3395fe64d7$export$e51a686c67fdaa2d as TabList, $b4f18e3395fe64d7$export$5dae8d435677f210 as TabPanels, $b4f18e3395fe64d7$export$3d96ec278d3efce4 as TabPanel, $b4f18e3395fe64d7$export$3e41faf802a29e71 as Tab, $b4f18e3395fe64d7$export$cfa7aa87c26e7d1f as TabsContext, $b4f18e3395fe64d7$export$364712098d2aa57c as TabListStateContext, $91fe5e721c7f36c1$export$17f80983afe4e444 as SelectionIndicator};
