// react-aria-components/dist/exports/TagGroup.js
import {Tag as $ada7ba7890ac1d93$export$3288d34c523a1192, TagGroup as $ada7ba7890ac1d93$export$67ea30858aaf75e3, TagGroupContext as $ada7ba7890ac1d93$export$5b07b5dd2cbd96e3, TagList as $ada7ba7890ac1d93$export$f9fef0f55402315b, TagListContext as $ada7ba7890ac1d93$export$e755ce3685dd0ca9} from "../private/TagGroup.js";
import {Button as $fc203795b9b363cd$export$353f5b6fc5456de1} from "../private/Button.js";
import {Label as $3e4839e5b30e7b17$export$b04be29aa201d4f5} from "../private/Label.js";
import {Text as $20d769b1e2b13352$export$5f1af8db9871e1d6} from "../private/Text.js";
import {SelectionIndicator as $0d6f83ad40839938$export$17f80983afe4e444} from "../private/SelectionIndicator.js";
import "client-only";
import {Collection as $1eb185e1216ad463$re_export$Collection} from "react-aria/Collection";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.









export {$ada7ba7890ac1d93$export$67ea30858aaf75e3 as TagGroup, $ada7ba7890ac1d93$export$5b07b5dd2cbd96e3 as TagGroupContext, $ada7ba7890ac1d93$export$f9fef0f55402315b as TagList, $ada7ba7890ac1d93$export$e755ce3685dd0ca9 as TagListContext, $ada7ba7890ac1d93$export$3288d34c523a1192 as Tag, $1eb185e1216ad463$re_export$Collection as Collection, $fc203795b9b363cd$export$353f5b6fc5456de1 as Button, $3e4839e5b30e7b17$export$b04be29aa201d4f5 as Label, $20d769b1e2b13352$export$5f1af8db9871e1d6 as Text, $0d6f83ad40839938$export$17f80983afe4e444 as SelectionIndicator};
