// react-aria-components/dist/exports/Tooltip.mjs
import {Tooltip as $05a50f7d78b03ad9$export$28c660c63b792dea, TooltipContext as $05a50f7d78b03ad9$export$39ae08fa83328b12, TooltipTrigger as $05a50f7d78b03ad9$export$8c610744efcf8a1d, TooltipTriggerStateContext as $05a50f7d78b03ad9$export$7a7623236eec67fa} from "../private/Tooltip.mjs";
import {OverlayArrow as $4fcfe18fac72dabd$export$746d02f47f4d381} from "../private/OverlayArrow.mjs";
import "client-only";
import {Focusable as $140c4ab847cd78d4$re_export$Focusable} from "react-aria/Focusable";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.






export {$05a50f7d78b03ad9$export$8c610744efcf8a1d as TooltipTrigger, $05a50f7d78b03ad9$export$28c660c63b792dea as Tooltip, $05a50f7d78b03ad9$export$7a7623236eec67fa as TooltipTriggerStateContext, $05a50f7d78b03ad9$export$39ae08fa83328b12 as TooltipContext, $4fcfe18fac72dabd$export$746d02f47f4d381 as OverlayArrow, $140c4ab847cd78d4$re_export$Focusable as Focusable};
