// react-aria-components/dist/exports/useDragAndDrop.mjs
import {DragAndDropContext as $f9554a667e4f0374$export$d188a835a7bc5783, DropIndicator as $f9554a667e4f0374$export$62ed72bc21f6b8a6, DropIndicatorContext as $f9554a667e4f0374$export$f55761759794cf55} from "../private/DragAndDrop.mjs";
import {useDragAndDrop as $32ef394310635fd5$export$2cfc5be7a55829f6} from "../private/useDragAndDrop.mjs";
import "client-only";
import {DIRECTORY_DRAG_TYPE as $0954a51c8f00854e$re_export$DIRECTORY_DRAG_TYPE, isDirectoryDropItem as $0954a51c8f00854e$re_export$isDirectoryDropItem, isFileDropItem as $0954a51c8f00854e$re_export$isFileDropItem, isTextDropItem as $0954a51c8f00854e$re_export$isTextDropItem} from "react-aria/useDrop";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.





export {$32ef394310635fd5$export$2cfc5be7a55829f6 as useDragAndDrop, $f9554a667e4f0374$export$62ed72bc21f6b8a6 as DropIndicator, $f9554a667e4f0374$export$f55761759794cf55 as DropIndicatorContext, $f9554a667e4f0374$export$d188a835a7bc5783 as DragAndDropContext, $0954a51c8f00854e$re_export$DIRECTORY_DRAG_TYPE as DIRECTORY_DRAG_TYPE, $0954a51c8f00854e$re_export$isDirectoryDropItem as isDirectoryDropItem, $0954a51c8f00854e$re_export$isFileDropItem as isFileDropItem, $0954a51c8f00854e$re_export$isTextDropItem as isTextDropItem};
