// @adobe/react-spectrum/dist/private/tabs/Tabs.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $4ab2867caa392e8e$exports = require("../picker/Picker.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
require("../tabs_vars.css");
var $d196b45f15f7fd5c$exports = require("../tabs_vars_css.cjs");
var $15e3b68ec42125a9$exports = require("../text/Text.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $lo7c6$reactariauseTabList = require("react-aria/useTabList");
var $lo7c6$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $lo7c6$reactariaFocusRing = require("react-aria/FocusRing");
var $lo7c6$reactstatelyItem = require("react-stately/Item");
var $lo7c6$reactstatelyprivatelistListCollection = require("react-stately/private/list/ListCollection");
var $lo7c6$reactariamergeProps = require("react-aria/mergeProps");
var $lo7c6$react = require("react");
var $lo7c6$reactstatelyuseTabListState = require("react-stately/useTabListState");
var $lo7c6$reactstatelyprivatecollectionsuseCollection = require("react-stately/private/collections/useCollection");
var $lo7c6$reactariauseHover = require("react-aria/useHover");
var $lo7c6$reactariauseId = require("react-aria/useId");
var $lo7c6$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $lo7c6$reactariaI18nProvider = require("react-aria/I18nProvider");
var $lo7c6$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Tabs", function () { return $29c175e2cd645f35$export$b2539bed5023c21c; });
$parcel$export(module.exports, "TabList", function () { return $29c175e2cd645f35$export$e51a686c67fdaa2d; });
$parcel$export(module.exports, "TabPanels", function () { return $29c175e2cd645f35$export$5dae8d435677f210; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





















const $29c175e2cd645f35$var$TabContext = /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createContext(null);
const $29c175e2cd645f35$export$b2539bed5023c21c = /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).forwardRef(function Tabs(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    let { orientation: orientation = 'horizontal', density: density = 'regular', children: children, ...otherProps } = props;
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let tablistRef = (0, $lo7c6$react.useRef)(null);
    let wrapperRef = (0, $lo7c6$react.useRef)(null);
    let { direction: direction } = (0, $lo7c6$reactariaI18nProvider.useLocale)();
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    let [collapsed, setCollapsed] = (0, $lo7c6$react.useState)(false);
    let [selectedTab, setSelectedTab] = (0, $lo7c6$react.useState)(null);
    const [tabListState, setTabListState] = (0, $lo7c6$react.useState)(null);
    let [tabPositions, setTabPositions] = (0, $lo7c6$react.useState)([]);
    let prevTabPositions = (0, $lo7c6$react.useRef)(tabPositions);
    (0, $lo7c6$react.useEffect)(()=>{
        if (tablistRef.current) {
            let selectedTab = tablistRef.current.querySelector(`[data-key="${CSS.escape(tabListState?.selectedKey?.toString() ?? '')}"]`);
            if (selectedTab != null) setSelectedTab(selectedTab);
        }
    // collapse is in the dep array so selectedTab can be updated for TabLine positioning
    }, [
        children,
        tabListState?.selectedKey,
        collapsed,
        tablistRef
    ]);
    let checkShouldCollapse = (0, $lo7c6$react.useCallback)(()=>{
        if (wrapperRef.current && orientation !== 'vertical') {
            let tabsComponent = wrapperRef.current;
            let tabs = tablistRef.current?.querySelectorAll('[role="tab"]') ?? new NodeList();
            let tabDimensions = [
                ...tabs
            ].map((tab)=>tab.getBoundingClientRect());
            let end = direction === 'rtl' ? 'left' : 'right';
            let farEdgeTabList = tabsComponent.getBoundingClientRect()[end];
            let farEdgeLastTab = tabDimensions[tabDimensions.length - 1][end];
            let shouldCollapse = direction === 'rtl' ? farEdgeLastTab < farEdgeTabList : farEdgeTabList < farEdgeLastTab;
            setCollapsed(shouldCollapse);
            if (tabDimensions.length !== prevTabPositions.current.length || tabDimensions.some((box, index)=>box?.left !== prevTabPositions.current[index]?.left || box?.right !== prevTabPositions.current[index]?.right)) {
                setTabPositions(tabDimensions);
                prevTabPositions.current = tabDimensions;
            }
        }
    }, [
        tablistRef,
        wrapperRef,
        direction,
        orientation,
        setCollapsed,
        prevTabPositions,
        setTabPositions
    ]);
    (0, $lo7c6$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        checkShouldCollapse();
    }, [
        children,
        checkShouldCollapse
    ]);
    (0, $lo7c6$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: wrapperRef,
        onResize: checkShouldCollapse
    });
    let tabPanelProps = {
        'aria-labelledby': undefined
    };
    // When the tabs are collapsed, the tabPanel should be labelled by the Picker button element.
    let collapsibleTabListId = (0, $lo7c6$reactariauseId.useId)();
    if (collapsed && orientation !== 'vertical') tabPanelProps['aria-labelledby'] = collapsibleTabListId;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement($29c175e2cd645f35$var$TabContext.Provider, {
        value: {
            tabProps: {
                ...props,
                orientation: orientation,
                density: density
            },
            tabState: {
                tabListState: tabListState,
                setTabListState: setTabListState,
                selectedTab: selectedTab,
                collapsed: collapsed
            },
            refs: {
                tablistRef: tablistRef,
                wrapperRef: wrapperRef
            },
            tabPanelProps: tabPanelProps,
            tabLineState: tabPositions
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement("div", {
        ...(0, $lo7c6$reactariafilterDOMProps.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-TabsPanel', `spectrum-TabsPanel--${orientation}`, styleProps.className)
    }, props.children));
});
// @private
function $29c175e2cd645f35$var$Tab(props) {
    let { item: item, state: state } = props;
    let { key: key, rendered: rendered } = item;
    let ref = (0, $lo7c6$react.useRef)(undefined);
    let { tabProps: tabProps, isSelected: isSelected, isDisabled: isDisabled } = (0, $lo7c6$reactariauseTabList.useTab)({
        key: key
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $lo7c6$reactariauseHover.useHover)({
        ...props
    });
    let ElementType = item.props.href ? 'a' : 'div';
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement((0, $lo7c6$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement(ElementType, {
        ...(0, $lo7c6$reactariamergeProps.mergeProps)(tabProps, hoverProps),
        ref: ref,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Tabs-item', {
            'is-selected': isSelected,
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Icon')
            },
            text: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Tabs-itemLabel')
            }
        }
    }, typeof rendered === 'string' ? /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement((0, $15e3b68ec42125a9$exports.Text), null, rendered) : rendered)));
}
// @private
function $29c175e2cd645f35$var$TabLine(props) {
    let { orientation: orientation, selectedTab: // Is either the tab node (non-collapsed) or the picker node (collapsed)
    selectedTab, selectedKey: // selectedKey is provided so that the TabLine styles are updated when the TabPicker's width updates from a selection change
    selectedKey } = props;
    let { direction: direction } = (0, $lo7c6$reactariaI18nProvider.useLocale)();
    let { scale: scale } = (0, $544fc82701fc93e9$exports.useProvider)();
    let { tabLineState: tabLineState } = (0, $lo7c6$react.useContext)($29c175e2cd645f35$var$TabContext);
    let [style, setStyle] = (0, $lo7c6$react.useState)({
        width: undefined,
        height: undefined
    });
    let onResize = (0, $lo7c6$react.useCallback)(()=>{
        if (selectedTab) {
            let styleObj = {
                transform: undefined,
                width: undefined,
                height: undefined
            };
            // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
            let offset = direction === 'rtl' ? -1 * (selectedTab.offsetParent?.offsetWidth - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
            styleObj.transform = orientation === 'vertical' ? `translateY(${selectedTab.offsetTop}px)` : `translateX(${offset}px)`;
            if (orientation === 'horizontal') styleObj.width = `${selectedTab.offsetWidth}px`;
            else styleObj.height = `${selectedTab.offsetHeight}px`;
            setStyle(styleObj);
        }
    }, [
        direction,
        setStyle,
        selectedTab,
        orientation
    ]);
    (0, $lo7c6$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        onResize();
    }, [
        onResize,
        scale,
        selectedKey,
        tabLineState
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Tabs-selectionIndicator'),
        role: "presentation",
        style: style
    });
}
function $29c175e2cd645f35$export$e51a686c67fdaa2d(props) {
    const tabContext = (0, $lo7c6$react.useContext)($29c175e2cd645f35$var$TabContext);
    const { refs: refs, tabState: tabState, tabProps: tabProps, tabPanelProps: tabPanelProps } = tabContext;
    const { isQuiet: isQuiet, density: density, isEmphasized: isEmphasized, orientation: orientation } = tabProps;
    const { selectedTab: selectedTab, collapsed: collapsed, setTabListState: setTabListState } = tabState;
    const { tablistRef: tablistRef, wrapperRef: wrapperRef } = refs;
    // Pass original Tab props but override children to create the collection.
    const state = (0, $lo7c6$reactstatelyuseTabListState.useTabListState)({
        ...tabProps,
        children: props.children
    });
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    const { tabListProps: tabListProps } = (0, $lo7c6$reactariauseTabList.useTabList)({
        ...tabProps,
        ...props
    }, state, tablistRef);
    (0, $lo7c6$react.useEffect)(()=>{
        // Passing back to root as useTabPanel needs the TabListState
        setTabListState(state);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        state.disabledKeys,
        state.selectedItem,
        state.selectedKey,
        props.children
    ]);
    let collapseStyle = collapsed && orientation !== 'vertical' ? {
        maxWidth: 'calc(100% + 1px)',
        overflow: 'hidden',
        visibility: 'hidden',
        position: 'absolute'
    } : {
        maxWidth: 'calc(100% + 1px)'
    };
    let stylePropsFinal = orientation === 'vertical' ? styleProps : {
        style: collapseStyle
    };
    if (collapsed && orientation !== 'vertical') tabListProps['aria-hidden'] = true;
    let tabListclassName = (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-TabsPanel-tabs');
    const tabContent = /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement("div", {
        ...stylePropsFinal,
        ...tabListProps,
        ref: tablistRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Tabs', `spectrum-Tabs--${orientation}`, tabListclassName, {
            'spectrum-Tabs--quiet': isQuiet,
            'spectrum-Tabs--emphasized': isEmphasized,
            ['spectrum-Tabs--compact']: density === 'compact'
        }, orientation === 'vertical' && styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>/*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement($29c175e2cd645f35$var$Tab, {
            key: item.key,
            item: item,
            state: state,
            orientation: orientation
        })), /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement($29c175e2cd645f35$var$TabLine, {
        orientation: orientation,
        selectedTab: selectedTab
    }));
    if (orientation === 'vertical') return tabContent;
    else return /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement("div", {
        ...styleProps,
        ref: wrapperRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-TabsPanel-collapseWrapper', styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement($29c175e2cd645f35$var$TabPicker, {
        ...props,
        ...tabProps,
        visible: collapsed,
        id: tabPanelProps['aria-labelledby'],
        state: state,
        className: tabListclassName
    }), tabContent);
}
function $29c175e2cd645f35$export$5dae8d435677f210(props) {
    const { tabState: tabState, tabProps: tabProps } = (0, $lo7c6$react.useContext)($29c175e2cd645f35$var$TabContext);
    const { tabListState: tabListState } = tabState;
    const factory = (0, $lo7c6$react.useCallback)((nodes)=>new (0, $lo7c6$reactstatelyprivatelistListCollection.ListCollection)(nodes), []);
    const collection = (0, $lo7c6$reactstatelyprivatecollectionsuseCollection.useCollection)({
        items: tabProps.items,
        ...props
    }, factory, {
        suppressTextValueWarning: true
    });
    const selectedItem = tabListState && tabListState.selectedKey != null ? collection.getItem(tabListState.selectedKey) : null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement($29c175e2cd645f35$var$TabPanel, {
        ...props,
        key: tabListState?.selectedKey
    }, selectedItem && selectedItem.props.children);
}
// @private
function $29c175e2cd645f35$var$TabPanel(props) {
    const { tabState: tabState, tabPanelProps: ctxTabPanelProps } = (0, $lo7c6$react.useContext)($29c175e2cd645f35$var$TabContext);
    const { tabListState: tabListState } = tabState;
    let ref = (0, $lo7c6$react.useRef)(null);
    const { tabPanelProps: tabPanelProps } = (0, $lo7c6$reactariauseTabList.useTabPanel)(props, tabListState, ref);
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    if (ctxTabPanelProps['aria-labelledby']) tabPanelProps['aria-labelledby'] = ctxTabPanelProps['aria-labelledby'];
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement((0, $lo7c6$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement("div", {
        ...styleProps,
        ...tabPanelProps,
        ref: ref,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-TabsPanel-tabpanel', styleProps.className)
    }, props.children));
}
function $29c175e2cd645f35$var$TabPicker(props) {
    let { isDisabled: isDisabled, isEmphasized: isEmphasized, isQuiet: isQuiet, state: state, 'aria-labelledby': ariaLabeledBy, 'aria-label': ariaLabel, density: density, className: className, id: id, visible: visible } = props;
    let ref = (0, $lo7c6$react.useRef)(null);
    let [pickerNode, setPickerNode] = (0, $lo7c6$react.useState)(null);
    (0, $lo7c6$react.useEffect)(()=>{
        let node = (0, $65aea7b37663976b$exports.unwrapDOMRef)(ref);
        setPickerNode(node.current);
    }, [
        ref
    ]);
    let items = [
        ...state.collection
    ];
    let pickerProps = {
        'aria-labelledby': ariaLabeledBy,
        'aria-label': ariaLabel
    };
    const style = visible ? {} : {
        visibility: 'hidden',
        position: 'absolute'
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Tabs', 'spectrum-Tabs--horizontal', 'spectrum-Tabs--isCollapsed', {
            'spectrum-Tabs--quiet': isQuiet,
            ['spectrum-Tabs--compact']: density === 'compact',
            'spectrum-Tabs--emphasized': isEmphasized
        }, className),
        style: style,
        "aria-hidden": visible ? undefined : true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Icon')
            },
            button: {
                focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'focus-ring')
            }
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement((0, $4ab2867caa392e8e$exports.Picker), {
        ...pickerProps,
        id: id,
        items: items,
        ref: ref,
        isQuiet: true,
        isDisabled: !visible || isDisabled,
        selectedKey: state.selectedKey,
        disabledKeys: state.disabledKeys,
        onSelectionChange: (key)=>{
            if (key != null) state.setSelectedKey(key);
        },
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d196b45f15f7fd5c$exports))), 'spectrum-Tabs-picker')
    }, (item)=>/*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement((0, $lo7c6$reactstatelyItem.Item), item.props, item.rendered)), pickerNode && /*#__PURE__*/ (0, ($parcel$interopDefault($lo7c6$react))).createElement($29c175e2cd645f35$var$TabLine, {
        orientation: "horizontal",
        selectedTab: pickerNode,
        selectedKey: state.selectedKey
    })));
}
