// @adobe/react-spectrum/dist/private/tabs/Tabs.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {Picker as $fcdeb62019c30c53$export$ba25329847403e11} from "../picker/Picker.js";
import {SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686} from "../utils/Slots.js";
import "../tabs_vars.css";
import $b4Lwj$tabs_vars_cssmjs from "../tabs_vars_css.mjs";
import {Text as $42dd7396e689e4e6$export$5f1af8db9871e1d6} from "../text/Text.js";
import {unwrapDOMRef as $c234463e9ef56637$export$c7e28c72a4823176, useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useProvider as $089943c7a219141c$export$693cdb10cec23617, useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {useTab as $b4Lwj$useTab, useTabList as $b4Lwj$useTabList, useTabPanel as $b4Lwj$useTabPanel} from "react-aria/useTabList";
import {filterDOMProps as $b4Lwj$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusRing as $b4Lwj$FocusRing} from "react-aria/FocusRing";
import {Item as $b4Lwj$Item} from "react-stately/Item";
import {ListCollection as $b4Lwj$ListCollection} from "react-stately/private/list/ListCollection";
import {mergeProps as $b4Lwj$mergeProps} from "react-aria/mergeProps";
import $b4Lwj$react, {useRef as $b4Lwj$useRef, useState as $b4Lwj$useState, useEffect as $b4Lwj$useEffect, useCallback as $b4Lwj$useCallback, useContext as $b4Lwj$useContext} from "react";
import {useTabListState as $b4Lwj$useTabListState} from "react-stately/useTabListState";
import {useCollection as $b4Lwj$useCollection} from "react-stately/private/collections/useCollection";
import {useHover as $b4Lwj$useHover} from "react-aria/useHover";
import {useId as $b4Lwj$useId} from "react-aria/useId";
import {useLayoutEffect as $b4Lwj$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useLocale as $b4Lwj$useLocale} from "react-aria/I18nProvider";
import {useResizeObserver as $b4Lwj$useResizeObserver} from "react-aria/private/utils/useResizeObserver";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





















const $89b03460ad791d07$var$TabContext = /*#__PURE__*/ (0, $b4Lwj$react).createContext(null);
const $89b03460ad791d07$export$b2539bed5023c21c = /*#__PURE__*/ (0, $b4Lwj$react).forwardRef(function Tabs(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    let { orientation: orientation = 'horizontal', density: density = 'regular', children: children, ...otherProps } = props;
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let tablistRef = (0, $b4Lwj$useRef)(null);
    let wrapperRef = (0, $b4Lwj$useRef)(null);
    let { direction: direction } = (0, $b4Lwj$useLocale)();
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(otherProps);
    let [collapsed, setCollapsed] = (0, $b4Lwj$useState)(false);
    let [selectedTab, setSelectedTab] = (0, $b4Lwj$useState)(null);
    const [tabListState, setTabListState] = (0, $b4Lwj$useState)(null);
    let [tabPositions, setTabPositions] = (0, $b4Lwj$useState)([]);
    let prevTabPositions = (0, $b4Lwj$useRef)(tabPositions);
    (0, $b4Lwj$useEffect)(()=>{
        if (tablistRef.current) {
            var _tabListState_selectedKey;
            var _tabListState_selectedKey_toString;
            let selectedTab = tablistRef.current.querySelector(`[data-key="${CSS.escape((_tabListState_selectedKey_toString = tabListState === null || tabListState === void 0 ? void 0 : (_tabListState_selectedKey = tabListState.selectedKey) === null || _tabListState_selectedKey === void 0 ? void 0 : _tabListState_selectedKey.toString()) !== null && _tabListState_selectedKey_toString !== void 0 ? _tabListState_selectedKey_toString : '')}"]`);
            if (selectedTab != null) setSelectedTab(selectedTab);
        }
    // collapse is in the dep array so selectedTab can be updated for TabLine positioning
    }, [
        children,
        tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey,
        collapsed,
        tablistRef
    ]);
    let checkShouldCollapse = (0, $b4Lwj$useCallback)(()=>{
        if (wrapperRef.current && orientation !== 'vertical') {
            var _tablistRef_current;
            let tabsComponent = wrapperRef.current;
            var _tablistRef_current_querySelectorAll;
            let tabs = (_tablistRef_current_querySelectorAll = (_tablistRef_current = tablistRef.current) === null || _tablistRef_current === void 0 ? void 0 : _tablistRef_current.querySelectorAll('[role="tab"]')) !== null && _tablistRef_current_querySelectorAll !== void 0 ? _tablistRef_current_querySelectorAll : new NodeList();
            let tabDimensions = [
                ...tabs
            ].map((tab)=>tab.getBoundingClientRect());
            let end = direction === 'rtl' ? 'left' : 'right';
            let farEdgeTabList = tabsComponent.getBoundingClientRect()[end];
            let farEdgeLastTab = tabDimensions[tabDimensions.length - 1][end];
            let shouldCollapse = direction === 'rtl' ? farEdgeLastTab < farEdgeTabList : farEdgeTabList < farEdgeLastTab;
            setCollapsed(shouldCollapse);
            if (tabDimensions.length !== prevTabPositions.current.length || tabDimensions.some((box, index)=>{
                var _prevTabPositions_current_index, _prevTabPositions_current_index1;
                return (box === null || box === void 0 ? void 0 : box.left) !== ((_prevTabPositions_current_index = prevTabPositions.current[index]) === null || _prevTabPositions_current_index === void 0 ? void 0 : _prevTabPositions_current_index.left) || (box === null || box === void 0 ? void 0 : box.right) !== ((_prevTabPositions_current_index1 = prevTabPositions.current[index]) === null || _prevTabPositions_current_index1 === void 0 ? void 0 : _prevTabPositions_current_index1.right);
            })) {
                setTabPositions(tabDimensions);
                prevTabPositions.current = tabDimensions;
            }
        }
    }, [
        tablistRef,
        wrapperRef,
        direction,
        orientation,
        setCollapsed,
        prevTabPositions,
        setTabPositions
    ]);
    (0, $b4Lwj$useLayoutEffect)(()=>{
        checkShouldCollapse();
    }, [
        children,
        checkShouldCollapse
    ]);
    (0, $b4Lwj$useResizeObserver)({
        ref: wrapperRef,
        onResize: checkShouldCollapse
    });
    let tabPanelProps = {
        'aria-labelledby': undefined
    };
    // When the tabs are collapsed, the tabPanel should be labelled by the Picker button element.
    let collapsibleTabListId = (0, $b4Lwj$useId)();
    if (collapsed && orientation !== 'vertical') tabPanelProps['aria-labelledby'] = collapsibleTabListId;
    return /*#__PURE__*/ (0, $b4Lwj$react).createElement($89b03460ad791d07$var$TabContext.Provider, {
        value: {
            tabProps: {
                ...props,
                orientation: orientation,
                density: density
            },
            tabState: {
                tabListState: tabListState,
                setTabListState: setTabListState,
                selectedTab: selectedTab,
                collapsed: collapsed
            },
            refs: {
                tablistRef: tablistRef,
                wrapperRef: wrapperRef
            },
            tabPanelProps: tabPanelProps,
            tabLineState: tabPositions
        }
    }, /*#__PURE__*/ (0, $b4Lwj$react).createElement("div", {
        ...(0, $b4Lwj$filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-TabsPanel', `spectrum-TabsPanel--${orientation}`, styleProps.className)
    }, props.children));
});
// @private
function $89b03460ad791d07$var$Tab(props) {
    let { item: item, state: state } = props;
    let { key: key, rendered: rendered } = item;
    let ref = (0, $b4Lwj$useRef)(undefined);
    let { tabProps: tabProps, isSelected: isSelected, isDisabled: isDisabled } = (0, $b4Lwj$useTab)({
        key: key
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $b4Lwj$useHover)({
        ...props
    });
    let ElementType = item.props.href ? 'a' : 'div';
    return /*#__PURE__*/ (0, $b4Lwj$react).createElement((0, $b4Lwj$FocusRing), {
        focusRingClass: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'focus-ring')
    }, /*#__PURE__*/ (0, $b4Lwj$react).createElement(ElementType, {
        ...(0, $b4Lwj$mergeProps)(tabProps, hoverProps),
        ref: ref,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Tabs-item', {
            'is-selected': isSelected,
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ (0, $b4Lwj$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Icon')
            },
            text: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Tabs-itemLabel')
            }
        }
    }, typeof rendered === 'string' ? /*#__PURE__*/ (0, $b4Lwj$react).createElement((0, $42dd7396e689e4e6$export$5f1af8db9871e1d6), null, rendered) : rendered)));
}
// @private
function $89b03460ad791d07$var$TabLine(props) {
    let { orientation: orientation, selectedTab: // Is either the tab node (non-collapsed) or the picker node (collapsed)
    selectedTab, selectedKey: // selectedKey is provided so that the TabLine styles are updated when the TabPicker's width updates from a selection change
    selectedKey } = props;
    let { direction: direction } = (0, $b4Lwj$useLocale)();
    let { scale: scale } = (0, $089943c7a219141c$export$693cdb10cec23617)();
    let { tabLineState: tabLineState } = (0, $b4Lwj$useContext)($89b03460ad791d07$var$TabContext);
    let [style, setStyle] = (0, $b4Lwj$useState)({
        width: undefined,
        height: undefined
    });
    let onResize = (0, $b4Lwj$useCallback)(()=>{
        if (selectedTab) {
            var _selectedTab_offsetParent;
            let styleObj = {
                transform: undefined,
                width: undefined,
                height: undefined
            };
            // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
            let offset = direction === 'rtl' ? -1 * (((_selectedTab_offsetParent = selectedTab.offsetParent) === null || _selectedTab_offsetParent === void 0 ? void 0 : _selectedTab_offsetParent.offsetWidth) - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
            styleObj.transform = orientation === 'vertical' ? `translateY(${selectedTab.offsetTop}px)` : `translateX(${offset}px)`;
            if (orientation === 'horizontal') styleObj.width = `${selectedTab.offsetWidth}px`;
            else styleObj.height = `${selectedTab.offsetHeight}px`;
            setStyle(styleObj);
        }
    }, [
        direction,
        setStyle,
        selectedTab,
        orientation
    ]);
    (0, $b4Lwj$useLayoutEffect)(()=>{
        onResize();
    }, [
        onResize,
        scale,
        selectedKey,
        tabLineState
    ]);
    return /*#__PURE__*/ (0, $b4Lwj$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Tabs-selectionIndicator'),
        role: "presentation",
        style: style
    });
}
function $89b03460ad791d07$export$e51a686c67fdaa2d(props) {
    const tabContext = (0, $b4Lwj$useContext)($89b03460ad791d07$var$TabContext);
    const { refs: refs, tabState: tabState, tabProps: tabProps, tabPanelProps: tabPanelProps } = tabContext;
    const { isQuiet: isQuiet, density: density, isEmphasized: isEmphasized, orientation: orientation } = tabProps;
    const { selectedTab: selectedTab, collapsed: collapsed, setTabListState: setTabListState } = tabState;
    const { tablistRef: tablistRef, wrapperRef: wrapperRef } = refs;
    // Pass original Tab props but override children to create the collection.
    const state = (0, $b4Lwj$useTabListState)({
        ...tabProps,
        children: props.children
    });
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    const { tabListProps: tabListProps } = (0, $b4Lwj$useTabList)({
        ...tabProps,
        ...props
    }, state, tablistRef);
    (0, $b4Lwj$useEffect)(()=>{
        // Passing back to root as useTabPanel needs the TabListState
        setTabListState(state);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        state.disabledKeys,
        state.selectedItem,
        state.selectedKey,
        props.children
    ]);
    let collapseStyle = collapsed && orientation !== 'vertical' ? {
        maxWidth: 'calc(100% + 1px)',
        overflow: 'hidden',
        visibility: 'hidden',
        position: 'absolute'
    } : {
        maxWidth: 'calc(100% + 1px)'
    };
    let stylePropsFinal = orientation === 'vertical' ? styleProps : {
        style: collapseStyle
    };
    if (collapsed && orientation !== 'vertical') tabListProps['aria-hidden'] = true;
    let tabListclassName = (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-TabsPanel-tabs');
    const tabContent = /*#__PURE__*/ (0, $b4Lwj$react).createElement("div", {
        ...stylePropsFinal,
        ...tabListProps,
        ref: tablistRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Tabs', `spectrum-Tabs--${orientation}`, tabListclassName, {
            'spectrum-Tabs--quiet': isQuiet,
            'spectrum-Tabs--emphasized': isEmphasized,
            ['spectrum-Tabs--compact']: density === 'compact'
        }, orientation === 'vertical' && styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>/*#__PURE__*/ (0, $b4Lwj$react).createElement($89b03460ad791d07$var$Tab, {
            key: item.key,
            item: item,
            state: state,
            orientation: orientation
        })), /*#__PURE__*/ (0, $b4Lwj$react).createElement($89b03460ad791d07$var$TabLine, {
        orientation: orientation,
        selectedTab: selectedTab
    }));
    if (orientation === 'vertical') return tabContent;
    else return /*#__PURE__*/ (0, $b4Lwj$react).createElement("div", {
        ...styleProps,
        ref: wrapperRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-TabsPanel-collapseWrapper', styleProps.className)
    }, /*#__PURE__*/ (0, $b4Lwj$react).createElement($89b03460ad791d07$var$TabPicker, {
        ...props,
        ...tabProps,
        visible: collapsed,
        id: tabPanelProps['aria-labelledby'],
        state: state,
        className: tabListclassName
    }), tabContent);
}
function $89b03460ad791d07$export$5dae8d435677f210(props) {
    const { tabState: tabState, tabProps: tabProps } = (0, $b4Lwj$useContext)($89b03460ad791d07$var$TabContext);
    const { tabListState: tabListState } = tabState;
    const factory = (0, $b4Lwj$useCallback)((nodes)=>new (0, $b4Lwj$ListCollection)(nodes), []);
    const collection = (0, $b4Lwj$useCollection)({
        items: tabProps.items,
        ...props
    }, factory, {
        suppressTextValueWarning: true
    });
    const selectedItem = tabListState && tabListState.selectedKey != null ? collection.getItem(tabListState.selectedKey) : null;
    return /*#__PURE__*/ (0, $b4Lwj$react).createElement($89b03460ad791d07$var$TabPanel, {
        ...props,
        key: tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey
    }, selectedItem && selectedItem.props.children);
}
// @private
function $89b03460ad791d07$var$TabPanel(props) {
    const { tabState: tabState, tabPanelProps: ctxTabPanelProps } = (0, $b4Lwj$useContext)($89b03460ad791d07$var$TabContext);
    const { tabListState: tabListState } = tabState;
    let ref = (0, $b4Lwj$useRef)(null);
    const { tabPanelProps: tabPanelProps } = (0, $b4Lwj$useTabPanel)(props, tabListState, ref);
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    if (ctxTabPanelProps['aria-labelledby']) tabPanelProps['aria-labelledby'] = ctxTabPanelProps['aria-labelledby'];
    return /*#__PURE__*/ (0, $b4Lwj$react).createElement((0, $b4Lwj$FocusRing), {
        focusRingClass: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'focus-ring')
    }, /*#__PURE__*/ (0, $b4Lwj$react).createElement("div", {
        ...styleProps,
        ...tabPanelProps,
        ref: ref,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-TabsPanel-tabpanel', styleProps.className)
    }, props.children));
}
function $89b03460ad791d07$var$TabPicker(props) {
    let { isDisabled: isDisabled, isEmphasized: isEmphasized, isQuiet: isQuiet, state: state, 'aria-labelledby': ariaLabeledBy, 'aria-label': ariaLabel, density: density, className: className, id: id, visible: visible } = props;
    let ref = (0, $b4Lwj$useRef)(null);
    let [pickerNode, setPickerNode] = (0, $b4Lwj$useState)(null);
    (0, $b4Lwj$useEffect)(()=>{
        let node = (0, $c234463e9ef56637$export$c7e28c72a4823176)(ref);
        setPickerNode(node.current);
    }, [
        ref
    ]);
    let items = [
        ...state.collection
    ];
    let pickerProps = {
        'aria-labelledby': ariaLabeledBy,
        'aria-label': ariaLabel
    };
    const style = visible ? {} : {
        visibility: 'hidden',
        position: 'absolute'
    };
    return /*#__PURE__*/ (0, $b4Lwj$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Tabs', 'spectrum-Tabs--horizontal', 'spectrum-Tabs--isCollapsed', {
            'spectrum-Tabs--quiet': isQuiet,
            ['spectrum-Tabs--compact']: density === 'compact',
            'spectrum-Tabs--emphasized': isEmphasized
        }, className),
        style: style,
        "aria-hidden": visible ? undefined : true
    }, /*#__PURE__*/ (0, $b4Lwj$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Icon')
            },
            button: {
                focusRingClass: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'focus-ring')
            }
        }
    }, /*#__PURE__*/ (0, $b4Lwj$react).createElement((0, $fcdeb62019c30c53$export$ba25329847403e11), {
        ...pickerProps,
        id: id,
        items: items,
        ref: ref,
        isQuiet: true,
        isDisabled: !visible || isDisabled,
        selectedKey: state.selectedKey,
        disabledKeys: state.disabledKeys,
        onSelectionChange: (key)=>{
            if (key != null) state.setSelectedKey(key);
        },
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b4Lwj$tabs_vars_cssmjs))), 'spectrum-Tabs-picker')
    }, (item)=>/*#__PURE__*/ (0, $b4Lwj$react).createElement((0, $b4Lwj$Item), item.props, item.rendered)), pickerNode && /*#__PURE__*/ (0, $b4Lwj$react).createElement($89b03460ad791d07$var$TabLine, {
        orientation: "horizontal",
        selectedTab: pickerNode,
        selectedKey: state.selectedKey
    })));
}


export {$89b03460ad791d07$export$b2539bed5023c21c as Tabs, $89b03460ad791d07$export$e51a686c67fdaa2d as TabList, $89b03460ad791d07$export$5dae8d435677f210 as TabPanels};
