// @adobe/react-spectrum/dist/private/utils/Slots.cjs
var $fsa3r$reactariamergeProps = require("react-aria/mergeProps");
var $fsa3r$react = require("react");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "useSlotProps", function () { return $feede71cddc0c5f3$export$1e5c9e6e4e15efe3; });
$parcel$export(module.exports, "cssModuleToSlots", function () { return $feede71cddc0c5f3$export$365cf34cda9978e2; });
$parcel$export(module.exports, "SlotProvider", function () { return $feede71cddc0c5f3$export$8107b24b91795686; });
$parcel$export(module.exports, "ClearSlots", function () { return $feede71cddc0c5f3$export$ceb145244332b7a2; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

let $feede71cddc0c5f3$var$SlotContext = /*#__PURE__*/ (0, ($parcel$interopDefault($fsa3r$react))).createContext(null);
function $feede71cddc0c5f3$export$1e5c9e6e4e15efe3(props, defaultSlot) {
    let slot = props.slot || defaultSlot;
    // @ts-ignore TODO why is slot an object and not just string or undefined?
    let { [slot]: slotProps = {} } = (0, $fsa3r$react.useContext)($feede71cddc0c5f3$var$SlotContext) || {};
    return (0, $fsa3r$reactariamergeProps.mergeProps)(props, (0, $fsa3r$reactariamergeProps.mergeProps)(slotProps, {
        id: props.id
    }));
}
function $feede71cddc0c5f3$export$365cf34cda9978e2(cssModule) {
    return Object.keys(cssModule).reduce((acc, slot)=>{
        acc[slot] = {
            UNSAFE_className: cssModule[slot]
        };
        return acc;
    }, {});
}
function $feede71cddc0c5f3$export$8107b24b91795686(props) {
    const emptyObj = (0, $fsa3r$react.useMemo)(()=>({}), []);
    let parentSlots = (0, $fsa3r$react.useContext)($feede71cddc0c5f3$var$SlotContext) || emptyObj;
    let { slots: slots = emptyObj, children: children } = props;
    // Merge props for each slot from parent context and props
    let value = (0, $fsa3r$react.useMemo)(()=>Object.keys(parentSlots).concat(Object.keys(slots)).reduce((o, p)=>({
                ...o,
                [p]: (0, $fsa3r$reactariamergeProps.mergeProps)(parentSlots[p] || {}, slots[p] || {})
            }), {}), [
        parentSlots,
        slots
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($fsa3r$react))).createElement($feede71cddc0c5f3$var$SlotContext.Provider, {
        value: value
    }, children);
}
function $feede71cddc0c5f3$export$ceb145244332b7a2(props) {
    let { children: children, ...otherProps } = props;
    const emptyObj = (0, $fsa3r$react.useMemo)(()=>({}), []);
    let content = children;
    if ((0, ($parcel$interopDefault($fsa3r$react))).Children.toArray(children).length <= 1) {
        if (typeof children === 'function') // need to know if the node is a string or something else that react can render that doesn't get props
        content = /*#__PURE__*/ (0, ($parcel$interopDefault($fsa3r$react))).cloneElement((0, ($parcel$interopDefault($fsa3r$react))).Children.only(children), otherProps);
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($fsa3r$react))).createElement($feede71cddc0c5f3$var$SlotContext.Provider, {
        value: emptyObj
    }, content);
}
