// @adobe/react-spectrum/dist/private/layout/Grid.cjs
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $6iA1c$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $6iA1c$react = require("react");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Grid", function () { return $d6479700d21b596b$export$ef2184bd89960b14; });
$parcel$export(module.exports, "repeat", function () { return $d6479700d21b596b$export$76d90c956114f2c2; });
$parcel$export(module.exports, "minmax", function () { return $d6479700d21b596b$export$9c1b655deaca4988; });
$parcel$export(module.exports, "fitContent", function () { return $d6479700d21b596b$export$2f0b47b0911ce698; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



const $d6479700d21b596b$var$gridStyleProps = {
    ...(0, $b8f90d51c4908137$exports.baseStyleProps),
    autoFlow: [
        'gridAutoFlow',
        (0, $b8f90d51c4908137$exports.passthroughStyle)
    ],
    autoColumns: [
        'gridAutoColumns',
        $d6479700d21b596b$var$gridDimensionValue
    ],
    autoRows: [
        'gridAutoRows',
        $d6479700d21b596b$var$gridDimensionValue
    ],
    areas: [
        'gridTemplateAreas',
        $d6479700d21b596b$var$gridTemplateAreasValue
    ],
    columns: [
        'gridTemplateColumns',
        $d6479700d21b596b$var$gridTemplateValue
    ],
    rows: [
        'gridTemplateRows',
        $d6479700d21b596b$var$gridTemplateValue
    ],
    gap: [
        'gap',
        (0, $b8f90d51c4908137$exports.dimensionValue)
    ],
    rowGap: [
        'rowGap',
        (0, $b8f90d51c4908137$exports.dimensionValue)
    ],
    columnGap: [
        'columnGap',
        (0, $b8f90d51c4908137$exports.dimensionValue)
    ],
    justifyItems: [
        'justifyItems',
        (0, $b8f90d51c4908137$exports.passthroughStyle)
    ],
    justifyContent: [
        'justifyContent',
        (0, $b8f90d51c4908137$exports.passthroughStyle)
    ],
    alignItems: [
        'alignItems',
        (0, $b8f90d51c4908137$exports.passthroughStyle)
    ],
    alignContent: [
        'alignContent',
        (0, $b8f90d51c4908137$exports.passthroughStyle)
    ]
};
const $d6479700d21b596b$export$ef2184bd89960b14 = /*#__PURE__*/ (0, $6iA1c$react.forwardRef)(function Grid(props, ref) {
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps, $d6479700d21b596b$var$gridStyleProps);
    if (styleProps.style) styleProps.style.display = 'grid'; // inline-grid?
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6iA1c$react))).createElement("div", {
        ...(0, $6iA1c$reactariafilterDOMProps.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, children);
});
function $d6479700d21b596b$export$76d90c956114f2c2(count, repeat) {
    return `repeat(${count}, ${$d6479700d21b596b$var$gridTemplateValue(repeat)})`;
}
function $d6479700d21b596b$export$9c1b655deaca4988(min, max) {
    return `minmax(${$d6479700d21b596b$var$gridDimensionValue(min)}, ${$d6479700d21b596b$var$gridDimensionValue(max)})`;
}
function $d6479700d21b596b$export$2f0b47b0911ce698(dimension) {
    return `fit-content(${$d6479700d21b596b$var$gridDimensionValue(dimension)})`;
}
function $d6479700d21b596b$var$gridTemplateAreasValue(value) {
    return value.map((v)=>`"${v}"`).join('\n');
}
function $d6479700d21b596b$var$gridDimensionValue(value) {
    if (/^max-content|min-content|minmax|auto|fit-content|repeat|subgrid/.test(value)) return value;
    return (0, $b8f90d51c4908137$exports.dimensionValue)(value);
}
function $d6479700d21b596b$var$gridTemplateValue(value) {
    if (Array.isArray(value)) return value.map($d6479700d21b596b$var$gridDimensionValue).join(' ');
    return $d6479700d21b596b$var$gridDimensionValue(value);
}
