// @adobe/react-spectrum/dist/private/color/ColorWheel.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {ColorThumb as $7f568464139e11ee$export$a3cc47cee1c1ccc} from "./ColorThumb.js";
import {dimensionValue as $120fbea2d95e11ed$export$abc24f5b99744ea6, useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import "../colorwheel_vars.css";
import $3WWXI$colorwheel_vars_cssmjs from "../colorwheel_vars_css.mjs";
import {useFocusableRef as $c234463e9ef56637$export$96a734597687c040} from "../utils/useDOMRef.js";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useColorWheel as $3WWXI$useColorWheel} from "react-aria/useColorWheel";
import {ColorWheelContext as $3WWXI$ColorWheelContext} from "react-aria-components/ColorWheel";
import $3WWXI$react, {useRef as $3WWXI$useRef, useState as $3WWXI$useState, useCallback as $3WWXI$useCallback} from "react";
import {useColorWheelState as $3WWXI$useColorWheelState} from "react-stately/useColorWheelState";
import {useContextProps as $3WWXI$useContextProps} from "react-aria-components/slots";
import {useFocusRing as $3WWXI$useFocusRing} from "react-aria/useFocusRing";
import {useLayoutEffect as $3WWXI$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useResizeObserver as $3WWXI$useResizeObserver} from "react-aria/private/utils/useResizeObserver";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 













const $4f4444fad9e6c791$var$WHEEL_THICKNESS = 24;
const $4f4444fad9e6c791$export$f80663f808113381 = /*#__PURE__*/ (0, $3WWXI$react).forwardRef(function ColorWheel(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    let inputRef = (0, $3WWXI$useRef)(null);
    let containerRef = (0, $c234463e9ef56637$export$96a734597687c040)(ref, inputRef);
    [props, containerRef] = (0, $3WWXI$useContextProps)(props, containerRef, (0, $3WWXI$ColorWheelContext));
    let { isDisabled: isDisabled } = props;
    let size = props.size && (0, $120fbea2d95e11ed$export$abc24f5b99744ea6)(props.size);
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    let [wheelRadius, setWheelRadius] = (0, $3WWXI$useState)(0);
    let [wheelThickness, setWheelThickness] = (0, $3WWXI$useState)($4f4444fad9e6c791$var$WHEEL_THICKNESS);
    let resizeHandler = (0, $3WWXI$useCallback)(()=>{
        if (containerRef.current) {
            setWheelRadius(containerRef.current.offsetWidth / 2);
            let thickness = window.getComputedStyle(containerRef.current).getPropertyValue('--spectrum-colorwheel-track-thickness');
            if (thickness) setWheelThickness(parseInt(thickness, 10));
        }
    }, [
        containerRef,
        setWheelRadius,
        setWheelThickness
    ]);
    (0, $3WWXI$useLayoutEffect)(()=>{
        // the size observer's fallback to the window resize event doesn't fire on mount
        if (wheelRadius === 0) resizeHandler();
    }, [
        wheelRadius,
        resizeHandler
    ]);
    (0, $3WWXI$useResizeObserver)({
        ref: containerRef,
        onResize: resizeHandler
    });
    let state = (0, $3WWXI$useColorWheelState)(props);
    let { trackProps: trackProps, inputProps: inputProps, thumbProps: thumbProps } = (0, $3WWXI$useColorWheel)({
        ...props,
        innerRadius: wheelRadius - wheelThickness,
        outerRadius: wheelRadius
    }, state, inputRef);
    let { focusProps: focusProps, isFocusVisible: isFocusVisible } = (0, $3WWXI$useFocusRing)();
    return /*#__PURE__*/ (0, $3WWXI$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3WWXI$colorwheel_vars_cssmjs))), 'spectrum-ColorWheel', {
            'is-disabled': isDisabled
        }, styleProps.className),
        ref: containerRef,
        style: {
            ...styleProps.style,
            // Workaround around https://github.com/adobe/spectrum-css/issues/1032
            width: size,
            height: size
        }
    }, /*#__PURE__*/ (0, $3WWXI$react).createElement("div", {
        ...trackProps,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3WWXI$colorwheel_vars_cssmjs))), 'spectrum-ColorWheel-gradient')
    }), /*#__PURE__*/ (0, $3WWXI$react).createElement((0, $7f568464139e11ee$export$a3cc47cee1c1ccc), {
        value: state.getDisplayColor(),
        isFocused: isFocusVisible,
        isDisabled: isDisabled,
        isDragging: state.isDragging,
        containerRef: containerRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3WWXI$colorwheel_vars_cssmjs))), 'spectrum-ColorWheel-handle'),
        ...thumbProps
    }, /*#__PURE__*/ (0, $3WWXI$react).createElement("input", {
        ...focusProps,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3WWXI$colorwheel_vars_cssmjs))), 'spectrum-ColorWheel-slider'),
        ...inputProps,
        ref: inputRef
    })));
});


export {$4f4444fad9e6c791$export$f80663f808113381 as ColorWheel};
