// @adobe/react-spectrum/dist/private/color/ColorField.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
require("./colorfield.css");
var $7c2b171ef9085c63$exports = require("./colorfield_css.cjs");
var $827dbb466e199966$exports = require("../textfield/TextFieldBase.cjs");
var $1af2ca8553741739$exports = require("../form/Form.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $5QiVu$reactariauseColorField = require("react-aria/useColorField");
var $5QiVu$reactariacomponentsColorField = require("react-aria-components/ColorField");
var $5QiVu$react = require("react");
var $5QiVu$reactstatelyuseColorFieldState = require("react-stately/useColorFieldState");
var $5QiVu$reactariacomponentsslots = require("react-aria-components/slots");
var $5QiVu$reactariaI18nProvider = require("react-aria/I18nProvider");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ColorField", function () { return $cd43964140eb8bb2$export$b865d4358897bb17; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 










const $cd43964140eb8bb2$export$b865d4358897bb17 = /*#__PURE__*/ (0, ($parcel$interopDefault($5QiVu$react))).forwardRef(function ColorField(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $1af2ca8553741739$exports.useFormProps)(props);
    [props] = (0, $5QiVu$reactariacomponentsslots.useContextProps)(props, null, (0, $5QiVu$reactariacomponentsColorField.ColorFieldContext));
    let hasWarned = (0, $5QiVu$react.useRef)(false);
    (0, $5QiVu$react.useEffect)(()=>{
        if (props.placeholder && !hasWarned.current && process.env.NODE_ENV !== 'production') {
            console.warn('Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/ColorField.html#help-text');
            hasWarned.current = true;
        }
    }, [
        props.placeholder
    ]);
    if (props.channel) return /*#__PURE__*/ (0, ($parcel$interopDefault($5QiVu$react))).createElement($cd43964140eb8bb2$var$ColorChannelField, {
        ...props,
        channel: props.channel,
        forwardedRef: ref
    });
    else return /*#__PURE__*/ (0, ($parcel$interopDefault($5QiVu$react))).createElement($cd43964140eb8bb2$var$HexColorField, {
        ...props,
        forwardedRef: ref
    });
});
function $cd43964140eb8bb2$var$ColorChannelField(props) {
    let { value: // These disabled props are handled by the state hook
    value, defaultValue: defaultValue, onChange: onChange, validate: validate, forwardedRef: forwardedRef, ...otherProps } = props;
    let { locale: locale } = (0, $5QiVu$reactariaI18nProvider.useLocale)();
    let state = (0, $5QiVu$reactstatelyuseColorFieldState.useColorChannelFieldState)({
        ...props,
        locale: locale
    });
    let inputRef = (0, $5QiVu$react.useRef)(null);
    let result = (0, $5QiVu$reactariauseColorField.useColorChannelField)(otherProps, state, inputRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($5QiVu$react))).createElement((0, ($parcel$interopDefault($5QiVu$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($5QiVu$react))).createElement((0, $827dbb466e199966$exports.TextFieldBase), {
        ...otherProps,
        ref: forwardedRef,
        inputRef: inputRef,
        ...result,
        inputClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($7c2b171ef9085c63$exports))), 'react-spectrum-ColorField-input')
    }), props.name && /*#__PURE__*/ (0, ($parcel$interopDefault($5QiVu$react))).createElement("input", {
        type: "hidden",
        name: props.name,
        form: props.form,
        value: isNaN(state.numberValue) ? '' : state.numberValue
    }));
}
function $cd43964140eb8bb2$var$HexColorField(props) {
    let { value: // These disabled props are handled by the state hook
    value, defaultValue: defaultValue, onChange: onChange, forwardedRef: forwardedRef, ...otherProps } = props;
    let state = (0, $5QiVu$reactstatelyuseColorFieldState.useColorFieldState)(props);
    let inputRef = (0, $5QiVu$react.useRef)(null);
    let result = (0, $5QiVu$reactariauseColorField.useColorField)(otherProps, state, inputRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($5QiVu$react))).createElement((0, $827dbb466e199966$exports.TextFieldBase), {
        ...otherProps,
        ref: forwardedRef,
        inputRef: inputRef,
        ...result,
        inputClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($7c2b171ef9085c63$exports))), 'react-spectrum-ColorField-input')
    });
}
