// @adobe/react-spectrum/dist/private/color/ColorWheel.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $9b9b2ae635bd46b3$exports = require("./ColorThumb.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
require("../colorwheel_vars.css");
var $6ceb63f2e23f11c9$exports = require("../colorwheel_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $eKF6D$reactariauseColorWheel = require("react-aria/useColorWheel");
var $eKF6D$reactariacomponentsColorWheel = require("react-aria-components/ColorWheel");
var $eKF6D$react = require("react");
var $eKF6D$reactstatelyuseColorWheelState = require("react-stately/useColorWheelState");
var $eKF6D$reactariacomponentsslots = require("react-aria-components/slots");
var $eKF6D$reactariauseFocusRing = require("react-aria/useFocusRing");
var $eKF6D$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $eKF6D$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ColorWheel", function () { return $488ea514274761b9$export$f80663f808113381; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 













const $488ea514274761b9$var$WHEEL_THICKNESS = 24;
const $488ea514274761b9$export$f80663f808113381 = /*#__PURE__*/ (0, ($parcel$interopDefault($eKF6D$react))).forwardRef(function ColorWheel(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    let inputRef = (0, $eKF6D$react.useRef)(null);
    let containerRef = (0, $65aea7b37663976b$exports.useFocusableRef)(ref, inputRef);
    [props, containerRef] = (0, $eKF6D$reactariacomponentsslots.useContextProps)(props, containerRef, (0, $eKF6D$reactariacomponentsColorWheel.ColorWheelContext));
    let { isDisabled: isDisabled } = props;
    let size = props.size && (0, $b8f90d51c4908137$exports.dimensionValue)(props.size);
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    let [wheelRadius, setWheelRadius] = (0, $eKF6D$react.useState)(0);
    let [wheelThickness, setWheelThickness] = (0, $eKF6D$react.useState)($488ea514274761b9$var$WHEEL_THICKNESS);
    let resizeHandler = (0, $eKF6D$react.useCallback)(()=>{
        if (containerRef.current) {
            setWheelRadius(containerRef.current.offsetWidth / 2);
            let thickness = window.getComputedStyle(containerRef.current).getPropertyValue('--spectrum-colorwheel-track-thickness');
            if (thickness) setWheelThickness(parseInt(thickness, 10));
        }
    }, [
        containerRef,
        setWheelRadius,
        setWheelThickness
    ]);
    (0, $eKF6D$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        // the size observer's fallback to the window resize event doesn't fire on mount
        if (wheelRadius === 0) resizeHandler();
    }, [
        wheelRadius,
        resizeHandler
    ]);
    (0, $eKF6D$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: containerRef,
        onResize: resizeHandler
    });
    let state = (0, $eKF6D$reactstatelyuseColorWheelState.useColorWheelState)(props);
    let { trackProps: trackProps, inputProps: inputProps, thumbProps: thumbProps } = (0, $eKF6D$reactariauseColorWheel.useColorWheel)({
        ...props,
        innerRadius: wheelRadius - wheelThickness,
        outerRadius: wheelRadius
    }, state, inputRef);
    let { focusProps: focusProps, isFocusVisible: isFocusVisible } = (0, $eKF6D$reactariauseFocusRing.useFocusRing)();
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eKF6D$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6ceb63f2e23f11c9$exports))), 'spectrum-ColorWheel', {
            'is-disabled': isDisabled
        }, styleProps.className),
        ref: containerRef,
        style: {
            ...styleProps.style,
            // Workaround around https://github.com/adobe/spectrum-css/issues/1032
            width: size,
            height: size
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eKF6D$react))).createElement("div", {
        ...trackProps,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6ceb63f2e23f11c9$exports))), 'spectrum-ColorWheel-gradient')
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($eKF6D$react))).createElement((0, $9b9b2ae635bd46b3$exports.ColorThumb), {
        value: state.getDisplayColor(),
        isFocused: isFocusVisible,
        isDisabled: isDisabled,
        isDragging: state.isDragging,
        containerRef: containerRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6ceb63f2e23f11c9$exports))), 'spectrum-ColorWheel-handle'),
        ...thumbProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eKF6D$react))).createElement("input", {
        ...focusProps,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6ceb63f2e23f11c9$exports))), 'spectrum-ColorWheel-slider'),
        ...inputProps,
        ref: inputRef
    })));
});
