// @adobe/react-spectrum/dist/private/color/ColorField.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import "./colorfield.css";
import $5a4rp$colorfield_cssmjs from "./colorfield_css.mjs";
import {TextFieldBase as $1f88830e88ee8f61$export$d22444a338b6e3c2} from "../textfield/TextFieldBase.js";
import {useFormProps as $d23ca6800ac02cf1$export$a6b5be5c6b451665} from "../form/Form.js";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useColorChannelField as $5a4rp$useColorChannelField, useColorField as $5a4rp$useColorField} from "react-aria/useColorField";
import {ColorFieldContext as $5a4rp$ColorFieldContext} from "react-aria-components/ColorField";
import $5a4rp$react, {useRef as $5a4rp$useRef, useEffect as $5a4rp$useEffect} from "react";
import {useColorChannelFieldState as $5a4rp$useColorChannelFieldState, useColorFieldState as $5a4rp$useColorFieldState} from "react-stately/useColorFieldState";
import {useContextProps as $5a4rp$useContextProps} from "react-aria-components/slots";
import {useLocale as $5a4rp$useLocale} from "react-aria/I18nProvider";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 










const $5b07f9ba6aa1e89e$export$b865d4358897bb17 = /*#__PURE__*/ (0, $5a4rp$react).forwardRef(function ColorField(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    props = (0, $d23ca6800ac02cf1$export$a6b5be5c6b451665)(props);
    [props] = (0, $5a4rp$useContextProps)(props, null, (0, $5a4rp$ColorFieldContext));
    let hasWarned = (0, $5a4rp$useRef)(false);
    (0, $5a4rp$useEffect)(()=>{
        if (props.placeholder && !hasWarned.current && process.env.NODE_ENV !== 'production') {
            console.warn('Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/ColorField.html#help-text');
            hasWarned.current = true;
        }
    }, [
        props.placeholder
    ]);
    if (props.channel) return /*#__PURE__*/ (0, $5a4rp$react).createElement($5b07f9ba6aa1e89e$var$ColorChannelField, {
        ...props,
        channel: props.channel,
        forwardedRef: ref
    });
    else return /*#__PURE__*/ (0, $5a4rp$react).createElement($5b07f9ba6aa1e89e$var$HexColorField, {
        ...props,
        forwardedRef: ref
    });
});
function $5b07f9ba6aa1e89e$var$ColorChannelField(props) {
    let { value: // These disabled props are handled by the state hook
    value, defaultValue: defaultValue, onChange: onChange, validate: validate, forwardedRef: forwardedRef, ...otherProps } = props;
    let { locale: locale } = (0, $5a4rp$useLocale)();
    let state = (0, $5a4rp$useColorChannelFieldState)({
        ...props,
        locale: locale
    });
    let inputRef = (0, $5a4rp$useRef)(null);
    let result = (0, $5a4rp$useColorChannelField)(otherProps, state, inputRef);
    return /*#__PURE__*/ (0, $5a4rp$react).createElement((0, $5a4rp$react).Fragment, null, /*#__PURE__*/ (0, $5a4rp$react).createElement((0, $1f88830e88ee8f61$export$d22444a338b6e3c2), {
        ...otherProps,
        ref: forwardedRef,
        inputRef: inputRef,
        ...result,
        inputClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5a4rp$colorfield_cssmjs))), 'react-spectrum-ColorField-input')
    }), props.name && /*#__PURE__*/ (0, $5a4rp$react).createElement("input", {
        type: "hidden",
        name: props.name,
        form: props.form,
        value: isNaN(state.numberValue) ? '' : state.numberValue
    }));
}
function $5b07f9ba6aa1e89e$var$HexColorField(props) {
    let { value: // These disabled props are handled by the state hook
    value, defaultValue: defaultValue, onChange: onChange, forwardedRef: forwardedRef, ...otherProps } = props;
    let state = (0, $5a4rp$useColorFieldState)(props);
    let inputRef = (0, $5a4rp$useRef)(null);
    let result = (0, $5a4rp$useColorField)(otherProps, state, inputRef);
    return /*#__PURE__*/ (0, $5a4rp$react).createElement((0, $1f88830e88ee8f61$export$d22444a338b6e3c2), {
        ...otherProps,
        ref: forwardedRef,
        inputRef: inputRef,
        ...result,
        inputClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5a4rp$colorfield_cssmjs))), 'react-spectrum-ColorField-input')
    });
}


export {$5b07f9ba6aa1e89e$export$b865d4358897bb17 as ColorField};
