// @adobe/react-spectrum/dist/private/combobox/ComboBox.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
require("./combobox.css");
var $bd481a8d9a232d25$exports = require("./combobox_css.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $b93966d678e0af07$exports = require("../label/Field.cjs");
var $23798a2a76e33abb$exports = require("../button/FieldButton.cjs");
var $3bfb46fc68ccce33$exports = require("./intlStrings.cjs");
var $cb7ee1d9d5613db9$exports = require("../listbox/ListBoxBase.cjs");
var $72900fdaeb95a933$exports = require("./MobileComboBox.cjs");
var $39ed1c805b59752f$exports = require("../overlays/Popover.cjs");
var $948c2416aa3a9507$exports = require("../progress/ProgressCircle.cjs");
require("../inputgroup_vars.css");
var $6c88ab5aea804b3c$exports = require("../inputgroup_vars_css.cjs");
var $827dbb466e199966$exports = require("../textfield/TextFieldBase.cjs");
require("../textfield_vars.css");
var $5d389146adc85829$exports = require("../textfield_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $1af2ca8553741739$exports = require("../form/Form.cjs");
var $0b97cdf6ccc1e502$exports = require("../utils/useIsMobileDevice.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $h6nAh$reactariauseComboBox = require("react-aria/useComboBox");
var $h6nAh$spectrumiconsuiChevronDownMedium = require("@spectrum-icons/ui/ChevronDownMedium");
var $h6nAh$reactstatelyuseComboBoxState = require("react-stately/useComboBoxState");
var $h6nAh$reactariaFocusRing = require("react-aria/FocusRing");
var $h6nAh$reactariaprivateinteractionsPressResponder = require("react-aria/private/interactions/PressResponder");
var $h6nAh$react = require("react");
var $h6nAh$reactariauseFilter = require("react-aria/useFilter");
var $h6nAh$reactariauseHover = require("react-aria/useHover");
var $h6nAh$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $h6nAh$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $h6nAh$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ComboBox", function () { return $79361449ae3dc5a0$export$72b9695b8216309a; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



























const $79361449ae3dc5a0$export$72b9695b8216309a = /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).forwardRef(function ComboBox(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $1af2ca8553741739$exports.useFormProps)(props);
    let hasWarned = (0, $h6nAh$react.useRef)(false);
    (0, $h6nAh$react.useEffect)(()=>{
        if (props.placeholder && !hasWarned.current && process.env.NODE_ENV !== 'production') {
            console.warn('Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/ComboBox.html#help-text');
            hasWarned.current = true;
        }
    }, [
        props.placeholder
    ]);
    let isMobile = (0, $0b97cdf6ccc1e502$exports.useIsMobileDevice)();
    if (isMobile) // menuTrigger=focus/manual don't apply to mobile combobox
    return /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $72900fdaeb95a933$exports.MobileComboBox), {
        ...props,
        menuTrigger: "input",
        ref: ref
    });
    else return /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement($79361449ae3dc5a0$var$ComboBoxBase, {
        ...props,
        ref: ref
    });
});
const $79361449ae3dc5a0$var$ComboBoxBase = /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).forwardRef(function ComboBoxBase(props, ref) {
    let { menuTrigger: menuTrigger = 'input', shouldFlip: shouldFlip = true, direction: direction = 'bottom', align: align = 'start', isQuiet: isQuiet, loadingState: loadingState, onLoadMore: onLoadMore, allowsCustomValue: allowsCustomValue, menuWidth: customMenuWidth, name: name, formValue: formValue = 'text' } = props;
    if (allowsCustomValue) formValue = 'text';
    let stringFormatter = (0, $h6nAh$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($3bfb46fc68ccce33$exports))), '@react-spectrum/combobox');
    let isAsync = loadingState != null;
    let popoverRef = (0, $h6nAh$react.useRef)(null);
    let unwrappedPopoverRef = (0, $65aea7b37663976b$exports.useUnwrapDOMRef)(popoverRef);
    let buttonRef = (0, $h6nAh$react.useRef)(null);
    let unwrappedButtonRef = (0, $65aea7b37663976b$exports.useUnwrapDOMRef)(buttonRef);
    let listBoxRef = (0, $h6nAh$react.useRef)(null);
    let inputRef = (0, $h6nAh$react.useRef)(null);
    // serve as the new popover `triggerRef` instead of `unwrappedButtonRef` before for better positioning.
    let inputGroupRef = (0, $h6nAh$react.useRef)(null);
    let domRef = (0, $65aea7b37663976b$exports.useFocusableRef)(ref, inputRef);
    let { contains: contains } = (0, $h6nAh$reactariauseFilter.useFilter)({
        sensitivity: 'base'
    });
    let state = (0, $h6nAh$reactstatelyuseComboBoxState.useComboBoxState)({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: isAsync
    });
    let layout = (0, $cb7ee1d9d5613db9$exports.useListBoxLayout)();
    let { buttonProps: buttonProps, inputProps: inputProps, listBoxProps: listBoxProps, labelProps: labelProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $h6nAh$reactariauseComboBox.useComboBox)({
        ...props,
        layoutDelegate: layout,
        buttonRef: unwrappedButtonRef,
        popoverRef: unwrappedPopoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        menuTrigger: menuTrigger,
        name: formValue === 'text' ? name : undefined
    }, state);
    // Measure the width of the inputfield and the button to inform the width of the menu (below).
    let [menuWidth, setMenuWidth] = (0, $h6nAh$react.useState)(undefined);
    let { scale: scale } = (0, $544fc82701fc93e9$exports.useProvider)();
    let onResize = (0, $h6nAh$react.useCallback)(()=>{
        if (unwrappedButtonRef.current && inputRef.current) {
            let buttonWidth = unwrappedButtonRef.current.offsetWidth;
            let inputWidth = inputRef.current.offsetWidth;
            setMenuWidth(buttonWidth + inputWidth);
        }
    }, [
        unwrappedButtonRef,
        inputRef,
        setMenuWidth
    ]);
    (0, $h6nAh$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: domRef,
        onResize: onResize
    });
    (0, $h6nAh$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(onResize, [
        scale,
        onResize
    ]);
    let width = isQuiet ? undefined : menuWidth;
    let style = {
        width: customMenuWidth ? (0, $b8f90d51c4908137$exports.dimensionValue)(customMenuWidth) : width,
        minWidth: isQuiet ? `calc(${menuWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : menuWidth
    };
    let cbInputProps = {
        ...props,
        children: null
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, ($parcel$interopDefault($h6nAh$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $b93966d678e0af07$exports.Field), {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        labelProps: labelProps,
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement($79361449ae3dc5a0$var$ComboBoxInput, {
        ...cbInputProps,
        isOpen: state.isOpen,
        loadingState: loadingState,
        inputProps: inputProps,
        inputRef: inputRef,
        triggerProps: buttonProps,
        triggerRef: buttonRef,
        validationState: props.validationState || (isInvalid ? 'invalid' : undefined),
        ref: inputGroupRef
    })), name && formValue === 'key' && /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement("input", {
        type: "hidden",
        name: name,
        form: props.form,
        value: state.selectedKey ?? ''
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $39ed1c805b59752f$exports.Popover), {
        state: state,
        UNSAFE_style: style,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-popover', {
            'spectrum-InputGroup-popover--quiet': isQuiet
        }),
        ref: popoverRef,
        triggerRef: inputGroupRef,
        scrollRef: listBoxRef,
        placement: `${direction} ${align}`,
        hideArrow: true,
        isNonModal: true,
        shouldFlip: shouldFlip
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $cb7ee1d9d5613db9$exports.ListBoxBase), {
        ...listBoxProps,
        ref: listBoxRef,
        disallowEmptySelection: true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        isLoading: loadingState === 'loading' || loadingState === 'loadingMore',
        showLoadingSpinner: loadingState === 'loadingMore',
        onLoadMore: onLoadMore,
        renderEmptyState: ()=>isAsync && /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement("span", {
                className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'no-results')
            }, loadingState === 'loading' ? stringFormatter.format('loading') : stringFormatter.format('noResults'))
    })));
});
const $79361449ae3dc5a0$var$ComboBoxInput = /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).forwardRef(function ComboBoxInput(props, ref) {
    let { isQuiet: isQuiet, isDisabled: isDisabled, validationState: validationState, inputProps: inputProps, inputRef: inputRef, triggerProps: triggerProps, triggerRef: triggerRef, autoFocus: autoFocus, style: style, className: className, loadingState: loadingState, isOpen: isOpen, menuTrigger: menuTrigger } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $h6nAh$reactariauseHover.useHover)({});
    let stringFormatter = (0, $h6nAh$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($3bfb46fc68ccce33$exports))), '@react-spectrum/combobox');
    let timeout = (0, $h6nAh$react.useRef)(null);
    let [showLoading, setShowLoading] = (0, $h6nAh$react.useState)(false);
    let loadingCircle = /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $948c2416aa3a9507$exports.ProgressCircle), {
        "aria-label": stringFormatter.format('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-circleLoader', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input-circleLoader'))
    });
    let isLoading = loadingState === 'loading' || loadingState === 'filtering';
    let inputValue = inputProps.value;
    let lastInputValue = (0, $h6nAh$react.useRef)(inputValue);
    (0, $h6nAh$react.useEffect)(()=>{
        if (isLoading && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (!isLoading) {
            if (timeout.current) clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        isLoading,
        showLoading,
        inputValue
    ]);
    let [prevIsLoading, setPrevIsLoading] = (0, $h6nAh$react.useState)(isLoading);
    if (prevIsLoading !== isLoading && !isLoading) {
        setShowLoading(false);
        setPrevIsLoading(isLoading);
    }
    (0, $h6nAh$react.useEffect)(()=>{
        return ()=>{
            if (timeout.current) clearTimeout(timeout.current);
            timeout.current = null;
        };
    }, []);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $h6nAh$reactariaFocusRing.FocusRing), {
        within: true,
        isTextInput: true,
        focusClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'is-focused'),
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement("div", {
        ...hoverProps,
        ref: ref,
        style: style,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid' && !isDisabled,
            'is-hovered': isHovered
        }, className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $827dbb466e199966$exports.TextFieldBase), {
        inputProps: inputProps,
        inputRef: inputRef,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-field'),
        inputClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input'),
        validationIconClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input-validationIcon'),
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        // loading circle should only be displayed if menu is open, if menuTrigger is "manual", or first time load (to stop circle from showing up when user selects an option)
        // TODO: add special case for completionMode: complete as well
        isLoading: showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'),
        loadingIndicator: loadingState != null ? loadingCircle : undefined,
        disableFocusRing: true
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $h6nAh$reactariaprivateinteractionsPressResponder.PressResponder), {
        preventFocusOnPress: true,
        isPressed: isOpen
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, $23798a2a76e33abb$exports.FieldButton), {
        ...triggerProps,
        ref: triggerRef,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: validationState
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($h6nAh$react))).createElement((0, ($parcel$interopDefault($h6nAh$spectrumiconsuiChevronDownMedium))), {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-Dropdown-chevron')
    })))));
});
