// @adobe/react-spectrum/dist/private/combobox/MobileComboBox.cjs
require("../button_vars.css");
var $869138cbe3b599dc$exports = require("../button_vars_css.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $0fc8553a4214494f$exports = require("../button/ClearButton.cjs");
require("./combobox.css");
var $bd481a8d9a232d25$exports = require("./combobox_css.cjs");
var $b93966d678e0af07$exports = require("../label/Field.cjs");
var $3bfb46fc68ccce33$exports = require("./intlStrings.cjs");
require("../fieldlabel_vars.css");
var $53185441bef09fa8$exports = require("../fieldlabel_vars_css.cjs");
var $cb7ee1d9d5613db9$exports = require("../listbox/ListBoxBase.cjs");
var $948c2416aa3a9507$exports = require("../progress/ProgressCircle.cjs");
require("../search_vars.css");
var $d2d4ce3e4a6482f9$exports = require("../search_vars_css.cjs");
require("../inputgroup_vars.css");
var $6c88ab5aea804b3c$exports = require("../inputgroup_vars_css.cjs");
var $827dbb466e199966$exports = require("../textfield/TextFieldBase.cjs");
require("../textfield_vars.css");
var $5d389146adc85829$exports = require("../textfield_vars_css.cjs");
var $378dee1409fe2937$exports = require("../overlays/Tray.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $9dfug$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $9dfug$reactariauseButton = require("react-aria/useButton");
var $9dfug$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $9dfug$spectrumiconsuiChevronDownMedium = require("@spectrum-icons/ui/ChevronDownMedium");
var $9dfug$reactstatelyuseComboBoxState = require("react-stately/useComboBoxState");
var $9dfug$reactariaOverlay = require("react-aria/Overlay");
var $9dfug$reactariaFocusRing = require("react-aria/FocusRing");
var $9dfug$reactariaprivateinteractionsfocusSafely = require("react-aria/private/interactions/focusSafely");
var $9dfug$reactariaFocusScope = require("react-aria/FocusScope");
var $9dfug$reactariaprivateutilsshadowdomDOMFunctions = require("react-aria/private/utils/shadowdom/DOMFunctions");
var $9dfug$reactariamergeProps = require("react-aria/mergeProps");
var $9dfug$react = require("react");
var $9dfug$reactariaprivateinteractionsuseFocusVisible = require("react-aria/private/interactions/useFocusVisible");
var $9dfug$reactariauseComboBox = require("react-aria/useComboBox");
var $9dfug$reactariauseDialog = require("react-aria/useDialog");
var $9dfug$reactariauseField = require("react-aria/useField");
var $9dfug$reactariauseFilter = require("react-aria/useFilter");
var $9dfug$reactariaprivateutilsuseFormReset = require("react-aria/private/utils/useFormReset");
var $9dfug$reactariaprivateformuseFormValidation = require("react-aria/private/form/useFormValidation");
var $9dfug$reactariauseHover = require("react-aria/useHover");
var $9dfug$reactariauseId = require("react-aria/useId");
var $9dfug$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $9dfug$reactariauseObjectRef = require("react-aria/useObjectRef");
var $9dfug$reactariauseOverlayTrigger = require("react-aria/useOverlayTrigger");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "MobileComboBox", function () { return $72900fdaeb95a933$export$7637df911c069b4d; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







































const $72900fdaeb95a933$export$7637df911c069b4d = /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).forwardRef(function MobileComboBox(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    let { isQuiet: isQuiet, isDisabled: isDisabled, isReadOnly: isReadOnly, isRequired: isRequired, validationBehavior: validationBehavior, name: name, formValue: formValue = 'text', allowsCustomValue: allowsCustomValue } = props;
    if (allowsCustomValue) formValue = 'text';
    let { contains: contains } = (0, $9dfug$reactariauseFilter.useFilter)({
        sensitivity: 'base'
    });
    let state = (0, $9dfug$reactstatelyuseComboBoxState.useComboBoxState)({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: true,
        // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
        // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
        // then the closing of the tray will call setFocused(false) again due to cleanup effect)
        shouldCloseOnBlur: false
    });
    let buttonRef = (0, $9dfug$react.useRef)(null);
    let domRef = (0, $65aea7b37663976b$exports.useFocusableRef)(ref, buttonRef);
    let { triggerProps: triggerProps, overlayProps: overlayProps } = (0, $9dfug$reactariauseOverlayTrigger.useOverlayTrigger)({
        type: 'listbox'
    }, state, buttonRef);
    let inputRef = (0, $9dfug$react.useRef)(null);
    (0, $9dfug$reactariaprivateformuseFormValidation.useFormValidation)({
        ...props,
        focus: ()=>buttonRef.current?.focus()
    }, state, inputRef);
    let { isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = state.displayValidation;
    let validationState = props.validationState || (isInvalid ? 'invalid' : undefined);
    let errorMessage = props.errorMessage ?? validationErrors.join(' ');
    let { labelProps: labelProps, fieldProps: fieldProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = (0, $9dfug$reactariauseField.useField)({
        ...props,
        labelElementType: 'span',
        isInvalid: isInvalid,
        errorMessage: errorMessage
    });
    // Focus the button and show focus ring when clicking on the label
    labelProps.onClick = ()=>{
        if (!props.isDisabled) {
            buttonRef.current?.focus();
            (0, $9dfug$reactariaprivateinteractionsuseFocusVisible.setInteractionModality)('keyboard');
        }
    };
    let inputProps = {
        type: 'hidden',
        name: name,
        value: formValue === 'text' ? state.inputValue : String(state.selectedKey)
    };
    if (validationBehavior === 'native') {
        // Use a hidden <input type="text"> rather than <input type="hidden">
        // so that an empty value blocks HTML form submission when the field is required.
        inputProps.type = 'text';
        inputProps.hidden = true;
        inputProps.required = isRequired;
        // Ignore react warning.
        inputProps.onChange = ()=>{};
    }
    (0, $9dfug$reactariaprivateutilsuseFormReset.useFormReset)(inputRef, formValue === 'text' ? state.defaultInputValue : state.defaultSelectedKey, formValue === 'text' ? state.setInputValue : state.setSelectedKey);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, ($parcel$interopDefault($9dfug$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $b93966d678e0af07$exports.Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: validationState,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        elementType: "span",
        ref: domRef,
        includeNecessityIndicatorInAccessibilityName: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement($72900fdaeb95a933$export$adfa0abcd5972f7e, {
        ...(0, $9dfug$reactariamergeProps.mergeProps)(triggerProps, fieldProps, {
            autoFocus: props.autoFocus
        }),
        ref: buttonRef,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        isPlaceholder: !state.inputValue,
        validationState: validationState,
        onPress: ()=>!isReadOnly && state.open(null, 'manual')
    }, state.inputValue || props.placeholder || '')), /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("input", {
        ...inputProps,
        ref: inputRef
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $378dee1409fe2937$exports.Tray), {
        state: state,
        isFixedHeight: true,
        ...overlayProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement($72900fdaeb95a933$var$ComboBoxTray, {
        ...props,
        onClose: state.close,
        overlayProps: overlayProps,
        state: state
    })));
});
const $72900fdaeb95a933$export$adfa0abcd5972f7e = /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).forwardRef(function ComboBoxButton(props, ref) {
    let { isQuiet: isQuiet, isDisabled: isDisabled, isPlaceholder: isPlaceholder, validationState: validationState, children: children, style: style, className: className } = props;
    let stringFormatter = (0, $9dfug$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($3bfb46fc68ccce33$exports))), '@react-spectrum/combobox');
    let valueId = (0, $9dfug$reactariauseId.useId)();
    let invalidId = (0, $9dfug$reactariauseId.useId)();
    let validId = (0, $9dfug$reactariauseId.useId)();
    let validationIcon = validationState === 'invalid' ? /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, ($parcel$interopDefault($9dfug$spectrumiconsuiAlertMedium))), {
        id: invalidId,
        "aria-label": stringFormatter.format('invalid')
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, ($parcel$interopDefault($9dfug$spectrumiconsuiCheckmarkMedium))), {
        id: validId,
        "aria-label": stringFormatter.format('valid')
    });
    let validation = /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).cloneElement(validationIcon, {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-validationIcon', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input-validationIcon'))
    });
    let objRef = (0, $9dfug$reactariauseObjectRef.useObjectRef)(ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $9dfug$reactariauseHover.useHover)({});
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $9dfug$reactariauseButton.useButton)({
        ...props,
        'aria-labelledby': [
            props['aria-labelledby'],
            props['aria-label'] && !props['aria-labelledby'] ? props.id : null,
            valueId,
            validationState === 'invalid' ? invalidId : null,
            validationState === 'valid' ? validId : null
        ].filter(Boolean).join(' '),
        elementType: 'div'
    }, objRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $9dfug$reactariaFocusRing.FocusRing), {
        focusClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'is-focused'),
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("div", {
        ...(0, $9dfug$reactariamergeProps.mergeProps)(hoverProps, buttonProps),
        "aria-haspopup": "dialog",
        ref: objRef,
        style: {
            ...style,
            outline: 'none'
        },
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid' && !isDisabled,
            'is-hovered': isHovered
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'mobile-combobox'), className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Textfield--valid': validationState === 'valid' && !isDisabled,
            'spectrum-Textfield--quiet': isQuiet
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-field'))
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-input', {
            'is-hovered': isHovered,
            'is-placeholder': isPlaceholder,
            'is-disabled': isDisabled
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($53185441bef09fa8$exports))), 'spectrum-Field-field')), (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'mobile-input'))
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("span", {
        id: valueId,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'mobile-value')
    }, children)), validationState && !isDisabled ? validation : null), /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'spectrum-FieldButton', {
            'spectrum-FieldButton--quiet': isQuiet,
            'is-active': isPressed,
            'is-disabled': isDisabled,
            'spectrum-FieldButton--invalid': validationState === 'invalid' && !isDisabled,
            'is-hovered': isHovered
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-FieldButton'))
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, ($parcel$interopDefault($9dfug$spectrumiconsuiChevronDownMedium))), {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-Dropdown-chevron')
    }))));
});
function $72900fdaeb95a933$var$ComboBoxTray(props) {
    let { state: // completionMode = 'suggest',
    state, isDisabled: isDisabled, validationState: validationState, label: label, overlayProps: overlayProps, loadingState: loadingState, onLoadMore: onLoadMore, onClose: onClose } = props;
    let timeout = (0, $9dfug$react.useRef)(null);
    let [showLoading, setShowLoading] = (0, $9dfug$react.useState)(false);
    let inputRef = (0, $9dfug$react.useRef)(null);
    let buttonRef = (0, $9dfug$react.useRef)(null);
    let popoverRef = (0, $9dfug$react.useRef)(null);
    let listBoxRef = (0, $9dfug$react.useRef)(null);
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let layout = (0, $cb7ee1d9d5613db9$exports.useListBoxLayout)();
    let stringFormatter = (0, $9dfug$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($3bfb46fc68ccce33$exports))), '@react-spectrum/combobox');
    let { inputProps: inputProps, listBoxProps: listBoxProps, labelProps: labelProps } = (0, $9dfug$reactariauseComboBox.useComboBox)({
        ...props,
        // completionMode,
        layoutDelegate: layout,
        buttonRef: (0, $65aea7b37663976b$exports.unwrapDOMRef)(buttonRef),
        popoverRef: popoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        // Handled outside the tray.
        name: undefined
    }, state);
    (0, ($parcel$interopDefault($9dfug$react))).useEffect(()=>{
        if (inputRef.current) (0, $9dfug$reactariaprivateinteractionsfocusSafely.focusSafely)(inputRef.current);
    }, []);
    (0, ($parcel$interopDefault($9dfug$react))).useEffect(()=>{
        // When the tray closes, set state.isFocused (i.e. the tray input's focus tracker) to false.
        // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
        // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
        // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363
        if (!state.isOpen && state.isFocused) state.setFocused(false);
    });
    let { dialogProps: dialogProps } = (0, $9dfug$reactariauseDialog.useDialog)({
        'aria-labelledby': (0, $9dfug$reactariauseId.useId)(labelProps.id)
    }, popoverRef);
    // Override the role of the input to "searchbox" instead of "combobox".
    // Since the listbox is always visible, the combobox role doesn't really give us anything.
    // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
    // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
    // open the virtual keyboard rather than closing the tray.
    // Unlike "combobox", "aria-expanded" is not a valid attribute on "searchbox".
    inputProps.role = 'searchbox';
    inputProps['aria-haspopup'] = 'listbox';
    delete inputProps['aria-expanded'];
    delete inputProps.onTouchEnd;
    let clearButton = /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $0fc8553a4214494f$exports.ClearButton), {
        preventFocus: true,
        "aria-label": stringFormatter.format('clear'),
        excludeFromTabOrder: true,
        onPress: ()=>{
            state.setInputValue('');
            inputRef.current?.focus();
        },
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let loadingCircle = /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $948c2416aa3a9507$exports.ProgressCircle), {
        "aria-label": stringFormatter.format('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-circleLoader', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-circleLoader'))
    });
    // Close the software keyboard on scroll to give the user a bigger area to scroll.
    // But only do this if scrolling with touch, otherwise it can cause issues with touch
    // screen readers.
    let isTouchDown = (0, $9dfug$react.useRef)(false);
    let onTouchStart = ()=>{
        isTouchDown.current = true;
    };
    let onTouchEnd = ()=>{
        isTouchDown.current = false;
    };
    let onScroll = (0, $9dfug$react.useCallback)(()=>{
        if (!inputRef.current || (0, $9dfug$reactariaprivateutilsshadowdomDOMFunctions.getActiveElement)() !== inputRef.current || !isTouchDown.current) return;
        popoverRef.current?.focus();
    }, [
        inputRef,
        popoverRef,
        isTouchDown
    ]);
    let inputValue = inputProps.value;
    let lastInputValue = (0, $9dfug$react.useRef)(inputValue);
    (0, $9dfug$react.useEffect)(()=>{
        if (loadingState === 'filtering' && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (loadingState !== 'filtering') {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            if (timeout.current) clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        loadingState,
        inputValue,
        showLoading
    ]);
    let onKeyDown = (e)=>{
        // Close virtual keyboard if user hits Enter w/o any focused options
        if (e.key === 'Enter' && state.selectionManager.focusedKey == null) popoverRef.current?.focus();
        else inputProps.onKeyDown?.(e);
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $9dfug$reactariaFocusScope.FocusScope), {
        restoreFocus: true,
        contain: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("div", {
        ...(0, $9dfug$reactariamergeProps.mergeProps)(overlayProps, dialogProps),
        ref: popoverRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'tray-dialog')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $9dfug$reactariaOverlay.DismissButton), {
        onDismiss: onClose
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $827dbb466e199966$exports.TextFieldBase), {
        label: label,
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            onKeyDown: onKeyDown
        },
        inputRef: inputRef,
        isDisabled: isDisabled,
        isLoading: showLoading && loadingState === 'filtering',
        loadingIndicator: loadingState != null ? loadingCircle : undefined,
        validationState: validationState,
        labelAlign: "start",
        labelPosition: "top",
        wrapperChildren: (state.inputValue !== '' || loadingState === 'filtering' || validationState != null) && !props.isReadOnly ? clearButton : undefined,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search', 'spectrum-Textfield', 'spectrum-Search--loadable', {
            'spectrum-Search--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Search--valid': validationState === 'valid' && !isDisabled
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'tray-textfield', {
            'has-label': !!props.label
        })),
        inputClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'tray-textfield-input', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-input')),
        validationIconClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-validationIcon')
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $cb7ee1d9d5613db9$exports.ListBoxBase), {
        ...listBoxProps,
        domProps: {
            onTouchStart: onTouchStart,
            onTouchEnd: onTouchEnd
        },
        disallowEmptySelection: true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        renderEmptyState: ()=>loadingState !== 'loading' && /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement("span", {
                className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'no-results')
            }, stringFormatter.format('noResults')),
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($bd481a8d9a232d25$exports))), 'tray-listbox'),
        ref: listBoxRef,
        onScroll: onScroll,
        onLoadMore: onLoadMore,
        isLoading: isLoading
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9dfug$react))).createElement((0, $9dfug$reactariaOverlay.DismissButton), {
        onDismiss: onClose
    })));
}
