// @adobe/react-spectrum/dist/private/buttongroup/ButtonGroup.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
require("../buttongroup_vars.css");
var $ecd450464147f757$exports = require("../buttongroup_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $48ux6$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $48ux6$react = require("react");
var $48ux6$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $48ux6$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");
var $48ux6$reactariaprivateutilsuseValueEffect = require("react-aria/private/utils/useValueEffect");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ButtonGroup", function () { return $cae6b34e4dffcb70$export$69b1032f2ecdf404; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 










const $cae6b34e4dffcb70$export$69b1032f2ecdf404 = /*#__PURE__*/ (0, ($parcel$interopDefault($48ux6$react))).forwardRef(function ButtonGroup(props, ref) {
    let { scale: scale } = (0, $544fc82701fc93e9$exports.useProvider)();
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $feede71cddc0c5f3$exports.useSlotProps)(props, 'buttonGroup');
    let { children: children, orientation: orientation = 'horizontal', isDisabled: isDisabled, align: align = 'start', ...otherProps } = props;
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let [hasOverflow, setHasOverflow] = (0, $48ux6$reactariaprivateutilsuseValueEffect.useValueEffect)(false);
    let checkForOverflow = (0, $48ux6$react.useCallback)(()=>{
        let computeHasOverflow = ()=>{
            if (domRef.current && orientation === 'horizontal') {
                let buttonGroupChildren = Array.from(domRef.current.children);
                let maxX = domRef.current.offsetWidth + 1; // + 1 to account for rounding errors
                // If any buttons have negative X positions (align="end") or extend beyond
                // the width of the button group (align="start"), then switch to vertical.
                if (buttonGroupChildren.some((child)=>child.offsetLeft < 0 || child.offsetLeft + child.offsetWidth > maxX)) return true;
                return false;
            }
        };
        if (orientation === 'horizontal') setHasOverflow(function*() {
            // Force to horizontal for measurement.
            yield false;
            // Measure, and update if there is overflow.
            yield computeHasOverflow();
        });
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        domRef,
        orientation,
        scale,
        setHasOverflow,
        children
    ]);
    // There are two main reasons we need to remeasure:
    // 1. Internal changes: Check for initial overflow or when orientation/scale/children change (from checkForOverflow dep array)
    (0, $48ux6$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        checkForOverflow();
    }, [
        checkForOverflow
    ]);
    // 2. External changes: buttongroup won't change size due to any parents changing size, so listen to its container for size changes to figure out if we should remeasure
    let parent = (0, $48ux6$react.useRef)(undefined);
    (0, $48ux6$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        if (domRef.current) parent.current = domRef.current.parentElement;
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        domRef.current
    ]);
    (0, $48ux6$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: parent,
        onResize: checkForOverflow
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($48ux6$react))).createElement("div", {
        ...(0, $48ux6$reactariafilterDOMProps.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($ecd450464147f757$exports))), 'spectrum-ButtonGroup', {
            'spectrum-ButtonGroup--vertical': orientation === 'vertical' || hasOverflow,
            'spectrum-ButtonGroup--alignEnd': align === 'end',
            'spectrum-ButtonGroup--alignCenter': align === 'center'
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($48ux6$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            button: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($ecd450464147f757$exports))), 'spectrum-ButtonGroup-Button')
            }
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($48ux6$react))).createElement((0, $544fc82701fc93e9$exports.Provider), {
        isDisabled: isDisabled
    }, children)));
});
