// @adobe/react-spectrum/dist/private/buttongroup/ButtonGroup.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {Provider as $089943c7a219141c$export$2881499e37b75b9a, useProvider as $089943c7a219141c$export$693cdb10cec23617, useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686, useSlotProps as $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3} from "../utils/Slots.js";
import "../buttongroup_vars.css";
import $8aBSe$buttongroup_vars_cssmjs from "../buttongroup_vars_css.mjs";
import {useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {filterDOMProps as $8aBSe$filterDOMProps} from "react-aria/filterDOMProps";
import $8aBSe$react, {useCallback as $8aBSe$useCallback, useRef as $8aBSe$useRef} from "react";
import {useLayoutEffect as $8aBSe$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useResizeObserver as $8aBSe$useResizeObserver} from "react-aria/private/utils/useResizeObserver";
import {useValueEffect as $8aBSe$useValueEffect} from "react-aria/private/utils/useValueEffect";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 










const $020e7479c60446a5$export$69b1032f2ecdf404 = /*#__PURE__*/ (0, $8aBSe$react).forwardRef(function ButtonGroup(props, ref) {
    let { scale: scale } = (0, $089943c7a219141c$export$693cdb10cec23617)();
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    props = (0, $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3)(props, 'buttonGroup');
    let { children: children, orientation: orientation = 'horizontal', isDisabled: isDisabled, align: align = 'start', ...otherProps } = props;
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(otherProps);
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let [hasOverflow, setHasOverflow] = (0, $8aBSe$useValueEffect)(false);
    let checkForOverflow = (0, $8aBSe$useCallback)(()=>{
        let computeHasOverflow = ()=>{
            if (domRef.current && orientation === 'horizontal') {
                let buttonGroupChildren = Array.from(domRef.current.children);
                let maxX = domRef.current.offsetWidth + 1; // + 1 to account for rounding errors
                // If any buttons have negative X positions (align="end") or extend beyond
                // the width of the button group (align="start"), then switch to vertical.
                if (buttonGroupChildren.some((child)=>child.offsetLeft < 0 || child.offsetLeft + child.offsetWidth > maxX)) return true;
                return false;
            }
        };
        if (orientation === 'horizontal') setHasOverflow(function*() {
            // Force to horizontal for measurement.
            yield false;
            // Measure, and update if there is overflow.
            yield computeHasOverflow();
        });
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        domRef,
        orientation,
        scale,
        setHasOverflow,
        children
    ]);
    // There are two main reasons we need to remeasure:
    // 1. Internal changes: Check for initial overflow or when orientation/scale/children change (from checkForOverflow dep array)
    (0, $8aBSe$useLayoutEffect)(()=>{
        checkForOverflow();
    }, [
        checkForOverflow
    ]);
    // 2. External changes: buttongroup won't change size due to any parents changing size, so listen to its container for size changes to figure out if we should remeasure
    let parent = (0, $8aBSe$useRef)(undefined);
    (0, $8aBSe$useLayoutEffect)(()=>{
        if (domRef.current) parent.current = domRef.current.parentElement;
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        domRef.current
    ]);
    (0, $8aBSe$useResizeObserver)({
        ref: parent,
        onResize: checkForOverflow
    });
    return /*#__PURE__*/ (0, $8aBSe$react).createElement("div", {
        ...(0, $8aBSe$filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($8aBSe$buttongroup_vars_cssmjs))), 'spectrum-ButtonGroup', {
            'spectrum-ButtonGroup--vertical': orientation === 'vertical' || hasOverflow,
            'spectrum-ButtonGroup--alignEnd': align === 'end',
            'spectrum-ButtonGroup--alignCenter': align === 'center'
        }, styleProps.className)
    }, /*#__PURE__*/ (0, $8aBSe$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: {
            button: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($8aBSe$buttongroup_vars_cssmjs))), 'spectrum-ButtonGroup-Button')
            }
        }
    }, /*#__PURE__*/ (0, $8aBSe$react).createElement((0, $089943c7a219141c$export$2881499e37b75b9a), {
        isDisabled: isDisabled
    }, children)));
});


export {$020e7479c60446a5$export$69b1032f2ecdf404 as ButtonGroup};
