// @adobe/react-spectrum/dist/private/actionbar/ActionBar.cjs
var $183c5173677598aa$exports = require("../button/ActionButton.cjs");
var $1f2a1f451a6aa23a$exports = require("../actiongroup/ActionGroup.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $16e4ff10c839d8fe$exports = require("./intlStrings.cjs");
var $1048bdce1c849903$exports = require("../overlays/OpenTransition.cjs");
require("./actionbar.css");
var $b462d0874eb51e82$exports = require("./actionbar_css.cjs");
var $15e3b68ec42125a9$exports = require("../text/Text.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $6UHGv$reactariaprivateliveannouncerLiveAnnouncer = require("react-aria/private/live-announcer/LiveAnnouncer");
var $6UHGv$spectrumiconsuiCrossLarge = require("@spectrum-icons/ui/CrossLarge");
var $6UHGv$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $6UHGv$reactariaFocusScope = require("react-aria/FocusScope");
var $6UHGv$react = require("react");
var $6UHGv$reactariauseKeyboard = require("react-aria/useKeyboard");
var $6UHGv$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ActionBar", function () { return $d8369c3f1737954d$export$e213cebad6250b4a; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
















const $d8369c3f1737954d$export$e213cebad6250b4a = /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).forwardRef(function ActionBar(props, ref) {
    let isOpen = props.selectedItemCount !== 0;
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement((0, $1048bdce1c849903$exports.OpenTransition), {
        nodeRef: domRef,
        in: isOpen,
        mountOnEnter: true,
        unmountOnExit: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement($d8369c3f1737954d$var$ActionBarInnerWithRef, {
        ...props,
        ref: domRef
    }));
});
function $d8369c3f1737954d$var$ActionBarInner(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    let { children: children, isEmphasized: isEmphasized, onAction: onAction, onClearSelection: onClearSelection, selectedItemCount: selectedItemCount, isOpen: isOpen, buttonLabelBehavior: buttonLabelBehavior = 'collapse', items: items, disabledKeys: disabledKeys } = props;
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    let stringFormatter = (0, $6UHGv$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($16e4ff10c839d8fe$exports))), '@react-spectrum/actionbar');
    // Store the last count greater than zero in a ref so that we can retain it while rendering the fade-out animation.
    let [lastCount, setLastCount] = (0, $6UHGv$react.useState)(selectedItemCount);
    if ((selectedItemCount === 'all' || selectedItemCount > 0) && selectedItemCount !== lastCount) setLastCount(selectedItemCount);
    let { keyboardProps: keyboardProps } = (0, $6UHGv$reactariauseKeyboard.useKeyboard)({
        onKeyDown (e) {
            if (e.key === 'Escape') {
                e.preventDefault();
                onClearSelection();
            }
        }
    });
    // Announce "actions available" on mount.
    let isInitial = (0, $6UHGv$react.useRef)(true);
    (0, $6UHGv$react.useEffect)(()=>{
        if (isInitial.current) {
            isInitial.current = false;
            (0, $6UHGv$reactariaprivateliveannouncerLiveAnnouncer.announce)(stringFormatter.format('actionsAvailable'));
        }
    }, [
        stringFormatter
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement((0, $6UHGv$reactariaFocusScope.FocusScope), {
        restoreFocus: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement("div", {
        ...(0, $6UHGv$reactariafilterDOMProps.filterDOMProps)(props),
        ...styleProps,
        ...keyboardProps,
        ref: ref,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($b462d0874eb51e82$exports))), 'react-spectrum-ActionBar', {
            'react-spectrum-ActionBar--emphasized': isEmphasized,
            'is-open': isOpen
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($b462d0874eb51e82$exports))), 'react-spectrum-ActionBar-bar')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement((0, $1f2a1f451a6aa23a$exports.ActionGroup), {
        items: items,
        "aria-label": stringFormatter.format('actions'),
        isQuiet: true,
        staticColor: isEmphasized ? 'white' : undefined,
        overflowMode: "collapse",
        buttonLabelBehavior: buttonLabelBehavior,
        onAction: onAction,
        disabledKeys: disabledKeys,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($b462d0874eb51e82$exports))), 'react-spectrum-ActionBar-actionGroup')
    }, children), /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement((0, $183c5173677598aa$exports.ActionButton), {
        gridArea: (0, ($parcel$interopDefault($b462d0874eb51e82$exports))).clear,
        "aria-label": stringFormatter.format('clearSelection'),
        onPress: ()=>onClearSelection(),
        isQuiet: true,
        staticColor: isEmphasized ? 'white' : undefined
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement((0, ($parcel$interopDefault($6UHGv$spectrumiconsuiCrossLarge))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).createElement((0, $15e3b68ec42125a9$exports.Text), {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($b462d0874eb51e82$exports))), 'react-spectrum-ActionBar-selectedCount')
    }, lastCount === 'all' ? stringFormatter.format('selectedAll') : stringFormatter.format('selected', {
        count: lastCount
    })))));
}
const $d8369c3f1737954d$var$ActionBarInnerWithRef = /*#__PURE__*/ (0, ($parcel$interopDefault($6UHGv$react))).forwardRef($d8369c3f1737954d$var$ActionBarInner);
