// @adobe/react-spectrum/dist/private/actionbar/ActionBar.js
import {ActionButton as $c265dbb41bfd0210$export$cfc7921d29ef7b80} from "../button/ActionButton.js";
import {ActionGroup as $78c8311cc10fd6f1$export$c21a5597f732a168} from "../actiongroup/ActionGroup.js";
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import $4oqpT$intlStringsjs from "./intlStrings.js";
import {OpenTransition as $dd2af037c5de1a3e$export$b847a40ee92eff38} from "../overlays/OpenTransition.js";
import "./actionbar.css";
import $4oqpT$actionbar_cssmjs from "./actionbar_css.mjs";
import {Text as $42dd7396e689e4e6$export$5f1af8db9871e1d6} from "../text/Text.js";
import {useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {announce as $4oqpT$announce} from "react-aria/private/live-announcer/LiveAnnouncer";
import $4oqpT$spectrumiconsuiCrossLarge from "@spectrum-icons/ui/CrossLarge";
import {filterDOMProps as $4oqpT$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusScope as $4oqpT$FocusScope} from "react-aria/FocusScope";
import $4oqpT$react, {useState as $4oqpT$useState, useRef as $4oqpT$useRef, useEffect as $4oqpT$useEffect} from "react";
import {useKeyboard as $4oqpT$useKeyboard} from "react-aria/useKeyboard";
import {useLocalizedStringFormatter as $4oqpT$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
















const $524588d80192d966$export$e213cebad6250b4a = /*#__PURE__*/ (0, $4oqpT$react).forwardRef(function ActionBar(props, ref) {
    let isOpen = props.selectedItemCount !== 0;
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    return /*#__PURE__*/ (0, $4oqpT$react).createElement((0, $dd2af037c5de1a3e$export$b847a40ee92eff38), {
        nodeRef: domRef,
        in: isOpen,
        mountOnEnter: true,
        unmountOnExit: true
    }, /*#__PURE__*/ (0, $4oqpT$react).createElement($524588d80192d966$var$ActionBarInnerWithRef, {
        ...props,
        ref: domRef
    }));
});
function $524588d80192d966$var$ActionBarInner(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    let { children: children, isEmphasized: isEmphasized, onAction: onAction, onClearSelection: onClearSelection, selectedItemCount: selectedItemCount, isOpen: isOpen, buttonLabelBehavior: buttonLabelBehavior = 'collapse', items: items, disabledKeys: disabledKeys } = props;
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    let stringFormatter = (0, $4oqpT$useLocalizedStringFormatter)((0, ($parcel$interopDefault($4oqpT$intlStringsjs))), '@react-spectrum/actionbar');
    // Store the last count greater than zero in a ref so that we can retain it while rendering the fade-out animation.
    let [lastCount, setLastCount] = (0, $4oqpT$useState)(selectedItemCount);
    if ((selectedItemCount === 'all' || selectedItemCount > 0) && selectedItemCount !== lastCount) setLastCount(selectedItemCount);
    let { keyboardProps: keyboardProps } = (0, $4oqpT$useKeyboard)({
        onKeyDown (e) {
            if (e.key === 'Escape') {
                e.preventDefault();
                onClearSelection();
            }
        }
    });
    // Announce "actions available" on mount.
    let isInitial = (0, $4oqpT$useRef)(true);
    (0, $4oqpT$useEffect)(()=>{
        if (isInitial.current) {
            isInitial.current = false;
            (0, $4oqpT$announce)(stringFormatter.format('actionsAvailable'));
        }
    }, [
        stringFormatter
    ]);
    return /*#__PURE__*/ (0, $4oqpT$react).createElement((0, $4oqpT$FocusScope), {
        restoreFocus: true
    }, /*#__PURE__*/ (0, $4oqpT$react).createElement("div", {
        ...(0, $4oqpT$filterDOMProps)(props),
        ...styleProps,
        ...keyboardProps,
        ref: ref,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4oqpT$actionbar_cssmjs))), 'react-spectrum-ActionBar', {
            'react-spectrum-ActionBar--emphasized': isEmphasized,
            'is-open': isOpen
        }, styleProps.className)
    }, /*#__PURE__*/ (0, $4oqpT$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4oqpT$actionbar_cssmjs))), 'react-spectrum-ActionBar-bar')
    }, /*#__PURE__*/ (0, $4oqpT$react).createElement((0, $78c8311cc10fd6f1$export$c21a5597f732a168), {
        items: items,
        "aria-label": stringFormatter.format('actions'),
        isQuiet: true,
        staticColor: isEmphasized ? 'white' : undefined,
        overflowMode: "collapse",
        buttonLabelBehavior: buttonLabelBehavior,
        onAction: onAction,
        disabledKeys: disabledKeys,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4oqpT$actionbar_cssmjs))), 'react-spectrum-ActionBar-actionGroup')
    }, children), /*#__PURE__*/ (0, $4oqpT$react).createElement((0, $c265dbb41bfd0210$export$cfc7921d29ef7b80), {
        gridArea: (0, ($parcel$interopDefault($4oqpT$actionbar_cssmjs))).clear,
        "aria-label": stringFormatter.format('clearSelection'),
        onPress: ()=>onClearSelection(),
        isQuiet: true,
        staticColor: isEmphasized ? 'white' : undefined
    }, /*#__PURE__*/ (0, $4oqpT$react).createElement((0, $4oqpT$spectrumiconsuiCrossLarge), null)), /*#__PURE__*/ (0, $4oqpT$react).createElement((0, $42dd7396e689e4e6$export$5f1af8db9871e1d6), {
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4oqpT$actionbar_cssmjs))), 'react-spectrum-ActionBar-selectedCount')
    }, lastCount === 'all' ? stringFormatter.format('selectedAll') : stringFormatter.format('selected', {
        count: lastCount
    })))));
}
const $524588d80192d966$var$ActionBarInnerWithRef = /*#__PURE__*/ (0, $4oqpT$react).forwardRef($524588d80192d966$var$ActionBarInner);


export {$524588d80192d966$export$e213cebad6250b4a as ActionBar};
