// @adobe/react-spectrum/dist/private/theme-express/expressTheme.cjs
var $817d9c4715e2f636$exports = require("../theme-default/defaultTheme.cjs");
require("../vars_express.css");
var $d3bb744e95191474$exports = require("../vars_express_css.cjs");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "expressTheme", function () { return $c3e57b6bb12370aa$export$84d107d9d68a7850; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />


let $c3e57b6bb12370aa$export$84d107d9d68a7850 = {
    ...(0, $817d9c4715e2f636$exports.defaultTheme),
    global: {
        ...(0, $817d9c4715e2f636$exports.defaultTheme).global,
        express: (0, ($parcel$interopDefault($d3bb744e95191474$exports))).express
    },
    medium: {
        ...(0, $817d9c4715e2f636$exports.defaultTheme).medium,
        express: (0, ($parcel$interopDefault($d3bb744e95191474$exports))).medium
    },
    large: {
        ...(0, $817d9c4715e2f636$exports.defaultTheme).large,
        express: (0, ($parcel$interopDefault($d3bb744e95191474$exports))).large
    }
};
