// @adobe/react-spectrum/dist/private/theme-express/expressTheme.mjs
import {defaultTheme as $cdf1a36dc96a08db$export$164de7ab8df77ef0} from "../theme-default/defaultTheme.mjs";
import "../vars_express.css";
import $iCqbH$vars_express_cssmjs from "../vars_express_css.mjs";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />


let $199c2414a81b1b54$export$84d107d9d68a7850 = {
    ...(0, $cdf1a36dc96a08db$export$164de7ab8df77ef0),
    global: {
        ...(0, $cdf1a36dc96a08db$export$164de7ab8df77ef0).global,
        express: (0, ($parcel$interopDefault($iCqbH$vars_express_cssmjs))).express
    },
    medium: {
        ...(0, $cdf1a36dc96a08db$export$164de7ab8df77ef0).medium,
        express: (0, ($parcel$interopDefault($iCqbH$vars_express_cssmjs))).medium
    },
    large: {
        ...(0, $cdf1a36dc96a08db$export$164de7ab8df77ef0).large,
        express: (0, ($parcel$interopDefault($iCqbH$vars_express_cssmjs))).large
    }
};


export {$199c2414a81b1b54$export$84d107d9d68a7850 as expressTheme};
