// @adobe/react-spectrum/dist/private/dialog/AlertDialog.js
import {Button as $88f996b7df3752a1$export$353f5b6fc5456de1} from "../button/Button.js";
import {ButtonGroup as $020e7479c60446a5$export$69b1032f2ecdf404} from "../buttongroup/ButtonGroup.js";
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {Content as $558e2ad48297783c$export$7c6e2c02157bb7d2} from "../view/Content.js";
import {Dialog as $89418a3659cad0c7$export$3ddf2d174ce01153} from "./Dialog.js";
import {DialogContext as $abd082d14fc11575$export$8b93a07348a7730c} from "./context.js";
import {Divider as $7848cbd4e1a92d87$export$2e0a83ec2e27ecbb} from "../divider/Divider.js";
import {Heading as $ddc09b0bc61c28b1$export$a8a3e93435678ff9} from "../text/Heading.js";
import $eB83Y$intlStringsjs from "./intlStrings.js";
import "../dialog_vars.css";
import $eB83Y$dialog_vars_cssmjs from "../dialog_vars_css.mjs";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import $eB83Y$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import {chain as $eB83Y$chain} from "react-aria/chain";
import {filterDOMProps as $eB83Y$filterDOMProps} from "react-aria/filterDOMProps";
import $eB83Y$react, {forwardRef as $eB83Y$forwardRef, useContext as $eB83Y$useContext} from "react";
import {useLocalizedStringFormatter as $eB83Y$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 















const $0c3b86927117991f$export$de466dd8317b0b75 = /*#__PURE__*/ (0, $eB83Y$forwardRef)(function AlertDialog(props, ref) {
    let { onClose: onClose = ()=>{} } = (0, $eB83Y$useContext)((0, $abd082d14fc11575$export$8b93a07348a7730c)) || {};
    let { variant: variant, children: children, primaryActionLabel: primaryActionLabel, secondaryActionLabel: secondaryActionLabel, cancelLabel: cancelLabel, autoFocusButton: autoFocusButton, title: title, isPrimaryActionDisabled: isPrimaryActionDisabled, isSecondaryActionDisabled: isSecondaryActionDisabled, onCancel: onCancel = ()=>{}, onPrimaryAction: onPrimaryAction = ()=>{}, onSecondaryAction: onSecondaryAction = ()=>{}, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(otherProps);
    let stringFormatter = (0, $eB83Y$useLocalizedStringFormatter)((0, ($parcel$interopDefault($eB83Y$intlStringsjs))), '@react-spectrum/dialog');
    let confirmVariant = 'primary';
    if (variant) {
        if (variant === 'confirmation') confirmVariant = 'cta';
        else if (variant === 'destructive') confirmVariant = 'negative';
    }
    return /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $89418a3659cad0c7$export$3ddf2d174ce01153), {
        UNSAFE_style: styleProps.style,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($eB83Y$dialog_vars_cssmjs))), {
            [`spectrum-Dialog--${variant}`]: variant
        }, styleProps.className),
        isHidden: styleProps.hidden,
        size: "M",
        role: "alertdialog",
        ref: ref,
        ...(0, $eB83Y$filterDOMProps)(props)
    }, /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $ddc09b0bc61c28b1$export$a8a3e93435678ff9), null, title), (variant === 'error' || variant === 'warning') && /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $eB83Y$spectrumiconsuiAlertMedium), {
        slot: "typeIcon",
        "aria-label": stringFormatter.format('alert')
    }), /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $7848cbd4e1a92d87$export$2e0a83ec2e27ecbb), null), /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $558e2ad48297783c$export$7c6e2c02157bb7d2), null, children), /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $020e7479c60446a5$export$69b1032f2ecdf404), {
        align: "end"
    }, cancelLabel && /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $88f996b7df3752a1$export$353f5b6fc5456de1), {
        variant: "secondary",
        onPress: ()=>(0, $eB83Y$chain)(onClose(), onCancel()),
        autoFocus: autoFocusButton === 'cancel',
        "data-testid": "rsp-AlertDialog-cancelButton"
    }, cancelLabel), secondaryActionLabel && /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $88f996b7df3752a1$export$353f5b6fc5456de1), {
        variant: "secondary",
        onPress: ()=>(0, $eB83Y$chain)(onClose(), onSecondaryAction()),
        isDisabled: isSecondaryActionDisabled,
        autoFocus: autoFocusButton === 'secondary',
        "data-testid": "rsp-AlertDialog-secondaryButton"
    }, secondaryActionLabel), /*#__PURE__*/ (0, $eB83Y$react).createElement((0, $88f996b7df3752a1$export$353f5b6fc5456de1), {
        variant: confirmVariant,
        onPress: ()=>(0, $eB83Y$chain)(onClose(), onPrimaryAction()),
        isDisabled: isPrimaryActionDisabled,
        autoFocus: autoFocusButton === 'primary',
        "data-testid": "rsp-AlertDialog-confirmButton"
    }, primaryActionLabel)));
});


export {$0c3b86927117991f$export$de466dd8317b0b75 as AlertDialog};
