// @adobe/react-spectrum/dist/private/dialog/AlertDialog.cjs
var $92256f4fe9ec9f59$exports = require("../button/Button.cjs");
var $cae6b34e4dffcb70$exports = require("../buttongroup/ButtonGroup.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $685ef7ad6d6d547f$exports = require("../view/Content.cjs");
var $db50fa4488be370e$exports = require("./Dialog.cjs");
var $4965a9907649f3b8$exports = require("./context.cjs");
var $70687492d9e04f58$exports = require("../divider/Divider.cjs");
var $ef0b1cd7dce2b6c2$exports = require("../text/Heading.cjs");
var $8d2681652f6a64b7$exports = require("./intlStrings.cjs");
require("../dialog_vars.css");
var $5f6caa7677856121$exports = require("../dialog_vars_css.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $7B7fC$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $7B7fC$reactariachain = require("react-aria/chain");
var $7B7fC$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $7B7fC$react = require("react");
var $7B7fC$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "AlertDialog", function () { return $c82f495a1ede320e$export$de466dd8317b0b75; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 















const $c82f495a1ede320e$export$de466dd8317b0b75 = /*#__PURE__*/ (0, $7B7fC$react.forwardRef)(function AlertDialog(props, ref) {
    let { onClose: onClose = ()=>{} } = (0, $7B7fC$react.useContext)((0, $4965a9907649f3b8$exports.DialogContext)) || {};
    let { variant: variant, children: children, primaryActionLabel: primaryActionLabel, secondaryActionLabel: secondaryActionLabel, cancelLabel: cancelLabel, autoFocusButton: autoFocusButton, title: title, isPrimaryActionDisabled: isPrimaryActionDisabled, isSecondaryActionDisabled: isSecondaryActionDisabled, onCancel: onCancel = ()=>{}, onPrimaryAction: onPrimaryAction = ()=>{}, onSecondaryAction: onSecondaryAction = ()=>{}, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    let stringFormatter = (0, $7B7fC$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($8d2681652f6a64b7$exports))), '@react-spectrum/dialog');
    let confirmVariant = 'primary';
    if (variant) {
        if (variant === 'confirmation') confirmVariant = 'cta';
        else if (variant === 'destructive') confirmVariant = 'negative';
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $db50fa4488be370e$exports.Dialog), {
        UNSAFE_style: styleProps.style,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5f6caa7677856121$exports))), {
            [`spectrum-Dialog--${variant}`]: variant
        }, styleProps.className),
        isHidden: styleProps.hidden,
        size: "M",
        role: "alertdialog",
        ref: ref,
        ...(0, $7B7fC$reactariafilterDOMProps.filterDOMProps)(props)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $ef0b1cd7dce2b6c2$exports.Heading), null, title), (variant === 'error' || variant === 'warning') && /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, ($parcel$interopDefault($7B7fC$spectrumiconsuiAlertMedium))), {
        slot: "typeIcon",
        "aria-label": stringFormatter.format('alert')
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $70687492d9e04f58$exports.Divider), null), /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $685ef7ad6d6d547f$exports.Content), null, children), /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $cae6b34e4dffcb70$exports.ButtonGroup), {
        align: "end"
    }, cancelLabel && /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $92256f4fe9ec9f59$exports.Button), {
        variant: "secondary",
        onPress: ()=>(0, $7B7fC$reactariachain.chain)(onClose(), onCancel()),
        autoFocus: autoFocusButton === 'cancel',
        "data-testid": "rsp-AlertDialog-cancelButton"
    }, cancelLabel), secondaryActionLabel && /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $92256f4fe9ec9f59$exports.Button), {
        variant: "secondary",
        onPress: ()=>(0, $7B7fC$reactariachain.chain)(onClose(), onSecondaryAction()),
        isDisabled: isSecondaryActionDisabled,
        autoFocus: autoFocusButton === 'secondary',
        "data-testid": "rsp-AlertDialog-secondaryButton"
    }, secondaryActionLabel), /*#__PURE__*/ (0, ($parcel$interopDefault($7B7fC$react))).createElement((0, $92256f4fe9ec9f59$exports.Button), {
        variant: confirmVariant,
        onPress: ()=>(0, $7B7fC$reactariachain.chain)(onClose(), onPrimaryAction()),
        isDisabled: isPrimaryActionDisabled,
        autoFocus: autoFocusButton === 'primary',
        "data-testid": "rsp-AlertDialog-confirmButton"
    }, primaryActionLabel)));
});
