// @adobe/react-spectrum/dist/private/dialog/Dialog.js
import {ActionButton as $c265dbb41bfd0210$export$cfc7921d29ef7b80} from "../button/ActionButton.js";
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {DialogContext as $abd082d14fc11575$export$8b93a07348a7730c} from "./context.js";
import {Grid as $727c1a1d9e8b8d73$export$ef2184bd89960b14} from "../layout/Grid.js";
import $cgl0K$intlStringsjs from "./intlStrings.js";
import {SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686, useSlotProps as $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3} from "../utils/Slots.js";
import "../dialog_vars.css";
import $cgl0K$dialog_vars_cssmjs from "../dialog_vars_css.mjs";
import {unwrapDOMRef as $c234463e9ef56637$export$c7e28c72a4823176, useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useHasChild as $584638b763a93bff$export$e52e2242b6d0f1d4} from "../utils/useHasChild.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {useDialog as $cgl0K$useDialog} from "react-aria/useDialog";
import $cgl0K$spectrumiconsuiCrossLarge from "@spectrum-icons/ui/CrossLarge";
import {mergeProps as $cgl0K$mergeProps} from "react-aria/mergeProps";
import $cgl0K$react, {useContext as $cgl0K$useContext, useRef as $cgl0K$useRef, useMemo as $cgl0K$useMemo} from "react";
import {useLocalizedStringFormatter as $cgl0K$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 














let $89418a3659cad0c7$var$sizeMap = {
    S: 'small',
    M: 'medium',
    L: 'large',
    fullscreen: 'fullscreen',
    fullscreenTakeover: 'fullscreenTakeover'
};
const $89418a3659cad0c7$export$3ddf2d174ce01153 = /*#__PURE__*/ (0, $cgl0K$react).forwardRef(function Dialog(props, ref) {
    props = (0, $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3)(props, 'dialog');
    let { type: type = 'modal', ...contextProps } = (0, $cgl0K$useContext)((0, $abd082d14fc11575$export$8b93a07348a7730c)) || {};
    let { children: children, isDismissable: isDismissable = contextProps.isDismissable, onDismiss: onDismiss = contextProps.onClose, size: size, ...otherProps } = props;
    let stringFormatter = (0, $cgl0K$useLocalizedStringFormatter)((0, ($parcel$interopDefault($cgl0K$intlStringsjs))), '@react-spectrum/dialog');
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(otherProps);
    size = type === 'popover' ? size || 'S' : size || 'L';
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let gridRef = (0, $cgl0K$useRef)(null);
    let sizeVariant = $89418a3659cad0c7$var$sizeMap[type] || $89418a3659cad0c7$var$sizeMap[size];
    let { dialogProps: dialogProps, titleProps: titleProps } = (0, $cgl0K$useDialog)((0, $cgl0K$mergeProps)(contextProps, props), domRef);
    let hasHeader = (0, $584638b763a93bff$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-header']}`, (0, $c234463e9ef56637$export$c7e28c72a4823176)(gridRef));
    let hasHeading = (0, $584638b763a93bff$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-heading']}`, (0, $c234463e9ef56637$export$c7e28c72a4823176)(gridRef));
    let hasFooter = (0, $584638b763a93bff$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-footer']}`, (0, $c234463e9ef56637$export$c7e28c72a4823176)(gridRef));
    let hasTypeIcon = (0, $584638b763a93bff$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-typeIcon']}`, (0, $c234463e9ef56637$export$c7e28c72a4823176)(gridRef));
    let slots = (0, $cgl0K$useMemo)(()=>({
            hero: {
                UNSAFE_className: (0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-hero']
            },
            heading: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs))), 'spectrum-Dialog-heading', {
                    'spectrum-Dialog-heading--noHeader': !hasHeader,
                    'spectrum-Dialog-heading--noTypeIcon': !hasTypeIcon
                }),
                level: 2,
                ...titleProps
            },
            header: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs))), 'spectrum-Dialog-header', {
                    'spectrum-Dialog-header--noHeading': !hasHeading,
                    'spectrum-Dialog-header--noTypeIcon': !hasTypeIcon
                })
            },
            typeIcon: {
                UNSAFE_className: (0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-typeIcon']
            },
            divider: {
                UNSAFE_className: (0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-divider'],
                size: 'M'
            },
            content: {
                UNSAFE_className: (0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-content']
            },
            footer: {
                UNSAFE_className: (0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-footer']
            },
            buttonGroup: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs))), 'spectrum-Dialog-buttonGroup', {
                    'spectrum-Dialog-buttonGroup--noFooter': !hasFooter
                }),
                align: 'end'
            }
        }), // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        hasFooter,
        hasHeader,
        titleProps
    ]);
    return /*#__PURE__*/ (0, $cgl0K$react).createElement("section", {
        ...styleProps,
        ...dialogProps,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs))), 'spectrum-Dialog', {
            [`spectrum-Dialog--${sizeVariant}`]: sizeVariant,
            'spectrum-Dialog--dismissable': isDismissable
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ (0, $cgl0K$react).createElement((0, $727c1a1d9e8b8d73$export$ef2184bd89960b14), {
        ref: gridRef,
        UNSAFE_className: (0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-grid']
    }, /*#__PURE__*/ (0, $cgl0K$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: slots
    }, children), isDismissable && /*#__PURE__*/ (0, $cgl0K$react).createElement((0, $c265dbb41bfd0210$export$cfc7921d29ef7b80), {
        UNSAFE_className: (0, ($parcel$interopDefault($cgl0K$dialog_vars_cssmjs)))['spectrum-Dialog-closeButton'],
        isQuiet: true,
        "aria-label": stringFormatter.format('dismiss'),
        onPress: onDismiss
    }, /*#__PURE__*/ (0, $cgl0K$react).createElement((0, $cgl0K$spectrumiconsuiCrossLarge), null))));
});


export {$89418a3659cad0c7$export$3ddf2d174ce01153 as Dialog};
