// @adobe/react-spectrum/dist/private/dialog/Dialog.cjs
var $183c5173677598aa$exports = require("../button/ActionButton.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $4965a9907649f3b8$exports = require("./context.cjs");
var $d6479700d21b596b$exports = require("../layout/Grid.cjs");
var $8d2681652f6a64b7$exports = require("./intlStrings.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
require("../dialog_vars.css");
var $5f6caa7677856121$exports = require("../dialog_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $dd6348d4a1a51ff9$exports = require("../utils/useHasChild.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $hi7zK$reactariauseDialog = require("react-aria/useDialog");
var $hi7zK$spectrumiconsuiCrossLarge = require("@spectrum-icons/ui/CrossLarge");
var $hi7zK$reactariamergeProps = require("react-aria/mergeProps");
var $hi7zK$react = require("react");
var $hi7zK$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Dialog", function () { return $db50fa4488be370e$export$3ddf2d174ce01153; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 














let $db50fa4488be370e$var$sizeMap = {
    S: 'small',
    M: 'medium',
    L: 'large',
    fullscreen: 'fullscreen',
    fullscreenTakeover: 'fullscreenTakeover'
};
const $db50fa4488be370e$export$3ddf2d174ce01153 = /*#__PURE__*/ (0, ($parcel$interopDefault($hi7zK$react))).forwardRef(function Dialog(props, ref) {
    props = (0, $feede71cddc0c5f3$exports.useSlotProps)(props, 'dialog');
    let { type: type = 'modal', ...contextProps } = (0, $hi7zK$react.useContext)((0, $4965a9907649f3b8$exports.DialogContext)) || {};
    let { children: children, isDismissable: isDismissable = contextProps.isDismissable, onDismiss: onDismiss = contextProps.onClose, size: size, ...otherProps } = props;
    let stringFormatter = (0, $hi7zK$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($8d2681652f6a64b7$exports))), '@react-spectrum/dialog');
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    size = type === 'popover' ? size || 'S' : size || 'L';
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let gridRef = (0, $hi7zK$react.useRef)(null);
    let sizeVariant = $db50fa4488be370e$var$sizeMap[type] || $db50fa4488be370e$var$sizeMap[size];
    let { dialogProps: dialogProps, titleProps: titleProps } = (0, $hi7zK$reactariauseDialog.useDialog)((0, $hi7zK$reactariamergeProps.mergeProps)(contextProps, props), domRef);
    let hasHeader = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-header']}`, (0, $65aea7b37663976b$exports.unwrapDOMRef)(gridRef));
    let hasHeading = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-heading']}`, (0, $65aea7b37663976b$exports.unwrapDOMRef)(gridRef));
    let hasFooter = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-footer']}`, (0, $65aea7b37663976b$exports.unwrapDOMRef)(gridRef));
    let hasTypeIcon = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-typeIcon']}`, (0, $65aea7b37663976b$exports.unwrapDOMRef)(gridRef));
    let slots = (0, $hi7zK$react.useMemo)(()=>({
            hero: {
                UNSAFE_className: (0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-hero']
            },
            heading: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5f6caa7677856121$exports))), 'spectrum-Dialog-heading', {
                    'spectrum-Dialog-heading--noHeader': !hasHeader,
                    'spectrum-Dialog-heading--noTypeIcon': !hasTypeIcon
                }),
                level: 2,
                ...titleProps
            },
            header: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5f6caa7677856121$exports))), 'spectrum-Dialog-header', {
                    'spectrum-Dialog-header--noHeading': !hasHeading,
                    'spectrum-Dialog-header--noTypeIcon': !hasTypeIcon
                })
            },
            typeIcon: {
                UNSAFE_className: (0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-typeIcon']
            },
            divider: {
                UNSAFE_className: (0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-divider'],
                size: 'M'
            },
            content: {
                UNSAFE_className: (0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-content']
            },
            footer: {
                UNSAFE_className: (0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-footer']
            },
            buttonGroup: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5f6caa7677856121$exports))), 'spectrum-Dialog-buttonGroup', {
                    'spectrum-Dialog-buttonGroup--noFooter': !hasFooter
                }),
                align: 'end'
            }
        }), // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        hasFooter,
        hasHeader,
        titleProps
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($hi7zK$react))).createElement("section", {
        ...styleProps,
        ...dialogProps,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5f6caa7677856121$exports))), 'spectrum-Dialog', {
            [`spectrum-Dialog--${sizeVariant}`]: sizeVariant,
            'spectrum-Dialog--dismissable': isDismissable
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($hi7zK$react))).createElement((0, $d6479700d21b596b$exports.Grid), {
        ref: gridRef,
        UNSAFE_className: (0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-grid']
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($hi7zK$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: slots
    }, children), isDismissable && /*#__PURE__*/ (0, ($parcel$interopDefault($hi7zK$react))).createElement((0, $183c5173677598aa$exports.ActionButton), {
        UNSAFE_className: (0, ($parcel$interopDefault($5f6caa7677856121$exports)))['spectrum-Dialog-closeButton'],
        isQuiet: true,
        "aria-label": stringFormatter.format('dismiss'),
        onPress: onDismiss
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($hi7zK$react))).createElement((0, ($parcel$interopDefault($hi7zK$spectrumiconsuiCrossLarge))), null))));
});
