// @adobe/react-spectrum/dist/private/link/Link.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {getWrappedElement as $79576b63c62941bb$export$a5f5a6912b18861c} from "../utils/getWrappedElement.js";
import "../link_vars.css";
import $ftupv$link_vars_cssmjs from "../link_vars_css.mjs";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useSlotProps as $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3} from "../utils/Slots.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {useLink as $ftupv$useLink} from "react-aria/useLink";
import {FocusRing as $ftupv$FocusRing} from "react-aria/FocusRing";
import {mergeProps as $ftupv$mergeProps} from "react-aria/mergeProps";
import {mergeRefs as $ftupv$mergeRefs} from "react-aria/mergeRefs";
import $ftupv$react, {useRef as $ftupv$useRef} from "react";
import {useHover as $ftupv$useHover} from "react-aria/useHover";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 











let $9003d7dbbb34db95$var$isOldReact = parseInt((0, $ftupv$react).version, 10) <= 18;
function $9003d7dbbb34db95$export$a6c7ac8248d6e38a(props) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    props = (0, $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3)(props, 'link');
    let { variant: variant = 'primary', isQuiet: isQuiet, children: children, href: // @ts-ignore
    href } = props;
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $ftupv$useHover)({});
    let ref = (0, $ftupv$useRef)(null);
    let { linkProps: linkProps } = (0, $ftupv$useLink)({
        ...props,
        elementType: !href && typeof children === 'string' ? 'span' : 'a'
    }, ref);
    let domProps = {
        ...styleProps,
        ...(0, $ftupv$mergeProps)(linkProps, hoverProps),
        ref: ref,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($ftupv$link_vars_cssmjs))), 'spectrum-Link', {
            'spectrum-Link--quiet': isQuiet,
            [`spectrum-Link--${variant}`]: variant,
            'is-hovered': isHovered
        }, styleProps.className)
    };
    let link;
    if (href) link = /*#__PURE__*/ (0, $ftupv$react).createElement("a", domProps, children);
    else {
        // Backward compatibility.
        let wrappedChild = (0, $79576b63c62941bb$export$a5f5a6912b18861c)(children);
        let mergedRef = ref;
        if ($9003d7dbbb34db95$var$isOldReact) // @ts-ignore
        mergedRef = (0, $ftupv$mergeRefs)(ref, wrappedChild.ref);
        else // @ts-ignore
        mergedRef = (0, $ftupv$mergeRefs)(ref, wrappedChild.props.ref);
        link = /*#__PURE__*/ (0, $ftupv$react).cloneElement(wrappedChild, {
            ...(0, $ftupv$mergeProps)(wrappedChild.props, domProps),
            // @ts-ignore https://github.com/facebook/react/issues/8873
            ref: mergedRef
        });
    }
    return /*#__PURE__*/ (0, $ftupv$react).createElement((0, $ftupv$FocusRing), {
        focusRingClass: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($ftupv$link_vars_cssmjs))), 'focus-ring')
    }, link);
}


export {$9003d7dbbb34db95$export$a6c7ac8248d6e38a as Link};
