// @adobe/react-spectrum/dist/private/link/Link.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $9d2e0c8a06f8c833$exports = require("../utils/getWrappedElement.cjs");
require("../link_vars.css");
var $4262988de14a1a3b$exports = require("../link_vars_css.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $lyheQ$reactariauseLink = require("react-aria/useLink");
var $lyheQ$reactariaFocusRing = require("react-aria/FocusRing");
var $lyheQ$reactariamergeProps = require("react-aria/mergeProps");
var $lyheQ$reactariamergeRefs = require("react-aria/mergeRefs");
var $lyheQ$react = require("react");
var $lyheQ$reactariauseHover = require("react-aria/useHover");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Link", function () { return $37d87e020d601bb1$export$a6c7ac8248d6e38a; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 











let $37d87e020d601bb1$var$isOldReact = parseInt((0, ($parcel$interopDefault($lyheQ$react))).version, 10) <= 18;
function $37d87e020d601bb1$export$a6c7ac8248d6e38a(props) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $feede71cddc0c5f3$exports.useSlotProps)(props, 'link');
    let { variant: variant = 'primary', isQuiet: isQuiet, children: children, href: // @ts-ignore
    href } = props;
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $lyheQ$reactariauseHover.useHover)({});
    let ref = (0, $lyheQ$react.useRef)(null);
    let { linkProps: linkProps } = (0, $lyheQ$reactariauseLink.useLink)({
        ...props,
        elementType: !href && typeof children === 'string' ? 'span' : 'a'
    }, ref);
    let domProps = {
        ...styleProps,
        ...(0, $lyheQ$reactariamergeProps.mergeProps)(linkProps, hoverProps),
        ref: ref,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($4262988de14a1a3b$exports))), 'spectrum-Link', {
            'spectrum-Link--quiet': isQuiet,
            [`spectrum-Link--${variant}`]: variant,
            'is-hovered': isHovered
        }, styleProps.className)
    };
    let link;
    if (href) link = /*#__PURE__*/ (0, ($parcel$interopDefault($lyheQ$react))).createElement("a", domProps, children);
    else {
        // Backward compatibility.
        let wrappedChild = (0, $9d2e0c8a06f8c833$exports.getWrappedElement)(children);
        let mergedRef = ref;
        if ($37d87e020d601bb1$var$isOldReact) // @ts-ignore
        mergedRef = (0, $lyheQ$reactariamergeRefs.mergeRefs)(ref, wrappedChild.ref);
        else // @ts-ignore
        mergedRef = (0, $lyheQ$reactariamergeRefs.mergeRefs)(ref, wrappedChild.props.ref);
        link = /*#__PURE__*/ (0, ($parcel$interopDefault($lyheQ$react))).cloneElement(wrappedChild, {
            ...(0, $lyheQ$reactariamergeProps.mergeProps)(wrappedChild.props, domProps),
            // @ts-ignore https://github.com/facebook/react/issues/8873
            ref: mergedRef
        });
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lyheQ$react))).createElement((0, $lyheQ$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($4262988de14a1a3b$exports))), 'focus-ring')
    }, link);
}
