// @adobe/react-spectrum/dist/private/table/InsertionIndicator.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
require("./table.css");
var $a82a0782618e719b$exports = require("./table_css.cjs");
var $783600b7b364e0bf$exports = require("./TableViewBase.cjs");
var $3rpEw$react = require("react");
var $3rpEw$reactariaVisuallyHidden = require("react-aria/VisuallyHidden");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "InsertionIndicator", function () { return $e534bab1179262d5$export$2c0bab5914a9d088; });
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $e534bab1179262d5$export$2c0bab5914a9d088(props) {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks } = (0, $783600b7b364e0bf$exports.useTableContext)();
    const { target: target, rowProps: rowProps } = props;
    let ref = (0, $3rpEw$react.useRef)(null);
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator(props, dropState, ref);
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $3rpEw$reactariaVisuallyHidden.useVisuallyHidden)();
    let isDropTarget = dropState.isDropTarget(target);
    if (!isDropTarget && dropIndicatorProps['aria-hidden']) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3rpEw$react))).createElement("div", {
        style: {
            position: 'absolute',
            top: typeof rowProps.style?.top === 'number' && typeof rowProps.style?.height === 'number' ? rowProps.style.top + (target.dropPosition === 'after' ? rowProps.style.height : 0) : 0,
            width: rowProps.style?.width
        },
        role: "row",
        "aria-hidden": dropIndicatorProps['aria-hidden']
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3rpEw$react))).createElement("div", {
        role: "gridcell",
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($a82a0782618e719b$exports))), 'react-spectrum-Table-InsertionIndicator', {
            'react-spectrum-Table-InsertionIndicator--dropTarget': isDropTarget
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3rpEw$react))).createElement("div", {
        ...visuallyHiddenProps,
        role: "button",
        ...dropIndicatorProps,
        ref: ref
    })));
}
