// @adobe/react-spectrum/dist/private/progress/ProgressCircle.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
require("../circleloader_vars.css");
var $2e83e0b7417d39d2$exports = require("../circleloader_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $NgywW$reactstatelyprivateutilsnumber = require("react-stately/private/utils/number");
var $NgywW$react = require("react");
var $NgywW$reactariauseProgressBar = require("react-aria/useProgressBar");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ProgressCircle", function () { return $948c2416aa3a9507$export$c79b9d6b4cc92af7; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






const $948c2416aa3a9507$export$c79b9d6b4cc92af7 = /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).forwardRef(function ProgressCircle(props, ref) {
    let { value: value = 0, minValue: minValue = 0, maxValue: maxValue = 100, size: size = 'M', staticColor: staticColor, variant: variant, isIndeterminate: isIndeterminate = false, 'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledby, ...otherProps } = props;
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    value = (0, $NgywW$reactstatelyprivateutilsnumber.clamp)(value, minValue, maxValue);
    let { progressBarProps: progressBarProps } = (0, $NgywW$reactariauseProgressBar.useProgressBar)({
        ...props,
        value: value
    });
    let subMask1Style = {};
    let subMask2Style = {};
    if (!isIndeterminate) {
        let percentage = (value - minValue) / (maxValue - minValue) * 100;
        let angle;
        if (percentage > 0 && percentage <= 50) {
            angle = -180 + percentage / 50 * 180;
            subMask1Style.transform = `rotate(${angle}deg)`;
            subMask2Style.transform = 'rotate(-180deg)';
        } else if (percentage > 50) {
            angle = -180 + (percentage - 50) / 50 * 180;
            subMask1Style.transform = 'rotate(0deg)';
            subMask2Style.transform = `rotate(${angle}deg)`;
        }
    }
    if (!ariaLabel && !ariaLabelledby && process.env.NODE_ENV !== 'production') console.warn('ProgressCircle requires an aria-label or aria-labelledby attribute for accessibility');
    return /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        ...styleProps,
        ...progressBarProps,
        ref: domRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader', {
            'spectrum-CircleLoader--indeterminate': isIndeterminate,
            'spectrum-CircleLoader--small': size === 'S',
            'spectrum-CircleLoader--large': size === 'L',
            'spectrum-CircleLoader--overBackground': variant === 'overBackground',
            'spectrum-CircleLoader--staticWhite': staticColor === 'white',
            'spectrum-CircleLoader--staticBlack': staticColor === 'black'
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-track')
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-fills')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-fillMask1')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-fillSubMask1'),
        "data-testid": "fillSubMask1",
        style: subMask1Style
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-fill')
    }))), /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-fillMask2')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-fillSubMask2'),
        "data-testid": "fillSubMask2",
        style: subMask2Style
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($NgywW$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($2e83e0b7417d39d2$exports))), 'spectrum-CircleLoader-fill')
    })))));
});
