// @adobe/react-spectrum/dist/exports/index.js
import {ActionGroup as $78c8311cc10fd6f1$export$c21a5597f732a168} from "../private/actiongroup/ActionGroup.js";
import {Badge as $b1b5eaed19db7dfb$export$37acb3580601e69a} from "../private/badge/Badge.js";
import {Breadcrumbs as $5fa86543125c475c$export$2dc68d50d56fbbd} from "../private/breadcrumbs/Breadcrumbs.js";
import {Accordion as $d33033465c302d3d$export$a766cd26d0d69044, Disclosure as $d33033465c302d3d$export$74a362b31437ec83, DisclosurePanel as $d33033465c302d3d$export$feabaa331e1d464c, DisclosureTitle as $d33033465c302d3d$export$7843c6a5b3e340a2} from "../private/accordion/Accordion.js";
import {ActionBar as $524588d80192d966$export$e213cebad6250b4a} from "../private/actionbar/ActionBar.js";
import {ActionBarContainer as $d998ea5bb37d669c$export$ac2eb07f267e434c} from "../private/actionbar/ActionBarContainer.js";
import {ActionButton as $c265dbb41bfd0210$export$cfc7921d29ef7b80} from "../private/button/ActionButton.js";
import {Button as $88f996b7df3752a1$export$353f5b6fc5456de1} from "../private/button/Button.js";
import {LogicButton as $9bb0441a39d7224a$export$9b0b80fed00ba8b1} from "../private/button/LogicButton.js";
import {ToggleButton as $e0864cdc7011a0bd$export$d2b052e7b4be1756} from "../private/button/ToggleButton.js";
import {Avatar as $e11bc02d3528b12b$export$e2255cf6045e8d47} from "../private/avatar/Avatar.js";
import {ButtonGroup as $020e7479c60446a5$export$69b1032f2ecdf404} from "../private/buttongroup/ButtonGroup.js";
import {Calendar as $056e9579a2d4c8dd$export$e1aef45b828286de} from "../private/calendar/Calendar.js";
import {RangeCalendar as $ebb070240f5bf202$export$a4f5c8b89d277a8d} from "../private/calendar/RangeCalendar.js";
import {Checkbox as $986e1e93e04146a6$export$48513f6b9f8ce62d} from "../private/checkbox/Checkbox.js";
import {CheckboxGroup as $a060f1760eaa1b2e$export$4aa08d5625cb8ead} from "../private/checkbox/CheckboxGroup.js";
import {ColorArea as $d6409d59c6d41934$export$b2103f68a961418e} from "../private/color/ColorArea.js";
import {ColorEditor as $464d9fb69b03568b$export$5aa54fd21eb08d23} from "../private/color/ColorEditor.js";
import {ColorField as $5b07f9ba6aa1e89e$export$b865d4358897bb17} from "../private/color/ColorField.js";
import {ColorPicker as $c04167cb12cd5428$export$9feb1bc2e5f1ccb3} from "../private/color/ColorPicker.js";
import {ColorSlider as $5234de165996dd10$export$44fd664bcca5b6fb} from "../private/color/ColorSlider.js";
import {ColorSwatch as $471078cf7ddd2506$export$cae13e90592f246a} from "../private/color/ColorSwatch.js";
import {ColorSwatchPicker as $7e1c53aba5966059$export$b46792416e3d8515} from "../private/color/ColorSwatchPicker.js";
import {ColorWheel as $4f4444fad9e6c791$export$f80663f808113381} from "../private/color/ColorWheel.js";
import {ComboBox as $74a97b68e98f70c1$export$72b9695b8216309a} from "../private/combobox/ComboBox.js";
import {ContextualHelp as $e74df8e6258adf4e$export$7d3cdb256c2ba320} from "../private/contextualhelp/ContextualHelp.js";
import {AlertDialog as $0c3b86927117991f$export$de466dd8317b0b75} from "../private/dialog/AlertDialog.js";
import {Dialog as $89418a3659cad0c7$export$3ddf2d174ce01153} from "../private/dialog/Dialog.js";
import {DialogTrigger as $bcff05049955156f$export$2e1e1122cf0cba88} from "../private/dialog/DialogTrigger.js";
import {DialogContainer as $929c2962c9d61dce$export$547754aada6e339f} from "../private/dialog/DialogContainer.js";
import {useDialogContainer as $754411bdeab736f8$export$a2f2d8fa6720dab1} from "../private/dialog/useDialogContainer.js";
import {DateField as $e2601972aa3002a4$export$d9781c7894a82487} from "../private/datepicker/DateField.js";
import {DatePicker as $b155a4096c914b21$export$5109c6dd95d8fb00} from "../private/datepicker/DatePicker.js";
import {DateRangePicker as $62a249cadcc61296$export$17334619f3ac2224} from "../private/datepicker/DateRangePicker.js";
import {TimeField as $404615aff7fb8653$export$5eaee2322dd727eb} from "../private/datepicker/TimeField.js";
import {Divider as $7848cbd4e1a92d87$export$2e0a83ec2e27ecbb} from "../private/divider/Divider.js";
import {DropZone as $9b7e00ab669c9f2e$export$3c6489d84dc98b6} from "../private/dropzone/DropZone.js";
import {Form as $d23ca6800ac02cf1$export$a7fed597f4b8afd8} from "../private/form/Form.js";
import {Icon as $68df32599618a4b7$export$f04a61298a47a40f} from "../private/icon/Icon.js";
import {IllustratedMessage as $0c0ec59e9de032cd$export$406dbc84c317ece0} from "../private/illustratedmessage/IllustratedMessage.js";
import {InlineAlert as $c59531e91448aebd$export$a3b2c96db9b0eb71} from "../private/inlinealert/InlineAlert.js";
import {Image as $8d4f5af52a313241$export$3e431a229df88919} from "../private/image/Image.js";
import {Flex as $9b6884c982c0954a$export$f51f4c4ede09e011} from "../private/layout/Flex.js";
import {fitContent as $727c1a1d9e8b8d73$export$2f0b47b0911ce698, Grid as $727c1a1d9e8b8d73$export$ef2184bd89960b14, minmax as $727c1a1d9e8b8d73$export$9c1b655deaca4988, repeat as $727c1a1d9e8b8d73$export$76d90c956114f2c2} from "../private/layout/Grid.js";
import {LabeledValue as $4f61dbf9ff82452c$export$d1328f67a56fa517} from "../private/labeledvalue/LabeledValue.js";
import {Link as $9003d7dbbb34db95$export$a6c7ac8248d6e38a} from "../private/link/Link.js";
import {ListBox as $e6f9f1de53801217$export$41f133550aa26f48} from "../private/listbox/ListBox.js";
import {ListView as $bcd1a74211acbd51$export$84d0dd190d551cd1} from "../private/list/ListView.js";
import {ActionMenu as $db2bd940d589c867$export$ed57a210b9d13bb6} from "../private/menu/ActionMenu.js";
import {ContextualHelpTrigger as $6bc666e61b0562d8$export$5413b169fff83e61} from "../private/menu/ContextualHelpTrigger.js";
import {Menu as $79ddee63a726ea3d$export$d9b273488cd8ce6f} from "../private/menu/Menu.js";
import {MenuTrigger as $9f6ebde23392f425$export$27d2ad3c5815583e} from "../private/menu/MenuTrigger.js";
import {SubmenuTrigger as $0d6e4ea605197c86$export$ecabc99eeffab7ca} from "../private/menu/SubmenuTrigger.js";
import {Meter as $a9930872c4fbddbb$export$62e3ae2a4090b879} from "../private/meter/Meter.js";
import {NumberField as $e8146d8fdc8bfe0f$export$63c5fa0b2fdccd2e} from "../private/numberfield/NumberField.js";
import {Picker as $fcdeb62019c30c53$export$ba25329847403e11} from "../private/picker/Picker.js";
import {ProgressBar as $49f1c6d050e398e0$export$c17561cb55d4db30} from "../private/progress/ProgressBar.js";
import {ProgressCircle as $277696409c391eff$export$c79b9d6b4cc92af7} from "../private/progress/ProgressCircle.js";
import {Provider as $089943c7a219141c$export$2881499e37b75b9a, useProvider as $089943c7a219141c$export$693cdb10cec23617} from "../private/provider/Provider.js";
import {Radio as $27a2aa405605847c$export$d7b12c4107be0d61} from "../private/radio/Radio.js";
import {RadioGroup as $a592e2b53348b1bb$export$a98f0dcb43a68a25} from "../private/radio/RadioGroup.js";
import {RangeSlider as $fe33cb66aef2fd43$export$826424dabc3dd705} from "../private/slider/RangeSlider.js";
import {Slider as $d9eec645e047af67$export$472062a354075cee} from "../private/slider/Slider.js";
import {SearchField as $f73cad999d6c6ed8$export$b94867ecbd698f21} from "../private/searchfield/SearchField.js";
import {StatusLight as $2bbb23c6b3bb5597$export$5f84c37a31c6e41c} from "../private/statuslight/StatusLight.js";
import {Switch as $0d6506a6fb520294$export$b5d5cf8927ab7262} from "../private/switch/Switch.js";
import {Heading as $ddc09b0bc61c28b1$export$a8a3e93435678ff9} from "../private/text/Heading.js";
import {Keyboard as $aab4a3f5e32e20d3$export$16e4d70cc375e707} from "../private/text/Keyboard.js";
import {Text as $42dd7396e689e4e6$export$5f1af8db9871e1d6} from "../private/text/Text.js";
import {Cell as $e9d8704413caaded$re_export$Cell, Column as $e9d8704413caaded$export$816b5d811295e6bc, Row as $e9d8704413caaded$re_export$Row, TableBody as $e9d8704413caaded$re_export$TableBody, TableHeader as $e9d8704413caaded$re_export$TableHeader, TableView as $e9d8704413caaded$export$b3c27e869d856b7} from "../private/table/TableView.js";
import {TabList as $89b03460ad791d07$export$e51a686c67fdaa2d, TabPanels as $89b03460ad791d07$export$5dae8d435677f210, Tabs as $89b03460ad791d07$export$b2539bed5023c21c} from "../private/tabs/Tabs.js";
import {TagGroup as $48b99bf0df8f69a7$export$67ea30858aaf75e3} from "../private/tag/TagGroup.js";
import {TextArea as $fee5ca43c2eb72c2$export$f5c9f3c2c4054eec} from "../private/textfield/TextArea.js";
import {TextField as $79f51739ebc77ec5$export$2c73285ae9390cec} from "../private/textfield/TextField.js";
import {darkTheme as $0ab08fc9e7ba60d4$export$3e936a8db52a10a0} from "../private/theme-dark/darkTheme.js";
import {defaultTheme as $3098a24c36a8df81$export$164de7ab8df77ef0} from "../private/theme-default/defaultTheme.js";
import {lightTheme as $5404d152e822cfd7$export$f30cb9bc4f736419} from "../private/theme-light/lightTheme.js";
import {ToastContainer as $afd6f21ce00ee063$export$f2815235e76a62b9, ToastQueue as $afd6f21ce00ee063$export$f1f8569633bbbec4} from "../private/toast/ToastContainer.js";
import {Tooltip as $3f07dad5c19b53b6$export$28c660c63b792dea} from "../private/tooltip/Tooltip.js";
import {TooltipTrigger as $3e76a1633f5aa2b7$export$8c610744efcf8a1d} from "../private/tooltip/TooltipTrigger.js";
import {TreeView as $110b5af1301d605f$export$6940b0d9c820eca7, TreeViewItem as $110b5af1301d605f$export$6e77ea6719814e9c, TreeViewItemContent as $110b5af1301d605f$export$9a5779ed3fade674} from "../private/tree/TreeView.js";
import {Content as $558e2ad48297783c$export$7c6e2c02157bb7d2} from "../private/view/Content.js";
import {Footer as $56c10d78b1922ef3$export$a06f1c675e846f6f} from "../private/view/Footer.js";
import {Header as $eba69d1aae786da8$export$8b251419efc915eb} from "../private/view/Header.js";
import {View as $04bf23ef16a59f4a$export$27a5bd065ad55220} from "../private/view/View.js";
import {Well as $a0f3ed92e1d5ff07$export$d89c679ee2f73242} from "../private/well/Well.js";
import {useDragAndDrop as $a835b70880df29d9$export$2cfc5be7a55829f6} from "../private/dnd/useDragAndDrop.js";
import "client-only";
import {parseColor as $923bcd35c1671d1b$re_export$parseColor, getColorChannels as $923bcd35c1671d1b$re_export$getColorChannels} from "react-stately/Color";
import {FileTrigger as $923bcd35c1671d1b$re_export$FileTrigger} from "react-aria-components/FileTrigger";
import {Item as $923bcd35c1671d1b$re_export$Item} from "react-stately/Item";
import {Section as $923bcd35c1671d1b$re_export$Section} from "react-stately/Section";
import {useAsyncList as $923bcd35c1671d1b$re_export$useAsyncList} from "react-stately/useAsyncList";
import {useListData as $923bcd35c1671d1b$re_export$useListData} from "react-stately/useListData";
import {useTreeData as $923bcd35c1671d1b$re_export$useTreeData} from "react-stately/useTreeData";
import {VisuallyHidden as $923bcd35c1671d1b$re_export$VisuallyHidden} from "react-aria/VisuallyHidden";
import {useCollator as $923bcd35c1671d1b$re_export$useCollator} from "react-aria/useCollator";
import {useDateFormatter as $923bcd35c1671d1b$re_export$useDateFormatter} from "react-aria/useDateFormatter";
import {useFilter as $923bcd35c1671d1b$re_export$useFilter} from "react-aria/useFilter";
import {useLocale as $923bcd35c1671d1b$re_export$useLocale} from "react-aria/I18nProvider";
import {useLocalizedStringFormatter as $923bcd35c1671d1b$re_export$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {useNumberFormatter as $923bcd35c1671d1b$re_export$useNumberFormatter} from "react-aria/useNumberFormatter";
import {SSRProvider as $923bcd35c1671d1b$re_export$SSRProvider} from "react-aria/SSRProvider";
import {DIRECTORY_DRAG_TYPE as $923bcd35c1671d1b$re_export$DIRECTORY_DRAG_TYPE, isDirectoryDropItem as $923bcd35c1671d1b$re_export$isDirectoryDropItem, isFileDropItem as $923bcd35c1671d1b$re_export$isFileDropItem, isTextDropItem as $923bcd35c1671d1b$re_export$isTextDropItem} from "react-aria/useDrop";
import {Collection as $923bcd35c1671d1b$re_export$Collection} from "react-aria/Collection";

/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ // Mark as a client only package. This will cause a build time error if you try
// to import it from a React Server Component in a framework like Next.js.











































































































export {$78c8311cc10fd6f1$export$c21a5597f732a168 as ActionGroup, $b1b5eaed19db7dfb$export$37acb3580601e69a as Badge, $5fa86543125c475c$export$2dc68d50d56fbbd as Breadcrumbs, $d33033465c302d3d$export$a766cd26d0d69044 as Accordion, $d33033465c302d3d$export$74a362b31437ec83 as Disclosure, $d33033465c302d3d$export$feabaa331e1d464c as DisclosurePanel, $d33033465c302d3d$export$7843c6a5b3e340a2 as DisclosureTitle, $524588d80192d966$export$e213cebad6250b4a as ActionBar, $d998ea5bb37d669c$export$ac2eb07f267e434c as ActionBarContainer, $c265dbb41bfd0210$export$cfc7921d29ef7b80 as ActionButton, $88f996b7df3752a1$export$353f5b6fc5456de1 as Button, $9bb0441a39d7224a$export$9b0b80fed00ba8b1 as LogicButton, $e0864cdc7011a0bd$export$d2b052e7b4be1756 as ToggleButton, $e11bc02d3528b12b$export$e2255cf6045e8d47 as Avatar, $020e7479c60446a5$export$69b1032f2ecdf404 as ButtonGroup, $056e9579a2d4c8dd$export$e1aef45b828286de as Calendar, $ebb070240f5bf202$export$a4f5c8b89d277a8d as RangeCalendar, $986e1e93e04146a6$export$48513f6b9f8ce62d as Checkbox, $a060f1760eaa1b2e$export$4aa08d5625cb8ead as CheckboxGroup, $d6409d59c6d41934$export$b2103f68a961418e as ColorArea, $464d9fb69b03568b$export$5aa54fd21eb08d23 as ColorEditor, $5b07f9ba6aa1e89e$export$b865d4358897bb17 as ColorField, $c04167cb12cd5428$export$9feb1bc2e5f1ccb3 as ColorPicker, $5234de165996dd10$export$44fd664bcca5b6fb as ColorSlider, $471078cf7ddd2506$export$cae13e90592f246a as ColorSwatch, $7e1c53aba5966059$export$b46792416e3d8515 as ColorSwatchPicker, $4f4444fad9e6c791$export$f80663f808113381 as ColorWheel, $923bcd35c1671d1b$re_export$parseColor as parseColor, $923bcd35c1671d1b$re_export$getColorChannels as getColorChannels, $74a97b68e98f70c1$export$72b9695b8216309a as ComboBox, $e74df8e6258adf4e$export$7d3cdb256c2ba320 as ContextualHelp, $0c3b86927117991f$export$de466dd8317b0b75 as AlertDialog, $89418a3659cad0c7$export$3ddf2d174ce01153 as Dialog, $bcff05049955156f$export$2e1e1122cf0cba88 as DialogTrigger, $929c2962c9d61dce$export$547754aada6e339f as DialogContainer, $754411bdeab736f8$export$a2f2d8fa6720dab1 as useDialogContainer, $e2601972aa3002a4$export$d9781c7894a82487 as DateField, $b155a4096c914b21$export$5109c6dd95d8fb00 as DatePicker, $62a249cadcc61296$export$17334619f3ac2224 as DateRangePicker, $404615aff7fb8653$export$5eaee2322dd727eb as TimeField, $7848cbd4e1a92d87$export$2e0a83ec2e27ecbb as Divider, $9b7e00ab669c9f2e$export$3c6489d84dc98b6 as DropZone, $923bcd35c1671d1b$re_export$FileTrigger as FileTrigger, $d23ca6800ac02cf1$export$a7fed597f4b8afd8 as Form, $68df32599618a4b7$export$f04a61298a47a40f as Icon, $0c0ec59e9de032cd$export$406dbc84c317ece0 as IllustratedMessage, $c59531e91448aebd$export$a3b2c96db9b0eb71 as InlineAlert, $8d4f5af52a313241$export$3e431a229df88919 as Image, $9b6884c982c0954a$export$f51f4c4ede09e011 as Flex, $727c1a1d9e8b8d73$export$ef2184bd89960b14 as Grid, $727c1a1d9e8b8d73$export$2f0b47b0911ce698 as fitContent, $727c1a1d9e8b8d73$export$9c1b655deaca4988 as minmax, $727c1a1d9e8b8d73$export$76d90c956114f2c2 as repeat, $4f61dbf9ff82452c$export$d1328f67a56fa517 as LabeledValue, $9003d7dbbb34db95$export$a6c7ac8248d6e38a as Link, $e6f9f1de53801217$export$41f133550aa26f48 as ListBox, $bcd1a74211acbd51$export$84d0dd190d551cd1 as ListView, $db2bd940d589c867$export$ed57a210b9d13bb6 as ActionMenu, $6bc666e61b0562d8$export$5413b169fff83e61 as ContextualHelpTrigger, $79ddee63a726ea3d$export$d9b273488cd8ce6f as Menu, $9f6ebde23392f425$export$27d2ad3c5815583e as MenuTrigger, $0d6e4ea605197c86$export$ecabc99eeffab7ca as SubmenuTrigger, $a9930872c4fbddbb$export$62e3ae2a4090b879 as Meter, $e8146d8fdc8bfe0f$export$63c5fa0b2fdccd2e as NumberField, $fcdeb62019c30c53$export$ba25329847403e11 as Picker, $49f1c6d050e398e0$export$c17561cb55d4db30 as ProgressBar, $277696409c391eff$export$c79b9d6b4cc92af7 as ProgressCircle, $089943c7a219141c$export$2881499e37b75b9a as Provider, $089943c7a219141c$export$693cdb10cec23617 as useProvider, $27a2aa405605847c$export$d7b12c4107be0d61 as Radio, $a592e2b53348b1bb$export$a98f0dcb43a68a25 as RadioGroup, $fe33cb66aef2fd43$export$826424dabc3dd705 as RangeSlider, $d9eec645e047af67$export$472062a354075cee as Slider, $f73cad999d6c6ed8$export$b94867ecbd698f21 as SearchField, $2bbb23c6b3bb5597$export$5f84c37a31c6e41c as StatusLight, $0d6506a6fb520294$export$b5d5cf8927ab7262 as Switch, $ddc09b0bc61c28b1$export$a8a3e93435678ff9 as Heading, $aab4a3f5e32e20d3$export$16e4d70cc375e707 as Keyboard, $42dd7396e689e4e6$export$5f1af8db9871e1d6 as Text, $e9d8704413caaded$export$b3c27e869d856b7 as TableView, $e9d8704413caaded$re_export$TableHeader as TableHeader, $e9d8704413caaded$re_export$TableBody as TableBody, $e9d8704413caaded$export$816b5d811295e6bc as Column, $e9d8704413caaded$re_export$Row as Row, $e9d8704413caaded$re_export$Cell as Cell, $89b03460ad791d07$export$b2539bed5023c21c as Tabs, $89b03460ad791d07$export$e51a686c67fdaa2d as TabList, $89b03460ad791d07$export$5dae8d435677f210 as TabPanels, $48b99bf0df8f69a7$export$67ea30858aaf75e3 as TagGroup, $fee5ca43c2eb72c2$export$f5c9f3c2c4054eec as TextArea, $79f51739ebc77ec5$export$2c73285ae9390cec as TextField, $0ab08fc9e7ba60d4$export$3e936a8db52a10a0 as darkTheme, $3098a24c36a8df81$export$164de7ab8df77ef0 as defaultTheme, $5404d152e822cfd7$export$f30cb9bc4f736419 as lightTheme, $afd6f21ce00ee063$export$f2815235e76a62b9 as ToastContainer, $afd6f21ce00ee063$export$f1f8569633bbbec4 as ToastQueue, $3f07dad5c19b53b6$export$28c660c63b792dea as Tooltip, $3e76a1633f5aa2b7$export$8c610744efcf8a1d as TooltipTrigger, $110b5af1301d605f$export$6940b0d9c820eca7 as TreeView, $110b5af1301d605f$export$6e77ea6719814e9c as TreeViewItem, $110b5af1301d605f$export$9a5779ed3fade674 as TreeViewItemContent, $558e2ad48297783c$export$7c6e2c02157bb7d2 as Content, $56c10d78b1922ef3$export$a06f1c675e846f6f as Footer, $eba69d1aae786da8$export$8b251419efc915eb as Header, $04bf23ef16a59f4a$export$27a5bd065ad55220 as View, $a0f3ed92e1d5ff07$export$d89c679ee2f73242 as Well, $923bcd35c1671d1b$re_export$Item as Item, $923bcd35c1671d1b$re_export$Section as Section, $923bcd35c1671d1b$re_export$useAsyncList as useAsyncList, $923bcd35c1671d1b$re_export$useListData as useListData, $923bcd35c1671d1b$re_export$useTreeData as useTreeData, $923bcd35c1671d1b$re_export$VisuallyHidden as VisuallyHidden, $923bcd35c1671d1b$re_export$useCollator as useCollator, $923bcd35c1671d1b$re_export$useDateFormatter as useDateFormatter, $923bcd35c1671d1b$re_export$useFilter as useFilter, $923bcd35c1671d1b$re_export$useLocale as useLocale, $923bcd35c1671d1b$re_export$useLocalizedStringFormatter as useLocalizedStringFormatter, $923bcd35c1671d1b$re_export$useNumberFormatter as useNumberFormatter, $923bcd35c1671d1b$re_export$SSRProvider as SSRProvider, $a835b70880df29d9$export$2cfc5be7a55829f6 as useDragAndDrop, $923bcd35c1671d1b$re_export$DIRECTORY_DRAG_TYPE as DIRECTORY_DRAG_TYPE, $923bcd35c1671d1b$re_export$isDirectoryDropItem as isDirectoryDropItem, $923bcd35c1671d1b$re_export$isFileDropItem as isFileDropItem, $923bcd35c1671d1b$re_export$isTextDropItem as isTextDropItem, $923bcd35c1671d1b$re_export$Collection as Collection};
