// @adobe/react-spectrum/dist/private/autocomplete/MobileSearchAutocomplete.mjs
import {classNames as $6e6392558d48dfec$export$ce4ab0c55987d1ff} from "../utils/classNames.mjs";
import {ClearButton as $ab14010a528467be$export$13ec83e50bf04290} from "../button/ClearButton.mjs";
import {Field as $adcd096854d27620$export$a455218a85c89869} from "../label/Field.mjs";
import $gWiNP$intlStringsmjs from "./intlStrings.mjs";
import {ListBoxBase as $ee13b4eccaed924f$export$1afdcf349979fb7e, useListBoxLayout as $ee13b4eccaed924f$export$25768ea656ae32a7} from "../listbox/ListBoxBase.mjs";
import {ProgressCircle as $1cfe37e7feefa23d$export$c79b9d6b4cc92af7} from "../progress/ProgressCircle.mjs";
import "./searchautocomplete.css";
import $gWiNP$searchautocomplete_cssmjs from "./searchautocomplete_css.mjs";
import "../search_vars.css";
import $gWiNP$search_vars_cssmjs from "../search_vars_css.mjs";
import "../inputgroup_vars.css";
import $gWiNP$inputgroup_vars_cssmjs from "../inputgroup_vars_css.mjs";
import {TextFieldBase as $b312f2102feb9487$export$d22444a338b6e3c2} from "../textfield/TextFieldBase.mjs";
import "../textfield_vars.css";
import $gWiNP$textfield_vars_cssmjs from "../textfield_vars_css.mjs";
import {Tray as $9fca089dca5508dc$export$4589ed81930b555c} from "../overlays/Tray.mjs";
import {useFocusableRef as $3c2c983d5210446c$export$96a734597687c040} from "../utils/useDOMRef.mjs";
import {useProviderProps as $71dfb0e0358a12de$export$521c373ccc32c300} from "../provider/Provider.mjs";
import $gWiNP$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import {useButton as $gWiNP$useButton} from "react-aria/useButton";
import $gWiNP$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {useComboBoxState as $gWiNP$useComboBoxState} from "react-stately/useComboBoxState";
import {DismissButton as $gWiNP$DismissButton} from "react-aria/Overlay";
import {focusSafely as $gWiNP$focusSafely} from "react-aria/private/interactions/focusSafely";
import {FocusScope as $gWiNP$FocusScope} from "react-aria/FocusScope";
import {getActiveElement as $gWiNP$getActiveElement} from "react-aria/private/utils/shadowdom/DOMFunctions";
import $gWiNP$spectrumiconsuiMagnifier from "@spectrum-icons/ui/Magnifier";
import {mergeProps as $gWiNP$mergeProps} from "react-aria/mergeProps";
import $gWiNP$react, {useCallback as $gWiNP$useCallback, useRef as $gWiNP$useRef, useState as $gWiNP$useState, useEffect as $gWiNP$useEffect} from "react";
import {setInteractionModality as $gWiNP$setInteractionModality} from "react-aria/private/interactions/useFocusVisible";
import {useDialog as $gWiNP$useDialog} from "react-aria/useDialog";
import {useField as $gWiNP$useField} from "react-aria/useField";
import {useFilter as $gWiNP$useFilter} from "react-aria/useFilter";
import {useFocusRing as $gWiNP$useFocusRing} from "react-aria/useFocusRing";
import {useFormReset as $gWiNP$useFormReset} from "react-aria/private/utils/useFormReset";
import {useFormValidation as $gWiNP$useFormValidation} from "react-aria/private/form/useFormValidation";
import {useHover as $gWiNP$useHover} from "react-aria/useHover";
import {useId as $gWiNP$useId} from "react-aria/useId";
import {useLocalizedStringFormatter as $gWiNP$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {useOverlayTrigger as $gWiNP$useOverlayTrigger} from "react-aria/useOverlayTrigger";
import {useSearchAutocomplete as $gWiNP$useSearchAutocomplete} from "react-aria/private/autocomplete/useSearchAutocomplete";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




































function $d9f9f86122f1f89b$var$ForwardMobileSearchAutocomplete(props, ref) {
    props = (0, $71dfb0e0358a12de$export$521c373ccc32c300)(props);
    let { isQuiet: isQuiet, isDisabled: isDisabled, isRequired: isRequired, validationBehavior: validationBehavior, validate: validate, name: name, isReadOnly: isReadOnly, onSubmit: onSubmit = ()=>{} } = props;
    let { contains: contains } = (0, $gWiNP$useFilter)({
        sensitivity: 'base'
    });
    let state = (0, $gWiNP$useComboBoxState)({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: true,
        // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
        // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
        // then the closing of the tray will call setFocused(false) again due to cleanup effect)
        shouldCloseOnBlur: false,
        allowsCustomValue: true,
        onSelectionChange: (key)=>key !== null && onSubmit(null, key),
        selectedKey: undefined,
        defaultSelectedKey: undefined,
        validate: (0, $gWiNP$useCallback)((v)=>validate?.(v.inputValue), [
            validate
        ])
    });
    let buttonRef = (0, $gWiNP$useRef)(null);
    let domRef = (0, $3c2c983d5210446c$export$96a734597687c040)(ref, buttonRef);
    let { triggerProps: triggerProps, overlayProps: overlayProps } = (0, $gWiNP$useOverlayTrigger)({
        type: 'listbox'
    }, state, buttonRef);
    let inputRef = (0, $gWiNP$useRef)(null);
    (0, $gWiNP$useFormValidation)({
        ...props,
        focus: ()=>buttonRef.current?.focus()
    }, state, inputRef);
    let { isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = state.displayValidation;
    let validationState = props.validationState || (isInvalid ? 'invalid' : undefined);
    let errorMessage = props.errorMessage ?? validationErrors.join(' ');
    let { labelProps: labelProps, fieldProps: fieldProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = (0, $gWiNP$useField)({
        ...props,
        labelElementType: 'span',
        isInvalid: isInvalid,
        errorMessage: errorMessage
    });
    // Focus the button and show focus ring when clicking on the label
    labelProps.onClick = ()=>{
        if (!props.isDisabled && buttonRef.current) {
            buttonRef.current.focus();
            (0, $gWiNP$setInteractionModality)('keyboard');
        }
    };
    let inputProps = {
        type: 'hidden',
        name: name,
        value: state.inputValue
    };
    if (validationBehavior === 'native') {
        // Use a hidden <input type="text"> rather than <input type="hidden">
        // so that an empty value blocks HTML form submission when the field is required.
        inputProps.type = 'text';
        inputProps.hidden = true;
        inputProps.required = isRequired;
        // Ignore react warning.
        inputProps.onChange = ()=>{};
    }
    (0, $gWiNP$useFormReset)(inputRef, state.defaultInputValue, state.setInputValue);
    return /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$react).Fragment, null, /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $adcd096854d27620$export$a455218a85c89869), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        elementType: "span",
        ref: domRef,
        includeNecessityIndicatorInAccessibilityName: true
    }, /*#__PURE__*/ (0, $gWiNP$react).createElement($d9f9f86122f1f89b$var$SearchAutocompleteButton, {
        ...(0, $gWiNP$mergeProps)(triggerProps, fieldProps, {
            autoFocus: props.autoFocus,
            icon: props.icon
        }),
        ref: buttonRef,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        isReadOnly: isReadOnly,
        isPlaceholder: !state.inputValue,
        validationState: validationState,
        inputValue: state.inputValue,
        clearInput: ()=>state.setInputValue(''),
        onPress: ()=>!isReadOnly && state.open(null, 'manual')
    }, state.inputValue || props.placeholder || '')), /*#__PURE__*/ (0, $gWiNP$react).createElement("input", {
        ...inputProps,
        ref: inputRef
    }), /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $9fca089dca5508dc$export$4589ed81930b555c), {
        state: state,
        isFixedHeight: true,
        ...overlayProps
    }, /*#__PURE__*/ (0, $gWiNP$react).createElement($d9f9f86122f1f89b$var$SearchAutocompleteTray, {
        ...props,
        onClose: state.close,
        overlayProps: overlayProps,
        state: state
    })));
}
let $d9f9f86122f1f89b$export$e7a90f7d6b078162 = /*#__PURE__*/ (0, $gWiNP$react).forwardRef($d9f9f86122f1f89b$var$ForwardMobileSearchAutocomplete);
// any type is because we don't want to call useObjectRef because this is an internal component and we know
// we are always passing an object ref
const $d9f9f86122f1f89b$var$SearchAutocompleteButton = /*#__PURE__*/ (0, $gWiNP$react).forwardRef(function SearchAutocompleteButton(props, ref) {
    let searchIcon = /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$spectrumiconsuiMagnifier), {
        "data-testid": "searchicon"
    });
    let { icon: icon = searchIcon, isQuiet: isQuiet, isDisabled: isDisabled, isReadOnly: isReadOnly, isPlaceholder: isPlaceholder, validationState: validationState, inputValue: inputValue, clearInput: clearInput, children: children, style: style, className: className } = props;
    let stringFormatter = (0, $gWiNP$useLocalizedStringFormatter)((0, ($parcel$interopDefault($gWiNP$intlStringsmjs))), '@react-spectrum/autocomplete');
    let valueId = (0, $gWiNP$useId)();
    let invalidId = (0, $gWiNP$useId)();
    let validationIcon = validationState === 'invalid' ? /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$spectrumiconsuiAlertMedium), {
        id: invalidId,
        "aria-label": stringFormatter.format('invalid')
    }) : /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$spectrumiconsuiCheckmarkMedium), null);
    if (icon) icon = /*#__PURE__*/ (0, $gWiNP$react).cloneElement(icon, {
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$textfield_vars_cssmjs))), 'spectrum-Textfield-icon'),
        size: 'S'
    });
    let clearButton = /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $ab14010a528467be$export$13ec83e50bf04290), {
        onPress: (e)=>{
            clearInput?.();
            props?.onPress?.(e);
        },
        preventFocus: true,
        "aria-label": stringFormatter.format('clear'),
        excludeFromTabOrder: true,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let validation = /*#__PURE__*/ (0, $gWiNP$react).cloneElement(validationIcon, {
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$textfield_vars_cssmjs))), 'spectrum-Textfield-validationIcon', (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-input-validationIcon'), (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-Search-validationIcon'))
    });
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gWiNP$useHover)({});
    let { isFocused: isFocused, isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $gWiNP$useFocusRing)();
    let { buttonProps: buttonProps } = (0, $gWiNP$useButton)({
        ...props,
        'aria-labelledby': [
            props['aria-labelledby'],
            props['aria-label'] && !props['aria-labelledby'] ? props.id : null,
            valueId,
            validationState === 'invalid' ? invalidId : null
        ].filter(Boolean).join(' '),
        elementType: 'div'
    }, ref);
    return /*#__PURE__*/ (0, $gWiNP$react).createElement("div", {
        ...(0, $gWiNP$mergeProps)(hoverProps, focusProps, buttonProps),
        "aria-haspopup": "dialog",
        ref: ref,
        style: {
            ...style,
            outline: 'none'
        },
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$inputgroup_vars_cssmjs))), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid' && !isDisabled,
            'is-hovered': isHovered,
            'is-focused': isFocused,
            'focus-ring': isFocusVisible
        }, (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$searchautocomplete_cssmjs))), 'searchautocomplete', 'mobile-searchautocomplete'), className)
    }, /*#__PURE__*/ (0, $gWiNP$react).createElement("div", {
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$textfield_vars_cssmjs))), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Textfield--valid': validationState === 'valid' && !isDisabled,
            'spectrum-Textfield--quiet': isQuiet
        }, (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-Search', 'spectrum-Search--loadable', {
            'is-disabled': isDisabled,
            'is-quiet': isQuiet,
            'spectrum-Search--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Search--valid': validationState === 'valid' && !isDisabled
        }), (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-field'))
    }, /*#__PURE__*/ (0, $gWiNP$react).createElement("div", {
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$textfield_vars_cssmjs))), 'spectrum-Textfield-input', {
            'spectrum-Textfield-inputIcon': !!icon,
            'is-hovered': isHovered,
            'is-placeholder': isPlaceholder,
            'is-disabled': isDisabled,
            'is-quiet': isQuiet,
            'is-focused': isFocused
        }, (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-Search-input'), (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$searchautocomplete_cssmjs))), 'mobile-input'))
    }, icon, /*#__PURE__*/ (0, $gWiNP$react).createElement("span", {
        id: valueId,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$searchautocomplete_cssmjs))), 'mobile-value')
    }, children)), validationState && !isDisabled ? validation : null, (inputValue !== '' || validationState != null) && !isReadOnly && clearButton));
});
function $d9f9f86122f1f89b$var$SearchAutocompleteTray(props) {
    let searchIcon = /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$spectrumiconsuiMagnifier), {
        "data-testid": "searchicon"
    });
    let { state: // completionMode = 'suggest',
    state, icon: icon = searchIcon, isDisabled: isDisabled, validationState: validationState, label: label, overlayProps: overlayProps, loadingState: loadingState, onLoadMore: onLoadMore, onClose: onClose, onSubmit: onSubmit } = props;
    let timeout = (0, $gWiNP$useRef)(null);
    let [showLoading, setShowLoading] = (0, $gWiNP$useState)(false);
    let inputRef = (0, $gWiNP$useRef)(null);
    let popoverRef = (0, $gWiNP$useRef)(null);
    let listBoxRef = (0, $gWiNP$useRef)(null);
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let layout = (0, $ee13b4eccaed924f$export$25768ea656ae32a7)();
    let stringFormatter = (0, $gWiNP$useLocalizedStringFormatter)((0, ($parcel$interopDefault($gWiNP$intlStringsmjs))), '@react-spectrum/autocomplete');
    let { inputProps: inputProps, listBoxProps: listBoxProps, labelProps: labelProps, clearButtonProps: clearButtonProps } = (0, $gWiNP$useSearchAutocomplete)({
        ...props,
        layoutDelegate: layout,
        popoverRef: popoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        // Handled outside the tray.
        name: undefined
    }, state);
    (0, $gWiNP$react).useEffect(()=>{
        if (inputRef.current) (0, $gWiNP$focusSafely)(inputRef.current);
    }, []);
    (0, $gWiNP$react).useEffect(()=>{
        // When the tray closes, set state.isFocused (i.e. the tray input's focus tracker) to false.
        // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
        // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
        // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363
        if (!state.isOpen && state.isFocused) state.setFocused(false);
    });
    let { dialogProps: dialogProps } = (0, $gWiNP$useDialog)({
        'aria-labelledby': (0, $gWiNP$useId)(labelProps.id)
    }, popoverRef);
    // Override the role of the input to "searchbox" instead of "combobox".
    // Since the listbox is always visible, the combobox role doesn't really give us anything.
    // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
    // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
    // open the virtual keyboard rather than closing the tray.
    inputProps.role = 'searchbox';
    inputProps['aria-haspopup'] = 'listbox';
    delete inputProps.onTouchEnd;
    let clearButton = /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $ab14010a528467be$export$13ec83e50bf04290), {
        ...clearButtonProps,
        preventFocus: true,
        "aria-label": stringFormatter.format('clear'),
        excludeFromTabOrder: true,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let loadingCircle = /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $1cfe37e7feefa23d$export$c79b9d6b4cc92af7), {
        "aria-label": stringFormatter.format('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-Search-circleLoader', (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$textfield_vars_cssmjs))), 'spectrum-Textfield-circleLoader'))
    });
    // Close the software keyboard on scroll to give the user a bigger area to scroll.
    // But only do this if scrolling with touch, otherwise it can cause issues with touch
    // screen readers.
    let isTouchDown = (0, $gWiNP$useRef)(false);
    let onTouchStart = ()=>{
        isTouchDown.current = true;
    };
    let onTouchEnd = ()=>{
        isTouchDown.current = false;
    };
    let onScroll = (0, $gWiNP$useCallback)(()=>{
        if (!inputRef.current || (0, $gWiNP$getActiveElement)() !== inputRef.current || !isTouchDown.current) return;
        if (popoverRef.current) popoverRef.current.focus();
    }, [
        inputRef,
        popoverRef,
        isTouchDown
    ]);
    let inputValue = inputProps.value;
    let lastInputValue = (0, $gWiNP$useRef)(inputValue);
    (0, $gWiNP$useEffect)(()=>{
        if (loadingState === 'filtering' && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (loadingState !== 'filtering') {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            if (timeout.current !== null) {
                clearTimeout(timeout.current);
                timeout.current = null;
            }
        }
        lastInputValue.current = inputValue;
    }, [
        loadingState,
        inputValue,
        showLoading
    ]);
    let onKeyDown = (e)=>{
        // Close virtual keyboard, close tray, and fire onSubmit if user hits Enter w/o any focused options
        if (e.key === 'Enter' && state.selectionManager.focusedKey == null) {
            popoverRef.current?.focus();
            if (onClose) onClose();
            if (onSubmit) onSubmit(inputValue == null ? null : inputValue.toString(), null);
        } else if (inputProps.onKeyDown) inputProps.onKeyDown(e);
    };
    if (icon) icon = /*#__PURE__*/ (0, $gWiNP$react).cloneElement(icon, {
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$textfield_vars_cssmjs))), 'spectrum-Textfield-icon'),
        size: 'S'
    });
    return /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$FocusScope), {
        restoreFocus: true,
        contain: true
    }, /*#__PURE__*/ (0, $gWiNP$react).createElement("div", {
        ...(0, $gWiNP$mergeProps)(overlayProps, dialogProps),
        ref: popoverRef,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$searchautocomplete_cssmjs))), 'tray-dialog')
    }, /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$DismissButton), {
        onDismiss: onClose
    }), /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $b312f2102feb9487$export$d22444a338b6e3c2), {
        label: label,
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            onKeyDown: onKeyDown
        },
        inputRef: inputRef,
        isDisabled: isDisabled,
        isLoading: showLoading && loadingState === 'filtering',
        loadingIndicator: loadingState != null ? loadingCircle : undefined,
        validationState: validationState,
        wrapperChildren: (state.inputValue !== '' || loadingState === 'filtering' || validationState != null) && !props.isReadOnly ? clearButton : undefined,
        icon: icon,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-Search', 'spectrum-Textfield', 'spectrum-Search--loadable', {
            'spectrum-Search--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Search--valid': validationState === 'valid' && !isDisabled
        }, (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$searchautocomplete_cssmjs))), 'tray-textfield', {
            'has-label': !!props.label
        })),
        inputClassName: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-Search-input'),
        validationIconClassName: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$search_vars_cssmjs))), 'spectrum-Search-validationIcon')
    }), /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $ee13b4eccaed924f$export$1afdcf349979fb7e), {
        ...listBoxProps,
        domProps: {
            onTouchStart: onTouchStart,
            onTouchEnd: onTouchEnd
        },
        disallowEmptySelection: true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        renderEmptyState: ()=>loadingState !== 'loading' && /*#__PURE__*/ (0, $gWiNP$react).createElement("span", {
                className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$searchautocomplete_cssmjs))), 'no-results')
            }, stringFormatter.format('noResults')),
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($gWiNP$searchautocomplete_cssmjs))), 'tray-listbox'),
        ref: listBoxRef,
        onScroll: onScroll,
        onLoadMore: onLoadMore,
        isLoading: isLoading
    }), /*#__PURE__*/ (0, $gWiNP$react).createElement((0, $gWiNP$DismissButton), {
        onDismiss: onClose
    })));
}


export {$d9f9f86122f1f89b$export$e7a90f7d6b078162 as MobileSearchAutocomplete};
