// @adobe/react-spectrum/dist/private/autocomplete/MobileSearchAutocomplete.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $0fc8553a4214494f$exports = require("../button/ClearButton.cjs");
var $b93966d678e0af07$exports = require("../label/Field.cjs");
var $7daa85d4b4165307$exports = require("./intlStrings.cjs");
var $cb7ee1d9d5613db9$exports = require("../listbox/ListBoxBase.cjs");
var $948c2416aa3a9507$exports = require("../progress/ProgressCircle.cjs");
require("./searchautocomplete.css");
var $13183d44bc7908f8$exports = require("./searchautocomplete_css.cjs");
require("../search_vars.css");
var $d2d4ce3e4a6482f9$exports = require("../search_vars_css.cjs");
require("../inputgroup_vars.css");
var $6c88ab5aea804b3c$exports = require("../inputgroup_vars_css.cjs");
var $827dbb466e199966$exports = require("../textfield/TextFieldBase.cjs");
require("../textfield_vars.css");
var $5d389146adc85829$exports = require("../textfield_vars_css.cjs");
var $378dee1409fe2937$exports = require("../overlays/Tray.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $9fmUh$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $9fmUh$reactariauseButton = require("react-aria/useButton");
var $9fmUh$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $9fmUh$reactstatelyuseComboBoxState = require("react-stately/useComboBoxState");
var $9fmUh$reactariaOverlay = require("react-aria/Overlay");
var $9fmUh$reactariaprivateinteractionsfocusSafely = require("react-aria/private/interactions/focusSafely");
var $9fmUh$reactariaFocusScope = require("react-aria/FocusScope");
var $9fmUh$reactariaprivateutilsshadowdomDOMFunctions = require("react-aria/private/utils/shadowdom/DOMFunctions");
var $9fmUh$spectrumiconsuiMagnifier = require("@spectrum-icons/ui/Magnifier");
var $9fmUh$reactariamergeProps = require("react-aria/mergeProps");
var $9fmUh$react = require("react");
var $9fmUh$reactariaprivateinteractionsuseFocusVisible = require("react-aria/private/interactions/useFocusVisible");
var $9fmUh$reactariauseDialog = require("react-aria/useDialog");
var $9fmUh$reactariauseField = require("react-aria/useField");
var $9fmUh$reactariauseFilter = require("react-aria/useFilter");
var $9fmUh$reactariauseFocusRing = require("react-aria/useFocusRing");
var $9fmUh$reactariaprivateutilsuseFormReset = require("react-aria/private/utils/useFormReset");
var $9fmUh$reactariaprivateformuseFormValidation = require("react-aria/private/form/useFormValidation");
var $9fmUh$reactariauseHover = require("react-aria/useHover");
var $9fmUh$reactariauseId = require("react-aria/useId");
var $9fmUh$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $9fmUh$reactariauseOverlayTrigger = require("react-aria/useOverlayTrigger");
var $9fmUh$reactariaprivateautocompleteuseSearchAutocomplete = require("react-aria/private/autocomplete/useSearchAutocomplete");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "MobileSearchAutocomplete", function () { return $501e31d6b6d1cb75$export$e7a90f7d6b078162; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




































function $501e31d6b6d1cb75$var$ForwardMobileSearchAutocomplete(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    let { isQuiet: isQuiet, isDisabled: isDisabled, isRequired: isRequired, validationBehavior: validationBehavior, validate: validate, name: name, isReadOnly: isReadOnly, onSubmit: onSubmit = ()=>{} } = props;
    let { contains: contains } = (0, $9fmUh$reactariauseFilter.useFilter)({
        sensitivity: 'base'
    });
    let state = (0, $9fmUh$reactstatelyuseComboBoxState.useComboBoxState)({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: true,
        // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
        // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
        // then the closing of the tray will call setFocused(false) again due to cleanup effect)
        shouldCloseOnBlur: false,
        allowsCustomValue: true,
        onSelectionChange: (key)=>key !== null && onSubmit(null, key),
        selectedKey: undefined,
        defaultSelectedKey: undefined,
        validate: (0, $9fmUh$react.useCallback)((v)=>validate?.(v.inputValue), [
            validate
        ])
    });
    let buttonRef = (0, $9fmUh$react.useRef)(null);
    let domRef = (0, $65aea7b37663976b$exports.useFocusableRef)(ref, buttonRef);
    let { triggerProps: triggerProps, overlayProps: overlayProps } = (0, $9fmUh$reactariauseOverlayTrigger.useOverlayTrigger)({
        type: 'listbox'
    }, state, buttonRef);
    let inputRef = (0, $9fmUh$react.useRef)(null);
    (0, $9fmUh$reactariaprivateformuseFormValidation.useFormValidation)({
        ...props,
        focus: ()=>buttonRef.current?.focus()
    }, state, inputRef);
    let { isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = state.displayValidation;
    let validationState = props.validationState || (isInvalid ? 'invalid' : undefined);
    let errorMessage = props.errorMessage ?? validationErrors.join(' ');
    let { labelProps: labelProps, fieldProps: fieldProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = (0, $9fmUh$reactariauseField.useField)({
        ...props,
        labelElementType: 'span',
        isInvalid: isInvalid,
        errorMessage: errorMessage
    });
    // Focus the button and show focus ring when clicking on the label
    labelProps.onClick = ()=>{
        if (!props.isDisabled && buttonRef.current) {
            buttonRef.current.focus();
            (0, $9fmUh$reactariaprivateinteractionsuseFocusVisible.setInteractionModality)('keyboard');
        }
    };
    let inputProps = {
        type: 'hidden',
        name: name,
        value: state.inputValue
    };
    if (validationBehavior === 'native') {
        // Use a hidden <input type="text"> rather than <input type="hidden">
        // so that an empty value blocks HTML form submission when the field is required.
        inputProps.type = 'text';
        inputProps.hidden = true;
        inputProps.required = isRequired;
        // Ignore react warning.
        inputProps.onChange = ()=>{};
    }
    (0, $9fmUh$reactariaprivateutilsuseFormReset.useFormReset)(inputRef, state.defaultInputValue, state.setInputValue);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, ($parcel$interopDefault($9fmUh$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $b93966d678e0af07$exports.Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        elementType: "span",
        ref: domRef,
        includeNecessityIndicatorInAccessibilityName: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement($501e31d6b6d1cb75$var$SearchAutocompleteButton, {
        ...(0, $9fmUh$reactariamergeProps.mergeProps)(triggerProps, fieldProps, {
            autoFocus: props.autoFocus,
            icon: props.icon
        }),
        ref: buttonRef,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        isReadOnly: isReadOnly,
        isPlaceholder: !state.inputValue,
        validationState: validationState,
        inputValue: state.inputValue,
        clearInput: ()=>state.setInputValue(''),
        onPress: ()=>!isReadOnly && state.open(null, 'manual')
    }, state.inputValue || props.placeholder || '')), /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement("input", {
        ...inputProps,
        ref: inputRef
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $378dee1409fe2937$exports.Tray), {
        state: state,
        isFixedHeight: true,
        ...overlayProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement($501e31d6b6d1cb75$var$SearchAutocompleteTray, {
        ...props,
        onClose: state.close,
        overlayProps: overlayProps,
        state: state
    })));
}
let $501e31d6b6d1cb75$export$e7a90f7d6b078162 = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).forwardRef($501e31d6b6d1cb75$var$ForwardMobileSearchAutocomplete);
// any type is because we don't want to call useObjectRef because this is an internal component and we know
// we are always passing an object ref
const $501e31d6b6d1cb75$var$SearchAutocompleteButton = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).forwardRef(function SearchAutocompleteButton(props, ref) {
    let searchIcon = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, ($parcel$interopDefault($9fmUh$spectrumiconsuiMagnifier))), {
        "data-testid": "searchicon"
    });
    let { icon: icon = searchIcon, isQuiet: isQuiet, isDisabled: isDisabled, isReadOnly: isReadOnly, isPlaceholder: isPlaceholder, validationState: validationState, inputValue: inputValue, clearInput: clearInput, children: children, style: style, className: className } = props;
    let stringFormatter = (0, $9fmUh$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($7daa85d4b4165307$exports))), '@react-spectrum/autocomplete');
    let valueId = (0, $9fmUh$reactariauseId.useId)();
    let invalidId = (0, $9fmUh$reactariauseId.useId)();
    let validationIcon = validationState === 'invalid' ? /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, ($parcel$interopDefault($9fmUh$spectrumiconsuiAlertMedium))), {
        id: invalidId,
        "aria-label": stringFormatter.format('invalid')
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, ($parcel$interopDefault($9fmUh$spectrumiconsuiCheckmarkMedium))), null);
    if (icon) icon = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).cloneElement(icon, {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-icon'),
        size: 'S'
    });
    let clearButton = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $0fc8553a4214494f$exports.ClearButton), {
        onPress: (e)=>{
            clearInput?.();
            props?.onPress?.(e);
        },
        preventFocus: true,
        "aria-label": stringFormatter.format('clear'),
        excludeFromTabOrder: true,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let validation = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).cloneElement(validationIcon, {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-validationIcon', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input-validationIcon'), (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-validationIcon'))
    });
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $9fmUh$reactariauseHover.useHover)({});
    let { isFocused: isFocused, isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $9fmUh$reactariauseFocusRing.useFocusRing)();
    let { buttonProps: buttonProps } = (0, $9fmUh$reactariauseButton.useButton)({
        ...props,
        'aria-labelledby': [
            props['aria-labelledby'],
            props['aria-label'] && !props['aria-labelledby'] ? props.id : null,
            valueId,
            validationState === 'invalid' ? invalidId : null
        ].filter(Boolean).join(' '),
        elementType: 'div'
    }, ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement("div", {
        ...(0, $9fmUh$reactariamergeProps.mergeProps)(hoverProps, focusProps, buttonProps),
        "aria-haspopup": "dialog",
        ref: ref,
        style: {
            ...style,
            outline: 'none'
        },
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid' && !isDisabled,
            'is-hovered': isHovered,
            'is-focused': isFocused,
            'focus-ring': isFocusVisible
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'searchautocomplete', 'mobile-searchautocomplete'), className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Textfield--valid': validationState === 'valid' && !isDisabled,
            'spectrum-Textfield--quiet': isQuiet
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search', 'spectrum-Search--loadable', {
            'is-disabled': isDisabled,
            'is-quiet': isQuiet,
            'spectrum-Search--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Search--valid': validationState === 'valid' && !isDisabled
        }), (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-field'))
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-input', {
            'spectrum-Textfield-inputIcon': !!icon,
            'is-hovered': isHovered,
            'is-placeholder': isPlaceholder,
            'is-disabled': isDisabled,
            'is-quiet': isQuiet,
            'is-focused': isFocused
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-input'), (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'mobile-input'))
    }, icon, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement("span", {
        id: valueId,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'mobile-value')
    }, children)), validationState && !isDisabled ? validation : null, (inputValue !== '' || validationState != null) && !isReadOnly && clearButton));
});
function $501e31d6b6d1cb75$var$SearchAutocompleteTray(props) {
    let searchIcon = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, ($parcel$interopDefault($9fmUh$spectrumiconsuiMagnifier))), {
        "data-testid": "searchicon"
    });
    let { state: // completionMode = 'suggest',
    state, icon: icon = searchIcon, isDisabled: isDisabled, validationState: validationState, label: label, overlayProps: overlayProps, loadingState: loadingState, onLoadMore: onLoadMore, onClose: onClose, onSubmit: onSubmit } = props;
    let timeout = (0, $9fmUh$react.useRef)(null);
    let [showLoading, setShowLoading] = (0, $9fmUh$react.useState)(false);
    let inputRef = (0, $9fmUh$react.useRef)(null);
    let popoverRef = (0, $9fmUh$react.useRef)(null);
    let listBoxRef = (0, $9fmUh$react.useRef)(null);
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let layout = (0, $cb7ee1d9d5613db9$exports.useListBoxLayout)();
    let stringFormatter = (0, $9fmUh$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($7daa85d4b4165307$exports))), '@react-spectrum/autocomplete');
    let { inputProps: inputProps, listBoxProps: listBoxProps, labelProps: labelProps, clearButtonProps: clearButtonProps } = (0, $9fmUh$reactariaprivateautocompleteuseSearchAutocomplete.useSearchAutocomplete)({
        ...props,
        layoutDelegate: layout,
        popoverRef: popoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        // Handled outside the tray.
        name: undefined
    }, state);
    (0, ($parcel$interopDefault($9fmUh$react))).useEffect(()=>{
        if (inputRef.current) (0, $9fmUh$reactariaprivateinteractionsfocusSafely.focusSafely)(inputRef.current);
    }, []);
    (0, ($parcel$interopDefault($9fmUh$react))).useEffect(()=>{
        // When the tray closes, set state.isFocused (i.e. the tray input's focus tracker) to false.
        // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
        // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
        // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363
        if (!state.isOpen && state.isFocused) state.setFocused(false);
    });
    let { dialogProps: dialogProps } = (0, $9fmUh$reactariauseDialog.useDialog)({
        'aria-labelledby': (0, $9fmUh$reactariauseId.useId)(labelProps.id)
    }, popoverRef);
    // Override the role of the input to "searchbox" instead of "combobox".
    // Since the listbox is always visible, the combobox role doesn't really give us anything.
    // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
    // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
    // open the virtual keyboard rather than closing the tray.
    inputProps.role = 'searchbox';
    inputProps['aria-haspopup'] = 'listbox';
    delete inputProps.onTouchEnd;
    let clearButton = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $0fc8553a4214494f$exports.ClearButton), {
        ...clearButtonProps,
        preventFocus: true,
        "aria-label": stringFormatter.format('clear'),
        excludeFromTabOrder: true,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let loadingCircle = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $948c2416aa3a9507$exports.ProgressCircle), {
        "aria-label": stringFormatter.format('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-circleLoader', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-circleLoader'))
    });
    // Close the software keyboard on scroll to give the user a bigger area to scroll.
    // But only do this if scrolling with touch, otherwise it can cause issues with touch
    // screen readers.
    let isTouchDown = (0, $9fmUh$react.useRef)(false);
    let onTouchStart = ()=>{
        isTouchDown.current = true;
    };
    let onTouchEnd = ()=>{
        isTouchDown.current = false;
    };
    let onScroll = (0, $9fmUh$react.useCallback)(()=>{
        if (!inputRef.current || (0, $9fmUh$reactariaprivateutilsshadowdomDOMFunctions.getActiveElement)() !== inputRef.current || !isTouchDown.current) return;
        if (popoverRef.current) popoverRef.current.focus();
    }, [
        inputRef,
        popoverRef,
        isTouchDown
    ]);
    let inputValue = inputProps.value;
    let lastInputValue = (0, $9fmUh$react.useRef)(inputValue);
    (0, $9fmUh$react.useEffect)(()=>{
        if (loadingState === 'filtering' && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (loadingState !== 'filtering') {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            if (timeout.current !== null) {
                clearTimeout(timeout.current);
                timeout.current = null;
            }
        }
        lastInputValue.current = inputValue;
    }, [
        loadingState,
        inputValue,
        showLoading
    ]);
    let onKeyDown = (e)=>{
        // Close virtual keyboard, close tray, and fire onSubmit if user hits Enter w/o any focused options
        if (e.key === 'Enter' && state.selectionManager.focusedKey == null) {
            popoverRef.current?.focus();
            if (onClose) onClose();
            if (onSubmit) onSubmit(inputValue == null ? null : inputValue.toString(), null);
        } else if (inputProps.onKeyDown) inputProps.onKeyDown(e);
    };
    if (icon) icon = /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).cloneElement(icon, {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-icon'),
        size: 'S'
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $9fmUh$reactariaFocusScope.FocusScope), {
        restoreFocus: true,
        contain: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement("div", {
        ...(0, $9fmUh$reactariamergeProps.mergeProps)(overlayProps, dialogProps),
        ref: popoverRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'tray-dialog')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $9fmUh$reactariaOverlay.DismissButton), {
        onDismiss: onClose
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $827dbb466e199966$exports.TextFieldBase), {
        label: label,
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            onKeyDown: onKeyDown
        },
        inputRef: inputRef,
        isDisabled: isDisabled,
        isLoading: showLoading && loadingState === 'filtering',
        loadingIndicator: loadingState != null ? loadingCircle : undefined,
        validationState: validationState,
        wrapperChildren: (state.inputValue !== '' || loadingState === 'filtering' || validationState != null) && !props.isReadOnly ? clearButton : undefined,
        icon: icon,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search', 'spectrum-Textfield', 'spectrum-Search--loadable', {
            'spectrum-Search--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Search--valid': validationState === 'valid' && !isDisabled
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'tray-textfield', {
            'has-label': !!props.label
        })),
        inputClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-input'),
        validationIconClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-validationIcon')
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $cb7ee1d9d5613db9$exports.ListBoxBase), {
        ...listBoxProps,
        domProps: {
            onTouchStart: onTouchStart,
            onTouchEnd: onTouchEnd
        },
        disallowEmptySelection: true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        renderEmptyState: ()=>loadingState !== 'loading' && /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement("span", {
                className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'no-results')
            }, stringFormatter.format('noResults')),
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'tray-listbox'),
        ref: listBoxRef,
        onScroll: onScroll,
        onLoadMore: onLoadMore,
        isLoading: isLoading
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9fmUh$react))).createElement((0, $9fmUh$reactariaOverlay.DismissButton), {
        onDismiss: onClose
    })));
}
