// @adobe/react-spectrum/dist/private/autocomplete/SearchAutocomplete.mjs
import {classNames as $6e6392558d48dfec$export$ce4ab0c55987d1ff} from "../utils/classNames.mjs";
import {ClearButton as $ab14010a528467be$export$13ec83e50bf04290} from "../button/ClearButton.mjs";
import {dimensionValue as $63d03c54ca5e4b88$export$abc24f5b99744ea6} from "../utils/styleProps.mjs";
import {Field as $adcd096854d27620$export$a455218a85c89869} from "../label/Field.mjs";
import $3RnBb$intlStringsmjs from "./intlStrings.mjs";
import {ListBoxBase as $ee13b4eccaed924f$export$1afdcf349979fb7e, useListBoxLayout as $ee13b4eccaed924f$export$25768ea656ae32a7} from "../listbox/ListBoxBase.mjs";
import {MobileSearchAutocomplete as $d9f9f86122f1f89b$export$e7a90f7d6b078162} from "./MobileSearchAutocomplete.mjs";
import {Popover as $3a473e3b7032f626$export$5b6b19405a83ff9d} from "../overlays/Popover.mjs";
import {ProgressCircle as $1cfe37e7feefa23d$export$c79b9d6b4cc92af7} from "../progress/ProgressCircle.mjs";
import "./searchautocomplete.css";
import $3RnBb$searchautocomplete_cssmjs from "./searchautocomplete_css.mjs";
import "../search_vars.css";
import $3RnBb$search_vars_cssmjs from "../search_vars_css.mjs";
import "../inputgroup_vars.css";
import $3RnBb$inputgroup_vars_cssmjs from "../inputgroup_vars_css.mjs";
import {TextFieldBase as $b312f2102feb9487$export$d22444a338b6e3c2} from "../textfield/TextFieldBase.mjs";
import "../textfield_vars.css";
import $3RnBb$textfield_vars_cssmjs from "../textfield_vars_css.mjs";
import {useFocusableRef as $3c2c983d5210446c$export$96a734597687c040, useUnwrapDOMRef as $3c2c983d5210446c$export$1d5cc31d9d8df817} from "../utils/useDOMRef.mjs";
import {useFormProps as $c29c48d4ef19ffc4$export$a6b5be5c6b451665} from "../form/Form.mjs";
import {useIsMobileDevice as $f357d4aae54bf1ff$export$736bf165441b18c7} from "../utils/useIsMobileDevice.mjs";
import {useProvider as $71dfb0e0358a12de$export$693cdb10cec23617, useProviderProps as $71dfb0e0358a12de$export$521c373ccc32c300} from "../provider/Provider.mjs";
import {useSearchAutocomplete as $3RnBb$useSearchAutocomplete} from "react-aria/private/autocomplete/useSearchAutocomplete";
import {filterDOMProps as $3RnBb$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusRing as $3RnBb$FocusRing} from "react-aria/FocusRing";
import $3RnBb$spectrumiconsuiMagnifier from "@spectrum-icons/ui/Magnifier";
import {useComboBoxState as $3RnBb$useComboBoxState} from "react-stately/useComboBoxState";
import $3RnBb$react, {useRef as $3RnBb$useRef, useEffect as $3RnBb$useEffect, useCallback as $3RnBb$useCallback, useState as $3RnBb$useState, forwardRef as $3RnBb$forwardRef} from "react";
import {useFilter as $3RnBb$useFilter} from "react-aria/useFilter";
import {useHover as $3RnBb$useHover} from "react-aria/useHover";
import {useLayoutEffect as $3RnBb$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useLocalizedStringFormatter as $3RnBb$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {useResizeObserver as $3RnBb$useResizeObserver} from "react-aria/private/utils/useResizeObserver";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




























function $c6c497b46ad7f794$var$SearchAutocomplete(props, ref) {
    props = (0, $71dfb0e0358a12de$export$521c373ccc32c300)(props);
    props = (0, $c29c48d4ef19ffc4$export$a6b5be5c6b451665)(props);
    let hasWarned = (0, $3RnBb$useRef)(false);
    (0, $3RnBb$useEffect)(()=>{
        if (props.placeholder && !hasWarned.current && process.env.NODE_ENV !== 'production') {
            console.warn('Placeholders are deprecated due to accessibility issues. Please use help text instead.');
            hasWarned.current = true;
        }
    }, [
        props.placeholder
    ]);
    let isMobile = (0, $f357d4aae54bf1ff$export$736bf165441b18c7)();
    if (isMobile) // menuTrigger=focus/manual don't apply to mobile searchwithin
    return /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $d9f9f86122f1f89b$export$e7a90f7d6b078162), {
        ...props,
        menuTrigger: "input",
        ref: ref
    });
    else return /*#__PURE__*/ (0, $3RnBb$react).createElement($c6c497b46ad7f794$var$SearchAutocompleteBase, {
        ...props,
        ref: ref
    });
}
function $c6c497b46ad7f794$var$ForwardSearchAutocompleteBase(props, ref) {
    props = (0, $71dfb0e0358a12de$export$521c373ccc32c300)(props);
    let { menuTrigger: menuTrigger = 'input', shouldFlip: shouldFlip = true, direction: direction = 'bottom', align: align = 'start', isQuiet: isQuiet, menuWidth: customMenuWidth, loadingState: loadingState, onLoadMore: onLoadMore, onSubmit: onSubmit = ()=>{}, validate: validate } = props;
    let stringFormatter = (0, $3RnBb$useLocalizedStringFormatter)((0, ($parcel$interopDefault($3RnBb$intlStringsmjs))), '@react-spectrum/autocomplete');
    let isAsync = loadingState != null;
    let popoverRef = (0, $3RnBb$useRef)(null);
    let unwrappedPopoverRef = (0, $3c2c983d5210446c$export$1d5cc31d9d8df817)(popoverRef);
    let listBoxRef = (0, $3RnBb$useRef)(null);
    let inputRef = (0, $3RnBb$useRef)(null);
    let domRef = (0, $3c2c983d5210446c$export$96a734597687c040)(ref, inputRef);
    let { contains: contains } = (0, $3RnBb$useFilter)({
        sensitivity: 'base'
    });
    let state = (0, $3RnBb$useComboBoxState)({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: isAsync,
        allowsCustomValue: true,
        onSelectionChange: (key)=>key !== null && onSubmit(null, key),
        selectedKey: undefined,
        defaultSelectedKey: undefined,
        validate: (0, $3RnBb$useCallback)((v)=>validate?.(v.inputValue), [
            validate
        ])
    });
    let layout = (0, $ee13b4eccaed924f$export$25768ea656ae32a7)();
    let { inputProps: inputProps, listBoxProps: listBoxProps, labelProps: labelProps, clearButtonProps: clearButtonProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $3RnBb$useSearchAutocomplete)({
        ...props,
        layoutDelegate: layout,
        popoverRef: unwrappedPopoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        menuTrigger: menuTrigger
    }, state);
    // Measure the width of the inputfield to inform the width of the menu (below).
    let [menuWidth, setMenuWidth] = (0, $3RnBb$useState)(0);
    let { scale: scale } = (0, $71dfb0e0358a12de$export$693cdb10cec23617)();
    let onResize = (0, $3RnBb$useCallback)(()=>{
        if (inputRef.current) {
            let inputWidth = inputRef.current.offsetWidth;
            setMenuWidth(inputWidth);
        }
    }, [
        inputRef,
        setMenuWidth
    ]);
    (0, $3RnBb$useResizeObserver)({
        ref: domRef,
        onResize: onResize
    });
    (0, $3RnBb$useLayoutEffect)(onResize, [
        scale,
        onResize
    ]);
    let width = isQuiet ? undefined : menuWidth;
    let style = {
        width: customMenuWidth ? (0, $63d03c54ca5e4b88$export$abc24f5b99744ea6)(customMenuWidth) : width,
        minWidth: isQuiet ? `calc(${menuWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : menuWidth
    };
    return /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $3RnBb$react).Fragment, null, /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $adcd096854d27620$export$a455218a85c89869), {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        labelProps: labelProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        ref: domRef
    }, /*#__PURE__*/ (0, $3RnBb$react).createElement($c6c497b46ad7f794$var$SearchAutocompleteInput, {
        ...props,
        isOpen: state.isOpen,
        loadingState: loadingState,
        inputProps: inputProps,
        inputRef: inputRef,
        clearButtonProps: clearButtonProps,
        validationState: props.validationState || (isInvalid ? 'invalid' : undefined)
    })), /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $3a473e3b7032f626$export$5b6b19405a83ff9d), {
        state: state,
        UNSAFE_style: style,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-popover', {
            'spectrum-InputGroup-popover--quiet': isQuiet
        }),
        ref: popoverRef,
        triggerRef: inputRef,
        placement: `${direction} ${align}`,
        hideArrow: true,
        isNonModal: true,
        shouldFlip: shouldFlip
    }, /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $ee13b4eccaed924f$export$1afdcf349979fb7e), {
        ...listBoxProps,
        ref: listBoxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy ?? undefined,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        isLoading: loadingState === 'loading' || loadingState === 'loadingMore',
        showLoadingSpinner: loadingState === 'loadingMore',
        onLoadMore: onLoadMore,
        renderEmptyState: ()=>isAsync && /*#__PURE__*/ (0, $3RnBb$react).createElement("span", {
                className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$searchautocomplete_cssmjs))), 'no-results')
            }, stringFormatter.format('noResults'))
    })));
}
let $c6c497b46ad7f794$var$SearchAutocompleteBase = /*#__PURE__*/ (0, $3RnBb$react).forwardRef($c6c497b46ad7f794$var$ForwardSearchAutocompleteBase);
// any type is because we don't want to call useObjectRef because this is an internal component and we know
// we are always passing an object ref
function $c6c497b46ad7f794$var$ForwardSearchAutocompleteInput(props, ref) {
    let searchIcon = /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $3RnBb$spectrumiconsuiMagnifier), {
        "data-testid": "searchicon"
    });
    let { icon: icon = searchIcon, isQuiet: isQuiet, isDisabled: isDisabled, isReadOnly: isReadOnly, validationState: validationState, inputProps: inputProps, inputRef: inputRef, autoFocus: autoFocus, style: style, className: className, loadingState: loadingState, isOpen: isOpen, menuTrigger: menuTrigger, clearButtonProps: clearButtonProps } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $3RnBb$useHover)({});
    let stringFormatter = (0, $3RnBb$useLocalizedStringFormatter)((0, ($parcel$interopDefault($3RnBb$intlStringsmjs))), '@react-spectrum/autocomplete');
    let domProps = (0, $3RnBb$filterDOMProps)(props);
    let timeout = (0, $3RnBb$useRef)(null);
    let [showLoading, setShowLoading] = (0, $3RnBb$useState)(false);
    let loadingCircle = /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $1cfe37e7feefa23d$export$c79b9d6b4cc92af7), {
        "aria-label": stringFormatter.format('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$textfield_vars_cssmjs))), 'spectrum-Textfield-circleLoader', (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-input-circleLoader'), (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$search_vars_cssmjs))), 'spectrum-Search-circleLoader'))
    });
    let clearButton = /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $ab14010a528467be$export$13ec83e50bf04290), {
        ...clearButtonProps,
        preventFocus: true,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$search_vars_cssmjs))), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let isLoading = loadingState === 'loading' || loadingState === 'filtering';
    let inputValue = inputProps.value;
    let lastInputValue = (0, $3RnBb$useRef)(inputValue);
    (0, $3RnBb$useEffect)(()=>{
        if (isLoading && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (!isLoading) // If loading is no longer happening, clear any timers and hide the loading circle
        {
            if (timeout.current != null) {
                clearTimeout(timeout.current);
                timeout.current = null;
            }
        }
        lastInputValue.current = inputValue;
    }, [
        isLoading,
        showLoading,
        inputValue
    ]);
    let [prevIsLoading, setPrevIsLoading] = (0, $3RnBb$useState)(isLoading);
    if (prevIsLoading !== isLoading && !isLoading) {
        setShowLoading(false);
        setPrevIsLoading(isLoading);
    }
    return /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $3RnBb$FocusRing), {
        within: true,
        isTextInput: true,
        focusClass: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$inputgroup_vars_cssmjs))), 'is-focused'),
        focusRingClass: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$inputgroup_vars_cssmjs))), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, $3RnBb$react).createElement("div", {
        ...hoverProps,
        ref: ref,
        style: style,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$inputgroup_vars_cssmjs))), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid' && !isDisabled,
            'is-hovered': isHovered
        }, (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$searchautocomplete_cssmjs))), 'searchautocomplete'), className)
    }, /*#__PURE__*/ (0, $3RnBb$react).createElement((0, $b312f2102feb9487$export$d22444a338b6e3c2), {
        ...domProps,
        inputProps: inputProps,
        inputRef: inputRef,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$search_vars_cssmjs))), 'spectrum-Search', 'spectrum-Search--loadable', 'spectrum-Textfield', {
            'is-disabled': isDisabled,
            'is-quiet': isQuiet,
            'spectrum-Search--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Search--valid': validationState === 'valid' && !isDisabled
        }, (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-field')),
        inputClassName: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$search_vars_cssmjs))), 'spectrum-Search-input'),
        validationIconClassName: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($3RnBb$search_vars_cssmjs))), 'spectrum-Search-validationIcon'),
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        isLoading: showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'),
        loadingIndicator: loadingState != null ? loadingCircle : undefined,
        icon: icon,
        wrapperChildren: (inputValue !== '' || loadingState === 'filtering' || validationState != null) && !isReadOnly ? clearButton : undefined,
        disableFocusRing: true
    })));
}
let $c6c497b46ad7f794$var$SearchAutocompleteInput = /*#__PURE__*/ (0, $3RnBb$react).forwardRef($c6c497b46ad7f794$var$ForwardSearchAutocompleteInput);
/**
 * A SearchAutocomplete is a searchfield that supports a dynamic list of suggestions.
 */ let $c6c497b46ad7f794$export$dd65332a5b19fa63 = /*#__PURE__*/ (0, $3RnBb$forwardRef)($c6c497b46ad7f794$var$SearchAutocomplete);


export {$c6c497b46ad7f794$export$dd65332a5b19fa63 as SearchAutocomplete};
