// @adobe/react-spectrum/dist/private/autocomplete/SearchAutocomplete.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $0fc8553a4214494f$exports = require("../button/ClearButton.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $b93966d678e0af07$exports = require("../label/Field.cjs");
var $7daa85d4b4165307$exports = require("./intlStrings.cjs");
var $cb7ee1d9d5613db9$exports = require("../listbox/ListBoxBase.cjs");
var $501e31d6b6d1cb75$exports = require("./MobileSearchAutocomplete.cjs");
var $39ed1c805b59752f$exports = require("../overlays/Popover.cjs");
var $948c2416aa3a9507$exports = require("../progress/ProgressCircle.cjs");
require("./searchautocomplete.css");
var $13183d44bc7908f8$exports = require("./searchautocomplete_css.cjs");
require("../search_vars.css");
var $d2d4ce3e4a6482f9$exports = require("../search_vars_css.cjs");
require("../inputgroup_vars.css");
var $6c88ab5aea804b3c$exports = require("../inputgroup_vars_css.cjs");
var $827dbb466e199966$exports = require("../textfield/TextFieldBase.cjs");
require("../textfield_vars.css");
var $5d389146adc85829$exports = require("../textfield_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $1af2ca8553741739$exports = require("../form/Form.cjs");
var $0b97cdf6ccc1e502$exports = require("../utils/useIsMobileDevice.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $6SuLA$reactariaprivateautocompleteuseSearchAutocomplete = require("react-aria/private/autocomplete/useSearchAutocomplete");
var $6SuLA$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $6SuLA$reactariaFocusRing = require("react-aria/FocusRing");
var $6SuLA$spectrumiconsuiMagnifier = require("@spectrum-icons/ui/Magnifier");
var $6SuLA$reactstatelyuseComboBoxState = require("react-stately/useComboBoxState");
var $6SuLA$react = require("react");
var $6SuLA$reactariauseFilter = require("react-aria/useFilter");
var $6SuLA$reactariauseHover = require("react-aria/useHover");
var $6SuLA$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $6SuLA$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $6SuLA$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "SearchAutocomplete", function () { return $02024e29736a20cf$export$dd65332a5b19fa63; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




























function $02024e29736a20cf$var$SearchAutocomplete(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $1af2ca8553741739$exports.useFormProps)(props);
    let hasWarned = (0, $6SuLA$react.useRef)(false);
    (0, $6SuLA$react.useEffect)(()=>{
        if (props.placeholder && !hasWarned.current && process.env.NODE_ENV !== 'production') {
            console.warn('Placeholders are deprecated due to accessibility issues. Please use help text instead.');
            hasWarned.current = true;
        }
    }, [
        props.placeholder
    ]);
    let isMobile = (0, $0b97cdf6ccc1e502$exports.useIsMobileDevice)();
    if (isMobile) // menuTrigger=focus/manual don't apply to mobile searchwithin
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $501e31d6b6d1cb75$exports.MobileSearchAutocomplete), {
        ...props,
        menuTrigger: "input",
        ref: ref
    });
    else return /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement($02024e29736a20cf$var$SearchAutocompleteBase, {
        ...props,
        ref: ref
    });
}
function $02024e29736a20cf$var$ForwardSearchAutocompleteBase(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    let { menuTrigger: menuTrigger = 'input', shouldFlip: shouldFlip = true, direction: direction = 'bottom', align: align = 'start', isQuiet: isQuiet, menuWidth: customMenuWidth, loadingState: loadingState, onLoadMore: onLoadMore, onSubmit: onSubmit = ()=>{}, validate: validate } = props;
    let stringFormatter = (0, $6SuLA$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($7daa85d4b4165307$exports))), '@react-spectrum/autocomplete');
    let isAsync = loadingState != null;
    let popoverRef = (0, $6SuLA$react.useRef)(null);
    let unwrappedPopoverRef = (0, $65aea7b37663976b$exports.useUnwrapDOMRef)(popoverRef);
    let listBoxRef = (0, $6SuLA$react.useRef)(null);
    let inputRef = (0, $6SuLA$react.useRef)(null);
    let domRef = (0, $65aea7b37663976b$exports.useFocusableRef)(ref, inputRef);
    let { contains: contains } = (0, $6SuLA$reactariauseFilter.useFilter)({
        sensitivity: 'base'
    });
    let state = (0, $6SuLA$reactstatelyuseComboBoxState.useComboBoxState)({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: isAsync,
        allowsCustomValue: true,
        onSelectionChange: (key)=>key !== null && onSubmit(null, key),
        selectedKey: undefined,
        defaultSelectedKey: undefined,
        validate: (0, $6SuLA$react.useCallback)((v)=>validate?.(v.inputValue), [
            validate
        ])
    });
    let layout = (0, $cb7ee1d9d5613db9$exports.useListBoxLayout)();
    let { inputProps: inputProps, listBoxProps: listBoxProps, labelProps: labelProps, clearButtonProps: clearButtonProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $6SuLA$reactariaprivateautocompleteuseSearchAutocomplete.useSearchAutocomplete)({
        ...props,
        layoutDelegate: layout,
        popoverRef: unwrappedPopoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        menuTrigger: menuTrigger
    }, state);
    // Measure the width of the inputfield to inform the width of the menu (below).
    let [menuWidth, setMenuWidth] = (0, $6SuLA$react.useState)(0);
    let { scale: scale } = (0, $544fc82701fc93e9$exports.useProvider)();
    let onResize = (0, $6SuLA$react.useCallback)(()=>{
        if (inputRef.current) {
            let inputWidth = inputRef.current.offsetWidth;
            setMenuWidth(inputWidth);
        }
    }, [
        inputRef,
        setMenuWidth
    ]);
    (0, $6SuLA$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: domRef,
        onResize: onResize
    });
    (0, $6SuLA$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(onResize, [
        scale,
        onResize
    ]);
    let width = isQuiet ? undefined : menuWidth;
    let style = {
        width: customMenuWidth ? (0, $b8f90d51c4908137$exports.dimensionValue)(customMenuWidth) : width,
        minWidth: isQuiet ? `calc(${menuWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : menuWidth
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, ($parcel$interopDefault($6SuLA$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $b93966d678e0af07$exports.Field), {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        labelProps: labelProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement($02024e29736a20cf$var$SearchAutocompleteInput, {
        ...props,
        isOpen: state.isOpen,
        loadingState: loadingState,
        inputProps: inputProps,
        inputRef: inputRef,
        clearButtonProps: clearButtonProps,
        validationState: props.validationState || (isInvalid ? 'invalid' : undefined)
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $39ed1c805b59752f$exports.Popover), {
        state: state,
        UNSAFE_style: style,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-popover', {
            'spectrum-InputGroup-popover--quiet': isQuiet
        }),
        ref: popoverRef,
        triggerRef: inputRef,
        placement: `${direction} ${align}`,
        hideArrow: true,
        isNonModal: true,
        shouldFlip: shouldFlip
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $cb7ee1d9d5613db9$exports.ListBoxBase), {
        ...listBoxProps,
        ref: listBoxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy ?? undefined,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        isLoading: loadingState === 'loading' || loadingState === 'loadingMore',
        showLoadingSpinner: loadingState === 'loadingMore',
        onLoadMore: onLoadMore,
        renderEmptyState: ()=>isAsync && /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement("span", {
                className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'no-results')
            }, stringFormatter.format('noResults'))
    })));
}
let $02024e29736a20cf$var$SearchAutocompleteBase = /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).forwardRef($02024e29736a20cf$var$ForwardSearchAutocompleteBase);
// any type is because we don't want to call useObjectRef because this is an internal component and we know
// we are always passing an object ref
function $02024e29736a20cf$var$ForwardSearchAutocompleteInput(props, ref) {
    let searchIcon = /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, ($parcel$interopDefault($6SuLA$spectrumiconsuiMagnifier))), {
        "data-testid": "searchicon"
    });
    let { icon: icon = searchIcon, isQuiet: isQuiet, isDisabled: isDisabled, isReadOnly: isReadOnly, validationState: validationState, inputProps: inputProps, inputRef: inputRef, autoFocus: autoFocus, style: style, className: className, loadingState: loadingState, isOpen: isOpen, menuTrigger: menuTrigger, clearButtonProps: clearButtonProps } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $6SuLA$reactariauseHover.useHover)({});
    let stringFormatter = (0, $6SuLA$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($7daa85d4b4165307$exports))), '@react-spectrum/autocomplete');
    let domProps = (0, $6SuLA$reactariafilterDOMProps.filterDOMProps)(props);
    let timeout = (0, $6SuLA$react.useRef)(null);
    let [showLoading, setShowLoading] = (0, $6SuLA$react.useState)(false);
    let loadingCircle = /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $948c2416aa3a9507$exports.ProgressCircle), {
        "aria-label": stringFormatter.format('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-circleLoader', (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input-circleLoader'), (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-circleLoader'))
    });
    let clearButton = /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $0fc8553a4214494f$exports.ClearButton), {
        ...clearButtonProps,
        preventFocus: true,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let isLoading = loadingState === 'loading' || loadingState === 'filtering';
    let inputValue = inputProps.value;
    let lastInputValue = (0, $6SuLA$react.useRef)(inputValue);
    (0, $6SuLA$react.useEffect)(()=>{
        if (isLoading && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (!isLoading) // If loading is no longer happening, clear any timers and hide the loading circle
        {
            if (timeout.current != null) {
                clearTimeout(timeout.current);
                timeout.current = null;
            }
        }
        lastInputValue.current = inputValue;
    }, [
        isLoading,
        showLoading,
        inputValue
    ]);
    let [prevIsLoading, setPrevIsLoading] = (0, $6SuLA$react.useState)(isLoading);
    if (prevIsLoading !== isLoading && !isLoading) {
        setShowLoading(false);
        setPrevIsLoading(isLoading);
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $6SuLA$reactariaFocusRing.FocusRing), {
        within: true,
        isTextInput: true,
        focusClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'is-focused'),
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement("div", {
        ...hoverProps,
        ref: ref,
        style: style,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid' && !isDisabled,
            'is-hovered': isHovered
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($13183d44bc7908f8$exports))), 'searchautocomplete'), className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).createElement((0, $827dbb466e199966$exports.TextFieldBase), {
        ...domProps,
        inputProps: inputProps,
        inputRef: inputRef,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search', 'spectrum-Search--loadable', 'spectrum-Textfield', {
            'is-disabled': isDisabled,
            'is-quiet': isQuiet,
            'spectrum-Search--invalid': validationState === 'invalid' && !isDisabled,
            'spectrum-Search--valid': validationState === 'valid' && !isDisabled
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-field')),
        inputClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-input'),
        validationIconClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($d2d4ce3e4a6482f9$exports))), 'spectrum-Search-validationIcon'),
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        isLoading: showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'),
        loadingIndicator: loadingState != null ? loadingCircle : undefined,
        icon: icon,
        wrapperChildren: (inputValue !== '' || loadingState === 'filtering' || validationState != null) && !isReadOnly ? clearButton : undefined,
        disableFocusRing: true
    })));
}
let $02024e29736a20cf$var$SearchAutocompleteInput = /*#__PURE__*/ (0, ($parcel$interopDefault($6SuLA$react))).forwardRef($02024e29736a20cf$var$ForwardSearchAutocompleteInput);
/**
 * A SearchAutocomplete is a searchfield that supports a dynamic list of suggestions.
 */ let $02024e29736a20cf$export$dd65332a5b19fa63 = /*#__PURE__*/ (0, $6SuLA$react.forwardRef)($02024e29736a20cf$var$SearchAutocomplete);
