// react-aria-components/dist/private/ListBox.js
import {DEFAULT_SLOT as $b7b7a92703138c9b$export$c62b8e45d58ddad9, dom as $b7b7a92703138c9b$export$df3a06d6289f983e, Provider as $b7b7a92703138c9b$export$2881499e37b75b9a, useContextProps as $b7b7a92703138c9b$export$29f1550f4b0d4415, useRenderProps as $b7b7a92703138c9b$export$4d86445c2cf5e3, useSlot as $b7b7a92703138c9b$export$9d4c57ee4c6ffdd8} from "./utils.js";
import {CollectionRendererContext as $a53f0f6636929daa$export$4feb769f8ddf26c5, SectionContext as $a53f0f6636929daa$export$d40e14dec8b060a8} from "./Collection.js";
import {DragAndDropContext as $49776fcddfd94ccc$export$d188a835a7bc5783, DropIndicatorContext as $49776fcddfd94ccc$export$f55761759794cf55, useDndPersistedKeys as $49776fcddfd94ccc$export$d1e8e3fbb7461f6, useRenderDropIndicator as $49776fcddfd94ccc$export$971707d8a129a1f7} from "./DragAndDrop.js";
import {HeaderContext as $68ee918b64e9e759$export$e0e4026c12a8bdbb} from "./Header.js";
import {SelectableCollectionContext as $8f09b710ef85b337$export$b0d3ecf7112093a7} from "./Autocomplete.js";
import {SelectionIndicatorContext as $0d6f83ad40839938$export$c9549807523555e0} from "./SelectionIndicator.js";
import {SeparatorContext as $469f9b725520971a$export$6615d83f6de245ce} from "./Separator.js";
import {SharedElementTransition as $347bc273c4058e94$export$758399f318e6385a} from "./SharedElementTransition.js";
import {TextContext as $20d769b1e2b13352$export$9afb8bc826b033ea} from "./Text.js";
import {useListBox as $86Im4$useListBox, useListBoxSection as $86Im4$useListBoxSection, useOption as $86Im4$useOption} from "react-aria/useListBox";
import {Collection as $86Im4$Collection} from "react-aria/Collection";
import {CollectionBuilder as $86Im4$CollectionBuilder, createBranchComponent as $86Im4$createBranchComponent, createLeafComponent as $86Im4$createLeafComponent} from "react-aria/CollectionBuilder";
import {filterDOMProps as $86Im4$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusScope as $86Im4$FocusScope} from "react-aria/FocusScope";
import {inertValue as $86Im4$inertValue} from "react-aria/private/utils/inertValue";
import {SectionNode as $86Im4$SectionNode, ItemNode as $86Im4$ItemNode, LoaderNode as $86Im4$LoaderNode} from "react-aria/private/collections/BaseCollection";
import {ListKeyboardDelegate as $86Im4$ListKeyboardDelegate} from "react-aria/ListKeyboardDelegate";
import {useListState as $86Im4$useListState, UNSTABLE_useFilteredListState as $86Im4$UNSTABLE_useFilteredListState} from "react-stately/useListState";
import {useLoadMoreSentinel as $86Im4$useLoadMoreSentinel} from "react-aria/private/utils/useLoadMoreSentinel";
import {mergeProps as $86Im4$mergeProps} from "react-aria/mergeProps";
import $86Im4$react, {createContext as $86Im4$createContext, forwardRef as $86Im4$forwardRef, useContext as $86Im4$useContext, useMemo as $86Im4$useMemo, useRef as $86Im4$useRef, useEffect as $86Im4$useEffect} from "react";
import {useCollator as $86Im4$useCollator} from "react-aria/useCollator";
import {useFocus as $86Im4$useFocus} from "react-aria/useFocus";
import {useFocusRing as $86Im4$useFocusRing} from "react-aria/useFocusRing";
import {useHover as $86Im4$useHover} from "react-aria/useHover";
import {useKeyboard as $86Im4$useKeyboard} from "react-aria/useKeyboard";
import {useLocale as $86Im4$useLocale} from "react-aria/I18nProvider";
import {useObjectRef as $86Im4$useObjectRef} from "react-aria/useObjectRef";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



























const $ba3142315b3e1149$export$7ff8f37d2d81a48d = /*#__PURE__*/ (0, $86Im4$createContext)(null);
const $ba3142315b3e1149$export$7c5906fe4f1f2af2 = /*#__PURE__*/ (0, $86Im4$createContext)(null);
const $ba3142315b3e1149$export$41f133550aa26f48 = /*#__PURE__*/ (0, $86Im4$forwardRef)(function ListBox(props, ref) {
    [props, ref] = (0, $b7b7a92703138c9b$export$29f1550f4b0d4415)(props, ref, $ba3142315b3e1149$export$7ff8f37d2d81a48d);
    let state = (0, $86Im4$useContext)($ba3142315b3e1149$export$7c5906fe4f1f2af2);
    // The structure of ListBox is a bit strange because it needs to work inside other components like ComboBox and Select.
    // Those components render two copies of their children so that the collection can be built even when the popover is closed.
    // The first copy sends a collection document via context which we render the collection portal into.
    // The second copy sends a ListState object via context which we use to render the ListBox without rebuilding the state.
    // Otherwise, we have a standalone ListBox, so we need to create a collection and state ourselves.
    if (state) return /*#__PURE__*/ (0, $86Im4$react).createElement($ba3142315b3e1149$var$ListBoxInner, {
        state: state,
        props: props,
        listBoxRef: ref
    });
    return /*#__PURE__*/ (0, $86Im4$react).createElement((0, $86Im4$CollectionBuilder), {
        content: /*#__PURE__*/ (0, $86Im4$react).createElement((0, $86Im4$Collection), props)
    }, (collection)=>/*#__PURE__*/ (0, $86Im4$react).createElement($ba3142315b3e1149$var$StandaloneListBox, {
            props: props,
            listBoxRef: ref,
            collection: collection
        }));
});
function $ba3142315b3e1149$var$StandaloneListBox({ props: props, listBoxRef: listBoxRef, collection: collection }) {
    props = {
        ...props,
        collection: collection,
        children: null,
        items: null
    };
    let { layoutDelegate: layoutDelegate } = (0, $86Im4$useContext)((0, $a53f0f6636929daa$export$4feb769f8ddf26c5));
    let state = (0, $86Im4$useListState)({
        ...props,
        layoutDelegate: layoutDelegate
    });
    return /*#__PURE__*/ (0, $86Im4$react).createElement($ba3142315b3e1149$var$ListBoxInner, {
        state: state,
        props: props,
        listBoxRef: listBoxRef
    });
}
function $ba3142315b3e1149$var$ListBoxInner({ state: inputState, props: props, listBoxRef: listBoxRef }) {
    [props, listBoxRef] = (0, $b7b7a92703138c9b$export$29f1550f4b0d4415)(props, listBoxRef, (0, $8f09b710ef85b337$export$b0d3ecf7112093a7));
    let { dragAndDropHooks: dragAndDropHooks, layout: layout = 'stack', orientation: orientation = 'vertical', filter: filter } = props;
    let state = (0, $86Im4$UNSTABLE_useFilteredListState)(inputState, filter);
    let { collection: collection, selectionManager: selectionManager } = state;
    let isListDraggable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDraggableCollectionState);
    let isListDroppable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDroppableCollectionState);
    let { direction: direction } = (0, $86Im4$useLocale)();
    let { disabledBehavior: disabledBehavior, disabledKeys: disabledKeys } = selectionManager;
    let collator = (0, $86Im4$useCollator)({
        usage: 'search',
        sensitivity: 'base'
    });
    let { isVirtualized: isVirtualized, layoutDelegate: layoutDelegate, dropTargetDelegate: ctxDropTargetDelegate, CollectionRoot: CollectionRoot } = (0, $86Im4$useContext)((0, $a53f0f6636929daa$export$4feb769f8ddf26c5));
    let keyboardDelegate = (0, $86Im4$useMemo)(()=>props.keyboardDelegate || new (0, $86Im4$ListKeyboardDelegate)({
            collection: collection,
            collator: collator,
            ref: listBoxRef,
            disabledKeys: disabledKeys,
            disabledBehavior: disabledBehavior,
            layout: layout,
            orientation: orientation,
            direction: direction,
            layoutDelegate: layoutDelegate
        }), [
        collection,
        collator,
        listBoxRef,
        disabledBehavior,
        disabledKeys,
        orientation,
        direction,
        props.keyboardDelegate,
        layout,
        layoutDelegate
    ]);
    let { listBoxProps: listBoxProps } = (0, $86Im4$useListBox)({
        ...props,
        shouldSelectOnPressUp: isListDraggable || props.shouldSelectOnPressUp,
        keyboardDelegate: keyboardDelegate,
        isVirtualized: isVirtualized
    }, state, listBoxRef);
    let dragHooksProvided = (0, $86Im4$useRef)(isListDraggable);
    let dropHooksProvided = (0, $86Im4$useRef)(isListDroppable);
    (0, $86Im4$useEffect)(()=>{
        if (process.env.NODE_ENV === 'production') return;
        if (dragHooksProvided.current !== isListDraggable) console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
        if (dropHooksProvided.current !== isListDroppable) console.warn('Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
    }, [
        isListDraggable,
        isListDroppable
    ]);
    let dragState = undefined;
    let dropState = undefined;
    let droppableCollection = undefined;
    let isRootDropTarget = false;
    let dragPreview = null;
    let preview = (0, $86Im4$useRef)(null);
    if (isListDraggable && dragAndDropHooks) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: collection,
            selectionManager: selectionManager,
            preview: dragAndDropHooks.renderDragPreview ? preview : undefined
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, listBoxRef);
        let DragPreview = dragAndDropHooks.DragPreview;
        dragPreview = dragAndDropHooks.renderDragPreview ? /*#__PURE__*/ (0, $86Im4$react).createElement(DragPreview, {
            ref: preview
        }, dragAndDropHooks.renderDragPreview) : null;
    }
    if (isListDroppable && dragAndDropHooks) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: collection,
            selectionManager: selectionManager
        });
        let dropTargetDelegate = dragAndDropHooks.dropTargetDelegate || ctxDropTargetDelegate || new dragAndDropHooks.ListDropTargetDelegate(collection, listBoxRef, {
            orientation: orientation,
            layout: layout,
            direction: direction
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: keyboardDelegate,
            dropTargetDelegate: dropTargetDelegate
        }, dropState, listBoxRef);
        isRootDropTarget = dropState.isDropTarget({
            type: 'root'
        });
    }
    let { focusProps: focusProps, isFocused: isFocused, isFocusVisible: isFocusVisible } = (0, $86Im4$useFocusRing)();
    let isEmpty = state.collection.size === 0;
    let renderValues = {
        isDropTarget: isRootDropTarget,
        isEmpty: isEmpty,
        isFocused: isFocused,
        isFocusVisible: isFocusVisible,
        layout: props.layout || 'stack',
        orientation: orientation,
        state: state
    };
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...props,
        children: undefined,
        defaultClassName: 'react-aria-ListBox',
        values: renderValues
    });
    let emptyState = null;
    if (isEmpty && props.renderEmptyState) emptyState = /*#__PURE__*/ (0, $86Im4$react).createElement("div", {
        // eslint-disable-next-line
        role: "option",
        style: {
            display: 'contents'
        }
    }, props.renderEmptyState(renderValues));
    let DOMProps = (0, $86Im4$filterDOMProps)(props, {
        global: true
    });
    return /*#__PURE__*/ (0, $86Im4$react).createElement((0, $86Im4$FocusScope), null, /*#__PURE__*/ (0, $86Im4$react).createElement((0, $b7b7a92703138c9b$export$df3a06d6289f983e).div, {
        ...(0, $86Im4$mergeProps)(DOMProps, renderProps, listBoxProps, focusProps, droppableCollection === null || droppableCollection === void 0 ? void 0 : droppableCollection.collectionProps),
        ref: listBoxRef,
        slot: props.slot || undefined,
        onScroll: props.onScroll,
        "data-drop-target": isRootDropTarget || undefined,
        "data-empty": isEmpty || undefined,
        "data-focused": isFocused || undefined,
        "data-focus-visible": isFocusVisible || undefined,
        "data-layout": props.layout || 'stack',
        "data-orientation": orientation
    }, /*#__PURE__*/ (0, $86Im4$react).createElement((0, $b7b7a92703138c9b$export$2881499e37b75b9a), {
        values: [
            [
                $ba3142315b3e1149$export$7ff8f37d2d81a48d,
                props
            ],
            [
                $ba3142315b3e1149$export$7c5906fe4f1f2af2,
                state
            ],
            [
                (0, $49776fcddfd94ccc$export$d188a835a7bc5783),
                {
                    dragAndDropHooks: dragAndDropHooks,
                    dragState: dragState,
                    dropState: dropState
                }
            ],
            [
                (0, $469f9b725520971a$export$6615d83f6de245ce),
                {
                    elementType: 'div'
                }
            ],
            [
                (0, $49776fcddfd94ccc$export$f55761759794cf55),
                {
                    render: $ba3142315b3e1149$var$ListBoxDropIndicatorWrapper
                }
            ],
            [
                (0, $a53f0f6636929daa$export$d40e14dec8b060a8),
                {
                    name: 'ListBoxSection',
                    render: $ba3142315b3e1149$var$ListBoxSectionInner
                }
            ]
        ]
    }, /*#__PURE__*/ (0, $86Im4$react).createElement((0, $347bc273c4058e94$export$758399f318e6385a), null, /*#__PURE__*/ (0, $86Im4$react).createElement(CollectionRoot, {
        collection: collection,
        scrollRef: listBoxRef,
        persistedKeys: (0, $49776fcddfd94ccc$export$d1e8e3fbb7461f6)(selectionManager, dragAndDropHooks, dropState),
        renderDropIndicator: (0, $49776fcddfd94ccc$export$971707d8a129a1f7)(dragAndDropHooks, dropState)
    }))), emptyState, dragPreview));
}
function $ba3142315b3e1149$var$ListBoxSectionInner(props, ref, section, className = 'react-aria-ListBoxSection') {
    let state = (0, $86Im4$useContext)($ba3142315b3e1149$export$7c5906fe4f1f2af2);
    let { dragAndDropHooks: dragAndDropHooks, dropState: dropState } = (0, $86Im4$useContext)((0, $49776fcddfd94ccc$export$d188a835a7bc5783));
    let { CollectionBranch: CollectionBranch } = (0, $86Im4$useContext)((0, $a53f0f6636929daa$export$4feb769f8ddf26c5));
    let [headingRef, heading] = (0, $b7b7a92703138c9b$export$9d4c57ee4c6ffdd8)();
    var _props_arialabel;
    let { headingProps: headingProps, groupProps: groupProps } = (0, $86Im4$useListBoxSection)({
        heading: heading,
        'aria-label': (_props_arialabel = props['aria-label']) !== null && _props_arialabel !== void 0 ? _props_arialabel : undefined
    });
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...props,
        id: undefined,
        children: undefined,
        defaultClassName: className,
        values: undefined
    });
    let DOMProps = (0, $86Im4$filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    return /*#__PURE__*/ (0, $86Im4$react).createElement((0, $b7b7a92703138c9b$export$df3a06d6289f983e).section, {
        ...(0, $86Im4$mergeProps)(DOMProps, renderProps, groupProps),
        ref: ref
    }, /*#__PURE__*/ (0, $86Im4$react).createElement((0, $68ee918b64e9e759$export$e0e4026c12a8bdbb).Provider, {
        value: {
            ...headingProps,
            ref: headingRef
        }
    }, /*#__PURE__*/ (0, $86Im4$react).createElement(CollectionBranch, {
        collection: state.collection,
        parent: section,
        renderDropIndicator: (0, $49776fcddfd94ccc$export$971707d8a129a1f7)(dragAndDropHooks, dropState)
    })));
}
const $ba3142315b3e1149$export$dca12b0bb56e4fc = /*#__PURE__*/ (0, $86Im4$createBranchComponent)((0, $86Im4$SectionNode), $ba3142315b3e1149$var$ListBoxSectionInner);
const $ba3142315b3e1149$export$a11e76429ed99b4 = /*#__PURE__*/ (0, $86Im4$createLeafComponent)((0, $86Im4$ItemNode), function ListBoxItem(props, forwardedRef, item) {
    let ref = (0, $86Im4$useObjectRef)(forwardedRef);
    let state = (0, $86Im4$useContext)($ba3142315b3e1149$export$7c5906fe4f1f2af2);
    let { dragAndDropHooks: dragAndDropHooks, dragState: dragState, dropState: dropState } = (0, $86Im4$useContext)((0, $49776fcddfd94ccc$export$d188a835a7bc5783));
    let isDraggable = dragState && !(dragState.isDisabled || dragState.selectionManager.isDisabled(item.key));
    let { optionProps: optionProps, labelProps: labelProps, descriptionProps: descriptionProps, ...states } = (0, $86Im4$useOption)({
        key: item.key,
        'aria-label': props === null || props === void 0 ? void 0 : props['aria-label']
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $86Im4$useHover)({
        isDisabled: !states.allowsSelection && !states.hasAction && !isDraggable,
        onHoverStart: item.props.onHoverStart,
        onHoverChange: item.props.onHoverChange,
        onHoverEnd: item.props.onHoverEnd
    });
    let { keyboardProps: keyboardProps } = (0, $86Im4$useKeyboard)(props);
    let { focusProps: focusProps } = (0, $86Im4$useFocus)(props);
    let draggableItem = null;
    if (dragState && dragAndDropHooks) draggableItem = dragAndDropHooks.useDraggableItem({
        key: item.key,
        hasAction: states.hasAction
    }, dragState);
    let droppableItem = null;
    if (dropState && dragAndDropHooks) droppableItem = dragAndDropHooks.useDroppableItem({
        target: {
            type: 'item',
            key: item.key,
            dropPosition: 'on'
        }
    }, dropState, ref);
    let isDragging = dragState && dragState.isDragging(item.key);
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...props,
        id: undefined,
        children: props.children,
        defaultClassName: 'react-aria-ListBoxItem',
        values: {
            ...states,
            isHovered: isHovered,
            selectionMode: state.selectionManager.selectionMode,
            selectionBehavior: state.selectionManager.selectionBehavior,
            allowsDragging: !!dragState,
            isDragging: isDragging,
            isDropTarget: droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.isDropTarget
        }
    });
    (0, $86Im4$useEffect)(()=>{
        if (!item.textValue && process.env.NODE_ENV !== 'production') console.warn('A `textValue` prop is required for <ListBoxItem> elements with non-plain text children in order to support accessibility features such as type to select.');
    }, [
        item.textValue
    ]);
    let ElementType = props.href ? (0, $b7b7a92703138c9b$export$df3a06d6289f983e).a : (0, $b7b7a92703138c9b$export$df3a06d6289f983e).div;
    let DOMProps = (0, $86Im4$filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    delete DOMProps.onClick;
    if (props.href && optionProps.tabIndex == null) optionProps.tabIndex = -1;
    return /*#__PURE__*/ (0, $86Im4$react).createElement(ElementType, {
        ...(0, $86Im4$mergeProps)(DOMProps, renderProps, optionProps, hoverProps, keyboardProps, focusProps, draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragProps, droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps),
        ref: ref,
        "data-allows-dragging": !!dragState || undefined,
        "data-selected": states.isSelected || undefined,
        "data-disabled": states.isDisabled || undefined,
        "data-hovered": isHovered || undefined,
        "data-focused": states.isFocused || undefined,
        "data-focus-visible": states.isFocusVisible || undefined,
        "data-pressed": states.isPressed || undefined,
        "data-dragging": isDragging || undefined,
        "data-drop-target": (droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.isDropTarget) || undefined,
        "data-selection-mode": state.selectionManager.selectionMode === 'none' ? undefined : state.selectionManager.selectionMode
    }, /*#__PURE__*/ (0, $86Im4$react).createElement((0, $b7b7a92703138c9b$export$2881499e37b75b9a), {
        values: [
            [
                (0, $20d769b1e2b13352$export$9afb8bc826b033ea),
                {
                    slots: {
                        [(0, $b7b7a92703138c9b$export$c62b8e45d58ddad9)]: labelProps,
                        label: labelProps,
                        description: descriptionProps
                    }
                }
            ],
            [
                (0, $0d6f83ad40839938$export$c9549807523555e0),
                {
                    isSelected: states.isSelected
                }
            ]
        ]
    }, renderProps.children));
});
function $ba3142315b3e1149$var$ListBoxDropIndicatorWrapper(props, ref) {
    ref = (0, $86Im4$useObjectRef)(ref);
    let { dragAndDropHooks: dragAndDropHooks, dropState: dropState } = (0, $86Im4$useContext)((0, $49776fcddfd94ccc$export$d188a835a7bc5783));
    let { dropIndicatorProps: dropIndicatorProps, isHidden: isHidden, isDropTarget: isDropTarget } = dragAndDropHooks.useDropIndicator(props, dropState, ref);
    if (isHidden) return null;
    return /*#__PURE__*/ (0, $86Im4$react).createElement($ba3142315b3e1149$var$ListBoxDropIndicatorForwardRef, {
        ...props,
        dropIndicatorProps: dropIndicatorProps,
        isDropTarget: isDropTarget,
        ref: ref
    });
}
function $ba3142315b3e1149$var$ListBoxDropIndicator(props, ref) {
    let { dropIndicatorProps: dropIndicatorProps, isDropTarget: isDropTarget, ...otherProps } = props;
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...otherProps,
        defaultClassName: 'react-aria-DropIndicator',
        values: {
            isDropTarget: isDropTarget
        }
    });
    return /*#__PURE__*/ (0, $86Im4$react).createElement((0, $86Im4$react).Fragment, null, /*#__PURE__*/ (0, $86Im4$react).createElement((0, $b7b7a92703138c9b$export$df3a06d6289f983e).div, {
        ...dropIndicatorProps,
        ...renderProps,
        role: "option",
        ref: ref,
        "data-drop-target": isDropTarget || undefined
    }));
}
const $ba3142315b3e1149$var$ListBoxDropIndicatorForwardRef = /*#__PURE__*/ (0, $86Im4$forwardRef)($ba3142315b3e1149$var$ListBoxDropIndicator);
const $ba3142315b3e1149$export$8e6d031a08cf56a1 = (0, $86Im4$createLeafComponent)((0, $86Im4$LoaderNode), function ListBoxLoadingIndicator(props, ref, item) {
    let state = (0, $86Im4$useContext)($ba3142315b3e1149$export$7c5906fe4f1f2af2);
    let { isLoading: isLoading, onLoadMore: onLoadMore, scrollOffset: scrollOffset, ...otherProps } = props;
    let sentinelRef = (0, $86Im4$useRef)(null);
    let memoedLoadMoreProps = (0, $86Im4$useMemo)(()=>({
            onLoadMore: onLoadMore,
            collection: state === null || state === void 0 ? void 0 : state.collection,
            sentinelRef: sentinelRef,
            scrollOffset: scrollOffset
        }), [
        onLoadMore,
        scrollOffset,
        state === null || state === void 0 ? void 0 : state.collection
    ]);
    (0, $86Im4$useLoadMoreSentinel)(memoedLoadMoreProps, sentinelRef);
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...otherProps,
        id: undefined,
        children: item.rendered,
        defaultClassName: 'react-aria-ListBoxLoadingIndicator',
        values: undefined
    });
    let optionProps = {
        // For Android talkback
        tabIndex: -1
    };
    return /*#__PURE__*/ (0, $86Im4$react).createElement((0, $86Im4$react).Fragment, null, /*#__PURE__*/ (0, $86Im4$react).createElement("div", {
        style: {
            position: 'relative',
            width: 0,
            height: 0
        },
        inert: (0, $86Im4$inertValue)(true)
    }, /*#__PURE__*/ (0, $86Im4$react).createElement("div", {
        "data-testid": "loadMoreSentinel",
        ref: sentinelRef,
        style: {
            position: 'absolute',
            height: 1,
            width: 1
        }
    })), isLoading && renderProps.children && /*#__PURE__*/ (0, $86Im4$react).createElement((0, $86Im4$react).Fragment, null, /*#__PURE__*/ (0, $86Im4$react).createElement((0, $b7b7a92703138c9b$export$df3a06d6289f983e).div, {
        ...(0, $86Im4$mergeProps)((0, $86Im4$filterDOMProps)(props, {
            global: true
        }), optionProps),
        ...renderProps,
        // aria-selected isn't needed here since this option is not selectable.
        role: "option",
        ref: ref
    }, renderProps.children)));
});


export {$ba3142315b3e1149$export$7ff8f37d2d81a48d as ListBoxContext, $ba3142315b3e1149$export$7c5906fe4f1f2af2 as ListStateContext, $ba3142315b3e1149$export$41f133550aa26f48 as ListBox, $ba3142315b3e1149$export$dca12b0bb56e4fc as ListBoxSection, $ba3142315b3e1149$export$a11e76429ed99b4 as ListBoxItem, $ba3142315b3e1149$export$8e6d031a08cf56a1 as ListBoxLoadMoreItem};
