// react-aria-components/dist/private/ListBox.cjs
var $048d76b84370f141$exports = require("./utils.cjs");
var $f7b82bedbb70abac$exports = require("./Collection.cjs");
var $d3d8871226fc64f2$exports = require("./DragAndDrop.cjs");
var $749e1f015a6d7f1a$exports = require("./Header.cjs");
var $433949643203e332$exports = require("./Autocomplete.cjs");
var $61557b2a9b2862a8$exports = require("./SelectionIndicator.cjs");
var $5a1b0036f8cbf051$exports = require("./Separator.cjs");
var $9a60bd90621ebc78$exports = require("./SharedElementTransition.cjs");
var $cab7d9a238d19c33$exports = require("./Text.cjs");
var $4vZEP$reactariauseListBox = require("react-aria/useListBox");
var $4vZEP$reactariaCollection = require("react-aria/Collection");
var $4vZEP$reactariaCollectionBuilder = require("react-aria/CollectionBuilder");
var $4vZEP$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $4vZEP$reactariaFocusScope = require("react-aria/FocusScope");
var $4vZEP$reactariaprivateutilsinertValue = require("react-aria/private/utils/inertValue");
var $4vZEP$reactariaprivatecollectionsBaseCollection = require("react-aria/private/collections/BaseCollection");
var $4vZEP$reactariaListKeyboardDelegate = require("react-aria/ListKeyboardDelegate");
var $4vZEP$reactstatelyuseListState = require("react-stately/useListState");
var $4vZEP$reactariaprivateutilsuseLoadMoreSentinel = require("react-aria/private/utils/useLoadMoreSentinel");
var $4vZEP$reactariamergeProps = require("react-aria/mergeProps");
var $4vZEP$react = require("react");
var $4vZEP$reactariauseCollator = require("react-aria/useCollator");
var $4vZEP$reactariauseFocus = require("react-aria/useFocus");
var $4vZEP$reactariauseFocusRing = require("react-aria/useFocusRing");
var $4vZEP$reactariauseHover = require("react-aria/useHover");
var $4vZEP$reactariauseKeyboard = require("react-aria/useKeyboard");
var $4vZEP$reactariaI18nProvider = require("react-aria/I18nProvider");
var $4vZEP$reactariauseObjectRef = require("react-aria/useObjectRef");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ListBoxContext", function () { return $537333b300f7e667$export$7ff8f37d2d81a48d; });
$parcel$export(module.exports, "ListStateContext", function () { return $537333b300f7e667$export$7c5906fe4f1f2af2; });
$parcel$export(module.exports, "ListBox", function () { return $537333b300f7e667$export$41f133550aa26f48; });
$parcel$export(module.exports, "ListBoxSection", function () { return $537333b300f7e667$export$dca12b0bb56e4fc; });
$parcel$export(module.exports, "ListBoxItem", function () { return $537333b300f7e667$export$a11e76429ed99b4; });
$parcel$export(module.exports, "ListBoxLoadMoreItem", function () { return $537333b300f7e667$export$8e6d031a08cf56a1; });
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



























const $537333b300f7e667$export$7ff8f37d2d81a48d = /*#__PURE__*/ (0, $4vZEP$react.createContext)(null);
const $537333b300f7e667$export$7c5906fe4f1f2af2 = /*#__PURE__*/ (0, $4vZEP$react.createContext)(null);
const $537333b300f7e667$export$41f133550aa26f48 = /*#__PURE__*/ (0, $4vZEP$react.forwardRef)(function ListBox(props, ref) {
    [props, ref] = (0, $048d76b84370f141$exports.useContextProps)(props, ref, $537333b300f7e667$export$7ff8f37d2d81a48d);
    let state = (0, $4vZEP$react.useContext)($537333b300f7e667$export$7c5906fe4f1f2af2);
    // The structure of ListBox is a bit strange because it needs to work inside other components like ComboBox and Select.
    // Those components render two copies of their children so that the collection can be built even when the popover is closed.
    // The first copy sends a collection document via context which we render the collection portal into.
    // The second copy sends a ListState object via context which we use to render the ListBox without rebuilding the state.
    // Otherwise, we have a standalone ListBox, so we need to create a collection and state ourselves.
    if (state) return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement($537333b300f7e667$var$ListBoxInner, {
        state: state,
        props: props,
        listBoxRef: ref
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $4vZEP$reactariaCollectionBuilder.CollectionBuilder), {
        content: /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $4vZEP$reactariaCollection.Collection), props)
    }, (collection)=>/*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement($537333b300f7e667$var$StandaloneListBox, {
            props: props,
            listBoxRef: ref,
            collection: collection
        }));
});
function $537333b300f7e667$var$StandaloneListBox({ props: props, listBoxRef: listBoxRef, collection: collection }) {
    props = {
        ...props,
        collection: collection,
        children: null,
        items: null
    };
    let { layoutDelegate: layoutDelegate } = (0, $4vZEP$react.useContext)((0, $f7b82bedbb70abac$exports.CollectionRendererContext));
    let state = (0, $4vZEP$reactstatelyuseListState.useListState)({
        ...props,
        layoutDelegate: layoutDelegate
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement($537333b300f7e667$var$ListBoxInner, {
        state: state,
        props: props,
        listBoxRef: listBoxRef
    });
}
function $537333b300f7e667$var$ListBoxInner({ state: inputState, props: props, listBoxRef: listBoxRef }) {
    [props, listBoxRef] = (0, $048d76b84370f141$exports.useContextProps)(props, listBoxRef, (0, $433949643203e332$exports.SelectableCollectionContext));
    let { dragAndDropHooks: dragAndDropHooks, layout: layout = 'stack', orientation: orientation = 'vertical', filter: filter } = props;
    let state = (0, $4vZEP$reactstatelyuseListState.UNSTABLE_useFilteredListState)(inputState, filter);
    let { collection: collection, selectionManager: selectionManager } = state;
    let isListDraggable = !!dragAndDropHooks?.useDraggableCollectionState;
    let isListDroppable = !!dragAndDropHooks?.useDroppableCollectionState;
    let { direction: direction } = (0, $4vZEP$reactariaI18nProvider.useLocale)();
    let { disabledBehavior: disabledBehavior, disabledKeys: disabledKeys } = selectionManager;
    let collator = (0, $4vZEP$reactariauseCollator.useCollator)({
        usage: 'search',
        sensitivity: 'base'
    });
    let { isVirtualized: isVirtualized, layoutDelegate: layoutDelegate, dropTargetDelegate: ctxDropTargetDelegate, CollectionRoot: CollectionRoot } = (0, $4vZEP$react.useContext)((0, $f7b82bedbb70abac$exports.CollectionRendererContext));
    let keyboardDelegate = (0, $4vZEP$react.useMemo)(()=>props.keyboardDelegate || new (0, $4vZEP$reactariaListKeyboardDelegate.ListKeyboardDelegate)({
            collection: collection,
            collator: collator,
            ref: listBoxRef,
            disabledKeys: disabledKeys,
            disabledBehavior: disabledBehavior,
            layout: layout,
            orientation: orientation,
            direction: direction,
            layoutDelegate: layoutDelegate
        }), [
        collection,
        collator,
        listBoxRef,
        disabledBehavior,
        disabledKeys,
        orientation,
        direction,
        props.keyboardDelegate,
        layout,
        layoutDelegate
    ]);
    let { listBoxProps: listBoxProps } = (0, $4vZEP$reactariauseListBox.useListBox)({
        ...props,
        shouldSelectOnPressUp: isListDraggable || props.shouldSelectOnPressUp,
        keyboardDelegate: keyboardDelegate,
        isVirtualized: isVirtualized
    }, state, listBoxRef);
    let dragHooksProvided = (0, $4vZEP$react.useRef)(isListDraggable);
    let dropHooksProvided = (0, $4vZEP$react.useRef)(isListDroppable);
    (0, $4vZEP$react.useEffect)(()=>{
        if (process.env.NODE_ENV === 'production') return;
        if (dragHooksProvided.current !== isListDraggable) console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
        if (dropHooksProvided.current !== isListDroppable) console.warn('Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
    }, [
        isListDraggable,
        isListDroppable
    ]);
    let dragState = undefined;
    let dropState = undefined;
    let droppableCollection = undefined;
    let isRootDropTarget = false;
    let dragPreview = null;
    let preview = (0, $4vZEP$react.useRef)(null);
    if (isListDraggable && dragAndDropHooks) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: collection,
            selectionManager: selectionManager,
            preview: dragAndDropHooks.renderDragPreview ? preview : undefined
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, listBoxRef);
        let DragPreview = dragAndDropHooks.DragPreview;
        dragPreview = dragAndDropHooks.renderDragPreview ? /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement(DragPreview, {
            ref: preview
        }, dragAndDropHooks.renderDragPreview) : null;
    }
    if (isListDroppable && dragAndDropHooks) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: collection,
            selectionManager: selectionManager
        });
        let dropTargetDelegate = dragAndDropHooks.dropTargetDelegate || ctxDropTargetDelegate || new dragAndDropHooks.ListDropTargetDelegate(collection, listBoxRef, {
            orientation: orientation,
            layout: layout,
            direction: direction
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: keyboardDelegate,
            dropTargetDelegate: dropTargetDelegate
        }, dropState, listBoxRef);
        isRootDropTarget = dropState.isDropTarget({
            type: 'root'
        });
    }
    let { focusProps: focusProps, isFocused: isFocused, isFocusVisible: isFocusVisible } = (0, $4vZEP$reactariauseFocusRing.useFocusRing)();
    let isEmpty = state.collection.size === 0;
    let renderValues = {
        isDropTarget: isRootDropTarget,
        isEmpty: isEmpty,
        isFocused: isFocused,
        isFocusVisible: isFocusVisible,
        layout: props.layout || 'stack',
        orientation: orientation,
        state: state
    };
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...props,
        children: undefined,
        defaultClassName: 'react-aria-ListBox',
        values: renderValues
    });
    let emptyState = null;
    if (isEmpty && props.renderEmptyState) emptyState = /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement("div", {
        // eslint-disable-next-line
        role: "option",
        style: {
            display: 'contents'
        }
    }, props.renderEmptyState(renderValues));
    let DOMProps = (0, $4vZEP$reactariafilterDOMProps.filterDOMProps)(props, {
        global: true
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $4vZEP$reactariaFocusScope.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...(0, $4vZEP$reactariamergeProps.mergeProps)(DOMProps, renderProps, listBoxProps, focusProps, droppableCollection?.collectionProps),
        ref: listBoxRef,
        slot: props.slot || undefined,
        onScroll: props.onScroll,
        "data-drop-target": isRootDropTarget || undefined,
        "data-empty": isEmpty || undefined,
        "data-focused": isFocused || undefined,
        "data-focus-visible": isFocusVisible || undefined,
        "data-layout": props.layout || 'stack',
        "data-orientation": orientation
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $048d76b84370f141$exports.Provider), {
        values: [
            [
                $537333b300f7e667$export$7ff8f37d2d81a48d,
                props
            ],
            [
                $537333b300f7e667$export$7c5906fe4f1f2af2,
                state
            ],
            [
                (0, $d3d8871226fc64f2$exports.DragAndDropContext),
                {
                    dragAndDropHooks: dragAndDropHooks,
                    dragState: dragState,
                    dropState: dropState
                }
            ],
            [
                (0, $5a1b0036f8cbf051$exports.SeparatorContext),
                {
                    elementType: 'div'
                }
            ],
            [
                (0, $d3d8871226fc64f2$exports.DropIndicatorContext),
                {
                    render: $537333b300f7e667$var$ListBoxDropIndicatorWrapper
                }
            ],
            [
                (0, $f7b82bedbb70abac$exports.SectionContext),
                {
                    name: 'ListBoxSection',
                    render: $537333b300f7e667$var$ListBoxSectionInner
                }
            ]
        ]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $9a60bd90621ebc78$exports.SharedElementTransition), null, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement(CollectionRoot, {
        collection: collection,
        scrollRef: listBoxRef,
        persistedKeys: (0, $d3d8871226fc64f2$exports.useDndPersistedKeys)(selectionManager, dragAndDropHooks, dropState),
        renderDropIndicator: (0, $d3d8871226fc64f2$exports.useRenderDropIndicator)(dragAndDropHooks, dropState)
    }))), emptyState, dragPreview));
}
function $537333b300f7e667$var$ListBoxSectionInner(props, ref, section, className = 'react-aria-ListBoxSection') {
    let state = (0, $4vZEP$react.useContext)($537333b300f7e667$export$7c5906fe4f1f2af2);
    let { dragAndDropHooks: dragAndDropHooks, dropState: dropState } = (0, $4vZEP$react.useContext)((0, $d3d8871226fc64f2$exports.DragAndDropContext));
    let { CollectionBranch: CollectionBranch } = (0, $4vZEP$react.useContext)((0, $f7b82bedbb70abac$exports.CollectionRendererContext));
    let [headingRef, heading] = (0, $048d76b84370f141$exports.useSlot)();
    let { headingProps: headingProps, groupProps: groupProps } = (0, $4vZEP$reactariauseListBox.useListBoxSection)({
        heading: heading,
        'aria-label': props['aria-label'] ?? undefined
    });
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...props,
        id: undefined,
        children: undefined,
        defaultClassName: className,
        values: undefined
    });
    let DOMProps = (0, $4vZEP$reactariafilterDOMProps.filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $048d76b84370f141$exports.dom).section, {
        ...(0, $4vZEP$reactariamergeProps.mergeProps)(DOMProps, renderProps, groupProps),
        ref: ref
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $749e1f015a6d7f1a$exports.HeaderContext).Provider, {
        value: {
            ...headingProps,
            ref: headingRef
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement(CollectionBranch, {
        collection: state.collection,
        parent: section,
        renderDropIndicator: (0, $d3d8871226fc64f2$exports.useRenderDropIndicator)(dragAndDropHooks, dropState)
    })));
}
const $537333b300f7e667$export$dca12b0bb56e4fc = /*#__PURE__*/ (0, $4vZEP$reactariaCollectionBuilder.createBranchComponent)((0, $4vZEP$reactariaprivatecollectionsBaseCollection.SectionNode), $537333b300f7e667$var$ListBoxSectionInner);
const $537333b300f7e667$export$a11e76429ed99b4 = /*#__PURE__*/ (0, $4vZEP$reactariaCollectionBuilder.createLeafComponent)((0, $4vZEP$reactariaprivatecollectionsBaseCollection.ItemNode), function ListBoxItem(props, forwardedRef, item) {
    let ref = (0, $4vZEP$reactariauseObjectRef.useObjectRef)(forwardedRef);
    let state = (0, $4vZEP$react.useContext)($537333b300f7e667$export$7c5906fe4f1f2af2);
    let { dragAndDropHooks: dragAndDropHooks, dragState: dragState, dropState: dropState } = (0, $4vZEP$react.useContext)((0, $d3d8871226fc64f2$exports.DragAndDropContext));
    let isDraggable = dragState && !(dragState.isDisabled || dragState.selectionManager.isDisabled(item.key));
    let { optionProps: optionProps, labelProps: labelProps, descriptionProps: descriptionProps, ...states } = (0, $4vZEP$reactariauseListBox.useOption)({
        key: item.key,
        'aria-label': props?.['aria-label']
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $4vZEP$reactariauseHover.useHover)({
        isDisabled: !states.allowsSelection && !states.hasAction && !isDraggable,
        onHoverStart: item.props.onHoverStart,
        onHoverChange: item.props.onHoverChange,
        onHoverEnd: item.props.onHoverEnd
    });
    let { keyboardProps: keyboardProps } = (0, $4vZEP$reactariauseKeyboard.useKeyboard)(props);
    let { focusProps: focusProps } = (0, $4vZEP$reactariauseFocus.useFocus)(props);
    let draggableItem = null;
    if (dragState && dragAndDropHooks) draggableItem = dragAndDropHooks.useDraggableItem({
        key: item.key,
        hasAction: states.hasAction
    }, dragState);
    let droppableItem = null;
    if (dropState && dragAndDropHooks) droppableItem = dragAndDropHooks.useDroppableItem({
        target: {
            type: 'item',
            key: item.key,
            dropPosition: 'on'
        }
    }, dropState, ref);
    let isDragging = dragState && dragState.isDragging(item.key);
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...props,
        id: undefined,
        children: props.children,
        defaultClassName: 'react-aria-ListBoxItem',
        values: {
            ...states,
            isHovered: isHovered,
            selectionMode: state.selectionManager.selectionMode,
            selectionBehavior: state.selectionManager.selectionBehavior,
            allowsDragging: !!dragState,
            isDragging: isDragging,
            isDropTarget: droppableItem?.isDropTarget
        }
    });
    (0, $4vZEP$react.useEffect)(()=>{
        if (!item.textValue && process.env.NODE_ENV !== 'production') console.warn('A `textValue` prop is required for <ListBoxItem> elements with non-plain text children in order to support accessibility features such as type to select.');
    }, [
        item.textValue
    ]);
    let ElementType = props.href ? (0, $048d76b84370f141$exports.dom).a : (0, $048d76b84370f141$exports.dom).div;
    let DOMProps = (0, $4vZEP$reactariafilterDOMProps.filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    delete DOMProps.onClick;
    if (props.href && optionProps.tabIndex == null) optionProps.tabIndex = -1;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement(ElementType, {
        ...(0, $4vZEP$reactariamergeProps.mergeProps)(DOMProps, renderProps, optionProps, hoverProps, keyboardProps, focusProps, draggableItem?.dragProps, droppableItem?.dropProps),
        ref: ref,
        "data-allows-dragging": !!dragState || undefined,
        "data-selected": states.isSelected || undefined,
        "data-disabled": states.isDisabled || undefined,
        "data-hovered": isHovered || undefined,
        "data-focused": states.isFocused || undefined,
        "data-focus-visible": states.isFocusVisible || undefined,
        "data-pressed": states.isPressed || undefined,
        "data-dragging": isDragging || undefined,
        "data-drop-target": droppableItem?.isDropTarget || undefined,
        "data-selection-mode": state.selectionManager.selectionMode === 'none' ? undefined : state.selectionManager.selectionMode
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $048d76b84370f141$exports.Provider), {
        values: [
            [
                (0, $cab7d9a238d19c33$exports.TextContext),
                {
                    slots: {
                        [(0, $048d76b84370f141$exports.DEFAULT_SLOT)]: labelProps,
                        label: labelProps,
                        description: descriptionProps
                    }
                }
            ],
            [
                (0, $61557b2a9b2862a8$exports.SelectionIndicatorContext),
                {
                    isSelected: states.isSelected
                }
            ]
        ]
    }, renderProps.children));
});
function $537333b300f7e667$var$ListBoxDropIndicatorWrapper(props, ref) {
    ref = (0, $4vZEP$reactariauseObjectRef.useObjectRef)(ref);
    let { dragAndDropHooks: dragAndDropHooks, dropState: dropState } = (0, $4vZEP$react.useContext)((0, $d3d8871226fc64f2$exports.DragAndDropContext));
    let { dropIndicatorProps: dropIndicatorProps, isHidden: isHidden, isDropTarget: isDropTarget } = dragAndDropHooks.useDropIndicator(props, dropState, ref);
    if (isHidden) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement($537333b300f7e667$var$ListBoxDropIndicatorForwardRef, {
        ...props,
        dropIndicatorProps: dropIndicatorProps,
        isDropTarget: isDropTarget,
        ref: ref
    });
}
function $537333b300f7e667$var$ListBoxDropIndicator(props, ref) {
    let { dropIndicatorProps: dropIndicatorProps, isDropTarget: isDropTarget, ...otherProps } = props;
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...otherProps,
        defaultClassName: 'react-aria-DropIndicator',
        values: {
            isDropTarget: isDropTarget
        }
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, ($parcel$interopDefault($4vZEP$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...dropIndicatorProps,
        ...renderProps,
        role: "option",
        ref: ref,
        "data-drop-target": isDropTarget || undefined
    }));
}
const $537333b300f7e667$var$ListBoxDropIndicatorForwardRef = /*#__PURE__*/ (0, $4vZEP$react.forwardRef)($537333b300f7e667$var$ListBoxDropIndicator);
const $537333b300f7e667$export$8e6d031a08cf56a1 = (0, $4vZEP$reactariaCollectionBuilder.createLeafComponent)((0, $4vZEP$reactariaprivatecollectionsBaseCollection.LoaderNode), function ListBoxLoadingIndicator(props, ref, item) {
    let state = (0, $4vZEP$react.useContext)($537333b300f7e667$export$7c5906fe4f1f2af2);
    let { isLoading: isLoading, onLoadMore: onLoadMore, scrollOffset: scrollOffset, ...otherProps } = props;
    let sentinelRef = (0, $4vZEP$react.useRef)(null);
    let memoedLoadMoreProps = (0, $4vZEP$react.useMemo)(()=>({
            onLoadMore: onLoadMore,
            collection: state?.collection,
            sentinelRef: sentinelRef,
            scrollOffset: scrollOffset
        }), [
        onLoadMore,
        scrollOffset,
        state?.collection
    ]);
    (0, $4vZEP$reactariaprivateutilsuseLoadMoreSentinel.useLoadMoreSentinel)(memoedLoadMoreProps, sentinelRef);
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...otherProps,
        id: undefined,
        children: item.rendered,
        defaultClassName: 'react-aria-ListBoxLoadingIndicator',
        values: undefined
    });
    let optionProps = {
        // For Android talkback
        tabIndex: -1
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, ($parcel$interopDefault($4vZEP$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement("div", {
        style: {
            position: 'relative',
            width: 0,
            height: 0
        },
        inert: (0, $4vZEP$reactariaprivateutilsinertValue.inertValue)(true)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement("div", {
        "data-testid": "loadMoreSentinel",
        ref: sentinelRef,
        style: {
            position: 'absolute',
            height: 1,
            width: 1
        }
    })), isLoading && renderProps.children && /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, ($parcel$interopDefault($4vZEP$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($4vZEP$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...(0, $4vZEP$reactariamergeProps.mergeProps)((0, $4vZEP$reactariafilterDOMProps.filterDOMProps)(props, {
            global: true
        }), optionProps),
        ...renderProps,
        // aria-selected isn't needed here since this option is not selectable.
        role: "option",
        ref: ref
    }, renderProps.children)));
});
