// react-aria-components/dist/private/Menu.js
import {DEFAULT_SLOT as $b7b7a92703138c9b$export$c62b8e45d58ddad9, dom as $b7b7a92703138c9b$export$df3a06d6289f983e, Provider as $b7b7a92703138c9b$export$2881499e37b75b9a, useContextProps as $b7b7a92703138c9b$export$29f1550f4b0d4415, useRenderProps as $b7b7a92703138c9b$export$4d86445c2cf5e3, useSlot as $b7b7a92703138c9b$export$9d4c57ee4c6ffdd8, useSlottedContext as $b7b7a92703138c9b$export$fabf2dc03a41866e} from "./utils.js";
import {CollectionRendererContext as $a53f0f6636929daa$export$4feb769f8ddf26c5, SectionContext as $a53f0f6636929daa$export$d40e14dec8b060a8, usePersistedKeys as $a53f0f6636929daa$export$90e00781bc59d8f9} from "./Collection.js";
import {FieldInputContext as $8f09b710ef85b337$export$698f465ec27e93df, SelectableCollectionContext as $8f09b710ef85b337$export$b0d3ecf7112093a7} from "./Autocomplete.js";
import {HeaderContext as $68ee918b64e9e759$export$e0e4026c12a8bdbb} from "./Header.js";
import {KeyboardContext as $ba03015127ce0a12$export$744d98a3b8a94e1c} from "./Keyboard.js";
import {OverlayTriggerStateContext as $acf8e70c2f419f18$export$d2f961adcb0afbe} from "./Dialog.js";
import {PopoverContext as $03df2f2d3cffb62f$export$9b9a0cd73afb7ca4} from "./Popover.js";
import {SelectionIndicatorContext as $0d6f83ad40839938$export$c9549807523555e0} from "./SelectionIndicator.js";
import {SeparatorContext as $469f9b725520971a$export$6615d83f6de245ce} from "./Separator.js";
import {SharedElementTransition as $347bc273c4058e94$export$758399f318e6385a} from "./SharedElementTransition.js";
import {TextContext as $20d769b1e2b13352$export$9afb8bc826b033ea} from "./Text.js";
import {useMenuTrigger as $gKmcL$useMenuTrigger, useSubmenuTrigger as $gKmcL$useSubmenuTrigger, useMenu as $gKmcL$useMenu, useMenuSection as $gKmcL$useMenuSection, useMenuItem as $gKmcL$useMenuItem} from "react-aria/useMenu";
import {CollectionNode as $gKmcL$CollectionNode, SectionNode as $gKmcL$SectionNode, ItemNode as $gKmcL$ItemNode} from "react-aria/private/collections/BaseCollection";
import {useMenuTriggerState as $gKmcL$useMenuTriggerState, useSubmenuTriggerState as $gKmcL$useSubmenuTriggerState} from "react-stately/useMenuTriggerState";
import {Collection as $gKmcL$Collection} from "react-aria/Collection";
import {createBranchComponent as $gKmcL$createBranchComponent, CollectionBuilder as $gKmcL$CollectionBuilder, createLeafComponent as $gKmcL$createLeafComponent} from "react-aria/CollectionBuilder";
import {filterDOMProps as $gKmcL$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusScope as $gKmcL$FocusScope} from "react-aria/FocusScope";
import {mergeProps as $gKmcL$mergeProps} from "react-aria/mergeProps";
import {PressResponder as $gKmcL$PressResponder} from "react-aria/private/interactions/PressResponder";
import $gKmcL$react, {createContext as $gKmcL$createContext, useRef as $gKmcL$useRef, useContext as $gKmcL$useContext, forwardRef as $gKmcL$forwardRef, useMemo as $gKmcL$useMemo} from "react";
import {SelectionManager as $gKmcL$SelectionManager} from "react-stately/private/selection/SelectionManager";
import {useTreeState as $gKmcL$useTreeState} from "react-stately/useTreeState";
import {useHover as $gKmcL$useHover} from "react-aria/useHover";
import {useIsHidden as $gKmcL$useIsHidden} from "react-aria/private/collections/Hidden";
import {useMultipleSelectionState as $gKmcL$useMultipleSelectionState} from "react-stately/useMultipleSelectionState";
import {useObjectRef as $gKmcL$useObjectRef} from "react-aria/useObjectRef";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


























const $113aa0613e727c6c$export$c7e742effb1c51e2 = /*#__PURE__*/ (0, $gKmcL$createContext)(null);
const $113aa0613e727c6c$export$24aad8519b95b41b = /*#__PURE__*/ (0, $gKmcL$createContext)(null);
const $113aa0613e727c6c$export$795aec4671cbae19 = /*#__PURE__*/ (0, $gKmcL$createContext)(null);
const $113aa0613e727c6c$var$SelectionManagerContext = /*#__PURE__*/ (0, $gKmcL$createContext)(null);
function $113aa0613e727c6c$export$27d2ad3c5815583e(props) {
    let state = (0, $gKmcL$useMenuTriggerState)(props);
    let ref = (0, $gKmcL$useRef)(null);
    let { menuTriggerProps: menuTriggerProps, menuProps: menuProps } = (0, $gKmcL$useMenuTrigger)({
        ...props,
        type: 'menu'
    }, state, ref);
    let scrollRef = (0, $gKmcL$useRef)(null);
    // If within a collection (e.g. Tabs), render nothing.
    // Not using createHideableComponent for this because that also creates a forwardRef.
    let isHidden = (0, $gKmcL$useIsHidden)();
    if (isHidden) return null;
    return /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $b7b7a92703138c9b$export$2881499e37b75b9a), {
        values: [
            [
                $113aa0613e727c6c$export$c7e742effb1c51e2,
                {
                    ...menuProps,
                    ref: scrollRef
                }
            ],
            [
                (0, $acf8e70c2f419f18$export$d2f961adcb0afbe),
                state
            ],
            [
                $113aa0613e727c6c$export$795aec4671cbae19,
                state
            ],
            [
                (0, $03df2f2d3cffb62f$export$9b9a0cd73afb7ca4),
                {
                    trigger: 'MenuTrigger',
                    triggerRef: ref,
                    scrollRef: scrollRef,
                    placement: 'bottom start',
                    'aria-labelledby': menuProps['aria-labelledby']
                }
            ]
        ]
    }, /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $gKmcL$PressResponder), {
        ...menuTriggerProps,
        ref: ref,
        isPressed: state.isOpen
    }, props.children));
}
const $113aa0613e727c6c$var$SubmenuTriggerContext = /*#__PURE__*/ (0, $gKmcL$createContext)(null);
class $113aa0613e727c6c$var$SubmenuTriggerNode extends (0, $gKmcL$CollectionNode) {
    filter(collection, newCollection, filterFn) {
        let triggerNode = collection.getItem(this.firstChildKey);
        if (triggerNode && filterFn(triggerNode.textValue, this)) {
            let clone = this.clone();
            newCollection.addDescendants(clone, collection);
            return clone;
        }
        return null;
    }
}
$113aa0613e727c6c$var$SubmenuTriggerNode.type = 'submenutrigger';
const $113aa0613e727c6c$export$ecabc99eeffab7ca = /*#__PURE__*/ (0, $gKmcL$createBranchComponent)($113aa0613e727c6c$var$SubmenuTriggerNode, (props, ref, item)=>{
    let { CollectionBranch: CollectionBranch } = (0, $gKmcL$useContext)((0, $a53f0f6636929daa$export$4feb769f8ddf26c5));
    let state = (0, $gKmcL$useContext)($113aa0613e727c6c$export$24aad8519b95b41b);
    let rootMenuTriggerState = (0, $gKmcL$useContext)($113aa0613e727c6c$export$795aec4671cbae19);
    let submenuTriggerState = (0, $gKmcL$useSubmenuTriggerState)({
        triggerKey: item.key
    }, rootMenuTriggerState);
    let submenuRef = (0, $gKmcL$useRef)(null);
    let itemRef = (0, $gKmcL$useObjectRef)(ref);
    let { parentMenuRef: parentMenuRef, shouldUseVirtualFocus: shouldUseVirtualFocus } = (0, $gKmcL$useContext)($113aa0613e727c6c$var$SubmenuTriggerContext);
    let { submenuTriggerProps: submenuTriggerProps, submenuProps: submenuProps, popoverProps: popoverProps } = (0, $gKmcL$useSubmenuTrigger)({
        parentMenuRef: parentMenuRef,
        submenuRef: submenuRef,
        delay: props.delay,
        shouldUseVirtualFocus: shouldUseVirtualFocus
    }, submenuTriggerState, itemRef);
    return /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $b7b7a92703138c9b$export$2881499e37b75b9a), {
        values: [
            [
                $113aa0613e727c6c$var$MenuItemContext,
                {
                    ...submenuTriggerProps,
                    onAction: undefined,
                    ref: itemRef
                }
            ],
            [
                $113aa0613e727c6c$export$c7e742effb1c51e2,
                {
                    ref: submenuRef,
                    ...submenuProps
                }
            ],
            [
                (0, $acf8e70c2f419f18$export$d2f961adcb0afbe),
                submenuTriggerState
            ],
            [
                (0, $03df2f2d3cffb62f$export$9b9a0cd73afb7ca4),
                {
                    trigger: 'SubmenuTrigger',
                    triggerRef: itemRef,
                    placement: 'end top',
                    'aria-labelledby': submenuProps['aria-labelledby'],
                    ...popoverProps
                }
            ]
        ]
    }, /*#__PURE__*/ (0, $gKmcL$react).createElement(CollectionBranch, {
        collection: state.collection,
        parent: item
    }), props.children[1]);
}, (props)=>props.children[0]);
const $113aa0613e727c6c$export$d9b273488cd8ce6f = /*#__PURE__*/ (0, $gKmcL$forwardRef)(function Menu(props, ref) {
    [props, ref] = (0, $b7b7a92703138c9b$export$29f1550f4b0d4415)(props, ref, $113aa0613e727c6c$export$c7e742effb1c51e2);
    // Delay rendering the actual menu until we have the collection so that auto focus works properly.
    return /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $gKmcL$CollectionBuilder), {
        content: /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $gKmcL$Collection), props)
    }, (collection)=>/*#__PURE__*/ (0, $gKmcL$react).createElement($113aa0613e727c6c$var$MenuInner, {
            props: props,
            collection: collection,
            menuRef: ref
        }));
});
function $113aa0613e727c6c$var$MenuInner({ props: props, collection: collection, menuRef: ref }) {
    [props, ref] = (0, $b7b7a92703138c9b$export$29f1550f4b0d4415)(props, ref, (0, $8f09b710ef85b337$export$b0d3ecf7112093a7));
    let { filter: filter, ...autocompleteMenuProps } = props;
    let filteredCollection = (0, $gKmcL$useMemo)(()=>filter ? collection.filter(filter) : collection, [
        collection,
        filter
    ]);
    let state = (0, $gKmcL$useTreeState)({
        ...props,
        collection: filteredCollection,
        children: undefined
    });
    let triggerState = (0, $gKmcL$useContext)($113aa0613e727c6c$export$795aec4671cbae19);
    let { isVirtualized: isVirtualized, CollectionRoot: CollectionRoot } = (0, $gKmcL$useContext)((0, $a53f0f6636929daa$export$4feb769f8ddf26c5));
    let { menuProps: menuProps } = (0, $gKmcL$useMenu)({
        ...props,
        isVirtualized: isVirtualized,
        onClose: props.onClose || (triggerState === null || triggerState === void 0 ? void 0 : triggerState.close)
    }, state, ref);
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...props,
        children: undefined,
        defaultClassName: 'react-aria-Menu',
        values: {
            isEmpty: state.collection.size === 0
        }
    });
    let emptyState = null;
    if (state.collection.size === 0 && props.renderEmptyState) emptyState = /*#__PURE__*/ (0, $gKmcL$react).createElement("div", {
        role: "menuitem",
        style: {
            display: 'contents'
        }
    }, props.renderEmptyState());
    let DOMProps = (0, $gKmcL$filterDOMProps)(props, {
        global: true
    });
    return /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $gKmcL$FocusScope), null, /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $b7b7a92703138c9b$export$df3a06d6289f983e).div, {
        ...(0, $gKmcL$mergeProps)(DOMProps, renderProps, menuProps),
        ref: ref,
        slot: props.slot || undefined,
        "data-empty": state.collection.size === 0 || undefined,
        onScroll: props.onScroll
    }, /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $b7b7a92703138c9b$export$2881499e37b75b9a), {
        values: [
            [
                $113aa0613e727c6c$export$24aad8519b95b41b,
                state
            ],
            [
                (0, $469f9b725520971a$export$6615d83f6de245ce),
                {
                    elementType: 'div'
                }
            ],
            [
                (0, $a53f0f6636929daa$export$d40e14dec8b060a8),
                {
                    name: 'MenuSection',
                    render: $113aa0613e727c6c$var$MenuSectionInner
                }
            ],
            [
                $113aa0613e727c6c$var$SubmenuTriggerContext,
                {
                    parentMenuRef: ref,
                    shouldUseVirtualFocus: autocompleteMenuProps === null || autocompleteMenuProps === void 0 ? void 0 : autocompleteMenuProps.shouldUseVirtualFocus
                }
            ],
            [
                $113aa0613e727c6c$var$MenuItemContext,
                {
                    shouldCloseOnSelect: props.shouldCloseOnSelect
                }
            ],
            [
                (0, $8f09b710ef85b337$export$b0d3ecf7112093a7),
                null
            ],
            [
                (0, $8f09b710ef85b337$export$698f465ec27e93df),
                null
            ],
            [
                $113aa0613e727c6c$var$SelectionManagerContext,
                state.selectionManager
            ],
            /* Ensure root MenuTriggerState is defined, in case Menu is rendered outside a MenuTrigger. */ /* We assume the context can never change between defined and undefined. */ /* eslint-disable-next-line react-hooks/rules-of-hooks */ [
                $113aa0613e727c6c$export$795aec4671cbae19,
                triggerState !== null && triggerState !== void 0 ? triggerState : (0, $gKmcL$useMenuTriggerState)({})
            ]
        ]
    }, /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $347bc273c4058e94$export$758399f318e6385a), null, /*#__PURE__*/ (0, $gKmcL$react).createElement(CollectionRoot, {
        collection: state.collection,
        persistedKeys: (0, $a53f0f6636929daa$export$90e00781bc59d8f9)(state.selectionManager.focusedKey),
        scrollRef: ref
    }))), emptyState));
}
// A subclass of SelectionManager that forwards focus-related properties to the parent,
// but has its own local selection state.
class $113aa0613e727c6c$var$GroupSelectionManager extends (0, $gKmcL$SelectionManager) {
    get focusedKey() {
        return this.parent.focusedKey;
    }
    get isFocused() {
        return this.parent.isFocused;
    }
    setFocusedKey(key, childFocusStrategy) {
        return this.parent.setFocusedKey(key, childFocusStrategy);
    }
    setFocused(isFocused) {
        this.parent.setFocused(isFocused);
    }
    get childFocusStrategy() {
        return this.parent.childFocusStrategy;
    }
    constructor(parent, state){
        super(parent.collection, state);
        this.parent = parent;
    }
}
function $113aa0613e727c6c$var$MenuSectionInner(props, ref, section, className = 'react-aria-MenuSection') {
    var _section_props, _section_props1, _useSlottedContext;
    let state = (0, $gKmcL$useContext)($113aa0613e727c6c$export$24aad8519b95b41b);
    let { CollectionBranch: CollectionBranch } = (0, $gKmcL$useContext)((0, $a53f0f6636929daa$export$4feb769f8ddf26c5));
    let [headingRef, heading] = (0, $b7b7a92703138c9b$export$9d4c57ee4c6ffdd8)();
    var _section_props_arialabel;
    let { headingProps: headingProps, groupProps: groupProps } = (0, $gKmcL$useMenuSection)({
        heading: heading,
        'aria-label': (_section_props_arialabel = section.props['aria-label']) !== null && _section_props_arialabel !== void 0 ? _section_props_arialabel : undefined
    });
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...props,
        id: undefined,
        children: undefined,
        defaultClassName: className,
        className: (_section_props = section.props) === null || _section_props === void 0 ? void 0 : _section_props.className,
        style: (_section_props1 = section.props) === null || _section_props1 === void 0 ? void 0 : _section_props1.style,
        values: undefined
    });
    let parent = (0, $gKmcL$useContext)($113aa0613e727c6c$var$SelectionManagerContext);
    let selectionState = (0, $gKmcL$useMultipleSelectionState)(props);
    let manager = props.selectionMode != null ? new $113aa0613e727c6c$var$GroupSelectionManager(parent, selectionState) : parent;
    let closeOnSelect = (_useSlottedContext = (0, $b7b7a92703138c9b$export$fabf2dc03a41866e)($113aa0613e727c6c$var$MenuItemContext)) === null || _useSlottedContext === void 0 ? void 0 : _useSlottedContext.shouldCloseOnSelect;
    let DOMProps = (0, $gKmcL$filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    var _props_shouldCloseOnSelect;
    return /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $b7b7a92703138c9b$export$df3a06d6289f983e).section, {
        ...(0, $gKmcL$mergeProps)(DOMProps, renderProps, groupProps),
        ref: ref
    }, /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $b7b7a92703138c9b$export$2881499e37b75b9a), {
        values: [
            [
                (0, $68ee918b64e9e759$export$e0e4026c12a8bdbb),
                {
                    ...headingProps,
                    ref: headingRef
                }
            ],
            [
                $113aa0613e727c6c$var$SelectionManagerContext,
                manager
            ],
            [
                $113aa0613e727c6c$var$MenuItemContext,
                {
                    shouldCloseOnSelect: (_props_shouldCloseOnSelect = props.shouldCloseOnSelect) !== null && _props_shouldCloseOnSelect !== void 0 ? _props_shouldCloseOnSelect : closeOnSelect
                }
            ]
        ]
    }, /*#__PURE__*/ (0, $gKmcL$react).createElement(CollectionBranch, {
        collection: state.collection,
        parent: section
    })));
}
const $113aa0613e727c6c$export$4b1545b4f2016d26 = /*#__PURE__*/ (0, $gKmcL$createBranchComponent)((0, $gKmcL$SectionNode), $113aa0613e727c6c$var$MenuSectionInner);
const $113aa0613e727c6c$var$MenuItemContext = /*#__PURE__*/ (0, $gKmcL$createContext)(null);
const $113aa0613e727c6c$export$2ce376c2cc3355c8 = /*#__PURE__*/ (0, $gKmcL$createLeafComponent)((0, $gKmcL$ItemNode), function MenuItem(props, forwardedRef, item) {
    var _useSlottedContext;
    [props, forwardedRef] = (0, $b7b7a92703138c9b$export$29f1550f4b0d4415)(props, forwardedRef, $113aa0613e727c6c$var$MenuItemContext);
    let id = (_useSlottedContext = (0, $b7b7a92703138c9b$export$fabf2dc03a41866e)($113aa0613e727c6c$var$MenuItemContext)) === null || _useSlottedContext === void 0 ? void 0 : _useSlottedContext.id;
    let state = (0, $gKmcL$useContext)($113aa0613e727c6c$export$24aad8519b95b41b);
    let ref = (0, $gKmcL$useObjectRef)(forwardedRef);
    let selectionManager = (0, $gKmcL$useContext)($113aa0613e727c6c$var$SelectionManagerContext);
    let { isVirtualized: isVirtualized } = (0, $gKmcL$useContext)((0, $a53f0f6636929daa$export$4feb769f8ddf26c5));
    let { menuItemProps: menuItemProps, labelProps: labelProps, descriptionProps: descriptionProps, keyboardShortcutProps: keyboardShortcutProps, ...states } = (0, $gKmcL$useMenuItem)({
        ...props,
        id: id,
        key: item.key,
        selectionManager: selectionManager,
        isVirtualized: isVirtualized
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gKmcL$useHover)({
        isDisabled: states.isDisabled
    });
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...props,
        id: undefined,
        children: item.rendered,
        defaultClassName: 'react-aria-MenuItem',
        values: {
            ...states,
            isHovered: isHovered,
            isFocusVisible: states.isFocusVisible,
            selectionMode: selectionManager.selectionMode,
            selectionBehavior: selectionManager.selectionBehavior,
            hasSubmenu: !!props['aria-haspopup'],
            isOpen: props['aria-expanded'] === 'true'
        }
    });
    let ElementType = props.href ? (0, $b7b7a92703138c9b$export$df3a06d6289f983e).a : (0, $b7b7a92703138c9b$export$df3a06d6289f983e).div;
    let DOMProps = (0, $gKmcL$filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    delete DOMProps.onClick;
    return /*#__PURE__*/ (0, $gKmcL$react).createElement(ElementType, {
        ...(0, $gKmcL$mergeProps)(DOMProps, renderProps, menuItemProps, hoverProps),
        ref: ref,
        "data-disabled": states.isDisabled || undefined,
        "data-hovered": isHovered || undefined,
        "data-focused": states.isFocused || undefined,
        "data-focus-visible": states.isFocusVisible || undefined,
        "data-pressed": states.isPressed || undefined,
        "data-selected": states.isSelected || undefined,
        "data-selection-mode": selectionManager.selectionMode === 'none' ? undefined : selectionManager.selectionMode,
        "data-has-submenu": !!props['aria-haspopup'] || undefined,
        "data-open": props['aria-expanded'] === 'true' || undefined
    }, /*#__PURE__*/ (0, $gKmcL$react).createElement((0, $b7b7a92703138c9b$export$2881499e37b75b9a), {
        values: [
            [
                (0, $20d769b1e2b13352$export$9afb8bc826b033ea),
                {
                    slots: {
                        [(0, $b7b7a92703138c9b$export$c62b8e45d58ddad9)]: labelProps,
                        label: labelProps,
                        description: descriptionProps
                    }
                }
            ],
            [
                (0, $ba03015127ce0a12$export$744d98a3b8a94e1c),
                keyboardShortcutProps
            ],
            [
                (0, $0d6f83ad40839938$export$c9549807523555e0),
                {
                    isSelected: states.isSelected
                }
            ]
        ]
    }, renderProps.children));
});


export {$113aa0613e727c6c$export$c7e742effb1c51e2 as MenuContext, $113aa0613e727c6c$export$24aad8519b95b41b as MenuStateContext, $113aa0613e727c6c$export$795aec4671cbae19 as RootMenuTriggerStateContext, $113aa0613e727c6c$export$27d2ad3c5815583e as MenuTrigger, $113aa0613e727c6c$export$ecabc99eeffab7ca as SubmenuTrigger, $113aa0613e727c6c$export$d9b273488cd8ce6f as Menu, $113aa0613e727c6c$export$4b1545b4f2016d26 as MenuSection, $113aa0613e727c6c$export$2ce376c2cc3355c8 as MenuItem};
