// react-aria-components/dist/private/Meter.cjs
var $048d76b84370f141$exports = require("./utils.cjs");
var $d5d46822336ca1e1$exports = require("./Label.cjs");
var $9PgQQ$reactariauseMeter = require("react-aria/useMeter");
var $9PgQQ$reactstatelyprivateutilsnumber = require("react-stately/private/utils/number");
var $9PgQQ$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $9PgQQ$reactariamergeProps = require("react-aria/mergeProps");
var $9PgQQ$react = require("react");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "MeterContext", function () { return $54e83bc7920648be$export$8b645da15a96b44f; });
$parcel$export(module.exports, "Meter", function () { return $54e83bc7920648be$export$62e3ae2a4090b879; });
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






const $54e83bc7920648be$export$8b645da15a96b44f = /*#__PURE__*/ (0, $9PgQQ$react.createContext)(null);
const $54e83bc7920648be$export$62e3ae2a4090b879 = /*#__PURE__*/ (0, $9PgQQ$react.forwardRef)(function Meter(props, ref) {
    [props, ref] = (0, $048d76b84370f141$exports.useContextProps)(props, ref, $54e83bc7920648be$export$8b645da15a96b44f);
    let { value: value = 0, minValue: minValue = 0, maxValue: maxValue = 100 } = props;
    value = (0, $9PgQQ$reactstatelyprivateutilsnumber.clamp)(value, minValue, maxValue);
    let [labelRef, label] = (0, $048d76b84370f141$exports.useSlot)(!props['aria-label'] && !props['aria-labelledby']);
    let { meterProps: meterProps, labelProps: labelProps } = (0, $9PgQQ$reactariauseMeter.useMeter)({
        ...props,
        label: label
    });
    // Calculate the width of the progress bar as a percentage
    let percentage = (value - minValue) / (maxValue - minValue) * 100;
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...props,
        defaultClassName: 'react-aria-Meter',
        values: {
            percentage: percentage,
            valueText: meterProps['aria-valuetext']
        }
    });
    let DOMProps = (0, $9PgQQ$reactariafilterDOMProps.filterDOMProps)(props, {
        global: true
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9PgQQ$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...(0, $9PgQQ$reactariamergeProps.mergeProps)(DOMProps, renderProps, meterProps),
        ref: ref,
        slot: props.slot || undefined
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9PgQQ$react))).createElement((0, $d5d46822336ca1e1$exports.LabelContext).Provider, {
        value: {
            ...labelProps,
            ref: labelRef,
            elementType: 'span'
        }
    }, renderProps.children));
});
