// react-aria-components/dist/private/Meter.js
import {dom as $b7b7a92703138c9b$export$df3a06d6289f983e, useContextProps as $b7b7a92703138c9b$export$29f1550f4b0d4415, useRenderProps as $b7b7a92703138c9b$export$4d86445c2cf5e3, useSlot as $b7b7a92703138c9b$export$9d4c57ee4c6ffdd8} from "./utils.js";
import {LabelContext as $3e4839e5b30e7b17$export$75b6ee27786ba447} from "./Label.js";
import {useMeter as $kCMGV$useMeter} from "react-aria/useMeter";
import {clamp as $kCMGV$clamp} from "react-stately/private/utils/number";
import {filterDOMProps as $kCMGV$filterDOMProps} from "react-aria/filterDOMProps";
import {mergeProps as $kCMGV$mergeProps} from "react-aria/mergeProps";
import $kCMGV$react, {createContext as $kCMGV$createContext, forwardRef as $kCMGV$forwardRef} from "react";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






const $4dcc8ca510f2b96b$export$8b645da15a96b44f = /*#__PURE__*/ (0, $kCMGV$createContext)(null);
const $4dcc8ca510f2b96b$export$62e3ae2a4090b879 = /*#__PURE__*/ (0, $kCMGV$forwardRef)(function Meter(props, ref) {
    [props, ref] = (0, $b7b7a92703138c9b$export$29f1550f4b0d4415)(props, ref, $4dcc8ca510f2b96b$export$8b645da15a96b44f);
    let { value: value = 0, minValue: minValue = 0, maxValue: maxValue = 100 } = props;
    value = (0, $kCMGV$clamp)(value, minValue, maxValue);
    let [labelRef, label] = (0, $b7b7a92703138c9b$export$9d4c57ee4c6ffdd8)(!props['aria-label'] && !props['aria-labelledby']);
    let { meterProps: meterProps, labelProps: labelProps } = (0, $kCMGV$useMeter)({
        ...props,
        label: label
    });
    // Calculate the width of the progress bar as a percentage
    let percentage = (value - minValue) / (maxValue - minValue) * 100;
    let renderProps = (0, $b7b7a92703138c9b$export$4d86445c2cf5e3)({
        ...props,
        defaultClassName: 'react-aria-Meter',
        values: {
            percentage: percentage,
            valueText: meterProps['aria-valuetext']
        }
    });
    let DOMProps = (0, $kCMGV$filterDOMProps)(props, {
        global: true
    });
    return /*#__PURE__*/ (0, $kCMGV$react).createElement((0, $b7b7a92703138c9b$export$df3a06d6289f983e).div, {
        ...(0, $kCMGV$mergeProps)(DOMProps, renderProps, meterProps),
        ref: ref,
        slot: props.slot || undefined
    }, /*#__PURE__*/ (0, $kCMGV$react).createElement((0, $3e4839e5b30e7b17$export$75b6ee27786ba447).Provider, {
        value: {
            ...labelProps,
            ref: labelRef,
            elementType: 'span'
        }
    }, renderProps.children));
});


export {$4dcc8ca510f2b96b$export$8b645da15a96b44f as MeterContext, $4dcc8ca510f2b96b$export$62e3ae2a4090b879 as Meter};
