// react-aria-components/dist/private/Tree.cjs
var $16c7f9b22cce3838$exports = require("./Button.cjs");
var $365d89633c2041bc$exports = require("./Checkbox.cjs");
var $048d76b84370f141$exports = require("./utils.cjs");
var $f7b82bedbb70abac$exports = require("./Collection.cjs");
var $d3d8871226fc64f2$exports = require("./DragAndDrop.cjs");
var $baa7ef94f966d95f$exports = require("./GridList.cjs");
var $61557b2a9b2862a8$exports = require("./SelectionIndicator.cjs");
var $9a60bd90621ebc78$exports = require("./SharedElementTransition.cjs");
var $56f56e0916461149$exports = require("./TreeDropTargetDelegate.cjs");
var $av2Q0$reactariauseTree = require("react-aria/useTree");
var $av2Q0$reactariaprivatecollectionsBaseCollection = require("react-aria/private/collections/BaseCollection");
var $av2Q0$reactariaCollection = require("react-aria/Collection");
var $av2Q0$reactariaCollectionBuilder = require("react-aria/CollectionBuilder");
var $av2Q0$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $av2Q0$reactariaFocusScope = require("react-aria/FocusScope");
var $av2Q0$reactariaprivateutilsinertValue = require("react-aria/private/utils/inertValue");
var $av2Q0$reactariaListKeyboardDelegate = require("react-aria/ListKeyboardDelegate");
var $av2Q0$reactariaprivateutilsuseLoadMoreSentinel = require("react-aria/private/utils/useLoadMoreSentinel");
var $av2Q0$reactariamergeProps = require("react-aria/mergeProps");
var $av2Q0$react = require("react");
var $av2Q0$reactstatelyuseTreeState = require("react-stately/useTreeState");
var $av2Q0$reactariaprivatecollectionsuseCachedChildren = require("react-aria/private/collections/useCachedChildren");
var $av2Q0$reactariauseCollator = require("react-aria/useCollator");
var $av2Q0$reactstatelyuseControlledState = require("react-stately/useControlledState");
var $av2Q0$reactariauseFocusRing = require("react-aria/useFocusRing");
var $av2Q0$reactariauseGridList = require("react-aria/useGridList");
var $av2Q0$reactariauseHover = require("react-aria/useHover");
var $av2Q0$reactariauseId = require("react-aria/useId");
var $av2Q0$reactariaI18nProvider = require("react-aria/I18nProvider");
var $av2Q0$reactariauseObjectRef = require("react-aria/useObjectRef");
var $av2Q0$reactariaVisuallyHidden = require("react-aria/VisuallyHidden");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "TreeContext", function () { return $8b1888e80597ea2d$export$dfae7d399eea2568; });
$parcel$export(module.exports, "TreeStateContext", function () { return $8b1888e80597ea2d$export$8953bccafd7bce87; });
$parcel$export(module.exports, "Tree", function () { return $8b1888e80597ea2d$export$7fbedc92909ed28e; });
$parcel$export(module.exports, "TreeItemContent", function () { return $8b1888e80597ea2d$export$4b687e3f663d618c; });
$parcel$export(module.exports, "TreeItem", function () { return $8b1888e80597ea2d$export$53d36ab85dc89436; });
$parcel$export(module.exports, "TreeLoadMoreItem", function () { return $8b1888e80597ea2d$export$533df5f8efd48cc9; });
$parcel$export(module.exports, "TreeSection", function () { return $8b1888e80597ea2d$export$1cd40fa16d4f033a; });
$parcel$export(module.exports, "TreeHeader", function () { return $8b1888e80597ea2d$export$2a73ea8963a8efd8; });
/*
 * Copyright 2024 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






























class $8b1888e80597ea2d$var$TreeCollection extends (0, $av2Q0$reactariaprivatecollectionsBaseCollection.BaseCollection) {
    withExpandedKeys(lastExpandedKeys, expandedKeys) {
        let collection = this.clone();
        collection.expandedKeys = expandedKeys;
        // Clone ancestor section nodes so React knows to re-render since the same item won't cause a new render but a clone creating a new object with the same value will
        // Without this change, the items won't expand and collapse when virtualized inside a section
        $8b1888e80597ea2d$var$TreeCollection.cloneAncestorSections(expandedKeys, lastExpandedKeys, collection);
        $8b1888e80597ea2d$var$TreeCollection.cloneAncestorSections(lastExpandedKeys, expandedKeys, collection);
        collection.frozen = this.frozen;
        return collection;
    }
    // diff lastExpandedKeys and expandedKeys so we only clone what has changed
    static cloneAncestorSections(keys, excludeSet, collection) {
        for (let key of keys)if (!excludeSet.has(key)) {
            let currentKey = key;
            while(currentKey != null){
                let item = collection.getItem(currentKey);
                if (item?.type === 'section') {
                    collection.keyMap.set(currentKey, item.clone());
                    break;
                } else currentKey = item?.parentKey ?? null;
            }
        }
    }
    *[Symbol.iterator]() {
        let firstKey = this.getFirstKey();
        let node = firstKey != null ? this.getItem(firstKey) : null;
        while(node){
            yield node;
            if (node.type === 'section') node = node.nextKey != null ? this.getItem(node.nextKey) : null;
            else {
                // This will include both item and content nodes
                // We handle the content nodes in useCollectionRenderer and ListLayout
                let key = this.getKeyAfter(node.key);
                node = key != null ? this.getItem(key) : null;
            }
        }
    }
    getLastKey() {
        // Find the deepest expanded child. We don't use collection.getLastKey() here
        // because that will return the deepest child regardless of expandedKeys.
        // Instead, start from the last top-level key and walk down.
        let key = this.lastKey;
        if (key == null) return null;
        let node = this.getItem(key);
        while(node?.lastChildKey != null && (node.type !== 'item' || this.expandedKeys.has(node.key)))node = this.getItem(node.lastChildKey);
        return node?.key;
    }
    getKeyAfter(key) {
        let node = this.getItem(key);
        if (!node) return null;
        if ((this.expandedKeys.has(node.key) || node.type !== 'item') && node.firstChildKey != null) return node.firstChildKey;
        while(node){
            if (node.nextKey != null) return node.nextKey;
            if (node.parentKey != null) node = this.getItem(node.parentKey);
            else return null;
        }
        return null;
    }
    getKeyBefore(key) {
        let node = this.getItem(key);
        if (!node) return null;
        if (node.prevKey != null) {
            node = this.getItem(node.prevKey);
            // If the lastChildKey is expanded, check its lastChildKey
            while(node && (node.type !== 'item' || this.expandedKeys.has(node.key)) && node.lastChildKey != null)node = this.getItem(node.lastChildKey);
            return node?.key ?? null;
        }
        return node.parentKey;
    }
    getChildren(key) {
        let self = this;
        return {
            *[Symbol.iterator] () {
                let parent = self.getItem(key);
                let node = parent?.firstChildKey != null ? self.getItem(parent.firstChildKey) : null;
                if (parent && parent.type === 'section' && node) // Stop once either the node is null or the node is the parent's sibling
                while(node && node.key !== parent.nextKey){
                    yield self.getItem(node.key);
                    // This will include content nodes which we skip in ListLayout
                    let key = self.getKeyAfter(node.key);
                    node = key != null ? self.getItem(key) : null;
                }
                else while(node){
                    yield node;
                    node = node.nextKey != null ? self.getItem(node.nextKey) : null;
                }
            }
        };
    }
    getTextValue(key) {
        let item = this.getItem(key);
        return item ? item.textValue : '';
    }
    constructor(...args){
        super(...args), this.expandedKeys = new Set();
    }
}
const $8b1888e80597ea2d$export$dfae7d399eea2568 = /*#__PURE__*/ (0, $av2Q0$react.createContext)(null);
const $8b1888e80597ea2d$export$8953bccafd7bce87 = /*#__PURE__*/ (0, $av2Q0$react.createContext)(null);
const $8b1888e80597ea2d$export$7fbedc92909ed28e = /*#__PURE__*/ (0, $av2Q0$react.forwardRef)(function Tree(props, ref) {
    // Render the portal first so that we have the collection by the time we render the DOM in SSR.
    [props, ref] = (0, $048d76b84370f141$exports.useContextProps)(props, ref, $8b1888e80597ea2d$export$dfae7d399eea2568);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $av2Q0$reactariaCollectionBuilder.CollectionBuilder), {
        content: /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $av2Q0$reactariaCollection.Collection), props),
        createCollection: ()=>new $8b1888e80597ea2d$var$TreeCollection()
    }, (collection)=>/*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement($8b1888e80597ea2d$var$TreeInner, {
            props: props,
            collection: collection,
            treeRef: ref
        }));
});
const $8b1888e80597ea2d$var$EXPANSION_KEYS = {
    expand: {
        ltr: 'ArrowRight',
        rtl: 'ArrowLeft'
    },
    collapse: {
        ltr: 'ArrowLeft',
        rtl: 'ArrowRight'
    }
};
function $8b1888e80597ea2d$var$TreeInner({ props: props, collection: collection, treeRef: ref }) {
    const { dragAndDropHooks: dragAndDropHooks } = props;
    let { direction: direction } = (0, $av2Q0$reactariaI18nProvider.useLocale)();
    let collator = (0, $av2Q0$reactariauseCollator.useCollator)({
        usage: 'search',
        sensitivity: 'base'
    });
    let hasDragHooks = !!dragAndDropHooks?.useDraggableCollectionState;
    let hasDropHooks = !!dragAndDropHooks?.useDroppableCollectionState;
    let dragHooksProvided = (0, $av2Q0$react.useRef)(hasDragHooks);
    let dropHooksProvided = (0, $av2Q0$react.useRef)(hasDropHooks);
    (0, $av2Q0$react.useEffect)(()=>{
        if (dragHooksProvided.current !== hasDragHooks) console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
        if (dropHooksProvided.current !== hasDropHooks) console.warn('Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
    }, [
        hasDragHooks,
        hasDropHooks
    ]);
    let { selectionMode: selectionMode = 'none', expandedKeys: propExpandedKeys, defaultExpandedKeys: propDefaultExpandedKeys, onExpandedChange: onExpandedChange, disabledBehavior: disabledBehavior = 'all' } = props;
    let { CollectionRoot: CollectionRoot, isVirtualized: isVirtualized, layoutDelegate: layoutDelegate, dropTargetDelegate: ctxDropTargetDelegate } = (0, $av2Q0$react.useContext)((0, $f7b82bedbb70abac$exports.CollectionRendererContext));
    // Kinda annoying that we have to replicate this code here as well as in useTreeState, but don't want to add
    // flattenCollection stuff to useTreeState. Think about this later
    let [expandedKeys, setExpandedKeys] = (0, $av2Q0$reactstatelyuseControlledState.useControlledState)(propExpandedKeys ? new Set(propExpandedKeys) : undefined, propDefaultExpandedKeys ? new Set(propDefaultExpandedKeys) : new Set(), onExpandedChange);
    let [lastCollection, setLastCollection] = (0, $av2Q0$react.useState)(collection);
    let [lastExpandedKeys, setLastExpandedKeys] = (0, $av2Q0$react.useState)(expandedKeys);
    let [flattenedCollection, setFlattenedCollection] = (0, $av2Q0$react.useState)(()=>collection.withExpandedKeys(lastExpandedKeys, expandedKeys));
    // if the lastExpandedKeys is not the same as the currentExpandedKeys or the collection has changed, then run this
    if (!$8b1888e80597ea2d$var$areSetsEqual(lastExpandedKeys, expandedKeys) || collection !== lastCollection) {
        setFlattenedCollection(collection.withExpandedKeys(lastExpandedKeys, expandedKeys));
        setLastCollection(collection);
        setLastExpandedKeys(expandedKeys);
    }
    let state = (0, $av2Q0$reactstatelyuseTreeState.useTreeState)({
        ...props,
        selectionMode: selectionMode,
        expandedKeys: expandedKeys,
        onExpandedChange: setExpandedKeys,
        collection: flattenedCollection,
        children: undefined,
        disabledBehavior: disabledBehavior
    });
    let { gridProps: gridProps } = (0, $av2Q0$reactariauseTree.useTree)({
        ...props,
        isVirtualized: isVirtualized,
        layoutDelegate: layoutDelegate
    }, state, ref);
    let dragState = undefined;
    let dropState = undefined;
    let droppableCollection = undefined;
    let isRootDropTarget = false;
    let dragPreview = null;
    let preview = (0, $av2Q0$react.useRef)(null);
    if (hasDragHooks && dragAndDropHooks) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: state.collection,
            selectionManager: state.selectionManager,
            preview: dragAndDropHooks.renderDragPreview ? preview : undefined
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, ref);
        let DragPreview = dragAndDropHooks.DragPreview;
        dragPreview = dragAndDropHooks.renderDragPreview ? /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement(DragPreview, {
            ref: preview
        }, dragAndDropHooks.renderDragPreview) : null;
    }
    let [treeDropTargetDelegate] = (0, $av2Q0$react.useState)(()=>new (0, $56f56e0916461149$exports.TreeDropTargetDelegate)());
    if (hasDropHooks && dragAndDropHooks) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: state.collection,
            selectionManager: state.selectionManager
        });
        let dropTargetDelegate = dragAndDropHooks.dropTargetDelegate || ctxDropTargetDelegate || new dragAndDropHooks.ListDropTargetDelegate(state.collection, ref, {
            direction: direction
        });
        treeDropTargetDelegate.setup(dropTargetDelegate, state, direction);
        let keyboardDelegate = new (0, $av2Q0$reactariaListKeyboardDelegate.ListKeyboardDelegate)({
            collection: state.collection,
            collator: collator,
            ref: ref,
            disabledKeys: state.selectionManager.disabledKeys,
            disabledBehavior: state.selectionManager.disabledBehavior,
            direction: direction,
            layoutDelegate: layoutDelegate
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: keyboardDelegate,
            dropTargetDelegate: treeDropTargetDelegate,
            onDropActivate: (e)=>{
                // Expand collapsed item when dragging over. For keyboard, allow collapsing.
                if (e.target.type === 'item') {
                    let key = e.target.key;
                    let item = state.collection.getItem(key);
                    let isExpanded = expandedKeys.has(key);
                    if (item && item.hasChildNodes && (!isExpanded || dragAndDropHooks?.isVirtualDragging?.())) state.toggleKey(key);
                }
            },
            onKeyDown: (e)=>{
                let target = dropState?.target;
                if (target && target.type === 'item' && target.dropPosition === 'on') {
                    let item = state.collection.getItem(target.key);
                    if (e.key === $8b1888e80597ea2d$var$EXPANSION_KEYS['expand'][direction] && item?.hasChildNodes && !state.expandedKeys.has(target.key)) state.toggleKey(target.key);
                    else if (e.key === $8b1888e80597ea2d$var$EXPANSION_KEYS['collapse'][direction] && item?.hasChildNodes && state.expandedKeys.has(target.key)) state.toggleKey(target.key);
                }
            }
        }, dropState, ref);
        isRootDropTarget = dropState.isDropTarget({
            type: 'root'
        });
    }
    let isTreeDraggable = !!(hasDragHooks && !dragState?.isDisabled);
    let { focusProps: focusProps, isFocused: isFocused, isFocusVisible: isFocusVisible } = (0, $av2Q0$reactariauseFocusRing.useFocusRing)();
    let renderValues = {
        isEmpty: state.collection.size === 0,
        isFocused: isFocused,
        isFocusVisible: isFocusVisible,
        isDropTarget: isRootDropTarget,
        selectionMode: state.selectionManager.selectionMode,
        allowsDragging: !!isTreeDraggable,
        state: state
    };
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...props,
        children: undefined,
        defaultClassName: 'react-aria-Tree',
        values: renderValues
    });
    let emptyState = null;
    if (state.collection.size === 0 && props.renderEmptyState) {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        let { isEmpty: isEmpty, ...values } = renderValues;
        let content = props.renderEmptyState({
            ...values
        });
        let treeGridRowProps = {
            'aria-level': 1
        };
        emptyState = /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
            role: "row",
            style: {
                display: 'contents'
            },
            ...treeGridRowProps
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
            role: "gridcell",
            style: {
                display: 'contents'
            }
        }, content));
    }
    let DOMProps = (0, $av2Q0$reactariafilterDOMProps.filterDOMProps)(props, {
        global: true
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, ($parcel$interopDefault($av2Q0$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $av2Q0$reactariaFocusScope.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...(0, $av2Q0$reactariamergeProps.mergeProps)(DOMProps, renderProps, gridProps, focusProps, droppableCollection?.collectionProps),
        ref: ref,
        slot: props.slot || undefined,
        "data-empty": state.collection.size === 0 || undefined,
        "data-focused": isFocused || undefined,
        "data-drop-target": isRootDropTarget || undefined,
        "data-focus-visible": isFocusVisible || undefined,
        "data-selection-mode": state.selectionManager.selectionMode === 'none' ? undefined : state.selectionManager.selectionMode,
        "data-allows-dragging": !!isTreeDraggable || undefined
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.Provider), {
        values: [
            [
                $8b1888e80597ea2d$export$8953bccafd7bce87,
                state
            ],
            [
                (0, $d3d8871226fc64f2$exports.DragAndDropContext),
                {
                    dragAndDropHooks: dragAndDropHooks,
                    dragState: dragState,
                    dropState: dropState
                }
            ],
            [
                (0, $d3d8871226fc64f2$exports.DropIndicatorContext),
                {
                    render: $8b1888e80597ea2d$var$TreeDropIndicatorWrapper
                }
            ]
        ]
    }, hasDropHooks && /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement($8b1888e80597ea2d$var$RootDropIndicator, null), /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $9a60bd90621ebc78$exports.SharedElementTransition), null, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement(CollectionRoot, {
        collection: state.collection,
        persistedKeys: (0, $d3d8871226fc64f2$exports.useDndPersistedKeys)(state.selectionManager, dragAndDropHooks, dropState),
        scrollRef: ref,
        renderDropIndicator: (0, $d3d8871226fc64f2$exports.useRenderDropIndicator)(dragAndDropHooks, dropState)
    }))), emptyState)), dragPreview);
}
class $8b1888e80597ea2d$var$TreeContentNode extends (0, $av2Q0$reactariaprivatecollectionsBaseCollection.CollectionNode) {
    static{
        this.type = 'content';
    }
}
const $8b1888e80597ea2d$export$4b687e3f663d618c = /*#__PURE__*/ (0, $av2Q0$reactariaCollectionBuilder.createLeafComponent)($8b1888e80597ea2d$var$TreeContentNode, function TreeItemContent(props) {
    let values = (0, $av2Q0$react.useContext)($8b1888e80597ea2d$export$36b5dda0d9bc8f78);
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        children: props.children,
        values: values
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $f7b82bedbb70abac$exports.CollectionRendererContext).Provider, {
        value: (0, $f7b82bedbb70abac$exports.DefaultCollectionRenderer)
    }, renderProps.children);
});
const $8b1888e80597ea2d$export$36b5dda0d9bc8f78 = /*#__PURE__*/ (0, $av2Q0$react.createContext)(null);
class $8b1888e80597ea2d$var$TreeItemNode extends (0, $av2Q0$reactariaprivatecollectionsBaseCollection.CollectionNode) {
    static{
        this.type = 'item';
    }
}
const $8b1888e80597ea2d$export$53d36ab85dc89436 = /*#__PURE__*/ (0, $av2Q0$reactariaCollectionBuilder.createBranchComponent)($8b1888e80597ea2d$var$TreeItemNode, (props, ref, item)=>{
    let state = (0, $av2Q0$react.useContext)($8b1888e80597ea2d$export$8953bccafd7bce87);
    ref = (0, $av2Q0$reactariauseObjectRef.useObjectRef)(ref);
    let { dragAndDropHooks: dragAndDropHooks, dragState: dragState, dropState: dropState } = (0, $av2Q0$react.useContext)((0, $d3d8871226fc64f2$exports.DragAndDropContext));
    let isDraggable = dragState && !(dragState.isDisabled || dragState.selectionManager.isDisabled(item.key));
    // TODO: remove this when we support description in tree row
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    let { rowProps: rowProps, gridCellProps: gridCellProps, expandButtonProps: expandButtonProps, descriptionProps: descriptionProps, ...states } = (0, $av2Q0$reactariauseTree.useTreeItem)({
        node: item,
        shouldSelectOnPressUp: !!dragState
    }, state, ref);
    let isExpanded = rowProps['aria-expanded'] === true;
    let hasChildItems = props.hasChildItems || [
        ...state.collection.getChildren(item.key)
    ]?.length > 1;
    let level = rowProps['aria-level'] || 1;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $av2Q0$reactariauseHover.useHover)({
        // because of https://bugs.webkit.org/show_bug.cgi?id=214609, supporting hover styles when a item is ONLY isDraggable
        // results in hover styles sticking around after a reorder/drop operation...
        isDisabled: !states.allowsSelection && !states.hasAction && !isDraggable,
        onHoverStart: props.onHoverStart,
        onHoverChange: props.onHoverChange,
        onHoverEnd: props.onHoverEnd
    });
    let { isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $av2Q0$reactariauseFocusRing.useFocusRing)();
    let { isFocusVisible: isFocusVisibleWithin, focusProps: focusWithinProps } = (0, $av2Q0$reactariauseFocusRing.useFocusRing)({
        within: true
    });
    let { checkboxProps: checkboxProps } = (0, $av2Q0$reactariauseGridList.useGridListSelectionCheckbox)({
        key: item.key
    }, state);
    let draggableItem = null;
    if (dragState && dragAndDropHooks) draggableItem = dragAndDropHooks.useDraggableItem({
        key: item.key,
        hasDragButton: true
    }, dragState);
    let dropIndicator = null;
    let expandButtonRef = (0, $av2Q0$react.useRef)(null);
    let dropIndicatorRef = (0, $av2Q0$react.useRef)(null);
    let activateButtonRef = (0, $av2Q0$react.useRef)(null);
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $av2Q0$reactariaVisuallyHidden.useVisuallyHidden)();
    if (dropState && dragAndDropHooks) dropIndicator = dragAndDropHooks.useDropIndicator({
        target: {
            type: 'item',
            key: item.key,
            dropPosition: 'on'
        },
        activateButtonRef: activateButtonRef
    }, dropState, dropIndicatorRef);
    let isDragging = dragState && dragState.isDragging(item.key);
    let isDropTarget = dropIndicator?.isDropTarget;
    let selectionMode = state.selectionManager.selectionMode;
    let selectionBehavior = state.selectionManager.selectionBehavior;
    let renderPropValues = (0, ($parcel$interopDefault($av2Q0$react))).useMemo(()=>({
            ...states,
            isHovered: isHovered,
            isFocusVisible: isFocusVisible,
            isExpanded: isExpanded,
            hasChildItems: hasChildItems,
            level: level,
            selectionMode: selectionMode,
            selectionBehavior: selectionBehavior,
            isFocusVisibleWithin: isFocusVisibleWithin,
            state: state,
            id: item.key,
            allowsDragging: !!dragState,
            isDragging: isDragging,
            isDropTarget: isDropTarget
        }), [
        states,
        isHovered,
        isFocusVisible,
        isExpanded,
        hasChildItems,
        level,
        isFocusVisibleWithin,
        state,
        item.key,
        dragState,
        isDragging,
        isDropTarget,
        selectionBehavior,
        selectionMode
    ]);
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...props,
        id: undefined,
        children: item.rendered,
        defaultClassName: 'react-aria-TreeItem',
        defaultStyle: {
            // @ts-ignore
            '--tree-item-level': level
        },
        values: renderPropValues
    });
    (0, $av2Q0$react.useEffect)(()=>{
        if (!item.textValue && process.env.NODE_ENV !== 'production') console.warn('A `textValue` prop is required for <TreeItem> elements in order to support accessibility features such as type to select.');
    }, [
        item.textValue
    ]);
    (0, $av2Q0$react.useEffect)(()=>{
        if (hasChildItems && !expandButtonRef.current && process.env.NODE_ENV !== 'production') console.warn('Expandable tree items must contain a expand button so screen reader users can expand/collapse the item.');
    // eslint-disable-next-line
    }, []);
    let dragButtonRef = (0, $av2Q0$react.useRef)(null);
    (0, $av2Q0$react.useEffect)(()=>{
        if (dragState && !dragButtonRef.current && process.env.NODE_ENV !== 'production') console.warn('Draggable items in a Tree must contain a <Button slot="drag"> element so that keyboard and screen reader users can drag them.');
    // eslint-disable-next-line
    }, []);
    let children = (0, $av2Q0$reactariaprivatecollectionsuseCachedChildren.useCachedChildren)({
        items: state.collection.getChildren(item.key),
        children: (item)=>{
            switch(item.type){
                case 'content':
                    return item.render(item);
                // Skip item since we don't render the nested rows as children of the parent row, the flattened collection
                // will render them each as siblings instead
                case 'loader':
                case 'item':
                    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, ($parcel$interopDefault($av2Q0$react))).Fragment, null);
                default:
                    throw new Error('Unsupported element type in TreeRow: ' + item.type);
            }
        }
    });
    let activateButtonId = (0, $av2Q0$reactariauseId.useId)();
    let DOMProps = (0, $av2Q0$reactariafilterDOMProps.filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    delete DOMProps.onClick;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, ($parcel$interopDefault($av2Q0$react))).Fragment, null, dropIndicator && !dropIndicator.isHidden && /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "row",
        "aria-level": rowProps['aria-level'],
        "aria-expanded": rowProps['aria-expanded'],
        "aria-label": dropIndicator.dropIndicatorProps['aria-label']
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "gridcell",
        "aria-colindex": 1,
        style: {
            display: 'contents'
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicator.dropIndicatorProps,
        ref: dropIndicatorRef
    }), rowProps['aria-expanded'] != null ? // Button to allow touch screen reader users to expand the item while dragging.
    /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        id: activateButtonId,
        "aria-label": expandButtonProps['aria-label'],
        "aria-labelledby": `${activateButtonId} ${rowProps.id}`,
        tabIndex: -1,
        ref: activateButtonRef
    }) : null)), /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...(0, $av2Q0$reactariamergeProps.mergeProps)(DOMProps, rowProps, focusProps, hoverProps, focusWithinProps, draggableItem?.dragProps),
        ...renderProps,
        ref: ref,
        // TODO: missing selectionBehavior, hasAction and allowsSelection data attribute equivalents (available in renderProps). Do we want those?
        "data-expanded": hasChildItems && isExpanded || undefined,
        "data-has-child-items": hasChildItems || undefined,
        "data-level": level,
        "data-selected": states.isSelected || undefined,
        "data-disabled": states.isDisabled || undefined,
        "data-hovered": isHovered || undefined,
        "data-focused": states.isFocused || undefined,
        "data-focus-visible": isFocusVisible || undefined,
        "data-pressed": states.isPressed || undefined,
        "data-selection-mode": state.selectionManager.selectionMode === 'none' ? undefined : state.selectionManager.selectionMode,
        "data-allows-dragging": !!dragState || undefined,
        "data-dragging": isDragging || undefined,
        "data-drop-target": isDropTarget || undefined
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        ...gridCellProps,
        style: {
            display: 'contents'
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.Provider), {
        values: [
            [
                (0, $365d89633c2041bc$exports.CheckboxContext),
                {
                    slots: {
                        selection: checkboxProps
                    }
                }
            ],
            [
                (0, $365d89633c2041bc$exports.CheckboxFieldContext),
                {
                    slots: {
                        selection: checkboxProps
                    }
                }
            ],
            // TODO: support description in the tree row
            // TODO: don't think I need to pass isExpanded to the button here since it can be sourced from the renderProps? Might be worthwhile passing it down?
            [
                (0, $16c7f9b22cce3838$exports.ButtonContext),
                {
                    slots: {
                        [(0, $048d76b84370f141$exports.DEFAULT_SLOT)]: {},
                        chevron: {
                            ...expandButtonProps,
                            ref: expandButtonRef
                        },
                        drag: {
                            ...draggableItem?.dragButtonProps,
                            ref: dragButtonRef,
                            style: {
                                pointerEvents: 'none'
                            }
                        }
                    }
                }
            ],
            [
                $8b1888e80597ea2d$export$36b5dda0d9bc8f78,
                {
                    ...renderPropValues
                }
            ],
            [
                (0, $61557b2a9b2862a8$exports.SelectionIndicatorContext),
                {
                    isSelected: states.isSelected
                }
            ]
        ]
    }, children))));
});
const $8b1888e80597ea2d$export$533df5f8efd48cc9 = (0, $av2Q0$reactariaCollectionBuilder.createLeafComponent)((0, $av2Q0$reactariaprivatecollectionsBaseCollection.LoaderNode), function TreeLoadingSentinel(props, ref, item) {
    let { isVirtualized: isVirtualized } = (0, $av2Q0$react.useContext)((0, $f7b82bedbb70abac$exports.CollectionRendererContext));
    let state = (0, $av2Q0$react.useContext)($8b1888e80597ea2d$export$8953bccafd7bce87);
    let { isLoading: isLoading, onLoadMore: onLoadMore, scrollOffset: scrollOffset, ...otherProps } = props;
    let sentinelRef = (0, $av2Q0$react.useRef)(null);
    let memoedLoadMoreProps = (0, $av2Q0$react.useMemo)(()=>({
            onLoadMore: onLoadMore,
            // this collection will update anytime a row is expanded/collapsed becaused the flattenedRows will change.
            // This means onLoadMore will trigger but that might be ok cause the user should have logic to handle multiple loadMore calls
            collection: state?.collection,
            sentinelRef: sentinelRef,
            scrollOffset: scrollOffset
        }), [
        onLoadMore,
        scrollOffset,
        state?.collection
    ]);
    (0, $av2Q0$reactariaprivateutilsuseLoadMoreSentinel.useLoadMoreSentinel)(memoedLoadMoreProps, sentinelRef);
    ref = (0, $av2Q0$reactariauseObjectRef.useObjectRef)(ref);
    let { rowProps: rowProps, gridCellProps: gridCellProps } = (0, $av2Q0$reactariauseTree.useTreeItem)({
        node: item
    }, state, ref);
    let level = rowProps['aria-level'] || 1;
    // For now don't include aria-posinset and aria-setsize on loader since they aren't keyboard focusable
    // Arguably shouldn't include them ever since it might be confusing to the user to include the loaders as part of the
    // item count
    let ariaProps = {
        role: 'row',
        'aria-level': rowProps['aria-level']
    };
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...otherProps,
        id: undefined,
        children: item.rendered,
        defaultClassName: 'react-aria-TreeLoader',
        values: {
            level: level
        }
    });
    let style = {};
    if (isVirtualized) style = {
        display: 'contents'
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, ($parcel$interopDefault($av2Q0$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        style: {
            position: 'relative',
            width: 0,
            height: 0
        },
        inert: (0, $av2Q0$reactariaprivateutilsinertValue.inertValue)(true)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        "data-testid": "loadMoreSentinel",
        ref: sentinelRef,
        style: {
            position: 'absolute',
            height: 1,
            width: 1
        }
    })), isLoading && renderProps.children && /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ref: ref,
        ...(0, $av2Q0$reactariamergeProps.mergeProps)((0, $av2Q0$reactariafilterDOMProps.filterDOMProps)(props), ariaProps),
        ...renderProps,
        "data-level": level
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        ...gridCellProps,
        style: style
    }, renderProps.children)));
});
function $8b1888e80597ea2d$var$TreeDropIndicatorWrapper(props, ref) {
    ref = (0, $av2Q0$reactariauseObjectRef.useObjectRef)(ref);
    let { dragAndDropHooks: dragAndDropHooks, dropState: dropState } = (0, $av2Q0$react.useContext)((0, $d3d8871226fc64f2$exports.DragAndDropContext));
    let buttonRef = (0, $av2Q0$react.useRef)(null);
    let { dropIndicatorProps: dropIndicatorProps, isHidden: isHidden, isDropTarget: isDropTarget } = dragAndDropHooks.useDropIndicator(props, dropState, buttonRef);
    if (isHidden) return null;
    let level = dropState && props.target.type === 'item' ? (dropState.collection.getItem(props.target.key)?.level || 0) + 1 : 1;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement($8b1888e80597ea2d$var$TreeDropIndicatorForwardRef, {
        ...props,
        dropIndicatorProps: dropIndicatorProps,
        isDropTarget: isDropTarget,
        ref: ref,
        buttonRef: buttonRef,
        level: level
    });
}
function $8b1888e80597ea2d$var$TreeDropIndicator(props, ref) {
    let { dropIndicatorProps: dropIndicatorProps, isDropTarget: isDropTarget, buttonRef: buttonRef, level: level, ...otherProps } = props;
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $av2Q0$reactariaVisuallyHidden.useVisuallyHidden)();
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...otherProps,
        defaultClassName: 'react-aria-DropIndicator',
        defaultStyle: {
            position: 'relative',
            // @ts-ignore
            '--tree-item-level': level
        },
        values: {
            isDropTarget: isDropTarget
        }
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...renderProps,
        role: "row",
        "aria-level": level,
        ref: ref,
        "data-drop-target": isDropTarget || undefined
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "gridcell"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        ...visuallyHiddenProps,
        role: "button",
        ...dropIndicatorProps,
        ref: buttonRef
    }), renderProps.children));
}
const $8b1888e80597ea2d$var$TreeDropIndicatorForwardRef = /*#__PURE__*/ (0, $av2Q0$react.forwardRef)($8b1888e80597ea2d$var$TreeDropIndicator);
function $8b1888e80597ea2d$var$RootDropIndicator() {
    let { dragAndDropHooks: dragAndDropHooks, dropState: dropState } = (0, $av2Q0$react.useContext)((0, $d3d8871226fc64f2$exports.DragAndDropContext));
    let ref = (0, $av2Q0$react.useRef)(null);
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator({
        target: {
            type: 'root'
        }
    }, dropState, ref);
    let isDropTarget = dropState.isDropTarget({
        type: 'root'
    });
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $av2Q0$reactariaVisuallyHidden.useVisuallyHidden)();
    if (!isDropTarget && dropIndicatorProps['aria-hidden']) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps['aria-hidden'],
        style: {
            position: 'absolute'
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "gridcell"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicatorProps,
        ref: ref
    })));
}
const $8b1888e80597ea2d$export$1cd40fa16d4f033a = /*#__PURE__*/ (0, $av2Q0$reactariaCollectionBuilder.createBranchComponent)((0, $av2Q0$reactariaprivatecollectionsBaseCollection.SectionNode), (props, ref, item)=>{
    let state = (0, $av2Q0$react.useContext)($8b1888e80597ea2d$export$8953bccafd7bce87);
    let { CollectionBranch: CollectionBranch } = (0, $av2Q0$react.useContext)((0, $f7b82bedbb70abac$exports.CollectionRendererContext));
    let headingRef = (0, $av2Q0$react.useRef)(null);
    ref = (0, $av2Q0$reactariauseObjectRef.useObjectRef)(ref);
    let { rowHeaderProps: rowHeaderProps, rowProps: rowProps, rowGroupProps: rowGroupProps } = (0, $av2Q0$reactariauseGridList.useGridListSection)({
        'aria-label': props['aria-label'] ?? undefined
    }, state, ref);
    let renderProps = (0, $048d76b84370f141$exports.useRenderProps)({
        ...props,
        id: undefined,
        children: undefined,
        defaultClassName: 'react-aria-TreeSection',
        values: undefined
    });
    let DOMProps = (0, $av2Q0$reactariafilterDOMProps.filterDOMProps)(props, {
        global: true
    });
    delete DOMProps.id;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.dom).div, {
        ...(0, $av2Q0$reactariamergeProps.mergeProps)(DOMProps, renderProps, rowGroupProps),
        ref: ref
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $048d76b84370f141$exports.Provider), {
        values: [
            [
                (0, $baa7ef94f966d95f$exports.GridListHeaderContext),
                {
                    ...rowProps,
                    ref: headingRef
                }
            ],
            [
                (0, $baa7ef94f966d95f$exports.GridListHeaderInnerContext),
                {
                    ...rowHeaderProps
                }
            ]
        ]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement(CollectionBranch, {
        collection: state.collection,
        parent: item
    })));
});
const $8b1888e80597ea2d$export$2a73ea8963a8efd8 = (props)=>{
    return /*#__PURE__*/ (0, ($parcel$interopDefault($av2Q0$react))).createElement((0, $baa7ef94f966d95f$exports.GridListHeader), {
        className: "react-aria-TreeHeader",
        ...props
    }, props.children);
};
function $8b1888e80597ea2d$var$areSetsEqual(a, b) {
    if (a.size !== b.size) return false;
    for (let item of a){
        if (!b.has(item)) return false;
    }
    return true;
}
