// @adobe/react-spectrum/dist/private/datepicker/DateRangePicker.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {Content as $558e2ad48297783c$export$7c6e2c02157bb7d2} from "../view/Content.js";
import {DatePickerField as $219f26d0f9cef050$export$34dc4cfa15ead1} from "./DatePickerField.js";
import "./styles.css";
import $4amrE$styles_cssmjs from "./styles_css.mjs";
import {Dialog as $89418a3659cad0c7$export$3ddf2d174ce01153} from "../dialog/Dialog.js";
import {DialogTrigger as $bcff05049955156f$export$2e1e1122cf0cba88} from "../dialog/DialogTrigger.js";
import {Field as $3967792f95357356$export$a455218a85c89869} from "../label/Field.js";
import {FieldButton as $1fa99bd0fd8b0a92$export$47dc48f595b075da} from "../button/FieldButton.js";
import {Flex as $9b6884c982c0954a$export$f51f4c4ede09e011} from "../layout/Flex.js";
import {Input as $f0b9f6972621ffb5$export$f5b8910cec6cf069} from "./Input.js";
import $4amrE$intlStringsjs from "./intlStrings.js";
import {RangeCalendar as $ebb070240f5bf202$export$a4f5c8b89d277a8d} from "../calendar/RangeCalendar.js";
import "../inputgroup_vars.css";
import $4amrE$inputgroup_vars_cssmjs from "../inputgroup_vars_css.mjs";
import {TimeField as $404615aff7fb8653$export$5eaee2322dd727eb} from "./TimeField.js";
import {useFocusManagerRef as $14b5acfdaf2344b2$export$71a23a36270e4bf0, useFormatHelpText as $14b5acfdaf2344b2$export$322f4580ccd8dde6, useFormattedDateWidth as $14b5acfdaf2344b2$export$31e22e3c931fc056, useVisibleMonths as $14b5acfdaf2344b2$export$12ce2869ce471b1f} from "./utils.js";
import {useFormProps as $d23ca6800ac02cf1$export$a6b5be5c6b451665} from "../form/Form.js";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useDateRangePicker as $4amrE$useDateRangePicker} from "react-aria/useDateRangePicker";
import $4amrE$spectrumiconsworkflowCalendar from "@spectrum-icons/workflow/Calendar";
import {mergeProps as $4amrE$mergeProps} from "react-aria/mergeProps";
import $4amrE$react, {useRef as $4amrE$useRef} from "react";
import {useDateRangePickerState as $4amrE$useDateRangePickerState} from "react-stately/useDateRangePickerState";
import {useFocusRing as $4amrE$useFocusRing} from "react-aria/useFocusRing";
import {useHover as $4amrE$useHover} from "react-aria/useHover";
import {useLocale as $4amrE$useLocale} from "react-aria/I18nProvider";
import {useLocalizedStringFormatter as $4amrE$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

























const $62a249cadcc61296$export$17334619f3ac2224 = /*#__PURE__*/ (0, $4amrE$react).forwardRef(function DateRangePicker(props, ref) {
    var _state_timeRange, _state_timeRange1;
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    props = (0, $d23ca6800ac02cf1$export$a6b5be5c6b451665)(props);
    let { isQuiet: isQuiet, isDisabled: isDisabled, autoFocus: autoFocus, placeholderValue: placeholderValue, maxVisibleMonths: maxVisibleMonths = 1 } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $4amrE$useHover)({
        isDisabled: isDisabled
    });
    let targetRef = (0, $4amrE$useRef)(null);
    let state = (0, $4amrE$useDateRangePickerState)({
        ...props,
        shouldCloseOnSelect: ()=>!state.hasTime
    });
    let { labelProps: labelProps, groupProps: groupProps, buttonProps: buttonProps, dialogProps: dialogProps, startFieldProps: startFieldProps, endFieldProps: endFieldProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, calendarProps: calendarProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $4amrE$useDateRangePicker)(props, state, targetRef);
    let { isOpen: isOpen, setOpen: setOpen } = state;
    let { direction: direction } = (0, $4amrE$useLocale)();
    let domRef = (0, $14b5acfdaf2344b2$export$71a23a36270e4bf0)(ref);
    let stringFormatter = (0, $4amrE$useLocalizedStringFormatter)((0, ($parcel$interopDefault($4amrE$intlStringsjs))), '@react-spectrum/datepicker');
    let { isFocused: isFocused, isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $4amrE$useFocusRing)({
        within: true,
        isTextInput: true,
        autoFocus: autoFocus
    });
    let { isFocused: isFocusedButton, focusProps: focusPropsButton } = (0, $4amrE$useFocusRing)({
        within: false,
        isTextInput: false,
        autoFocus: autoFocus
    });
    let className = (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$inputgroup_vars_cssmjs))), 'spectrum-InputGroup', {
        'spectrum-InputGroup--quiet': isQuiet,
        'spectrum-InputGroup--invalid': isInvalid && !isDisabled,
        'is-disabled': isDisabled,
        'is-hovered': isHovered,
        'is-focused': isFocused,
        'focus-ring': isFocusVisible && !isFocusedButton
    });
    let fieldClassName = (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-input', {
        'is-disabled': isDisabled,
        'is-invalid': isInvalid && !isDisabled
    });
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = (0, $14b5acfdaf2344b2$export$322f4580ccd8dde6)(props);
    if (description && !props.description) descriptionProps.id = undefined;
    let placeholder = placeholderValue;
    let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : null;
    let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : null;
    let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : null;
    let timeGranularity = state.granularity === 'hour' || state.granularity === 'minute' || state.granularity === 'second' ? state.granularity : null;
    let showTimeField = !!timeGranularity;
    let visibleMonths = (0, $14b5acfdaf2344b2$export$12ce2869ce471b1f)(maxVisibleMonths);
    let validationState = state.validationState || (isInvalid ? 'invalid' : null);
    // Multiplying by two for the two dates, adding one character for the dash, and then the padding around the dash
    let approximateWidth = `calc(${(0, $14b5acfdaf2344b2$export$31e22e3c931fc056)(state) * 2 + 1}ch + 2 * var(--spectrum-global-dimension-size-100))`;
    return /*#__PURE__*/ (0, $4amrE$react).createElement((0, $3967792f95357356$export$a455218a85c89869), {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: validationState,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        wrapperClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ (0, $4amrE$react).createElement("div", {
        ...(0, $4amrE$mergeProps)(groupProps, hoverProps, focusProps),
        className: className,
        ref: targetRef
    }, /*#__PURE__*/ (0, $4amrE$react).createElement("div", {
        style: {
            overflow: 'hidden',
            width: '100%'
        }
    }, /*#__PURE__*/ (0, $4amrE$react).createElement((0, $f0b9f6972621ffb5$export$f5b8910cec6cf069), {
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-field'),
        inputClassName: fieldClassName,
        disableFocusRing: true,
        minWidth: approximateWidth
    }, /*#__PURE__*/ (0, $4amrE$react).createElement((0, $219f26d0f9cef050$export$34dc4cfa15ead1), {
        ...startFieldProps,
        "data-testid": "start-date",
        isQuiet: props.isQuiet,
        inputClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-startField')
    }), /*#__PURE__*/ (0, $4amrE$react).createElement($62a249cadcc61296$var$DateRangeDash, null), /*#__PURE__*/ (0, $4amrE$react).createElement((0, $219f26d0f9cef050$export$34dc4cfa15ead1), {
        ...endFieldProps,
        "data-testid": "end-date",
        isQuiet: props.isQuiet,
        inputClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$inputgroup_vars_cssmjs))), 'spectrum-Datepicker-endField', (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-endField'))
    }))), /*#__PURE__*/ (0, $4amrE$react).createElement((0, $bcff05049955156f$export$2e1e1122cf0cba88), {
        type: "popover",
        mobileType: "tray",
        placement: direction === 'rtl' ? 'bottom right' : 'bottom left',
        targetRef: targetRef,
        hideArrow: true,
        isOpen: isOpen,
        onOpenChange: setOpen,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ (0, $4amrE$react).createElement((0, $1fa99bd0fd8b0a92$export$47dc48f595b075da), {
        ...(0, $4amrE$mergeProps)(buttonProps, focusPropsButton),
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$inputgroup_vars_cssmjs))), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: validationState
    }, /*#__PURE__*/ (0, $4amrE$react).createElement((0, $4amrE$spectrumiconsworkflowCalendar), null)), /*#__PURE__*/ (0, $4amrE$react).createElement((0, $89418a3659cad0c7$export$3ddf2d174ce01153), {
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-dialog'),
        ...dialogProps
    }, /*#__PURE__*/ (0, $4amrE$react).createElement((0, $558e2ad48297783c$export$7c6e2c02157bb7d2), null, /*#__PURE__*/ (0, $4amrE$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-dialogContent')
    }, /*#__PURE__*/ (0, $4amrE$react).createElement((0, $ebb070240f5bf202$export$a4f5c8b89d277a8d), {
        ...calendarProps,
        visibleMonths: visibleMonths,
        createCalendar: props.createCalendar,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-calendar', {
            'is-invalid': validationState === 'invalid'
        })
    }), showTimeField && /*#__PURE__*/ (0, $4amrE$react).createElement((0, $9b6884c982c0954a$export$f51f4c4ede09e011), {
        gap: "size-100",
        marginTop: "size-100",
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-timeFields')
    }, /*#__PURE__*/ (0, $4amrE$react).createElement((0, $404615aff7fb8653$export$5eaee2322dd727eb), {
        label: stringFormatter.format('startTime'),
        value: ((_state_timeRange = state.timeRange) === null || _state_timeRange === void 0 ? void 0 : _state_timeRange.start) || null,
        onChange: (v)=>state.setTime('start', v),
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        flex: true
    }), /*#__PURE__*/ (0, $4amrE$react).createElement((0, $404615aff7fb8653$export$5eaee2322dd727eb), {
        label: stringFormatter.format('endTime'),
        value: ((_state_timeRange1 = state.timeRange) === null || _state_timeRange1 === void 0 ? void 0 : _state_timeRange1.end) || null,
        onChange: (v)=>state.setTime('end', v),
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        flex: true
    }))))))));
});
function $62a249cadcc61296$var$DateRangeDash() {
    return /*#__PURE__*/ (0, $4amrE$react).createElement("span", {
        "aria-hidden": "true",
        "data-testid": "date-range-dash",
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($4amrE$styles_cssmjs))), 'react-spectrum-Datepicker-rangeDash')
    });
}


export {$62a249cadcc61296$export$17334619f3ac2224 as DateRangePicker};
