// @adobe/react-spectrum/dist/private/datepicker/DatePicker.cjs
var $943d9406146967ab$exports = require("../calendar/Calendar.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $685ef7ad6d6d547f$exports = require("../view/Content.cjs");
var $6a1721f36ff2e171$exports = require("./DatePickerField.cjs");
require("./styles.css");
var $25dd6e69bdd309d3$exports = require("./styles_css.cjs");
var $db50fa4488be370e$exports = require("../dialog/Dialog.cjs");
var $d4a85248c617d550$exports = require("../dialog/DialogTrigger.cjs");
var $b93966d678e0af07$exports = require("../label/Field.cjs");
var $23798a2a76e33abb$exports = require("../button/FieldButton.cjs");
var $5d83a0dbed853d9d$exports = require("./Input.cjs");
var $88d53cd9f4248e73$exports = require("./intlStrings.cjs");
require("../inputgroup_vars.css");
var $6c88ab5aea804b3c$exports = require("../inputgroup_vars_css.cjs");
var $54e7dc5129906ae8$exports = require("./TimeField.cjs");
require("../textfield_vars.css");
require("../textfield_vars_css.cjs");
var $7f5eff3a70a58c6f$exports = require("./utils.cjs");
var $1af2ca8553741739$exports = require("../form/Form.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $gdRX1$reactariauseDatePicker = require("react-aria/useDatePicker");
var $gdRX1$spectrumiconsworkflowCalendar = require("@spectrum-icons/workflow/Calendar");
var $gdRX1$reactariamergeProps = require("react-aria/mergeProps");
var $gdRX1$react = require("react");
var $gdRX1$reactstatelyuseDatePickerState = require("react-stately/useDatePickerState");
var $gdRX1$reactariauseFocusRing = require("react-aria/useFocusRing");
var $gdRX1$reactariauseHover = require("react-aria/useHover");
var $gdRX1$reactariaI18nProvider = require("react-aria/I18nProvider");
var $gdRX1$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "DatePicker", function () { return $eb116c03ace0a24b$export$5109c6dd95d8fb00; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

























const $eb116c03ace0a24b$export$5109c6dd95d8fb00 = /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).forwardRef(function DatePicker(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $1af2ca8553741739$exports.useFormProps)(props);
    let { autoFocus: autoFocus, isQuiet: isQuiet, isDisabled: isDisabled, placeholderValue: placeholderValue, maxVisibleMonths: maxVisibleMonths = 1 } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gdRX1$reactariauseHover.useHover)({
        isDisabled: isDisabled
    });
    let targetRef = (0, $gdRX1$react.useRef)(null);
    let state = (0, $gdRX1$reactstatelyuseDatePickerState.useDatePickerState)({
        ...props,
        shouldCloseOnSelect: ()=>!state.hasTime
    });
    let { groupProps: groupProps, labelProps: labelProps, fieldProps: fieldProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, buttonProps: buttonProps, dialogProps: dialogProps, calendarProps: calendarProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $gdRX1$reactariauseDatePicker.useDatePicker)(props, state, targetRef);
    let { isOpen: isOpen, setOpen: setOpen } = state;
    let { direction: direction } = (0, $gdRX1$reactariaI18nProvider.useLocale)();
    let domRef = (0, $7f5eff3a70a58c6f$exports.useFocusManagerRef)(ref);
    let stringFormatter = (0, $gdRX1$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($88d53cd9f4248e73$exports))), '@react-spectrum/datepicker');
    let { isFocused: isFocused, isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $gdRX1$reactariauseFocusRing.useFocusRing)({
        within: true,
        isTextInput: true,
        autoFocus: autoFocus
    });
    let { isFocused: isFocusedButton, focusProps: focusPropsButton } = (0, $gdRX1$reactariauseFocusRing.useFocusRing)({
        within: false,
        isTextInput: false,
        autoFocus: autoFocus
    });
    let className = (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup', {
        'spectrum-InputGroup--quiet': isQuiet,
        'spectrum-InputGroup--invalid': isInvalid && !isDisabled,
        'is-disabled': isDisabled,
        'is-hovered': isHovered,
        'is-focused': isFocused,
        'focus-ring': isFocusVisible && !isFocusedButton
    });
    let fieldClassName = (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-input', {
        'is-disabled': isDisabled,
        'is-invalid': isInvalid && !isDisabled
    });
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = (0, $7f5eff3a70a58c6f$exports.useFormatHelpText)(props);
    if (description && !props.description) descriptionProps.id = undefined;
    let placeholder = placeholderValue;
    let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : null;
    let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : null;
    let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : null;
    let timeGranularity = state.granularity === 'hour' || state.granularity === 'minute' || state.granularity === 'second' ? state.granularity : null;
    let showTimeField = !!timeGranularity;
    let visibleMonths = (0, $7f5eff3a70a58c6f$exports.useVisibleMonths)(maxVisibleMonths);
    let validationState = state.validationState || (isInvalid ? 'invalid' : null);
    let approximateWidth = (0, $7f5eff3a70a58c6f$exports.useFormattedDateWidth)(state) + 'ch';
    return /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $b93966d678e0af07$exports.Field), {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: validationState,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        wrapperClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($25dd6e69bdd309d3$exports))), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement("div", {
        ...(0, $gdRX1$reactariamergeProps.mergeProps)(groupProps, hoverProps, focusProps),
        className: className,
        ref: targetRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $5d83a0dbed853d9d$exports.Input), {
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-InputGroup-field'),
        inputClassName: fieldClassName,
        disableFocusRing: true,
        minWidth: approximateWidth
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $6a1721f36ff2e171$exports.DatePickerField), {
        ...fieldProps,
        "data-testid": "date-field",
        isQuiet: isQuiet
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $d4a85248c617d550$exports.DialogTrigger), {
        type: "popover",
        mobileType: "tray",
        placement: direction === 'rtl' ? 'bottom right' : 'bottom left',
        targetRef: targetRef,
        hideArrow: true,
        isOpen: isOpen,
        onOpenChange: setOpen,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $23798a2a76e33abb$exports.FieldButton), {
        ...(0, $gdRX1$reactariamergeProps.mergeProps)(buttonProps, focusPropsButton),
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($6c88ab5aea804b3c$exports))), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: validationState
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, ($parcel$interopDefault($gdRX1$spectrumiconsworkflowCalendar))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $db50fa4488be370e$exports.Dialog), {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($25dd6e69bdd309d3$exports))), 'react-spectrum-Datepicker-dialog'),
        ...dialogProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $685ef7ad6d6d547f$exports.Content), null, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($25dd6e69bdd309d3$exports))), 'react-spectrum-Datepicker-dialogContent')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $943d9406146967ab$exports.Calendar), {
        ...calendarProps,
        visibleMonths: visibleMonths,
        createCalendar: props.createCalendar,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($25dd6e69bdd309d3$exports))), 'react-spectrum-Datepicker-calendar', {
            'is-invalid': isInvalid
        })
    }), showTimeField && /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($25dd6e69bdd309d3$exports))), 'react-spectrum-Datepicker-timeFields')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gdRX1$react))).createElement((0, $54e7dc5129906ae8$exports.TimeField), {
        label: stringFormatter.format('time'),
        value: state.timeValue,
        onChange: state.setTimeValue,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        marginTop: "size-100"
    }))))))));
});
