// @adobe/react-spectrum/dist/private/datepicker/DatePicker.js
import {Calendar as $056e9579a2d4c8dd$export$e1aef45b828286de} from "../calendar/Calendar.js";
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {Content as $558e2ad48297783c$export$7c6e2c02157bb7d2} from "../view/Content.js";
import {DatePickerField as $219f26d0f9cef050$export$34dc4cfa15ead1} from "./DatePickerField.js";
import "./styles.css";
import $2BM5b$styles_cssmjs from "./styles_css.mjs";
import {Dialog as $89418a3659cad0c7$export$3ddf2d174ce01153} from "../dialog/Dialog.js";
import {DialogTrigger as $bcff05049955156f$export$2e1e1122cf0cba88} from "../dialog/DialogTrigger.js";
import {Field as $3967792f95357356$export$a455218a85c89869} from "../label/Field.js";
import {FieldButton as $1fa99bd0fd8b0a92$export$47dc48f595b075da} from "../button/FieldButton.js";
import {Input as $f0b9f6972621ffb5$export$f5b8910cec6cf069} from "./Input.js";
import $2BM5b$intlStringsjs from "./intlStrings.js";
import "../inputgroup_vars.css";
import $2BM5b$inputgroup_vars_cssmjs from "../inputgroup_vars_css.mjs";
import {TimeField as $404615aff7fb8653$export$5eaee2322dd727eb} from "./TimeField.js";
import "../textfield_vars.css";
import "../textfield_vars_css.mjs";
import {useFocusManagerRef as $14b5acfdaf2344b2$export$71a23a36270e4bf0, useFormatHelpText as $14b5acfdaf2344b2$export$322f4580ccd8dde6, useFormattedDateWidth as $14b5acfdaf2344b2$export$31e22e3c931fc056, useVisibleMonths as $14b5acfdaf2344b2$export$12ce2869ce471b1f} from "./utils.js";
import {useFormProps as $d23ca6800ac02cf1$export$a6b5be5c6b451665} from "../form/Form.js";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useDatePicker as $2BM5b$useDatePicker} from "react-aria/useDatePicker";
import $2BM5b$spectrumiconsworkflowCalendar from "@spectrum-icons/workflow/Calendar";
import {mergeProps as $2BM5b$mergeProps} from "react-aria/mergeProps";
import $2BM5b$react, {useRef as $2BM5b$useRef} from "react";
import {useDatePickerState as $2BM5b$useDatePickerState} from "react-stately/useDatePickerState";
import {useFocusRing as $2BM5b$useFocusRing} from "react-aria/useFocusRing";
import {useHover as $2BM5b$useHover} from "react-aria/useHover";
import {useLocale as $2BM5b$useLocale} from "react-aria/I18nProvider";
import {useLocalizedStringFormatter as $2BM5b$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

























const $b155a4096c914b21$export$5109c6dd95d8fb00 = /*#__PURE__*/ (0, $2BM5b$react).forwardRef(function DatePicker(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    props = (0, $d23ca6800ac02cf1$export$a6b5be5c6b451665)(props);
    let { autoFocus: autoFocus, isQuiet: isQuiet, isDisabled: isDisabled, placeholderValue: placeholderValue, maxVisibleMonths: maxVisibleMonths = 1 } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $2BM5b$useHover)({
        isDisabled: isDisabled
    });
    let targetRef = (0, $2BM5b$useRef)(null);
    let state = (0, $2BM5b$useDatePickerState)({
        ...props,
        shouldCloseOnSelect: ()=>!state.hasTime
    });
    let { groupProps: groupProps, labelProps: labelProps, fieldProps: fieldProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, buttonProps: buttonProps, dialogProps: dialogProps, calendarProps: calendarProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $2BM5b$useDatePicker)(props, state, targetRef);
    let { isOpen: isOpen, setOpen: setOpen } = state;
    let { direction: direction } = (0, $2BM5b$useLocale)();
    let domRef = (0, $14b5acfdaf2344b2$export$71a23a36270e4bf0)(ref);
    let stringFormatter = (0, $2BM5b$useLocalizedStringFormatter)((0, ($parcel$interopDefault($2BM5b$intlStringsjs))), '@react-spectrum/datepicker');
    let { isFocused: isFocused, isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $2BM5b$useFocusRing)({
        within: true,
        isTextInput: true,
        autoFocus: autoFocus
    });
    let { isFocused: isFocusedButton, focusProps: focusPropsButton } = (0, $2BM5b$useFocusRing)({
        within: false,
        isTextInput: false,
        autoFocus: autoFocus
    });
    let className = (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$inputgroup_vars_cssmjs))), 'spectrum-InputGroup', {
        'spectrum-InputGroup--quiet': isQuiet,
        'spectrum-InputGroup--invalid': isInvalid && !isDisabled,
        'is-disabled': isDisabled,
        'is-hovered': isHovered,
        'is-focused': isFocused,
        'focus-ring': isFocusVisible && !isFocusedButton
    });
    let fieldClassName = (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-input', {
        'is-disabled': isDisabled,
        'is-invalid': isInvalid && !isDisabled
    });
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = (0, $14b5acfdaf2344b2$export$322f4580ccd8dde6)(props);
    if (description && !props.description) descriptionProps.id = undefined;
    let placeholder = placeholderValue;
    let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : null;
    let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : null;
    let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : null;
    let timeGranularity = state.granularity === 'hour' || state.granularity === 'minute' || state.granularity === 'second' ? state.granularity : null;
    let showTimeField = !!timeGranularity;
    let visibleMonths = (0, $14b5acfdaf2344b2$export$12ce2869ce471b1f)(maxVisibleMonths);
    let validationState = state.validationState || (isInvalid ? 'invalid' : null);
    let approximateWidth = (0, $14b5acfdaf2344b2$export$31e22e3c931fc056)(state) + 'ch';
    return /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $3967792f95357356$export$a455218a85c89869), {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: validationState,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        wrapperClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$styles_cssmjs))), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement("div", {
        ...(0, $2BM5b$mergeProps)(groupProps, hoverProps, focusProps),
        className: className,
        ref: targetRef
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $f0b9f6972621ffb5$export$f5b8910cec6cf069), {
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$inputgroup_vars_cssmjs))), 'spectrum-InputGroup-field'),
        inputClassName: fieldClassName,
        disableFocusRing: true,
        minWidth: approximateWidth
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $219f26d0f9cef050$export$34dc4cfa15ead1), {
        ...fieldProps,
        "data-testid": "date-field",
        isQuiet: isQuiet
    })), /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $bcff05049955156f$export$2e1e1122cf0cba88), {
        type: "popover",
        mobileType: "tray",
        placement: direction === 'rtl' ? 'bottom right' : 'bottom left',
        targetRef: targetRef,
        hideArrow: true,
        isOpen: isOpen,
        onOpenChange: setOpen,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $1fa99bd0fd8b0a92$export$47dc48f595b075da), {
        ...(0, $2BM5b$mergeProps)(buttonProps, focusPropsButton),
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$inputgroup_vars_cssmjs))), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: validationState
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $2BM5b$spectrumiconsworkflowCalendar), null)), /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $89418a3659cad0c7$export$3ddf2d174ce01153), {
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$styles_cssmjs))), 'react-spectrum-Datepicker-dialog'),
        ...dialogProps
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $558e2ad48297783c$export$7c6e2c02157bb7d2), null, /*#__PURE__*/ (0, $2BM5b$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$styles_cssmjs))), 'react-spectrum-Datepicker-dialogContent')
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $056e9579a2d4c8dd$export$e1aef45b828286de), {
        ...calendarProps,
        visibleMonths: visibleMonths,
        createCalendar: props.createCalendar,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$styles_cssmjs))), 'react-spectrum-Datepicker-calendar', {
            'is-invalid': isInvalid
        })
    }), showTimeField && /*#__PURE__*/ (0, $2BM5b$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($2BM5b$styles_cssmjs))), 'react-spectrum-Datepicker-timeFields')
    }, /*#__PURE__*/ (0, $2BM5b$react).createElement((0, $404615aff7fb8653$export$5eaee2322dd727eb), {
        label: stringFormatter.format('time'),
        value: state.timeValue,
        onChange: state.setTimeValue,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        marginTop: "size-100"
    }))))))));
});


export {$b155a4096c914b21$export$5109c6dd95d8fb00 as DatePicker};
