// @adobe/react-spectrum/dist/private/datepicker/DateField.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {DatePickerSegment as $5bd9e661d85b8808$export$6388987c5223b54e} from "./DatePickerSegment.js";
import "./styles.css";
import $fKiD6$styles_cssmjs from "./styles_css.mjs";
import {Field as $3967792f95357356$export$a455218a85c89869} from "../label/Field.js";
import {Input as $f0b9f6972621ffb5$export$f5b8910cec6cf069} from "./Input.js";
import {useFocusManagerRef as $14b5acfdaf2344b2$export$71a23a36270e4bf0, useFormatHelpText as $14b5acfdaf2344b2$export$322f4580ccd8dde6, useFormattedDateWidth as $14b5acfdaf2344b2$export$31e22e3c931fc056} from "./utils.js";
import {useFormProps as $d23ca6800ac02cf1$export$a6b5be5c6b451665} from "../form/Form.js";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useDateField as $fKiD6$useDateField} from "react-aria/useDateField";
import {createCalendar as $fKiD6$createCalendar} from "@internationalized/date";
import $fKiD6$react, {useRef as $fKiD6$useRef} from "react";
import {useDateFieldState as $fKiD6$useDateFieldState} from "react-stately/useDateFieldState";
import {useLocale as $fKiD6$useLocale} from "react-aria/I18nProvider";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 












const $e2601972aa3002a4$export$d9781c7894a82487 = /*#__PURE__*/ (0, $fKiD6$react).forwardRef(function DateField(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    props = (0, $d23ca6800ac02cf1$export$a6b5be5c6b451665)(props);
    let { autoFocus: autoFocus, isDisabled: isDisabled, isReadOnly: isReadOnly, isRequired: isRequired, isQuiet: isQuiet } = props;
    let domRef = (0, $14b5acfdaf2344b2$export$71a23a36270e4bf0)(ref);
    let { locale: locale } = (0, $fKiD6$useLocale)();
    let state = (0, $fKiD6$useDateFieldState)({
        ...props,
        locale: locale,
        createCalendar: $fKiD6$createCalendar
    });
    let fieldRef = (0, $fKiD6$useRef)(null);
    let inputRef = (0, $fKiD6$useRef)(null);
    let { labelProps: labelProps, fieldProps: fieldProps, inputProps: inputProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $fKiD6$useDateField)({
        ...props,
        inputRef: inputRef
    }, state, fieldRef);
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = (0, $14b5acfdaf2344b2$export$322f4580ccd8dde6)(props);
    if (description && !props.description) descriptionProps.id = undefined;
    let validationState = state.validationState || (isInvalid ? 'invalid' : null);
    let approximateWidth = (0, $14b5acfdaf2344b2$export$31e22e3c931fc056)(state) + 'ch';
    return /*#__PURE__*/ (0, $fKiD6$react).createElement((0, $3967792f95357356$export$a455218a85c89869), {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: validationState !== null && validationState !== void 0 ? validationState : undefined,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        wrapperClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($fKiD6$styles_cssmjs))), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ (0, $fKiD6$react).createElement((0, $f0b9f6972621ffb5$export$f5b8910cec6cf069), {
        ref: fieldRef,
        fieldProps: fieldProps,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        autoFocus: autoFocus,
        validationState: validationState,
        minWidth: approximateWidth,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($fKiD6$styles_cssmjs))), 'react-spectrum-DateField')
    }, state.segments.map((segment, i)=>/*#__PURE__*/ (0, $fKiD6$react).createElement((0, $5bd9e661d85b8808$export$6388987c5223b54e), {
            key: i,
            segment: segment,
            state: state,
            isDisabled: isDisabled,
            isReadOnly: isReadOnly,
            isRequired: isRequired
        })), /*#__PURE__*/ (0, $fKiD6$react).createElement("input", {
        ...inputProps,
        ref: inputRef
    })));
});


export {$e2601972aa3002a4$export$d9781c7894a82487 as DateField};
