// @adobe/react-spectrum/dist/private/theme-dark/darkTheme.cjs
require("../vars_spectrum-dark.css");
var $89f17e66cecca39e$exports = require("../vars_spectrum-dark_css.cjs");
require("../vars_spectrum-darkest.css");
var $2ea0f847e8cad417$exports = require("../vars_spectrum-darkest_css.cjs");
require("../vars_spectrum-global.css");
var $863948f6068f7cfe$exports = require("../vars_spectrum-global_css.cjs");
require("../vars_spectrum-large.css");
var $5700823b600c3d54$exports = require("../vars_spectrum-large_css.cjs");
require("../vars_spectrum-medium.css");
var $7e589f15a90945e8$exports = require("../vars_spectrum-medium_css.cjs");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "darkTheme", function () { return $02cb1ca5db5534de$export$3e936a8db52a10a0; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />





let $02cb1ca5db5534de$export$3e936a8db52a10a0 = {
    global: ($parcel$interopDefault($863948f6068f7cfe$exports)),
    light: (0, ($parcel$interopDefault($89f17e66cecca39e$exports))),
    dark: (0, ($parcel$interopDefault($2ea0f847e8cad417$exports))),
    medium: ($parcel$interopDefault($7e589f15a90945e8$exports)),
    large: ($parcel$interopDefault($5700823b600c3d54$exports))
};
