// @adobe/react-spectrum/dist/private/menu/ContextualHelpTrigger.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
require("../contextualhelp_vars.css");
var $09ef91de04df24e0$exports = require("../contextualhelp_vars_css.cjs");
var $39ed1c805b59752f$exports = require("../overlays/Popover.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
require("../menu_vars.css");
var $35d34152ff885d5c$exports = require("../menu_vars_css.cjs");
var $827c4fa3df3c822d$exports = require("./context.cjs");
var $802fb5441f76e7b0$exports = require("./Menu.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $0b97cdf6ccc1e502$exports = require("../utils/useIsMobileDevice.cjs");
var $6kttE$reactariaFocusScope = require("react-aria/FocusScope");
var $6kttE$reactariaprivateinteractionsuseFocusVisible = require("react-aria/private/interactions/useFocusVisible");
var $6kttE$reactariaprivateutilsshadowdomDOMFunctions = require("react-aria/private/utils/shadowdom/DOMFunctions");
var $6kttE$react = require("react");
var $6kttE$reactdom = require("react-dom");
var $6kttE$reactariauseMenu = require("react-aria/useMenu");
var $6kttE$reactstatelyuseMenuTriggerState = require("react-stately/useMenuTriggerState");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ContextualHelpTrigger", function () { return $6b61ca38ae72d583$export$5413b169fff83e61; });
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 















function $6b61ca38ae72d583$var$ContextualHelpTrigger(props) {
    let { isUnavailable: isUnavailable = false, targetKey: targetKey } = props;
    let triggerRef = (0, $6kttE$react.useRef)(null);
    let popoverRef = (0, $6kttE$react.useRef)(null);
    let { popoverContainer: popoverContainer, trayContainerRef: trayContainerRef, rootMenuTriggerState: rootMenuTriggerState, menu: parentMenuRef, state: state } = (0, $827c4fa3df3c822d$exports.useMenuStateContext)();
    let submenuTriggerState = (0, $6kttE$reactstatelyuseMenuTriggerState.useSubmenuTriggerState)({
        triggerKey: targetKey
    }, {
        ...rootMenuTriggerState,
        ...state
    });
    let submenuRef = (0, $65aea7b37663976b$exports.unwrapDOMRef)(popoverRef);
    let { submenuTriggerProps: submenuTriggerProps, popoverProps: popoverProps } = (0, $6kttE$reactariauseMenu.useSubmenuTrigger)({
        parentMenuRef: parentMenuRef,
        submenuRef: submenuRef,
        type: 'dialog',
        isDisabled: !isUnavailable
    }, submenuTriggerState, triggerRef);
    let isMobile = (0, $0b97cdf6ccc1e502$exports.useIsMobileDevice)();
    let [traySubmenuAnimation, setTraySubmenuAnimation] = (0, $6kttE$react.useState)('');
    (0, $6kttE$react.useEffect)(()=>{
        if (submenuTriggerState.isOpen) setTraySubmenuAnimation('spectrum-TraySubmenu-enter');
    }, [
        submenuTriggerState.isOpen
    ]);
    let slots = {};
    if (isUnavailable) slots = {
        dialog: {
            UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($09ef91de04df24e0$exports))), 'react-spectrum-ContextualHelp-dialog', {
                'react-spectrum-ContextualHelp-dialog--isMobile': isMobile
            }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), {
                'spectrum-Menu-subdialog': !isMobile,
                [traySubmenuAnimation]: isMobile
            }))
        },
        content: {
            UNSAFE_className: (0, ($parcel$interopDefault($09ef91de04df24e0$exports)))['react-spectrum-ContextualHelp-content']
        },
        footer: {
            UNSAFE_className: (0, ($parcel$interopDefault($09ef91de04df24e0$exports)))['react-spectrum-ContextualHelp-footer']
        }
    };
    let [trigger] = (0, ($parcel$interopDefault($6kttE$react))).Children.toArray(props.children);
    let [, content] = props.children;
    let onBlurWithin = (e)=>{
        if (e.relatedTarget && popoverRef.current && !(0, $6kttE$reactariaprivateutilsshadowdomDOMFunctions.nodeContains)(popoverRef.current.UNSAFE_getDOMNode(), e.relatedTarget) && !(e.relatedTarget === triggerRef.current && (0, $6kttE$reactariaprivateinteractionsuseFocusVisible.getInteractionModality)() === 'pointer')) {
            if (submenuTriggerState.isOpen) submenuTriggerState.close();
        }
    };
    let overlay;
    let tray;
    let onBackButtonPress = ()=>{
        setTraySubmenuAnimation('spectrum-TraySubmenu-exit');
        setTimeout(()=>{
            submenuTriggerState.close();
            if (parentMenuRef.current && !(0, $6kttE$reactariaprivateutilsshadowdomDOMFunctions.isFocusWithin)(parentMenuRef.current)) parentMenuRef.current.focus();
        }, 220); // Matches transition duration
    };
    if (isMobile) {
        delete submenuTriggerProps.onBlur;
        delete submenuTriggerProps.onHoverChange;
        if (trayContainerRef.current && submenuTriggerState.isOpen) {
            let subDialogKeyDown = (e)=>{
                switch(e.key){
                    case 'Escape':
                        e.stopPropagation();
                        onBackButtonPress();
                        break;
                }
            };
            tray = /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).createElement((0, $802fb5441f76e7b0$exports.TrayHeaderWrapper), {
                isSubmenu: true,
                parentMenuTreeState: state,
                rootMenuTriggerState: rootMenuTriggerState,
                wrapperKeyDown: subDialogKeyDown,
                onBackButtonPress: onBackButtonPress
            }, content);
            overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$reactdom))).createPortal(tray, trayContainerRef.current);
        }
    } else {
        let onDismissButtonPress = ()=>{
            submenuTriggerState.close();
            parentMenuRef.current?.focus();
        };
        overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).createElement((0, $39ed1c805b59752f$exports.Popover), {
            ...popoverProps,
            UNSAFE_style: {
                clipPath: 'unset',
                overflow: 'visible',
                filter: 'unset',
                borderWidth: '0px'
            },
            UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), 'spectrum-Submenu-popover'),
            onDismissButtonPress: onDismissButtonPress,
            onBlurWithin: onBlurWithin,
            container: popoverContainer,
            state: submenuTriggerState,
            ref: popoverRef,
            triggerRef: triggerRef,
            placement: "end top",
            containerPadding: 0,
            hideArrow: true,
            enableBothDismissButtons: true
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).createElement((0, $6kttE$reactariaFocusScope.FocusScope), {
            restoreFocus: true,
            contain: true
        }, content));
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).createElement((0, ($parcel$interopDefault($6kttE$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).createElement((0, $827c4fa3df3c822d$exports.SubmenuTriggerContext).Provider, {
        value: {
            isUnavailable: isUnavailable,
            triggerRef: triggerRef,
            ...submenuTriggerProps
        }
    }, trigger), /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: slots
    }, submenuTriggerState.isOpen && overlay));
}
$6b61ca38ae72d583$var$ContextualHelpTrigger.getCollectionNode = function* getCollectionNode(props) {
    let childArray = [];
    (0, ($parcel$interopDefault($6kttE$react))).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).isValidElement(child)) childArray.push(child);
    });
    let [trigger] = childArray;
    let [, content] = props.children;
    yield {
        element: /*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).cloneElement(trigger, {
            ...trigger.props,
            hasChildItems: true,
            isTrigger: true
        }),
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($6kttE$react))).createElement($6b61ca38ae72d583$var$ContextualHelpTrigger, {
                key: element.key,
                targetKey: element.key,
                ...props
            }, element, content)
    };
};
let $6b61ca38ae72d583$export$5413b169fff83e61 = $6b61ca38ae72d583$var$ContextualHelpTrigger;
