// @adobe/react-spectrum/dist/private/menu/ContextualHelpTrigger.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import "../contextualhelp_vars.css";
import $b0zQ7$contextualhelp_vars_cssmjs from "../contextualhelp_vars_css.mjs";
import {Popover as $2fa1c97e743ad66b$export$5b6b19405a83ff9d} from "../overlays/Popover.js";
import {SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686} from "../utils/Slots.js";
import "../menu_vars.css";
import $b0zQ7$menu_vars_cssmjs from "../menu_vars_css.mjs";
import {SubmenuTriggerContext as $a4d1910f0ff9d033$export$8d97fe02339fc0e3, useMenuStateContext as $a4d1910f0ff9d033$export$efa3856fc0e85e7f} from "./context.js";
import {TrayHeaderWrapper as $79ddee63a726ea3d$export$3dfe97b5c32d8d8c} from "./Menu.js";
import {unwrapDOMRef as $c234463e9ef56637$export$c7e28c72a4823176} from "../utils/useDOMRef.js";
import {useIsMobileDevice as $196ab9279fe71c29$export$736bf165441b18c7} from "../utils/useIsMobileDevice.js";
import {FocusScope as $b0zQ7$FocusScope} from "react-aria/FocusScope";
import {getInteractionModality as $b0zQ7$getInteractionModality} from "react-aria/private/interactions/useFocusVisible";
import {nodeContains as $b0zQ7$nodeContains, isFocusWithin as $b0zQ7$isFocusWithin} from "react-aria/private/utils/shadowdom/DOMFunctions";
import $b0zQ7$react, {useRef as $b0zQ7$useRef, useState as $b0zQ7$useState, useEffect as $b0zQ7$useEffect} from "react";
import $b0zQ7$reactdom from "react-dom";
import {useSubmenuTrigger as $b0zQ7$useSubmenuTrigger} from "react-aria/useMenu";
import {useSubmenuTriggerState as $b0zQ7$useSubmenuTriggerState} from "react-stately/useMenuTriggerState";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 















function $6bc666e61b0562d8$var$ContextualHelpTrigger(props) {
    let { isUnavailable: isUnavailable = false, targetKey: targetKey } = props;
    let triggerRef = (0, $b0zQ7$useRef)(null);
    let popoverRef = (0, $b0zQ7$useRef)(null);
    let { popoverContainer: popoverContainer, trayContainerRef: trayContainerRef, rootMenuTriggerState: rootMenuTriggerState, menu: parentMenuRef, state: state } = (0, $a4d1910f0ff9d033$export$efa3856fc0e85e7f)();
    let submenuTriggerState = (0, $b0zQ7$useSubmenuTriggerState)({
        triggerKey: targetKey
    }, {
        ...rootMenuTriggerState,
        ...state
    });
    let submenuRef = (0, $c234463e9ef56637$export$c7e28c72a4823176)(popoverRef);
    let { submenuTriggerProps: submenuTriggerProps, popoverProps: popoverProps } = (0, $b0zQ7$useSubmenuTrigger)({
        parentMenuRef: parentMenuRef,
        submenuRef: submenuRef,
        type: 'dialog',
        isDisabled: !isUnavailable
    }, submenuTriggerState, triggerRef);
    let isMobile = (0, $196ab9279fe71c29$export$736bf165441b18c7)();
    let [traySubmenuAnimation, setTraySubmenuAnimation] = (0, $b0zQ7$useState)('');
    (0, $b0zQ7$useEffect)(()=>{
        if (submenuTriggerState.isOpen) setTraySubmenuAnimation('spectrum-TraySubmenu-enter');
    }, [
        submenuTriggerState.isOpen
    ]);
    let slots = {};
    if (isUnavailable) slots = {
        dialog: {
            UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b0zQ7$contextualhelp_vars_cssmjs))), 'react-spectrum-ContextualHelp-dialog', {
                'react-spectrum-ContextualHelp-dialog--isMobile': isMobile
            }, (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b0zQ7$menu_vars_cssmjs))), {
                'spectrum-Menu-subdialog': !isMobile,
                [traySubmenuAnimation]: isMobile
            }))
        },
        content: {
            UNSAFE_className: (0, ($parcel$interopDefault($b0zQ7$contextualhelp_vars_cssmjs)))['react-spectrum-ContextualHelp-content']
        },
        footer: {
            UNSAFE_className: (0, ($parcel$interopDefault($b0zQ7$contextualhelp_vars_cssmjs)))['react-spectrum-ContextualHelp-footer']
        }
    };
    let [trigger] = (0, $b0zQ7$react).Children.toArray(props.children);
    let [, content] = props.children;
    let onBlurWithin = (e)=>{
        if (e.relatedTarget && popoverRef.current && !(0, $b0zQ7$nodeContains)(popoverRef.current.UNSAFE_getDOMNode(), e.relatedTarget) && !(e.relatedTarget === triggerRef.current && (0, $b0zQ7$getInteractionModality)() === 'pointer')) {
            if (submenuTriggerState.isOpen) submenuTriggerState.close();
        }
    };
    let overlay;
    let tray;
    let onBackButtonPress = ()=>{
        setTraySubmenuAnimation('spectrum-TraySubmenu-exit');
        setTimeout(()=>{
            submenuTriggerState.close();
            if (parentMenuRef.current && !(0, $b0zQ7$isFocusWithin)(parentMenuRef.current)) parentMenuRef.current.focus();
        }, 220); // Matches transition duration
    };
    if (isMobile) {
        delete submenuTriggerProps.onBlur;
        delete submenuTriggerProps.onHoverChange;
        if (trayContainerRef.current && submenuTriggerState.isOpen) {
            let subDialogKeyDown = (e)=>{
                switch(e.key){
                    case 'Escape':
                        e.stopPropagation();
                        onBackButtonPress();
                        break;
                }
            };
            tray = /*#__PURE__*/ (0, $b0zQ7$react).createElement((0, $79ddee63a726ea3d$export$3dfe97b5c32d8d8c), {
                isSubmenu: true,
                parentMenuTreeState: state,
                rootMenuTriggerState: rootMenuTriggerState,
                wrapperKeyDown: subDialogKeyDown,
                onBackButtonPress: onBackButtonPress
            }, content);
            overlay = /*#__PURE__*/ (0, $b0zQ7$reactdom).createPortal(tray, trayContainerRef.current);
        }
    } else {
        let onDismissButtonPress = ()=>{
            var _parentMenuRef_current;
            submenuTriggerState.close();
            (_parentMenuRef_current = parentMenuRef.current) === null || _parentMenuRef_current === void 0 ? void 0 : _parentMenuRef_current.focus();
        };
        overlay = /*#__PURE__*/ (0, $b0zQ7$react).createElement((0, $2fa1c97e743ad66b$export$5b6b19405a83ff9d), {
            ...popoverProps,
            UNSAFE_style: {
                clipPath: 'unset',
                overflow: 'visible',
                filter: 'unset',
                borderWidth: '0px'
            },
            UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($b0zQ7$menu_vars_cssmjs))), 'spectrum-Submenu-popover'),
            onDismissButtonPress: onDismissButtonPress,
            onBlurWithin: onBlurWithin,
            container: popoverContainer,
            state: submenuTriggerState,
            ref: popoverRef,
            triggerRef: triggerRef,
            placement: "end top",
            containerPadding: 0,
            hideArrow: true,
            enableBothDismissButtons: true
        }, /*#__PURE__*/ (0, $b0zQ7$react).createElement((0, $b0zQ7$FocusScope), {
            restoreFocus: true,
            contain: true
        }, content));
    }
    return /*#__PURE__*/ (0, $b0zQ7$react).createElement((0, $b0zQ7$react).Fragment, null, /*#__PURE__*/ (0, $b0zQ7$react).createElement((0, $a4d1910f0ff9d033$export$8d97fe02339fc0e3).Provider, {
        value: {
            isUnavailable: isUnavailable,
            triggerRef: triggerRef,
            ...submenuTriggerProps
        }
    }, trigger), /*#__PURE__*/ (0, $b0zQ7$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: slots
    }, submenuTriggerState.isOpen && overlay));
}
$6bc666e61b0562d8$var$ContextualHelpTrigger.getCollectionNode = function* getCollectionNode(props) {
    let childArray = [];
    (0, $b0zQ7$react).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, $b0zQ7$react).isValidElement(child)) childArray.push(child);
    });
    let [trigger] = childArray;
    let [, content] = props.children;
    yield {
        element: /*#__PURE__*/ (0, $b0zQ7$react).cloneElement(trigger, {
            ...trigger.props,
            hasChildItems: true,
            isTrigger: true
        }),
        wrapper: (element)=>/*#__PURE__*/ (0, $b0zQ7$react).createElement($6bc666e61b0562d8$var$ContextualHelpTrigger, {
                key: element.key,
                targetKey: element.key,
                ...props
            }, element, content)
    };
};
let $6bc666e61b0562d8$export$5413b169fff83e61 = $6bc666e61b0562d8$var$ContextualHelpTrigger;


export {$6bc666e61b0562d8$export$5413b169fff83e61 as ContextualHelpTrigger};
