// @adobe/react-spectrum/dist/private/menu/MenuTrigger.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $827c4fa3df3c822d$exports = require("./context.cjs");
var $39ed1c805b59752f$exports = require("../overlays/Popover.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
require("../menu_vars.css");
var $35d34152ff885d5c$exports = require("../menu_vars_css.cjs");
var $378dee1409fe2937$exports = require("../overlays/Tray.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $0b97cdf6ccc1e502$exports = require("../utils/useIsMobileDevice.cjs");
var $5K69p$reactstatelyuseMenuTriggerState = require("react-stately/useMenuTriggerState");
var $5K69p$reactariaprivateinteractionsPressResponder = require("react-aria/private/interactions/PressResponder");
var $5K69p$react = require("react");
var $5K69p$reactariauseInteractOutside = require("react-aria/useInteractOutside");
var $5K69p$reactariauseMenu = require("react-aria/useMenu");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "MenuTrigger", function () { return $98227f5fd590c993$export$27d2ad3c5815583e; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 












const $98227f5fd590c993$export$27d2ad3c5815583e = /*#__PURE__*/ (0, $5K69p$react.forwardRef)(function MenuTrigger(props, ref) {
    let triggerRef = (0, $5K69p$react.useRef)(null);
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let menuTriggerRef = domRef || triggerRef;
    let menuRef = (0, $5K69p$react.useRef)(null);
    let { children: children, align: align = 'start', shouldFlip: shouldFlip = true, direction: direction = 'bottom', closeOnSelect: closeOnSelect, trigger: trigger = 'press' } = props;
    let [menuTrigger, menu] = (0, ($parcel$interopDefault($5K69p$react))).Children.toArray(children);
    let state = (0, $5K69p$reactstatelyuseMenuTriggerState.useMenuTriggerState)(props);
    let { menuTriggerProps: menuTriggerProps, menuProps: menuProps } = (0, $5K69p$reactariauseMenu.useMenuTrigger)({
        trigger: trigger
    }, state, menuTriggerRef);
    let initialPlacement;
    switch(direction){
        case 'left':
        case 'right':
        case 'start':
        case 'end':
            initialPlacement = `${direction} ${align === 'end' ? 'bottom' : 'top'}`;
            break;
        case 'bottom':
        case 'top':
        default:
            initialPlacement = `${direction} ${align}`;
    }
    let isMobile = (0, $0b97cdf6ccc1e502$exports.useIsMobileDevice)();
    let menuContext = {
        ...menuProps,
        ref: menuRef,
        onClose: state.close,
        closeOnSelect: closeOnSelect,
        autoFocus: state.focusStrategy || true,
        UNSAFE_style: isMobile ? {
            width: '100%',
            maxHeight: 'inherit'
        } : undefined,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), {
            'spectrum-Menu-popover': !isMobile
        }),
        state: state
    };
    // Close when clicking outside the root menu when a submenu is open.
    let rootOverlayRef = (0, $5K69p$react.useRef)(null);
    let rootOverlayDomRef = (0, $65aea7b37663976b$exports.unwrapDOMRef)(rootOverlayRef);
    (0, $5K69p$reactariauseInteractOutside.useInteractOutside)({
        ref: rootOverlayDomRef,
        onInteractOutside: ()=>{
            state?.close();
        },
        isDisabled: !state.isOpen || state.expandedKeysStack.length === 0
    });
    // On small screen devices, the menu is rendered in a tray, otherwise a popover.
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($5K69p$react))).createElement((0, $378dee1409fe2937$exports.Tray), {
        state: state,
        isFixedHeight: true,
        ref: rootOverlayRef
    }, menu);
    else overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($5K69p$react))).createElement((0, $39ed1c805b59752f$exports.Popover), {
        ref: rootOverlayRef,
        UNSAFE_style: {
            clipPath: 'unset',
            overflow: 'visible',
            filter: 'unset',
            borderWidth: '0px'
        },
        state: state,
        triggerRef: menuTriggerRef,
        scrollRef: menuRef,
        placement: initialPlacement,
        hideArrow: true,
        shouldFlip: shouldFlip,
        shouldContainFocus: true
    }, menu);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($5K69p$react))).createElement((0, $5K69p$react.Fragment), null, /*#__PURE__*/ (0, ($parcel$interopDefault($5K69p$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            actionButton: {
                holdAffordance: trigger === 'longPress'
            }
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($5K69p$react))).createElement((0, $5K69p$reactariaprivateinteractionsPressResponder.PressResponder), {
        ...menuTriggerProps,
        ref: menuTriggerRef,
        isPressed: state.isOpen
    }, menuTrigger)), /*#__PURE__*/ (0, ($parcel$interopDefault($5K69p$react))).createElement((0, $827c4fa3df3c822d$exports.MenuContext).Provider, {
        value: menuContext
    }, overlay));
});
