// @adobe/react-spectrum/dist/private/menu/MenuTrigger.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {MenuContext as $a4d1910f0ff9d033$export$c7e742effb1c51e2} from "./context.js";
import {Popover as $2fa1c97e743ad66b$export$5b6b19405a83ff9d} from "../overlays/Popover.js";
import {SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686} from "../utils/Slots.js";
import "../menu_vars.css";
import $kcXab$menu_vars_cssmjs from "../menu_vars_css.mjs";
import {Tray as $16b239851776d94c$export$4589ed81930b555c} from "../overlays/Tray.js";
import {unwrapDOMRef as $c234463e9ef56637$export$c7e28c72a4823176, useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useIsMobileDevice as $196ab9279fe71c29$export$736bf165441b18c7} from "../utils/useIsMobileDevice.js";
import {useMenuTriggerState as $kcXab$useMenuTriggerState} from "react-stately/useMenuTriggerState";
import {PressResponder as $kcXab$PressResponder} from "react-aria/private/interactions/PressResponder";
import $kcXab$react, {forwardRef as $kcXab$forwardRef, useRef as $kcXab$useRef, Fragment as $kcXab$Fragment} from "react";
import {useInteractOutside as $kcXab$useInteractOutside} from "react-aria/useInteractOutside";
import {useMenuTrigger as $kcXab$useMenuTrigger} from "react-aria/useMenu";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 












const $9f6ebde23392f425$export$27d2ad3c5815583e = /*#__PURE__*/ (0, $kcXab$forwardRef)(function MenuTrigger(props, ref) {
    let triggerRef = (0, $kcXab$useRef)(null);
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let menuTriggerRef = domRef || triggerRef;
    let menuRef = (0, $kcXab$useRef)(null);
    let { children: children, align: align = 'start', shouldFlip: shouldFlip = true, direction: direction = 'bottom', closeOnSelect: closeOnSelect, trigger: trigger = 'press' } = props;
    let [menuTrigger, menu] = (0, $kcXab$react).Children.toArray(children);
    let state = (0, $kcXab$useMenuTriggerState)(props);
    let { menuTriggerProps: menuTriggerProps, menuProps: menuProps } = (0, $kcXab$useMenuTrigger)({
        trigger: trigger
    }, state, menuTriggerRef);
    let initialPlacement;
    switch(direction){
        case 'left':
        case 'right':
        case 'start':
        case 'end':
            initialPlacement = `${direction} ${align === 'end' ? 'bottom' : 'top'}`;
            break;
        case 'bottom':
        case 'top':
        default:
            initialPlacement = `${direction} ${align}`;
    }
    let isMobile = (0, $196ab9279fe71c29$export$736bf165441b18c7)();
    let menuContext = {
        ...menuProps,
        ref: menuRef,
        onClose: state.close,
        closeOnSelect: closeOnSelect,
        autoFocus: state.focusStrategy || true,
        UNSAFE_style: isMobile ? {
            width: '100%',
            maxHeight: 'inherit'
        } : undefined,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($kcXab$menu_vars_cssmjs))), {
            'spectrum-Menu-popover': !isMobile
        }),
        state: state
    };
    // Close when clicking outside the root menu when a submenu is open.
    let rootOverlayRef = (0, $kcXab$useRef)(null);
    let rootOverlayDomRef = (0, $c234463e9ef56637$export$c7e28c72a4823176)(rootOverlayRef);
    (0, $kcXab$useInteractOutside)({
        ref: rootOverlayDomRef,
        onInteractOutside: ()=>{
            state === null || state === void 0 ? void 0 : state.close();
        },
        isDisabled: !state.isOpen || state.expandedKeysStack.length === 0
    });
    // On small screen devices, the menu is rendered in a tray, otherwise a popover.
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ (0, $kcXab$react).createElement((0, $16b239851776d94c$export$4589ed81930b555c), {
        state: state,
        isFixedHeight: true,
        ref: rootOverlayRef
    }, menu);
    else overlay = /*#__PURE__*/ (0, $kcXab$react).createElement((0, $2fa1c97e743ad66b$export$5b6b19405a83ff9d), {
        ref: rootOverlayRef,
        UNSAFE_style: {
            clipPath: 'unset',
            overflow: 'visible',
            filter: 'unset',
            borderWidth: '0px'
        },
        state: state,
        triggerRef: menuTriggerRef,
        scrollRef: menuRef,
        placement: initialPlacement,
        hideArrow: true,
        shouldFlip: shouldFlip,
        shouldContainFocus: true
    }, menu);
    return /*#__PURE__*/ (0, $kcXab$react).createElement((0, $kcXab$Fragment), null, /*#__PURE__*/ (0, $kcXab$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: {
            actionButton: {
                holdAffordance: trigger === 'longPress'
            }
        }
    }, /*#__PURE__*/ (0, $kcXab$react).createElement((0, $kcXab$PressResponder), {
        ...menuTriggerProps,
        ref: menuTriggerRef,
        isPressed: state.isOpen
    }, menuTrigger)), /*#__PURE__*/ (0, $kcXab$react).createElement((0, $a4d1910f0ff9d033$export$c7e742effb1c51e2).Provider, {
        value: menuContext
    }, overlay));
});


export {$9f6ebde23392f425$export$27d2ad3c5815583e as MenuTrigger};
