// @adobe/react-spectrum/dist/private/menu/MenuItem.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {ClearSlots as $68f4bc2c1abc5618$export$ceb145244332b7a2, SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686} from "../utils/Slots.js";
import {Grid as $727c1a1d9e8b8d73$export$ef2184bd89960b14} from "../layout/Grid.js";
import $dgEp4$intlStringsjs from "./intlStrings.js";
import "../menu_vars.css";
import $dgEp4$menu_vars_cssmjs from "../menu_vars_css.mjs";
import {Text as $42dd7396e689e4e6$export$5f1af8db9871e1d6} from "../text/Text.js";
import {useMenuContext as $a4d1910f0ff9d033$export$21c7ab35b39f78ec, useSubmenuTriggerContext as $a4d1910f0ff9d033$export$dc2161044aa0b36d} from "./context.js";
import $dgEp4$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import $dgEp4$spectrumiconsworkflowChevronLeft from "@spectrum-icons/workflow/ChevronLeft";
import $dgEp4$spectrumiconsworkflowChevronRight from "@spectrum-icons/workflow/ChevronRight";
import {FocusRing as $dgEp4$FocusRing} from "react-aria/FocusRing";
import $dgEp4$spectrumiconsworkflowInfoOutline from "@spectrum-icons/workflow/InfoOutline";
import {mergeRefs as $dgEp4$mergeRefs} from "react-aria/mergeRefs";
import $dgEp4$react, {useRef as $dgEp4$useRef, useMemo as $dgEp4$useMemo} from "react";
import {useLocale as $dgEp4$useLocale} from "react-aria/I18nProvider";
import {useLocalizedStringFormatter as $dgEp4$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {useMenuItem as $dgEp4$useMenuItem} from "react-aria/useMenu";
import {useObjectRef as $dgEp4$useObjectRef} from "react-aria/useObjectRef";
import {useSlotId as $dgEp4$useSlotId} from "react-aria/private/utils/useId";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


















function $53bbb287499fadf8$export$2ce376c2cc3355c8(props) {
    let { item: item, state: state, isVirtualized: isVirtualized } = props;
    let { closeOnSelect: closeOnSelect } = (0, $a4d1910f0ff9d033$export$21c7ab35b39f78ec)();
    let { rendered: rendered, key: key } = item;
    let stringFormatter = (0, $dgEp4$useLocalizedStringFormatter)((0, ($parcel$interopDefault($dgEp4$intlStringsjs))), '@react-spectrum/menu');
    let { direction: direction } = (0, $dgEp4$useLocale)();
    let submenuTriggerContext = (0, $a4d1910f0ff9d033$export$dc2161044aa0b36d)();
    let { triggerRef: triggerRef, ...submenuTriggerProps } = submenuTriggerContext || {};
    let isSubmenuTrigger = !!submenuTriggerContext;
    let isUnavailable;
    let ElementType = item.props.href ? 'a' : 'div';
    if (isSubmenuTrigger) isUnavailable = submenuTriggerContext.isUnavailable;
    let isDisabled = state.disabledKeys.has(key);
    let isContextualHelpTrigger = isSubmenuTrigger && isUnavailable !== undefined;
    let isSelectable = (isContextualHelpTrigger ? !isUnavailable : !isSubmenuTrigger) && state.selectionManager.selectionMode !== 'none';
    let isSelected = isSelectable && state.selectionManager.isSelected(key);
    let itemref = (0, $dgEp4$useRef)(null);
    let ref = (0, $dgEp4$useObjectRef)((0, $dgEp4$useMemo)(()=>(0, $dgEp4$mergeRefs)(itemref, triggerRef), [
        itemref,
        triggerRef
    ]));
    let { menuItemProps: menuItemProps, labelProps: labelProps, descriptionProps: descriptionProps, keyboardShortcutProps: keyboardShortcutProps } = (0, $dgEp4$useMenuItem)({
        isSelected: isSelected,
        isDisabled: isDisabled,
        'aria-label': item['aria-label'],
        key: key,
        closeOnSelect: closeOnSelect,
        isVirtualized: isVirtualized,
        ...submenuTriggerProps
    }, state, ref);
    let endId = (0, $dgEp4$useSlotId)();
    let endProps = {};
    if (endId) {
        endProps.id = endId;
        menuItemProps['aria-describedby'] = [
            menuItemProps['aria-describedby'],
            endId
        ].filter(Boolean).join(' ');
    }
    let contents = typeof rendered === 'string' ? /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $42dd7396e689e4e6$export$5f1af8db9871e1d6), null, rendered) : rendered;
    return /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $dgEp4$FocusRing), {
        focusRingClass: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs))), 'focus-ring')
    }, /*#__PURE__*/ (0, $dgEp4$react).createElement(ElementType, {
        ...menuItemProps,
        ref: ref,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs))), 'spectrum-Menu-item', {
            'is-disabled': isDisabled,
            'is-selected': isSelected,
            'is-selectable': isSelectable,
            'is-open': submenuTriggerProps.isOpen
        })
    }, /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $727c1a1d9e8b8d73$export$ef2184bd89960b14), {
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs))), 'spectrum-Menu-itemGrid')
    }, /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $68f4bc2c1abc5618$export$ceb145244332b7a2), null, /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: {
            text: {
                UNSAFE_className: (0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs)))['spectrum-Menu-itemLabel'],
                ...labelProps
            },
            end: {
                UNSAFE_className: (0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs)))['spectrum-Menu-end'],
                ...endProps
            },
            icon: {
                UNSAFE_className: (0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs)))['spectrum-Menu-icon'],
                size: 'S'
            },
            description: {
                UNSAFE_className: (0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs)))['spectrum-Menu-description'],
                ...descriptionProps
            },
            keyboard: {
                UNSAFE_className: (0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs)))['spectrum-Menu-keyboard'],
                ...keyboardShortcutProps
            },
            chevron: {
                UNSAFE_className: (0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs)))['spectrum-Menu-chevron'],
                size: 'S'
            }
        }
    }, contents, isSelected && /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $dgEp4$spectrumiconsuiCheckmarkMedium), {
        slot: "checkmark",
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($dgEp4$menu_vars_cssmjs))), 'spectrum-Menu-checkmark')
    }), isUnavailable && /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $dgEp4$spectrumiconsworkflowInfoOutline), {
        slot: "end",
        size: "XS",
        alignSelf: "center",
        "aria-label": stringFormatter.format('unavailable')
    }), isUnavailable == null && isSubmenuTrigger && (direction === 'rtl' ? /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $dgEp4$spectrumiconsworkflowChevronLeft), {
        slot: "chevron"
    }) : /*#__PURE__*/ (0, $dgEp4$react).createElement((0, $dgEp4$spectrumiconsworkflowChevronRight), {
        slot: "chevron"
    })))))));
}


export {$53bbb287499fadf8$export$2ce376c2cc3355c8 as MenuItem};
