// @adobe/react-spectrum/dist/private/menu/MenuItem.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
var $d6479700d21b596b$exports = require("../layout/Grid.cjs");
var $e609f7c27f409c35$exports = require("./intlStrings.cjs");
require("../menu_vars.css");
var $35d34152ff885d5c$exports = require("../menu_vars_css.cjs");
var $15e3b68ec42125a9$exports = require("../text/Text.cjs");
var $827c4fa3df3c822d$exports = require("./context.cjs");
var $lz9FT$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $lz9FT$spectrumiconsworkflowChevronLeft = require("@spectrum-icons/workflow/ChevronLeft");
var $lz9FT$spectrumiconsworkflowChevronRight = require("@spectrum-icons/workflow/ChevronRight");
var $lz9FT$reactariaFocusRing = require("react-aria/FocusRing");
var $lz9FT$spectrumiconsworkflowInfoOutline = require("@spectrum-icons/workflow/InfoOutline");
var $lz9FT$reactariamergeRefs = require("react-aria/mergeRefs");
var $lz9FT$react = require("react");
var $lz9FT$reactariaI18nProvider = require("react-aria/I18nProvider");
var $lz9FT$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $lz9FT$reactariauseMenu = require("react-aria/useMenu");
var $lz9FT$reactariauseObjectRef = require("react-aria/useObjectRef");
var $lz9FT$reactariaprivateutilsuseId = require("react-aria/private/utils/useId");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "MenuItem", function () { return $f98c72ac58c30ee0$export$2ce376c2cc3355c8; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


















function $f98c72ac58c30ee0$export$2ce376c2cc3355c8(props) {
    let { item: item, state: state, isVirtualized: isVirtualized } = props;
    let { closeOnSelect: closeOnSelect } = (0, $827c4fa3df3c822d$exports.useMenuContext)();
    let { rendered: rendered, key: key } = item;
    let stringFormatter = (0, $lz9FT$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($e609f7c27f409c35$exports))), '@react-spectrum/menu');
    let { direction: direction } = (0, $lz9FT$reactariaI18nProvider.useLocale)();
    let submenuTriggerContext = (0, $827c4fa3df3c822d$exports.useSubmenuTriggerContext)();
    let { triggerRef: triggerRef, ...submenuTriggerProps } = submenuTriggerContext || {};
    let isSubmenuTrigger = !!submenuTriggerContext;
    let isUnavailable;
    let ElementType = item.props.href ? 'a' : 'div';
    if (isSubmenuTrigger) isUnavailable = submenuTriggerContext.isUnavailable;
    let isDisabled = state.disabledKeys.has(key);
    let isContextualHelpTrigger = isSubmenuTrigger && isUnavailable !== undefined;
    let isSelectable = (isContextualHelpTrigger ? !isUnavailable : !isSubmenuTrigger) && state.selectionManager.selectionMode !== 'none';
    let isSelected = isSelectable && state.selectionManager.isSelected(key);
    let itemref = (0, $lz9FT$react.useRef)(null);
    let ref = (0, $lz9FT$reactariauseObjectRef.useObjectRef)((0, $lz9FT$react.useMemo)(()=>(0, $lz9FT$reactariamergeRefs.mergeRefs)(itemref, triggerRef), [
        itemref,
        triggerRef
    ]));
    let { menuItemProps: menuItemProps, labelProps: labelProps, descriptionProps: descriptionProps, keyboardShortcutProps: keyboardShortcutProps } = (0, $lz9FT$reactariauseMenu.useMenuItem)({
        isSelected: isSelected,
        isDisabled: isDisabled,
        'aria-label': item['aria-label'],
        key: key,
        closeOnSelect: closeOnSelect,
        isVirtualized: isVirtualized,
        ...submenuTriggerProps
    }, state, ref);
    let endId = (0, $lz9FT$reactariaprivateutilsuseId.useSlotId)();
    let endProps = {};
    if (endId) {
        endProps.id = endId;
        menuItemProps['aria-describedby'] = [
            menuItemProps['aria-describedby'],
            endId
        ].filter(Boolean).join(' ');
    }
    let contents = typeof rendered === 'string' ? /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, $15e3b68ec42125a9$exports.Text), null, rendered) : rendered;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, $lz9FT$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement(ElementType, {
        ...menuItemProps,
        ref: ref,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), 'spectrum-Menu-item', {
            'is-disabled': isDisabled,
            'is-selected': isSelected,
            'is-selectable': isSelectable,
            'is-open': submenuTriggerProps.isOpen
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, $d6479700d21b596b$exports.Grid), {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), 'spectrum-Menu-itemGrid')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, $feede71cddc0c5f3$exports.ClearSlots), null, /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, ($parcel$interopDefault($35d34152ff885d5c$exports)))['spectrum-Menu-itemLabel'],
                ...labelProps
            },
            end: {
                UNSAFE_className: (0, ($parcel$interopDefault($35d34152ff885d5c$exports)))['spectrum-Menu-end'],
                ...endProps
            },
            icon: {
                UNSAFE_className: (0, ($parcel$interopDefault($35d34152ff885d5c$exports)))['spectrum-Menu-icon'],
                size: 'S'
            },
            description: {
                UNSAFE_className: (0, ($parcel$interopDefault($35d34152ff885d5c$exports)))['spectrum-Menu-description'],
                ...descriptionProps
            },
            keyboard: {
                UNSAFE_className: (0, ($parcel$interopDefault($35d34152ff885d5c$exports)))['spectrum-Menu-keyboard'],
                ...keyboardShortcutProps
            },
            chevron: {
                UNSAFE_className: (0, ($parcel$interopDefault($35d34152ff885d5c$exports)))['spectrum-Menu-chevron'],
                size: 'S'
            }
        }
    }, contents, isSelected && /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, ($parcel$interopDefault($lz9FT$spectrumiconsuiCheckmarkMedium))), {
        slot: "checkmark",
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), 'spectrum-Menu-checkmark')
    }), isUnavailable && /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, ($parcel$interopDefault($lz9FT$spectrumiconsworkflowInfoOutline))), {
        slot: "end",
        size: "XS",
        alignSelf: "center",
        "aria-label": stringFormatter.format('unavailable')
    }), isUnavailable == null && isSubmenuTrigger && (direction === 'rtl' ? /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, ($parcel$interopDefault($lz9FT$spectrumiconsworkflowChevronLeft))), {
        slot: "chevron"
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($lz9FT$react))).createElement((0, ($parcel$interopDefault($lz9FT$spectrumiconsworkflowChevronRight))), {
        slot: "chevron"
    })))))));
}
