// @adobe/react-spectrum/dist/private/menu/SubmenuTrigger.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {MenuContext as $a4d1910f0ff9d033$export$c7e742effb1c51e2, SubmenuTriggerContext as $a4d1910f0ff9d033$export$8d97fe02339fc0e3, useMenuStateContext as $a4d1910f0ff9d033$export$efa3856fc0e85e7f} from "./context.js";
import {Popover as $2fa1c97e743ad66b$export$5b6b19405a83ff9d} from "../overlays/Popover.js";
import "../menu_vars.css";
import $jyqUi$menu_vars_cssmjs from "../menu_vars_css.mjs";
import {useIsMobileDevice as $196ab9279fe71c29$export$736bf165441b18c7} from "../utils/useIsMobileDevice.js";
import {isFocusWithin as $jyqUi$isFocusWithin} from "react-aria/private/utils/shadowdom/DOMFunctions";
import {mergeProps as $jyqUi$mergeProps} from "react-aria/mergeProps";
import $jyqUi$react, {useRef as $jyqUi$useRef} from "react";
import $jyqUi$reactdom from "react-dom";
import {useLocale as $jyqUi$useLocale} from "react-aria/I18nProvider";
import {useSubmenuTrigger as $jyqUi$useSubmenuTrigger} from "react-aria/useMenu";
import {useSubmenuTriggerState as $jyqUi$useSubmenuTriggerState} from "react-stately/useMenuTriggerState";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 











function $0d6e4ea605197c86$var$SubmenuTrigger(props) {
    let triggerRef = (0, $jyqUi$useRef)(null);
    let { children: children, targetKey: targetKey } = props;
    let [menuTrigger, menu] = (0, $jyqUi$react).Children.toArray(children);
    let { popoverContainer: popoverContainer, trayContainerRef: trayContainerRef, menu: parentMenuRef, submenu: menuRef, rootMenuTriggerState: rootMenuTriggerState } = (0, $a4d1910f0ff9d033$export$efa3856fc0e85e7f)();
    let submenuTriggerState = (0, $jyqUi$useSubmenuTriggerState)({
        triggerKey: targetKey
    }, rootMenuTriggerState);
    let { submenuTriggerProps: submenuTriggerProps, submenuProps: submenuProps, popoverProps: popoverProps } = (0, $jyqUi$useSubmenuTrigger)({
        parentMenuRef: parentMenuRef,
        submenuRef: menuRef
    }, submenuTriggerState, triggerRef);
    let isMobile = (0, $196ab9279fe71c29$export$736bf165441b18c7)();
    let onBackButtonPress = ()=>{
        submenuTriggerState.close();
        if (parentMenuRef.current && !(0, $jyqUi$isFocusWithin)(parentMenuRef.current)) parentMenuRef.current.focus();
    };
    let { direction: direction } = (0, $jyqUi$useLocale)();
    let mobileSubmenuKeyDown = (e)=>{
        switch(e.key){
            case 'ArrowLeft':
                var _triggerRef_current;
                if (direction === 'ltr') (_triggerRef_current = triggerRef.current) === null || _triggerRef_current === void 0 ? void 0 : _triggerRef_current.focus();
                break;
            case 'ArrowRight':
                var _triggerRef_current1;
                if (direction === 'rtl') (_triggerRef_current1 = triggerRef.current) === null || _triggerRef_current1 === void 0 ? void 0 : _triggerRef_current1.focus();
                break;
        }
    };
    let overlay;
    if (isMobile) {
        var _submenuProps;
        delete submenuTriggerProps.onBlur;
        delete submenuTriggerProps.onHoverChange;
        var _autoFocus;
        (_autoFocus = (_submenuProps = submenuProps).autoFocus) !== null && _autoFocus !== void 0 ? _autoFocus : _submenuProps.autoFocus = true;
        if (trayContainerRef.current && submenuTriggerState.isOpen) overlay = /*#__PURE__*/ (0, $jyqUi$reactdom).createPortal(menu, trayContainerRef.current);
    } else {
        let onDismissButtonPress = ()=>{
            var _parentMenuRef_current;
            submenuTriggerState.close();
            (_parentMenuRef_current = parentMenuRef.current) === null || _parentMenuRef_current === void 0 ? void 0 : _parentMenuRef_current.focus();
        };
        overlay = /*#__PURE__*/ (0, $jyqUi$react).createElement((0, $2fa1c97e743ad66b$export$5b6b19405a83ff9d), {
            ...popoverProps,
            onDismissButtonPress: onDismissButtonPress,
            UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($jyqUi$menu_vars_cssmjs))), 'spectrum-Submenu-popover'),
            container: popoverContainer,
            containerPadding: 0,
            enableBothDismissButtons: true,
            UNSAFE_style: {
                clipPath: 'unset',
                overflow: 'visible',
                borderWidth: '0px'
            },
            state: submenuTriggerState,
            triggerRef: triggerRef,
            scrollRef: menuRef,
            placement: "end top",
            hideArrow: true
        }, menu);
    }
    let menuContext = {
        ...(0, $jyqUi$mergeProps)(submenuProps, {
            ref: menuRef,
            UNSAFE_style: isMobile ? {
                width: '100%',
                maxHeight: 'inherit'
            } : undefined,
            UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($jyqUi$menu_vars_cssmjs))), {
                'spectrum-Menu-popover': !isMobile
            }),
            ...isMobile && {
                onBackButtonPress: onBackButtonPress,
                onKeyDown: mobileSubmenuKeyDown
            }
        })
    };
    return /*#__PURE__*/ (0, $jyqUi$react).createElement((0, $jyqUi$react).Fragment, null, /*#__PURE__*/ (0, $jyqUi$react).createElement((0, $a4d1910f0ff9d033$export$8d97fe02339fc0e3).Provider, {
        value: {
            triggerRef: triggerRef,
            ...submenuTriggerProps
        }
    }, menuTrigger), /*#__PURE__*/ (0, $jyqUi$react).createElement((0, $a4d1910f0ff9d033$export$c7e742effb1c51e2).Provider, {
        value: menuContext
    }, overlay));
}
$0d6e4ea605197c86$var$SubmenuTrigger.getCollectionNode = function*(props) {
    let childArray = [];
    (0, $jyqUi$react).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, $jyqUi$react).isValidElement(child)) childArray.push(child);
    });
    let [trigger] = childArray;
    let [, content] = props.children;
    yield {
        element: /*#__PURE__*/ (0, $jyqUi$react).cloneElement(trigger, {
            ...trigger.props,
            hasChildItems: true,
            isTrigger: true
        }),
        wrapper: (element)=>/*#__PURE__*/ (0, $jyqUi$react).createElement($0d6e4ea605197c86$var$SubmenuTrigger, {
                key: element.key,
                targetKey: element.key,
                ...props
            }, element, content)
    };
};
let $0d6e4ea605197c86$export$ecabc99eeffab7ca = $0d6e4ea605197c86$var$SubmenuTrigger;


export {$0d6e4ea605197c86$export$ecabc99eeffab7ca as SubmenuTrigger};
