// @adobe/react-spectrum/dist/private/menu/SubmenuTrigger.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $827c4fa3df3c822d$exports = require("./context.cjs");
var $39ed1c805b59752f$exports = require("../overlays/Popover.cjs");
require("../menu_vars.css");
var $35d34152ff885d5c$exports = require("../menu_vars_css.cjs");
var $0b97cdf6ccc1e502$exports = require("../utils/useIsMobileDevice.cjs");
var $8ibdT$reactariaprivateutilsshadowdomDOMFunctions = require("react-aria/private/utils/shadowdom/DOMFunctions");
var $8ibdT$reactariamergeProps = require("react-aria/mergeProps");
var $8ibdT$react = require("react");
var $8ibdT$reactdom = require("react-dom");
var $8ibdT$reactariaI18nProvider = require("react-aria/I18nProvider");
var $8ibdT$reactariauseMenu = require("react-aria/useMenu");
var $8ibdT$reactstatelyuseMenuTriggerState = require("react-stately/useMenuTriggerState");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "SubmenuTrigger", function () { return $e8f09a94a9b12d55$export$ecabc99eeffab7ca; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 











function $e8f09a94a9b12d55$var$SubmenuTrigger(props) {
    let triggerRef = (0, $8ibdT$react.useRef)(null);
    let { children: children, targetKey: targetKey } = props;
    let [menuTrigger, menu] = (0, ($parcel$interopDefault($8ibdT$react))).Children.toArray(children);
    let { popoverContainer: popoverContainer, trayContainerRef: trayContainerRef, menu: parentMenuRef, submenu: menuRef, rootMenuTriggerState: rootMenuTriggerState } = (0, $827c4fa3df3c822d$exports.useMenuStateContext)();
    let submenuTriggerState = (0, $8ibdT$reactstatelyuseMenuTriggerState.useSubmenuTriggerState)({
        triggerKey: targetKey
    }, rootMenuTriggerState);
    let { submenuTriggerProps: submenuTriggerProps, submenuProps: submenuProps, popoverProps: popoverProps } = (0, $8ibdT$reactariauseMenu.useSubmenuTrigger)({
        parentMenuRef: parentMenuRef,
        submenuRef: menuRef
    }, submenuTriggerState, triggerRef);
    let isMobile = (0, $0b97cdf6ccc1e502$exports.useIsMobileDevice)();
    let onBackButtonPress = ()=>{
        submenuTriggerState.close();
        if (parentMenuRef.current && !(0, $8ibdT$reactariaprivateutilsshadowdomDOMFunctions.isFocusWithin)(parentMenuRef.current)) parentMenuRef.current.focus();
    };
    let { direction: direction } = (0, $8ibdT$reactariaI18nProvider.useLocale)();
    let mobileSubmenuKeyDown = (e)=>{
        switch(e.key){
            case 'ArrowLeft':
                if (direction === 'ltr') triggerRef.current?.focus();
                break;
            case 'ArrowRight':
                if (direction === 'rtl') triggerRef.current?.focus();
                break;
        }
    };
    let overlay;
    if (isMobile) {
        delete submenuTriggerProps.onBlur;
        delete submenuTriggerProps.onHoverChange;
        submenuProps.autoFocus ??= true;
        if (trayContainerRef.current && submenuTriggerState.isOpen) overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$reactdom))).createPortal(menu, trayContainerRef.current);
    } else {
        let onDismissButtonPress = ()=>{
            submenuTriggerState.close();
            parentMenuRef.current?.focus();
        };
        overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$react))).createElement((0, $39ed1c805b59752f$exports.Popover), {
            ...popoverProps,
            onDismissButtonPress: onDismissButtonPress,
            UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), 'spectrum-Submenu-popover'),
            container: popoverContainer,
            containerPadding: 0,
            enableBothDismissButtons: true,
            UNSAFE_style: {
                clipPath: 'unset',
                overflow: 'visible',
                borderWidth: '0px'
            },
            state: submenuTriggerState,
            triggerRef: triggerRef,
            scrollRef: menuRef,
            placement: "end top",
            hideArrow: true
        }, menu);
    }
    let menuContext = {
        ...(0, $8ibdT$reactariamergeProps.mergeProps)(submenuProps, {
            ref: menuRef,
            UNSAFE_style: isMobile ? {
                width: '100%',
                maxHeight: 'inherit'
            } : undefined,
            UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($35d34152ff885d5c$exports))), {
                'spectrum-Menu-popover': !isMobile
            }),
            ...isMobile && {
                onBackButtonPress: onBackButtonPress,
                onKeyDown: mobileSubmenuKeyDown
            }
        })
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$react))).createElement((0, ($parcel$interopDefault($8ibdT$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$react))).createElement((0, $827c4fa3df3c822d$exports.SubmenuTriggerContext).Provider, {
        value: {
            triggerRef: triggerRef,
            ...submenuTriggerProps
        }
    }, menuTrigger), /*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$react))).createElement((0, $827c4fa3df3c822d$exports.MenuContext).Provider, {
        value: menuContext
    }, overlay));
}
$e8f09a94a9b12d55$var$SubmenuTrigger.getCollectionNode = function*(props) {
    let childArray = [];
    (0, ($parcel$interopDefault($8ibdT$react))).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$react))).isValidElement(child)) childArray.push(child);
    });
    let [trigger] = childArray;
    let [, content] = props.children;
    yield {
        element: /*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$react))).cloneElement(trigger, {
            ...trigger.props,
            hasChildItems: true,
            isTrigger: true
        }),
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($8ibdT$react))).createElement($e8f09a94a9b12d55$var$SubmenuTrigger, {
                key: element.key,
                targetKey: element.key,
                ...props
            }, element, content)
    };
};
let $e8f09a94a9b12d55$export$ecabc99eeffab7ca = $e8f09a94a9b12d55$var$SubmenuTrigger;
