// @adobe/react-spectrum/dist/private/list/InsertionIndicator.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
require("./styles.css");
var $9730d29fe3ac43ea$exports = require("./styles_css.cjs");
var $65a78aedcedf442b$exports = require("./ListView.cjs");
var $NMNEK$react = require("react");
var $NMNEK$reactariaVisuallyHidden = require("react-aria/VisuallyHidden");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "default", function () { return $fafdda45927e7fac$export$2e2bcd8739ae039; });





function $fafdda45927e7fac$export$2e2bcd8739ae039(props) {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks } = (0, $NMNEK$react.useContext)((0, $65a78aedcedf442b$exports.ListViewContext));
    const { target: target, isPresentationOnly: isPresentationOnly } = props;
    let ref = (0, $NMNEK$react.useRef)(null);
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator(props, dropState, ref);
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $NMNEK$reactariaVisuallyHidden.useVisuallyHidden)();
    let isDropTarget = dropState.isDropTarget(target);
    if (!isDropTarget && dropIndicatorProps['aria-hidden']) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($NMNEK$react))).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps['aria-hidden']
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($NMNEK$react))).createElement("div", {
        role: "gridcell",
        "aria-selected": "false",
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListViewInsertionIndicator', {
            'react-spectrum-ListViewInsertionIndicator--dropTarget': isDropTarget
        })
    }, !isPresentationOnly && /*#__PURE__*/ (0, ($parcel$interopDefault($NMNEK$react))).createElement("div", {
        ...visuallyHiddenProps,
        role: "button",
        ...dropIndicatorProps,
        ref: ref
    })));
}
