// @adobe/react-spectrum/dist/private/list/ListView.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $fafdda45927e7fac$exports = require("./InsertionIndicator.cjs");
var $236758a19a84d732$exports = require("./intlStrings.cjs");
require("./styles.css");
var $9730d29fe3ac43ea$exports = require("./styles_css.cjs");
var $425ebe0ef8a16c9d$exports = require("./ListViewItem.cjs");
var $d428a6aed80ea2ae$exports = require("./ListViewLayout.cjs");
var $948c2416aa3a9507$exports = require("../progress/ProgressCircle.cjs");
var $13cbcef955aa8766$exports = require("./RootDropIndicator.cjs");
var $bd2ba47d5f5b19a2$exports = require("./DragPreview.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $b4U5H$reactariauseGridList = require("react-aria/useGridList");
var $b4U5H$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $b4U5H$reactariaFocusRing = require("react-aria/FocusRing");
var $b4U5H$reactariaFocusScope = require("react-aria/FocusScope");
var $b4U5H$reactariaListKeyboardDelegate = require("react-aria/ListKeyboardDelegate");
var $b4U5H$reactstatelyuseListState = require("react-stately/useListState");
var $b4U5H$reactariamergeProps = require("react-aria/mergeProps");
var $b4U5H$react = require("react");
var $b4U5H$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $b4U5H$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $b4U5H$reactariaprivatevirtualizerVirtualizer = require("react-aria/private/virtualizer/Virtualizer");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ListViewContext", function () { return $65a78aedcedf442b$export$870039b0abfe3de0; });
$parcel$export(module.exports, "ListView", function () { return $65a78aedcedf442b$export$84d0dd190d551cd1; });
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






















const $65a78aedcedf442b$export$870039b0abfe3de0 = /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createContext(null);
const $65a78aedcedf442b$var$ROW_HEIGHTS = {
    compact: {
        medium: 32,
        large: 40
    },
    regular: {
        medium: 40,
        large: 50
    },
    spacious: {
        medium: 48,
        large: 60
    }
};
function $65a78aedcedf442b$var$useListLayout(state, density, overflowMode) {
    let { scale: scale } = (0, $544fc82701fc93e9$exports.useProvider)();
    let layout = (0, $b4U5H$react.useMemo)(()=>new (0, $d428a6aed80ea2ae$exports.ListViewLayout)({
            estimatedRowHeight: $65a78aedcedf442b$var$ROW_HEIGHTS[density || 'regular'][scale]
        }), // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        scale,
        density,
        overflowMode
    ]);
    return layout;
}
const $65a78aedcedf442b$export$84d0dd190d551cd1 = /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).forwardRef(function ListView(props, ref) {
    let { density: density = 'regular', loadingState: loadingState, onLoadMore: onLoadMore, isQuiet: isQuiet, overflowMode: overflowMode = 'truncate', onAction: onAction, dragAndDropHooks: dragAndDropHooks, renderEmptyState: renderEmptyState, ...otherProps } = props;
    let isListDraggable = !!dragAndDropHooks?.useDraggableCollectionState;
    let isListDroppable = !!dragAndDropHooks?.useDroppableCollectionState;
    let dragHooksProvided = (0, $b4U5H$react.useRef)(isListDraggable);
    let dropHooksProvided = (0, $b4U5H$react.useRef)(isListDroppable);
    (0, $b4U5H$react.useEffect)(()=>{
        if (dragHooksProvided.current !== isListDraggable && process.env.NODE_ENV !== 'production') console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
        if (dropHooksProvided.current !== isListDroppable && process.env.NODE_ENV !== 'production') console.warn('Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
    }, [
        isListDraggable,
        isListDroppable
    ]);
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let state = (0, $b4U5H$reactstatelyuseListState.useListState)({
        ...props,
        selectionBehavior: props.selectionStyle === 'highlight' ? 'replace' : 'toggle'
    });
    let { collection: collection, selectionManager: selectionManager } = state;
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    let dragState = null;
    let preview = (0, $b4U5H$react.useRef)(null);
    if (isListDraggable && dragAndDropHooks) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: collection,
            selectionManager: selectionManager,
            preview: preview
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, domRef);
    }
    let layout = $65a78aedcedf442b$var$useListLayout(state, props.density || 'regular', overflowMode);
    let DragPreview = dragAndDropHooks?.DragPreview;
    let dropState = null;
    let droppableCollection = null;
    let isRootDropTarget = false;
    if (isListDroppable && dragAndDropHooks) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: collection,
            selectionManager: selectionManager
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: new (0, $b4U5H$reactariaListKeyboardDelegate.ListKeyboardDelegate)({
                collection: collection,
                disabledKeys: dragState?.draggingKeys.size ? undefined : selectionManager.disabledKeys,
                ref: domRef,
                layoutDelegate: layout
            }),
            dropTargetDelegate: layout
        }, dropState, domRef);
        isRootDropTarget = dropState.isDropTarget({
            type: 'root'
        });
    }
    let { gridProps: gridProps } = (0, $b4U5H$reactariauseGridList.useGridList)({
        ...props,
        isVirtualized: true,
        layoutDelegate: layout,
        onAction: onAction
    }, state, domRef);
    let focusedKey = selectionManager.focusedKey;
    let dropTargetKey = null;
    if (dropState?.target?.type === 'item') {
        dropTargetKey = dropState.target.key;
        if (dropState.target.dropPosition === 'after') // Normalize to the "before" drop position since we only render those in the DOM.
        dropTargetKey = state.collection.getKeyAfter(dropTargetKey) ?? dropTargetKey;
    }
    let persistedKeys = (0, $b4U5H$react.useMemo)(()=>{
        return new Set([
            focusedKey,
            dropTargetKey
        ].filter((k)=>k !== null));
    }, [
        focusedKey,
        dropTargetKey
    ]);
    // wait for layout to get accurate measurements
    let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = (0, $b4U5H$react.useState)(false);
    let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = (0, $b4U5H$react.useState)(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $b4U5H$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        if (domRef.current) {
            // 2 is the width of the border which is not part of the box size
            setVerticalScollbarVisible(domRef.current.clientWidth + 2 < domRef.current.offsetWidth);
            setHorizontalScollbarVisible(domRef.current.clientHeight + 2 < domRef.current.offsetHeight);
        }
    });
    let hasAnyChildren = (0, $b4U5H$react.useMemo)(()=>[
            ...collection
        ].some((item)=>item.hasChildNodes), [
        collection
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement($65a78aedcedf442b$export$870039b0abfe3de0.Provider, {
        value: {
            state: state,
            dragState: dragState,
            dropState: dropState,
            dragAndDropHooks: dragAndDropHooks,
            onAction: onAction,
            isListDraggable: isListDraggable,
            isListDroppable: isListDroppable,
            layout: layout,
            loadingState: loadingState,
            renderEmptyState: renderEmptyState
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $b4U5H$reactariaFocusScope.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $b4U5H$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $b4U5H$reactariaprivatevirtualizerVirtualizer.Virtualizer), {
        ...(0, $b4U5H$reactariamergeProps.mergeProps)(isListDroppable ? droppableCollection?.collectionProps : null, gridProps),
        ...(0, $b4U5H$reactariafilterDOMProps.filterDOMProps)(otherProps),
        ...gridProps,
        ...styleProps,
        onScroll: undefined,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        ref: domRef,
        persistedKeys: persistedKeys,
        scrollDirection: "vertical",
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListView', `react-spectrum-ListView--${density}`, 'react-spectrum-ListView--emphasized', {
            'react-spectrum-ListView--quiet': isQuiet,
            'react-spectrum-ListView--loadingMore': loadingState === 'loadingMore',
            'react-spectrum-ListView--draggable': !!isListDraggable,
            'react-spectrum-ListView--dropTarget': !!isRootDropTarget,
            'react-spectrum-ListView--isVerticalScrollbarVisible': isVerticalScrollbarVisible,
            'react-spectrum-ListView--isHorizontalScrollbarVisible': isHorizontalScrollbarVisible,
            'react-spectrum-ListView--hasAnyChildren': hasAnyChildren,
            'react-spectrum-ListView--wrap': overflowMode === 'wrap'
        }, styleProps.className),
        layout: layout,
        layoutOptions: (0, $b4U5H$react.useMemo)(()=>({
                isLoading: isLoading
            }), [
            isLoading
        ]),
        collection: collection
    }, (0, $b4U5H$react.useCallback)((type, item)=>{
        if (type === 'item') return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement($65a78aedcedf442b$var$Item, {
            item: item
        });
        else if (type === 'loader') return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement($65a78aedcedf442b$var$LoadingView, null);
        else if (type === 'placeholder') return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement($65a78aedcedf442b$var$EmptyState, null);
    }, [])))), DragPreview && isListDraggable && dragAndDropHooks && dragState && /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement(DragPreview, {
        ref: preview
    }, ()=>{
        if (dragState.draggedKey == null) return null;
        if (dragAndDropHooks.renderPreview) return dragAndDropHooks.renderPreview(dragState.draggingKeys, dragState.draggedKey);
        let item = state.collection.getItem(dragState.draggedKey);
        if (!item) return null;
        let itemCount = dragState.draggingKeys.size;
        let itemHeight = layout.getLayoutInfo(dragState.draggedKey)?.rect.height ?? 0;
        return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $bd2ba47d5f5b19a2$exports.DragPreview), {
            item: item,
            itemCount: itemCount,
            itemHeight: itemHeight,
            density: density
        });
    }));
});
function $65a78aedcedf442b$var$Item({ item: item }) {
    let { isListDroppable: isListDroppable, state: state, onAction: onAction } = (0, $b4U5H$react.useContext)($65a78aedcedf442b$export$870039b0abfe3de0);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, ($parcel$interopDefault($b4U5H$react))).Fragment, null, isListDroppable && state.collection.getKeyBefore(item.key) == null && /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $13cbcef955aa8766$exports.default), {
        key: "root"
    }), isListDroppable && /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $fafdda45927e7fac$exports.default), {
        key: `${item.key}-before`,
        target: {
            key: item.key,
            type: 'item',
            dropPosition: 'before'
        }
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $425ebe0ef8a16c9d$exports.ListViewItem), {
        item: item,
        isEmphasized: true,
        hasActions: !!onAction
    }), isListDroppable && /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $fafdda45927e7fac$exports.default), {
        key: `${item.key}-after`,
        target: {
            key: item.key,
            type: 'item',
            dropPosition: 'after'
        },
        isPresentationOnly: state.collection.getKeyAfter(item.key) != null
    }));
}
function $65a78aedcedf442b$var$LoadingView() {
    let { state: state } = (0, $b4U5H$react.useContext)($65a78aedcedf442b$export$870039b0abfe3de0);
    let stringFormatter = (0, $b4U5H$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($236758a19a84d732$exports))), '@react-spectrum/list');
    return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement($65a78aedcedf442b$var$CenteredWrapper, null, /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement((0, $948c2416aa3a9507$exports.ProgressCircle), {
        isIndeterminate: true,
        "aria-label": state.collection.size > 0 ? stringFormatter.format('loadingMore') : stringFormatter.format('loading')
    }));
}
function $65a78aedcedf442b$var$EmptyState() {
    let { renderEmptyState: renderEmptyState } = (0, $b4U5H$react.useContext)($65a78aedcedf442b$export$870039b0abfe3de0);
    let emptyState = renderEmptyState ? renderEmptyState() : null;
    if (emptyState == null) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement($65a78aedcedf442b$var$CenteredWrapper, null, emptyState);
}
function $65a78aedcedf442b$var$CenteredWrapper({ children: children }) {
    let { state: state } = (0, $b4U5H$react.useContext)($65a78aedcedf442b$export$870039b0abfe3de0);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListView-centeredWrapper', {
            'react-spectrum-ListView-centeredWrapper--loadingMore': state.collection.size > 0
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($b4U5H$react))).createElement("div", {
        role: "gridcell"
    }, children));
}
