// @adobe/react-spectrum/dist/private/list/ListViewItem.cjs
var $9bc060484abc63af$exports = require("../checkbox/Checkbox.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
var $d6479700d21b596b$exports = require("../layout/Grid.cjs");
require("./styles.css");
var $9730d29fe3ac43ea$exports = require("./styles_css.cjs");
var $65a78aedcedf442b$exports = require("./ListView.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $15e3b68ec42125a9$exports = require("../text/Text.cjs");
var $dd6348d4a1a51ff9$exports = require("../utils/useHasChild.cjs");
var $jKEvN$spectrumiconsuiChevronLeftMedium = require("@spectrum-icons/ui/ChevronLeftMedium");
var $jKEvN$spectrumiconsuiChevronRightMedium = require("@spectrum-icons/ui/ChevronRightMedium");
var $jKEvN$reacttransitiongroup = require("react-transition-group");
var $jKEvN$reactariaFocusRing = require("react-aria/FocusRing");
var $jKEvN$reactariaprivateinteractionsuseFocusVisible = require("react-aria/private/interactions/useFocusVisible");
var $jKEvN$spectrumiconsuiListGripper = require("@spectrum-icons/ui/ListGripper");
var $jKEvN$reactariamergeProps = require("react-aria/mergeProps");
var $jKEvN$react = require("react");
var $jKEvN$reactariauseButton = require("react-aria/useButton");
var $jKEvN$reactariauseFocusRing = require("react-aria/useFocusRing");
var $jKEvN$reactariauseGridList = require("react-aria/useGridList");
var $jKEvN$reactariauseHover = require("react-aria/useHover");
var $jKEvN$reactariaI18nProvider = require("react-aria/I18nProvider");
var $jKEvN$reactariaVisuallyHidden = require("react-aria/VisuallyHidden");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ListViewItem", function () { return $425ebe0ef8a16c9d$export$c6bde0c04b033c0e; });
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






















function $425ebe0ef8a16c9d$export$c6bde0c04b033c0e(props) {
    let { item: item, isEmphasized: isEmphasized } = props;
    let { state: state, dragState: dragState, dropState: dropState, isListDraggable: isListDraggable, isListDroppable: isListDroppable, layout: layout, dragAndDropHooks: dragAndDropHooks, loadingState: loadingState } = (0, $jKEvN$react.useContext)((0, $65a78aedcedf442b$exports.ListViewContext));
    let { direction: direction } = (0, $jKEvN$reactariaI18nProvider.useLocale)();
    let rowRef = (0, $jKEvN$react.useRef)(null);
    let checkboxWrapperRef = (0, $jKEvN$react.useRef)(null);
    let { isFocusVisible: isFocusVisibleWithin, focusProps: focusWithinProps } = (0, $jKEvN$reactariauseFocusRing.useFocusRing)({
        within: true
    });
    let { isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $jKEvN$reactariauseFocusRing.useFocusRing)();
    let { rowProps: rowProps, gridCellProps: gridCellProps, isPressed: isPressed, descriptionProps: descriptionProps, isSelected: isSelected, isDisabled: isDisabled, allowsSelection: allowsSelection, hasAction: hasAction } = (0, $jKEvN$reactariauseGridList.useGridListItem)({
        node: item,
        isVirtualized: true,
        shouldSelectOnPressUp: isListDraggable
    }, state, rowRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $jKEvN$reactariauseHover.useHover)({
        isDisabled: !allowsSelection && !hasAction
    });
    let { checkboxProps: checkboxProps } = (0, $jKEvN$reactariauseGridList.useGridListSelectionCheckbox)({
        key: item.key
    }, state);
    let hasDescription = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-description']}`, rowRef);
    let draggableItem = null;
    if (isListDraggable && dragAndDropHooks && dragState) {
        draggableItem = dragAndDropHooks.useDraggableItem({
            key: item.key,
            hasDragButton: true
        }, dragState);
        if (isDisabled) draggableItem = null;
    }
    let isDropTarget = false;
    let dropIndicator = null;
    let dropIndicatorRef = (0, $jKEvN$react.useRef)(null);
    if (isListDroppable && dragAndDropHooks && dropState) {
        let target = {
            type: 'item',
            key: item.key,
            dropPosition: 'on'
        };
        isDropTarget = dropState.isDropTarget(target);
        dropIndicator = dragAndDropHooks.useDropIndicator({
            target: target
        }, dropState, dropIndicatorRef);
    }
    let dragButtonRef = (0, ($parcel$interopDefault($jKEvN$react))).useRef(null);
    let { buttonProps: buttonProps } = (0, $jKEvN$reactariauseButton.useButton)({
        ...draggableItem?.dragButtonProps,
        elementType: 'div'
    }, dragButtonRef);
    let chevron = direction === 'ltr' ? /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, ($parcel$interopDefault($jKEvN$spectrumiconsuiChevronRightMedium))), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListViewItem-parentIndicator', {
            'react-spectrum-ListViewItem-parentIndicator--hasChildItems': item.props.hasChildItems,
            'is-disabled': !hasAction
        })
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, ($parcel$interopDefault($jKEvN$spectrumiconsuiChevronLeftMedium))), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListViewItem-parentIndicator', {
            'react-spectrum-ListViewItem-parentIndicator--hasChildItems': item.props.hasChildItems,
            'is-disabled': !hasAction
        })
    });
    let showCheckbox = state.selectionManager.selectionMode !== 'none' && state.selectionManager.selectionBehavior === 'toggle';
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $jKEvN$reactariaVisuallyHidden.useVisuallyHidden)();
    const mergedProps = (0, $jKEvN$reactariamergeProps.mergeProps)(rowProps, draggableItem?.dragProps, hoverProps, focusWithinProps, focusProps);
    // Remove tab index from list row if performing a screenreader drag. This prevents TalkBack from focusing the row,
    // allowing for single swipe navigation between row drop indicator
    if (dragAndDropHooks?.isVirtualDragging?.()) mergedProps.tabIndex = undefined;
    let isFirstRow = item.prevKey == null;
    let isLastRow = item.nextKey == null;
    // Figure out if the ListView content is equal or greater in height to the container. If so, we'll need to round the bottom
    // border corners of the last row when selected and we can get rid of the bottom border if it isn't selected to avoid border overlap
    // with bottom border
    let isFlushWithContainerBottom = false;
    if (isLastRow && loadingState !== 'loadingMore') {
        if (layout.getContentSize()?.height >= (layout.virtualizer?.visibleRect.height ?? 0)) isFlushWithContainerBottom = true;
    }
    // previous item isn't selected
    // and the previous item isn't focused or, if it is focused, then if focus globally isn't visible or just focus isn't in the listview
    let roundTops = !(item.prevKey != null && state.selectionManager.isSelected(item.prevKey)) && (state.selectionManager.focusedKey !== item.prevKey || !((0, $jKEvN$reactariaprivateinteractionsuseFocusVisible.isFocusVisible)() && state.selectionManager.isFocused));
    let roundBottoms = !(item.nextKey != null && state.selectionManager.isSelected(item.nextKey)) && (state.selectionManager.focusedKey !== item.nextKey || !((0, $jKEvN$reactariaprivateinteractionsuseFocusVisible.isFocusVisible)() && state.selectionManager.isFocused));
    let content = typeof item.rendered === 'string' ? /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $15e3b68ec42125a9$exports.Text), null, item.rendered) : item.rendered;
    if (isDisabled) content = /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $544fc82701fc93e9$exports.Provider), {
        isDisabled: true
    }, content);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement("div", {
        ...mergedProps,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListView-row', {
            'focus-ring': isFocusVisible,
            'round-tops': roundTops || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key,
            'round-bottoms': roundBottoms || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key
        }),
        ref: rowRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement("div", {
        // TODO: refactor the css here now that we are focusing the row?
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListViewItem', {
            'is-active': isPressed,
            'is-focused': isFocusVisibleWithin,
            'focus-ring': isFocusVisible,
            'is-hovered': isHovered,
            'is-selected': isSelected,
            'is-disabled': isDisabled,
            'is-prev-selected': item.prevKey != null && state.selectionManager.isSelected(item.prevKey),
            'is-next-selected': item.nextKey != null && state.selectionManager.isSelected(item.nextKey),
            'react-spectrum-ListViewItem--highlightSelection': state.selectionManager.selectionBehavior === 'replace' && (isSelected || item.nextKey != null && state.selectionManager.isSelected(item.nextKey)),
            'react-spectrum-ListViewItem--dropTarget': !!isDropTarget,
            'react-spectrum-ListViewItem--firstRow': isFirstRow,
            'react-spectrum-ListViewItem--lastRow': isLastRow,
            'react-spectrum-ListViewItem--isFlushBottom': isFlushWithContainerBottom,
            'react-spectrum-ListViewItem--hasDescription': hasDescription
        }),
        ...gridCellProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $d6479700d21b596b$exports.Grid), {
        UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-grid']
    }, isListDraggable && /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement("div", {
        className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-draghandle-container']
    }, !isDisabled && /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $jKEvN$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement("div", {
        ...buttonProps,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($9730d29fe3ac43ea$exports))), 'react-spectrum-ListViewItem-draghandle-button'),
        style: !isFocusVisibleWithin ? {
            ...visuallyHiddenProps.style
        } : {},
        ref: dragButtonRef,
        draggable: "true"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, ($parcel$interopDefault($jKEvN$spectrumiconsuiListGripper))), null)))), isListDroppable && !dropIndicator?.isHidden && /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicator?.dropIndicatorProps,
        ref: dropIndicatorRef
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $jKEvN$reacttransitiongroup.CSSTransition), {
        in: showCheckbox,
        unmountOnExit: true,
        classNames: {
            enter: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-checkbox--enter'],
            enterActive: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-checkbox--enterActive'],
            exit: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-checkbox--exit'],
            exitActive: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-checkbox--exitActive']
        },
        timeout: 160,
        nodeRef: checkboxWrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement("div", {
        ref: checkboxWrapperRef,
        className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-checkboxWrapper']
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $9bc060484abc63af$exports.Checkbox), {
        ...checkboxProps,
        UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-checkbox'],
        isEmphasized: isEmphasized
    }))), /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-content']
            },
            description: {
                UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-description'],
                ...descriptionProps
            },
            illustration: {
                UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-thumbnail']
            },
            image: {
                UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-thumbnail']
            },
            actionButton: {
                UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-actions'],
                isQuiet: true
            },
            actionGroup: {
                UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-actions'],
                isQuiet: true,
                density: 'compact'
            },
            actionMenu: {
                UNSAFE_className: (0, ($parcel$interopDefault($9730d29fe3ac43ea$exports)))['react-spectrum-ListViewItem-actionmenu'],
                isQuiet: true
            }
        }
    }, content, /*#__PURE__*/ (0, ($parcel$interopDefault($jKEvN$react))).createElement((0, $feede71cddc0c5f3$exports.ClearSlots), null, chevron)))));
}
