// @adobe/react-spectrum/dist/private/list/ListView.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import $db18ac9b81a1c0a9$export$2e2bcd8739ae039 from "./InsertionIndicator.js";
import $jU4mA$intlStringsjs from "./intlStrings.js";
import "./styles.css";
import $jU4mA$styles_cssmjs from "./styles_css.mjs";
import {ListViewItem as $79675b2331570dd1$export$c6bde0c04b033c0e} from "./ListViewItem.js";
import {ListViewLayout as $d5bbdf752e3f3896$export$dab781655dfbb7d3} from "./ListViewLayout.js";
import {ProgressCircle as $277696409c391eff$export$c79b9d6b4cc92af7} from "../progress/ProgressCircle.js";
import $24b3e35414efede8$export$2e2bcd8739ae039 from "./RootDropIndicator.js";
import {DragPreview as $bf4b311ad7dda766$export$905ab40ac2179daa} from "./DragPreview.js";
import {useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useProvider as $089943c7a219141c$export$693cdb10cec23617} from "../provider/Provider.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {useGridList as $jU4mA$useGridList} from "react-aria/useGridList";
import {filterDOMProps as $jU4mA$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusRing as $jU4mA$FocusRing} from "react-aria/FocusRing";
import {FocusScope as $jU4mA$FocusScope} from "react-aria/FocusScope";
import {ListKeyboardDelegate as $jU4mA$ListKeyboardDelegate} from "react-aria/ListKeyboardDelegate";
import {useListState as $jU4mA$useListState} from "react-stately/useListState";
import {mergeProps as $jU4mA$mergeProps} from "react-aria/mergeProps";
import $jU4mA$react, {useMemo as $jU4mA$useMemo, useRef as $jU4mA$useRef, useEffect as $jU4mA$useEffect, useState as $jU4mA$useState, useCallback as $jU4mA$useCallback, useContext as $jU4mA$useContext} from "react";
import {useLayoutEffect as $jU4mA$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useLocalizedStringFormatter as $jU4mA$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {Virtualizer as $jU4mA$Virtualizer} from "react-aria/private/virtualizer/Virtualizer";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






















const $bcd1a74211acbd51$export$870039b0abfe3de0 = /*#__PURE__*/ (0, $jU4mA$react).createContext(null);
const $bcd1a74211acbd51$var$ROW_HEIGHTS = {
    compact: {
        medium: 32,
        large: 40
    },
    regular: {
        medium: 40,
        large: 50
    },
    spacious: {
        medium: 48,
        large: 60
    }
};
function $bcd1a74211acbd51$var$useListLayout(state, density, overflowMode) {
    let { scale: scale } = (0, $089943c7a219141c$export$693cdb10cec23617)();
    let layout = (0, $jU4mA$useMemo)(()=>new (0, $d5bbdf752e3f3896$export$dab781655dfbb7d3)({
            estimatedRowHeight: $bcd1a74211acbd51$var$ROW_HEIGHTS[density || 'regular'][scale]
        }), // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        scale,
        density,
        overflowMode
    ]);
    return layout;
}
const $bcd1a74211acbd51$export$84d0dd190d551cd1 = /*#__PURE__*/ (0, $jU4mA$react).forwardRef(function ListView(props, ref) {
    var _dropState_target;
    let { density: density = 'regular', loadingState: loadingState, onLoadMore: onLoadMore, isQuiet: isQuiet, overflowMode: overflowMode = 'truncate', onAction: onAction, dragAndDropHooks: dragAndDropHooks, renderEmptyState: renderEmptyState, ...otherProps } = props;
    let isListDraggable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDraggableCollectionState);
    let isListDroppable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDroppableCollectionState);
    let dragHooksProvided = (0, $jU4mA$useRef)(isListDraggable);
    let dropHooksProvided = (0, $jU4mA$useRef)(isListDroppable);
    (0, $jU4mA$useEffect)(()=>{
        if (dragHooksProvided.current !== isListDraggable && process.env.NODE_ENV !== 'production') console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
        if (dropHooksProvided.current !== isListDroppable && process.env.NODE_ENV !== 'production') console.warn('Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
    }, [
        isListDraggable,
        isListDroppable
    ]);
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let state = (0, $jU4mA$useListState)({
        ...props,
        selectionBehavior: props.selectionStyle === 'highlight' ? 'replace' : 'toggle'
    });
    let { collection: collection, selectionManager: selectionManager } = state;
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    let dragState = null;
    let preview = (0, $jU4mA$useRef)(null);
    if (isListDraggable && dragAndDropHooks) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: collection,
            selectionManager: selectionManager,
            preview: preview
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, domRef);
    }
    let layout = $bcd1a74211acbd51$var$useListLayout(state, props.density || 'regular', overflowMode);
    let DragPreview = dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.DragPreview;
    let dropState = null;
    let droppableCollection = null;
    let isRootDropTarget = false;
    if (isListDroppable && dragAndDropHooks) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: collection,
            selectionManager: selectionManager
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: new (0, $jU4mA$ListKeyboardDelegate)({
                collection: collection,
                disabledKeys: (dragState === null || dragState === void 0 ? void 0 : dragState.draggingKeys.size) ? undefined : selectionManager.disabledKeys,
                ref: domRef,
                layoutDelegate: layout
            }),
            dropTargetDelegate: layout
        }, dropState, domRef);
        isRootDropTarget = dropState.isDropTarget({
            type: 'root'
        });
    }
    let { gridProps: gridProps } = (0, $jU4mA$useGridList)({
        ...props,
        isVirtualized: true,
        layoutDelegate: layout,
        onAction: onAction
    }, state, domRef);
    let focusedKey = selectionManager.focusedKey;
    let dropTargetKey = null;
    if ((dropState === null || dropState === void 0 ? void 0 : (_dropState_target = dropState.target) === null || _dropState_target === void 0 ? void 0 : _dropState_target.type) === 'item') {
        dropTargetKey = dropState.target.key;
        var _state_collection_getKeyAfter;
        if (dropState.target.dropPosition === 'after') // Normalize to the "before" drop position since we only render those in the DOM.
        dropTargetKey = (_state_collection_getKeyAfter = state.collection.getKeyAfter(dropTargetKey)) !== null && _state_collection_getKeyAfter !== void 0 ? _state_collection_getKeyAfter : dropTargetKey;
    }
    let persistedKeys = (0, $jU4mA$useMemo)(()=>{
        return new Set([
            focusedKey,
            dropTargetKey
        ].filter((k)=>k !== null));
    }, [
        focusedKey,
        dropTargetKey
    ]);
    // wait for layout to get accurate measurements
    let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = (0, $jU4mA$useState)(false);
    let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = (0, $jU4mA$useState)(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $jU4mA$useLayoutEffect)(()=>{
        if (domRef.current) {
            // 2 is the width of the border which is not part of the box size
            setVerticalScollbarVisible(domRef.current.clientWidth + 2 < domRef.current.offsetWidth);
            setHorizontalScollbarVisible(domRef.current.clientHeight + 2 < domRef.current.offsetHeight);
        }
    });
    let hasAnyChildren = (0, $jU4mA$useMemo)(()=>[
            ...collection
        ].some((item)=>item.hasChildNodes), [
        collection
    ]);
    return /*#__PURE__*/ (0, $jU4mA$react).createElement($bcd1a74211acbd51$export$870039b0abfe3de0.Provider, {
        value: {
            state: state,
            dragState: dragState,
            dropState: dropState,
            dragAndDropHooks: dragAndDropHooks,
            onAction: onAction,
            isListDraggable: isListDraggable,
            isListDroppable: isListDroppable,
            layout: layout,
            loadingState: loadingState,
            renderEmptyState: renderEmptyState
        }
    }, /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $jU4mA$FocusScope), null, /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $jU4mA$FocusRing), {
        focusRingClass: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($jU4mA$styles_cssmjs))), 'focus-ring')
    }, /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $jU4mA$Virtualizer), {
        ...(0, $jU4mA$mergeProps)(isListDroppable ? droppableCollection === null || droppableCollection === void 0 ? void 0 : droppableCollection.collectionProps : null, gridProps),
        ...(0, $jU4mA$filterDOMProps)(otherProps),
        ...gridProps,
        ...styleProps,
        onScroll: undefined,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        ref: domRef,
        persistedKeys: persistedKeys,
        scrollDirection: "vertical",
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($jU4mA$styles_cssmjs))), 'react-spectrum-ListView', `react-spectrum-ListView--${density}`, 'react-spectrum-ListView--emphasized', {
            'react-spectrum-ListView--quiet': isQuiet,
            'react-spectrum-ListView--loadingMore': loadingState === 'loadingMore',
            'react-spectrum-ListView--draggable': !!isListDraggable,
            'react-spectrum-ListView--dropTarget': !!isRootDropTarget,
            'react-spectrum-ListView--isVerticalScrollbarVisible': isVerticalScrollbarVisible,
            'react-spectrum-ListView--isHorizontalScrollbarVisible': isHorizontalScrollbarVisible,
            'react-spectrum-ListView--hasAnyChildren': hasAnyChildren,
            'react-spectrum-ListView--wrap': overflowMode === 'wrap'
        }, styleProps.className),
        layout: layout,
        layoutOptions: (0, $jU4mA$useMemo)(()=>({
                isLoading: isLoading
            }), [
            isLoading
        ]),
        collection: collection
    }, (0, $jU4mA$useCallback)((type, item)=>{
        if (type === 'item') return /*#__PURE__*/ (0, $jU4mA$react).createElement($bcd1a74211acbd51$var$Item, {
            item: item
        });
        else if (type === 'loader') return /*#__PURE__*/ (0, $jU4mA$react).createElement($bcd1a74211acbd51$var$LoadingView, null);
        else if (type === 'placeholder') return /*#__PURE__*/ (0, $jU4mA$react).createElement($bcd1a74211acbd51$var$EmptyState, null);
    }, [])))), DragPreview && isListDraggable && dragAndDropHooks && dragState && /*#__PURE__*/ (0, $jU4mA$react).createElement(DragPreview, {
        ref: preview
    }, ()=>{
        var _layout_getLayoutInfo;
        if (dragState.draggedKey == null) return null;
        if (dragAndDropHooks.renderPreview) return dragAndDropHooks.renderPreview(dragState.draggingKeys, dragState.draggedKey);
        let item = state.collection.getItem(dragState.draggedKey);
        if (!item) return null;
        let itemCount = dragState.draggingKeys.size;
        var _layout_getLayoutInfo_rect_height;
        let itemHeight = (_layout_getLayoutInfo_rect_height = (_layout_getLayoutInfo = layout.getLayoutInfo(dragState.draggedKey)) === null || _layout_getLayoutInfo === void 0 ? void 0 : _layout_getLayoutInfo.rect.height) !== null && _layout_getLayoutInfo_rect_height !== void 0 ? _layout_getLayoutInfo_rect_height : 0;
        return /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $bf4b311ad7dda766$export$905ab40ac2179daa), {
            item: item,
            itemCount: itemCount,
            itemHeight: itemHeight,
            density: density
        });
    }));
});
function $bcd1a74211acbd51$var$Item({ item: item }) {
    let { isListDroppable: isListDroppable, state: state, onAction: onAction } = (0, $jU4mA$useContext)($bcd1a74211acbd51$export$870039b0abfe3de0);
    return /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $jU4mA$react).Fragment, null, isListDroppable && state.collection.getKeyBefore(item.key) == null && /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $24b3e35414efede8$export$2e2bcd8739ae039), {
        key: "root"
    }), isListDroppable && /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $db18ac9b81a1c0a9$export$2e2bcd8739ae039), {
        key: `${item.key}-before`,
        target: {
            key: item.key,
            type: 'item',
            dropPosition: 'before'
        }
    }), /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $79675b2331570dd1$export$c6bde0c04b033c0e), {
        item: item,
        isEmphasized: true,
        hasActions: !!onAction
    }), isListDroppable && /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $db18ac9b81a1c0a9$export$2e2bcd8739ae039), {
        key: `${item.key}-after`,
        target: {
            key: item.key,
            type: 'item',
            dropPosition: 'after'
        },
        isPresentationOnly: state.collection.getKeyAfter(item.key) != null
    }));
}
function $bcd1a74211acbd51$var$LoadingView() {
    let { state: state } = (0, $jU4mA$useContext)($bcd1a74211acbd51$export$870039b0abfe3de0);
    let stringFormatter = (0, $jU4mA$useLocalizedStringFormatter)((0, ($parcel$interopDefault($jU4mA$intlStringsjs))), '@react-spectrum/list');
    return /*#__PURE__*/ (0, $jU4mA$react).createElement($bcd1a74211acbd51$var$CenteredWrapper, null, /*#__PURE__*/ (0, $jU4mA$react).createElement((0, $277696409c391eff$export$c79b9d6b4cc92af7), {
        isIndeterminate: true,
        "aria-label": state.collection.size > 0 ? stringFormatter.format('loadingMore') : stringFormatter.format('loading')
    }));
}
function $bcd1a74211acbd51$var$EmptyState() {
    let { renderEmptyState: renderEmptyState } = (0, $jU4mA$useContext)($bcd1a74211acbd51$export$870039b0abfe3de0);
    let emptyState = renderEmptyState ? renderEmptyState() : null;
    if (emptyState == null) return null;
    return /*#__PURE__*/ (0, $jU4mA$react).createElement($bcd1a74211acbd51$var$CenteredWrapper, null, emptyState);
}
function $bcd1a74211acbd51$var$CenteredWrapper({ children: children }) {
    let { state: state } = (0, $jU4mA$useContext)($bcd1a74211acbd51$export$870039b0abfe3de0);
    return /*#__PURE__*/ (0, $jU4mA$react).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($jU4mA$styles_cssmjs))), 'react-spectrum-ListView-centeredWrapper', {
            'react-spectrum-ListView-centeredWrapper--loadingMore': state.collection.size > 0
        })
    }, /*#__PURE__*/ (0, $jU4mA$react).createElement("div", {
        role: "gridcell"
    }, children));
}


export {$bcd1a74211acbd51$export$870039b0abfe3de0 as ListViewContext, $bcd1a74211acbd51$export$84d0dd190d551cd1 as ListView};
