// @adobe/react-spectrum/dist/private/dnd/useDragAndDrop.js
import {useDraggableCollection as $9iyO4$useDraggableCollection, useDraggableItem as $9iyO4$useDraggableItem, DragPreview as $9iyO4$DragPreview} from "react-aria/useDraggableCollection";
import {useDraggableCollectionState as $9iyO4$useDraggableCollectionState} from "react-stately/useDraggableCollectionState";
import {useDroppableItem as $9iyO4$useDroppableItem, useDroppableCollection as $9iyO4$useDroppableCollection, useDropIndicator as $9iyO4$useDropIndicator} from "react-aria/useDroppableCollection";
import {useDroppableCollectionState as $9iyO4$useDroppableCollectionState} from "react-stately/useDroppableCollectionState";
import {isVirtualDragging as $9iyO4$isVirtualDragging} from "react-aria/private/dnd/DragManager";
import {useMemo as $9iyO4$useMemo} from "react";

/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





function $a835b70880df29d9$export$2cfc5be7a55829f6(options) {
    let dragAndDropHooks = (0, $9iyO4$useMemo)(()=>{
        let { onDrop: onDrop, onInsert: onInsert, onItemDrop: onItemDrop, onReorder: onReorder, onRootDrop: onRootDrop, getItems: getItems, renderPreview: renderPreview } = options;
        let isDraggable = !!getItems;
        let isDroppable = !!(onDrop || onInsert || onItemDrop || onReorder || onRootDrop);
        let hooks = {};
        if (isDraggable) {
            hooks.useDraggableCollectionState = function useDraggableCollectionStateOverride(props) {
                return (0, $9iyO4$useDraggableCollectionState)({
                    ...props,
                    ...options,
                    getItems: options.getItems
                });
            };
            hooks.useDraggableCollection = (0, $9iyO4$useDraggableCollection);
            hooks.useDraggableItem = (0, $9iyO4$useDraggableItem);
            hooks.DragPreview = (0, $9iyO4$DragPreview);
            hooks.renderPreview = renderPreview;
        }
        if (isDroppable) {
            hooks.useDroppableCollectionState = function useDroppableCollectionStateOverride(props) {
                return (0, $9iyO4$useDroppableCollectionState)({
                    ...props,
                    ...options
                });
            };
            hooks.useDroppableItem = (0, $9iyO4$useDroppableItem);
            hooks.useDroppableCollection = function useDroppableCollectionOverride(props, state, ref) {
                return (0, $9iyO4$useDroppableCollection)({
                    ...props,
                    ...options
                }, state, ref);
            };
            hooks.useDropIndicator = (0, $9iyO4$useDropIndicator);
        }
        if (isDraggable || isDroppable) hooks.isVirtualDragging = (0, $9iyO4$isVirtualDragging);
        return hooks;
    }, [
        options
    ]);
    return {
        dragAndDropHooks: dragAndDropHooks
    };
}


export {$a835b70880df29d9$export$2cfc5be7a55829f6 as useDragAndDrop};
