// @adobe/react-spectrum/dist/private/dnd/useDragAndDrop.cjs
var $gvwR2$reactariauseDraggableCollection = require("react-aria/useDraggableCollection");
var $gvwR2$reactstatelyuseDraggableCollectionState = require("react-stately/useDraggableCollectionState");
var $gvwR2$reactariauseDroppableCollection = require("react-aria/useDroppableCollection");
var $gvwR2$reactstatelyuseDroppableCollectionState = require("react-stately/useDroppableCollectionState");
var $gvwR2$reactariaprivatedndDragManager = require("react-aria/private/dnd/DragManager");
var $gvwR2$react = require("react");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "useDragAndDrop", function () { return $180c0494a69c5bd7$export$2cfc5be7a55829f6; });
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





function $180c0494a69c5bd7$export$2cfc5be7a55829f6(options) {
    let dragAndDropHooks = (0, $gvwR2$react.useMemo)(()=>{
        let { onDrop: onDrop, onInsert: onInsert, onItemDrop: onItemDrop, onReorder: onReorder, onRootDrop: onRootDrop, getItems: getItems, renderPreview: renderPreview } = options;
        let isDraggable = !!getItems;
        let isDroppable = !!(onDrop || onInsert || onItemDrop || onReorder || onRootDrop);
        let hooks = {};
        if (isDraggable) {
            hooks.useDraggableCollectionState = function useDraggableCollectionStateOverride(props) {
                return (0, $gvwR2$reactstatelyuseDraggableCollectionState.useDraggableCollectionState)({
                    ...props,
                    ...options,
                    getItems: options.getItems
                });
            };
            hooks.useDraggableCollection = (0, $gvwR2$reactariauseDraggableCollection.useDraggableCollection);
            hooks.useDraggableItem = (0, $gvwR2$reactariauseDraggableCollection.useDraggableItem);
            hooks.DragPreview = (0, $gvwR2$reactariauseDraggableCollection.DragPreview);
            hooks.renderPreview = renderPreview;
        }
        if (isDroppable) {
            hooks.useDroppableCollectionState = function useDroppableCollectionStateOverride(props) {
                return (0, $gvwR2$reactstatelyuseDroppableCollectionState.useDroppableCollectionState)({
                    ...props,
                    ...options
                });
            };
            hooks.useDroppableItem = (0, $gvwR2$reactariauseDroppableCollection.useDroppableItem);
            hooks.useDroppableCollection = function useDroppableCollectionOverride(props, state, ref) {
                return (0, $gvwR2$reactariauseDroppableCollection.useDroppableCollection)({
                    ...props,
                    ...options
                }, state, ref);
            };
            hooks.useDropIndicator = (0, $gvwR2$reactariauseDroppableCollection.useDropIndicator);
        }
        if (isDraggable || isDroppable) hooks.isVirtualDragging = (0, $gvwR2$reactariaprivatedndDragManager.isVirtualDragging);
        return hooks;
    }, [
        options
    ]);
    return {
        dragAndDropHooks: dragAndDropHooks
    };
}
