// @adobe/react-spectrum/dist/private/button/Button.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $adbda93dea711ffd$exports = require("./intlStrings.cjs");
var $948c2416aa3a9507$exports = require("../progress/ProgressCircle.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
require("../button_vars.css");
var $869138cbe3b599dc$exports = require("../button_vars_css.cjs");
var $15e3b68ec42125a9$exports = require("../text/Text.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $dd6348d4a1a51ff9$exports = require("../utils/useHasChild.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $d1lFD$reactariauseButton = require("react-aria/useButton");
var $d1lFD$reactariaFocusRing = require("react-aria/FocusRing");
var $d1lFD$reactariaprivateutilsplatform = require("react-aria/private/utils/platform");
var $d1lFD$reactariamergeProps = require("react-aria/mergeProps");
var $d1lFD$react = require("react");
var $d1lFD$reactariauseFocus = require("react-aria/useFocus");
var $d1lFD$reactariauseHover = require("react-aria/useHover");
var $d1lFD$reactariauseId = require("react-aria/useId");
var $d1lFD$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Button", function () { return $92256f4fe9ec9f59$export$353f5b6fc5456de1; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


















function $92256f4fe9ec9f59$var$disablePendingProps(props) {
    // Don't allow interaction while isPending is true
    if (props.isPending) {
        props.onPress = undefined;
        props.onPressStart = undefined;
        props.onPressEnd = undefined;
        props.onPressChange = undefined;
        props.onPressUp = undefined;
        props.onKeyDown = undefined;
        props.onKeyUp = undefined;
        props.onClick = undefined;
        props.href = undefined;
    }
    return props;
}
const $92256f4fe9ec9f59$export$353f5b6fc5456de1 = /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).forwardRef(function Button(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $feede71cddc0c5f3$exports.useSlotProps)(props, 'button');
    props = $92256f4fe9ec9f59$var$disablePendingProps(props);
    let { elementType: Element = 'button', children: children, variant: variant, style: style = variant === 'accent' || variant === 'cta' ? 'fill' : 'outline', staticColor: staticColor, isDisabled: isDisabled, isPending: isPending, autoFocus: autoFocus, ...otherProps } = props;
    let domRef = (0, $65aea7b37663976b$exports.useFocusableRef)(ref);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $d1lFD$reactariauseButton.useButton)(props, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $d1lFD$reactariauseHover.useHover)({
        isDisabled: isDisabled
    });
    let [isFocused, onFocusChange] = (0, $d1lFD$react.useState)(false);
    let { focusProps: focusProps } = (0, $d1lFD$reactariauseFocus.useFocus)({
        onFocusChange: onFocusChange,
        isDisabled: isDisabled
    });
    let stringFormatter = (0, $d1lFD$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($adbda93dea711ffd$exports))), '@react-spectrum/button');
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    let hasLabel = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($869138cbe3b599dc$exports)))['spectrum-Button-label']}`, domRef);
    let hasIcon = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($869138cbe3b599dc$exports)))['spectrum-Icon']}`, domRef);
    // an aria label will block children and their labels from being read, this is undesirable for pending state
    let hasAriaLabel = !!buttonProps['aria-label'] || !!buttonProps['aria-labelledby'];
    let [isProgressVisible, setIsProgressVisible] = (0, $d1lFD$react.useState)(false);
    let backupButtonId = (0, $d1lFD$reactariauseId.useId)();
    let buttonId = buttonProps.id || backupButtonId;
    let iconId = (0, $d1lFD$reactariauseId.useId)();
    let textId = (0, $d1lFD$reactariauseId.useId)();
    let spinnerId = (0, $d1lFD$reactariauseId.useId)();
    (0, $d1lFD$react.useEffect)(()=>{
        let timeout;
        if (isPending) // Start timer when isPending is set to true.
        timeout = setTimeout(()=>{
            setIsProgressVisible(true);
        }, 1000);
        else // Exit loading state when isPending is set to false. */
        setIsProgressVisible(false);
        return ()=>{
            // Clean up on unmount or when user removes isPending prop before entering loading state.
            clearTimeout(timeout);
        };
    }, [
        isPending
    ]);
    if (variant === 'cta') variant = 'accent';
    else if (variant === 'overBackground') {
        variant = 'primary';
        staticColor = 'white';
    }
    const isPendingAriaLiveLabel = `${hasAriaLabel ? buttonProps['aria-label'] : ''} ${stringFormatter.format('pending')}`.trim();
    const isPendingAriaLiveLabelledby = hasAriaLabel ? buttonProps['aria-labelledby']?.replace(buttonId, spinnerId) ?? spinnerId : `${hasIcon ? iconId : ''} ${hasLabel ? textId : ''} ${spinnerId}`.trim();
    let ariaLive = 'polite';
    if ((0, $d1lFD$reactariaprivateutilsplatform.isAppleDevice)() && (!hasAriaLabel || (0, $d1lFD$reactariaprivateutilsplatform.isFirefox)())) ariaLive = 'off';
    let isPendingProps = isPending ? {
        onClick: (e)=>{
            if (e.currentTarget instanceof HTMLButtonElement) e.preventDefault();
        }
    } : {
        // no-op.
        // Not sure why, but TypeScript wouldn't allow to have an empty object `{}`.
        onClick: ()=>{}
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement((0, $d1lFD$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement(Element, {
        ...styleProps,
        ...(0, $d1lFD$reactariamergeProps.mergeProps)(buttonProps, hoverProps, focusProps, isPendingProps),
        id: buttonId,
        ref: domRef,
        "data-variant": variant,
        "data-style": style,
        "data-static-color": staticColor || undefined,
        "aria-disabled": isPending ? 'true' : undefined,
        "aria-label": isPending ? isPendingAriaLiveLabel : buttonProps['aria-label'],
        "aria-labelledby": isPending ? isPendingAriaLiveLabelledby : buttonProps['aria-labelledby'],
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'spectrum-Button', {
            'spectrum-Button--iconOnly': hasIcon && !hasLabel,
            'is-disabled': isDisabled || isProgressVisible,
            'is-active': isPressed,
            'is-hovered': isHovered,
            'spectrum-Button--pending': isProgressVisible
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            icon: {
                id: iconId,
                size: 'S',
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'spectrum-Icon')
            },
            text: {
                id: textId,
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'spectrum-Button-label')
            }
        }
    }, typeof children === 'string' ? /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement((0, $15e3b68ec42125a9$exports.Text), null, children) : children, isPending && /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement("div", {
        "aria-hidden": "true",
        style: {
            visibility: isProgressVisible ? 'visible' : 'hidden'
        },
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'spectrum-Button-circleLoader')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement((0, $948c2416aa3a9507$exports.ProgressCircle), {
        "aria-label": isPendingAriaLiveLabel,
        isIndeterminate: true,
        size: "S",
        staticColor: staticColor
    })), isPending && /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement((0, ($parcel$interopDefault($d1lFD$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement("div", {
        "aria-live": isFocused ? ariaLive : 'off'
    }, isProgressVisible && /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement("div", {
        role: "img",
        "aria-labelledby": isPendingAriaLiveLabelledby
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($d1lFD$react))).createElement("div", {
        id: spinnerId,
        role: "img",
        "aria-label": isPendingAriaLiveLabel
    })))));
});
