// @adobe/react-spectrum/dist/private/button/ClearButton.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
require("../button_vars.css");
var $869138cbe3b599dc$exports = require("../button_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $fYCN7$reactariauseButton = require("react-aria/useButton");
var $fYCN7$spectrumiconsuiCrossSmall = require("@spectrum-icons/ui/CrossSmall");
var $fYCN7$reactariaFocusRing = require("react-aria/FocusRing");
var $fYCN7$reactariamergeProps = require("react-aria/mergeProps");
var $fYCN7$react = require("react");
var $fYCN7$reactariauseHover = require("react-aria/useHover");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ClearButton", function () { return $0fc8553a4214494f$export$13ec83e50bf04290; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 









const $0fc8553a4214494f$export$13ec83e50bf04290 = /*#__PURE__*/ (0, ($parcel$interopDefault($fYCN7$react))).forwardRef(function ClearButton(props, ref) {
    let { children: children = /*#__PURE__*/ (0, ($parcel$interopDefault($fYCN7$react))).createElement((0, ($parcel$interopDefault($fYCN7$spectrumiconsuiCrossSmall))), {
        UNSAFE_className: (0, ($parcel$interopDefault($869138cbe3b599dc$exports)))['spectrum-Icon']
    }), focusClassName: focusClassName, variant: variant, autoFocus: autoFocus, isDisabled: isDisabled, preventFocus: preventFocus, elementType: elementType = preventFocus ? 'div' : 'button', inset: inset = false, ...otherProps } = props;
    let domRef = (0, $65aea7b37663976b$exports.useFocusableRef)(ref);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $fYCN7$reactariauseButton.useButton)({
        ...props,
        elementType: elementType
    }, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $fYCN7$reactariauseHover.useHover)({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    // For cases like the clear button in a search field, remove the tabIndex so
    // iOS 14 with VoiceOver doesn't focus the button and hide the keyboard when
    // moving the cursor over the clear button.
    if (preventFocus) delete buttonProps.tabIndex;
    let ElementType = elementType;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($fYCN7$react))).createElement((0, $fYCN7$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'focus-ring', focusClassName),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($fYCN7$react))).createElement(ElementType, {
        ...styleProps,
        ...(0, $fYCN7$reactariamergeProps.mergeProps)(buttonProps, hoverProps),
        ref: domRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), 'spectrum-ClearButton', {
            [`spectrum-ClearButton--${variant}`]: variant,
            'is-disabled': isDisabled,
            'is-active': isPressed,
            'is-hovered': isHovered,
            'spectrum-ClearButton--inset': inset
        }, styleProps.className)
    }, children));
});
