// @adobe/react-spectrum/dist/private/label/Field.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $e04035822dddb314$exports = require("../layout/Flex.cjs");
var $2b77b98944e1735c$exports = require("./HelpText.cjs");
var $b881bddc71fd043e$exports = require("./Label.cjs");
require("../fieldlabel_vars.css");
var $53185441bef09fa8$exports = require("../fieldlabel_vars_css.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
var $1af2ca8553741739$exports = require("../form/Form.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $2jTpw$reactariamergeProps = require("react-aria/mergeProps");
var $2jTpw$react = require("react");
var $2jTpw$reactariauseId = require("react-aria/useId");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Field", function () { return $b93966d678e0af07$export$a455218a85c89869; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 










const $b93966d678e0af07$export$a455218a85c89869 = /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).forwardRef(function Field(props, ref) {
    let formProps = (0, $1af2ca8553741739$exports.useFormProps)(props);
    let isInForm = formProps !== props;
    props = formProps;
    let { label: label, labelPosition: labelPosition = 'top', labelAlign: labelAlign, isRequired: isRequired, necessityIndicator: necessityIndicator, includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName, validationState: validationState, isInvalid: isInvalid, description: description, errorMessage: errorMessage = (e)=>e.validationErrors.join(' '), validationErrors: validationErrors, validationDetails: validationDetails, isDisabled: isDisabled, showErrorIcon: showErrorIcon, contextualHelp: contextualHelp, children: children, labelProps: labelProps = {}, descriptionProps: // Not every component that uses <Field> supports help text.
    descriptionProps = {}, errorMessageProps: errorMessageProps = {}, elementType: elementType, wrapperClassName: wrapperClassName, wrapperProps: wrapperProps = {}, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(otherProps);
    let errorMessageString = null;
    if (typeof errorMessage === 'function') errorMessageString = isInvalid != null && validationErrors != null && validationDetails != null ? errorMessage({
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails
    }) : null;
    else errorMessageString = errorMessage;
    let hasHelpText = !!description || errorMessageString && (isInvalid || validationState === 'invalid');
    let contextualHelpId = (0, $2jTpw$reactariauseId.useId)();
    let fallbackLabelPropsId = (0, $2jTpw$reactariauseId.useId)();
    if (label && contextualHelp && !labelProps.id) labelProps.id = fallbackLabelPropsId;
    let labelWrapperClass = (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($53185441bef09fa8$exports))), 'spectrum-Field', {
        'spectrum-Field--positionTop': labelPosition === 'top',
        'spectrum-Field--positionSide': labelPosition === 'side',
        'spectrum-Field--alignEnd': labelAlign === 'end',
        'spectrum-Field--hasContextualHelp': !!props.contextualHelp
    }, styleProps.className, wrapperClassName);
    children = /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).cloneElement(children, (0, $2jTpw$reactariamergeProps.mergeProps)(children.props, {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($53185441bef09fa8$exports))), 'spectrum-Field-field')
    }));
    let renderHelpText = ()=>/*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement((0, $2b77b98944e1735c$exports.HelpText), {
            descriptionProps: descriptionProps,
            errorMessageProps: errorMessageProps,
            description: description,
            errorMessage: errorMessageString,
            validationState: validationState,
            isInvalid: isInvalid,
            isDisabled: isDisabled,
            showErrorIcon: showErrorIcon,
            gridArea: (0, ($parcel$interopDefault($53185441bef09fa8$exports))).helpText
        });
    let renderChildren = ()=>{
        if (labelPosition === 'side') return /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement((0, $e04035822dddb314$exports.Flex), {
            direction: "column",
            UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($53185441bef09fa8$exports))), 'spectrum-Field-wrapper')
        }, children, hasHelpText && renderHelpText());
        return /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement((0, ($parcel$interopDefault($2jTpw$react))).Fragment, null, children, hasHelpText && renderHelpText());
    };
    let labelAndContextualHelp = /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement((0, ($parcel$interopDefault($2jTpw$react))).Fragment, null, label && /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement((0, $b881bddc71fd043e$exports.Label), {
        ...labelProps,
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        isRequired: isRequired,
        necessityIndicator: necessityIndicator,
        includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName,
        elementType: elementType
    }, label), label && contextualHelp && /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            actionButton: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($53185441bef09fa8$exports))), 'spectrum-Field-contextualHelp'),
                id: contextualHelpId,
                'aria-labelledby': labelProps?.id ? `${labelProps.id} ${contextualHelpId}` : undefined
            }
        }
    }, contextualHelp));
    // Need to add an extra wrapper for the label and contextual help if labelPosition is side,
    // so that the table layout works inside forms.
    if (isInForm && labelPosition === 'side' && label && contextualHelp) labelAndContextualHelp = /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($53185441bef09fa8$exports))), 'spectrum-Field-labelCell')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($53185441bef09fa8$exports))), 'spectrum-Field-labelWrapper')
    }, labelAndContextualHelp));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($2jTpw$react))).createElement("div", {
        ...styleProps,
        ...wrapperProps,
        ref: ref,
        className: labelWrapperClass
    }, labelAndContextualHelp, renderChildren());
});
