// @adobe/react-spectrum/dist/private/tag/TagGroup.cjs
var $183c5173677598aa$exports = require("../button/ActionButton.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $b93966d678e0af07$exports = require("../label/Field.cjs");
var $77e6bffa322a0b1b$exports = require("./intlStrings.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
require("../tags_vars.css");
var $74103ae0a349d695$exports = require("../tags_vars_css.cjs");
var $5bdeed39ae3015c4$exports = require("./Tag.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $1af2ca8553741739$exports = require("../form/Form.cjs");
var $80FZR$reactariauseTagGroup = require("react-aria/useTagGroup");
var $80FZR$reactariaFocusRing = require("react-aria/FocusRing");
var $80FZR$reactariaFocusScope = require("react-aria/FocusScope");
var $80FZR$reactstatelyprivatelistListCollection = require("react-stately/private/list/ListCollection");
var $80FZR$reactariaListKeyboardDelegate = require("react-aria/ListKeyboardDelegate");
var $80FZR$react = require("react");
var $80FZR$reactariauseId = require("react-aria/useId");
var $80FZR$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $80FZR$reactstatelyuseListState = require("react-stately/useListState");
var $80FZR$reactariaI18nProvider = require("react-aria/I18nProvider");
var $80FZR$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $80FZR$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");
var $80FZR$reactariaprivateutilsuseValueEffect = require("react-aria/private/utils/useValueEffect");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "TagGroup", function () { return $4275f73de689771f$export$67ea30858aaf75e3; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





















const $4275f73de689771f$var$TAG_STYLES = {
    medium: {
        height: 24,
        margin: 4
    },
    large: {
        height: 30,
        margin: 5
    }
};
const $4275f73de689771f$export$67ea30858aaf75e3 = /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).forwardRef(function TagGroup(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $1af2ca8553741739$exports.useFormProps)(props);
    let { maxRows: maxRows, children: children, actionLabel: actionLabel, onAction: onAction, labelPosition: labelPosition, renderEmptyState: renderEmptyState = ()=>stringFormatter.format('noTags') } = props;
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let containerRef = (0, $80FZR$react.useRef)(null);
    let tagsRef = (0, $80FZR$react.useRef)(null);
    let { direction: direction } = (0, $80FZR$reactariaI18nProvider.useLocale)();
    let { scale: scale } = (0, $544fc82701fc93e9$exports.useProvider)();
    let stringFormatter = (0, $80FZR$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($77e6bffa322a0b1b$exports))), '@react-spectrum/tag');
    let [isCollapsed, setIsCollapsed] = (0, $80FZR$react.useState)(maxRows != null);
    let state = (0, $80FZR$reactstatelyuseListState.useListState)(props);
    let [tagState, setTagState] = (0, $80FZR$reactariaprivateutilsuseValueEffect.useValueEffect)({
        visibleTagCount: state.collection.size,
        showCollapseButton: false
    });
    let keyboardDelegate = (0, $80FZR$react.useMemo)(()=>{
        let collection = isCollapsed ? new (0, $80FZR$reactstatelyprivatelistListCollection.ListCollection)([
            ...state.collection
        ].slice(0, tagState.visibleTagCount)) : new (0, $80FZR$reactstatelyprivatelistListCollection.ListCollection)([
            ...state.collection
        ]);
        return new (0, $80FZR$reactariaListKeyboardDelegate.ListKeyboardDelegate)({
            collection: collection,
            ref: tagsRef,
            direction: direction,
            orientation: 'horizontal'
        });
    }, [
        direction,
        isCollapsed,
        state.collection,
        tagState.visibleTagCount,
        tagsRef
    ]);
    // Remove onAction from props so it doesn't make it into useGridList.
    delete props.onAction;
    let { gridProps: gridProps, labelProps: labelProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = (0, $80FZR$reactariauseTagGroup.useTagGroup)({
        ...props,
        keyboardDelegate: keyboardDelegate
    }, state, tagsRef);
    let actionsId = (0, $80FZR$reactariauseId.useId)();
    let actionsRef = (0, $80FZR$react.useRef)(null);
    let updateVisibleTagCount = (0, $80FZR$react.useCallback)(()=>{
        if (maxRows && maxRows > 0) {
            let computeVisibleTagCount = ()=>{
                // Refs can be null at runtime.
                let currContainerRef = containerRef.current;
                let currTagsRef = tagsRef.current;
                let currActionsRef = actionsRef.current;
                if (!currContainerRef || !currTagsRef || !currActionsRef || state.collection.size === 0) return {
                    visibleTagCount: 0,
                    showCollapseButton: false
                };
                // Count rows and show tags until we hit the maxRows.
                let tags = [
                    ...currTagsRef.children
                ];
                let currY = -Infinity;
                let rowCount = 0;
                let index = 0;
                let tagWidths = [];
                for (let tag of tags){
                    let { width: width, y: y } = tag.getBoundingClientRect();
                    if (y !== currY) {
                        currY = y;
                        rowCount++;
                    }
                    if (maxRows && rowCount > maxRows) break;
                    tagWidths.push(width);
                    index++;
                }
                // Remove tags until there is space for the collapse button and action button (if present) on the last row.
                let buttons = [
                    ...currActionsRef.children
                ];
                if (maxRows && buttons.length > 0 && rowCount >= maxRows && currContainerRef.parentElement) {
                    let buttonsWidth = buttons.reduce((acc, curr)=>acc += curr.getBoundingClientRect().width, 0);
                    buttonsWidth += $4275f73de689771f$var$TAG_STYLES[scale].margin * 2 * buttons.length;
                    let end = direction === 'ltr' ? 'right' : 'left';
                    let containerEnd = currContainerRef.parentElement.getBoundingClientRect()[end];
                    let lastTagEnd = tags[index - 1]?.getBoundingClientRect()[end];
                    lastTagEnd += $4275f73de689771f$var$TAG_STYLES[scale].margin;
                    let availableWidth = containerEnd - lastTagEnd;
                    while(availableWidth < buttonsWidth && index > 0){
                        availableWidth += tagWidths.pop();
                        index--;
                    }
                }
                return {
                    visibleTagCount: Math.max(index, 1),
                    showCollapseButton: index < state.collection.size
                };
            };
            setTagState(function*() {
                // Update to show all items.
                yield {
                    visibleTagCount: state.collection.size,
                    showCollapseButton: true
                };
                // Measure, and update to show the items until maxRows is reached.
                yield computeVisibleTagCount();
            });
        }
    }, [
        maxRows,
        setTagState,
        direction,
        scale,
        state.collection.size
    ]);
    (0, $80FZR$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: containerRef,
        onResize: updateVisibleTagCount
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $80FZR$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(updateVisibleTagCount, [
        children
    ]);
    (0, $80FZR$react.useEffect)(()=>{
        // Recalculate visible tags when fonts are loaded.
        document.fonts?.ready.then(()=>updateVisibleTagCount());
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    let visibleTags = (0, $80FZR$react.useMemo)(()=>[
            ...state.collection
        ].slice(0, isCollapsed ? tagState.visibleTagCount : state.collection.size), [
        isCollapsed,
        state.collection,
        tagState.visibleTagCount
    ]);
    let handlePressCollapse = ()=>{
        // Prevents button from losing focus if focusedKey got collapsed.
        state.selectionManager.setFocusedKey(null);
        setIsCollapsed((prevCollapsed)=>!prevCollapsed);
    };
    let showActions = tagState.showCollapseButton || actionLabel && onAction;
    let isEmpty = state.collection.size === 0;
    let containerStyle = (0, $80FZR$react.useMemo)(()=>{
        if (maxRows == null || !isCollapsed || isEmpty) return undefined;
        let maxHeight = ($4275f73de689771f$var$TAG_STYLES[scale].height + $4275f73de689771f$var$TAG_STYLES[scale].margin * 2) * maxRows;
        return {
            maxHeight: maxHeight,
            overflow: 'hidden'
        };
    }, [
        isCollapsed,
        maxRows,
        isEmpty,
        scale
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement((0, $80FZR$reactariaFocusScope.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement((0, $b93966d678e0af07$exports.Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: true,
        ref: domRef,
        elementType: "span",
        wrapperClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'spectrum-Tags-fieldWrapper', {
            'spectrum-Tags-fieldWrapper--positionSide': labelPosition === 'side'
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement("div", {
        ref: containerRef,
        style: containerStyle,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'spectrum-Tags-container', {
            'spectrum-Tags-container--empty': isEmpty
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement((0, $80FZR$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement("div", {
        ref: tagsRef,
        ...gridProps,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'spectrum-Tags')
    }, visibleTags.map((item)=>/*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement((0, $5bdeed39ae3015c4$exports.Tag), {
            ...item.props,
            key: item.key,
            item: item,
            state: state
        }, item.rendered)), isEmpty && /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'spectrum-Tags-empty-state')
    }, renderEmptyState()))), showActions && !isEmpty && /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement((0, $544fc82701fc93e9$exports.Provider), {
        isDisabled: false
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement("div", {
        role: "group",
        ref: actionsRef,
        id: actionsId,
        "aria-label": stringFormatter.format('actions'),
        "aria-labelledby": `${gridProps.id} ${actionsId}`,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'spectrum-Tags-actions')
    }, tagState.showCollapseButton && /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement((0, $183c5173677598aa$exports.ActionButton), {
        isQuiet: true,
        onPress: handlePressCollapse,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'spectrum-Tags-actionButton')
    }, isCollapsed ? stringFormatter.format('showAllButtonLabel', {
        tagCount: state.collection.size
    }) : stringFormatter.format('hideButtonLabel')), actionLabel && onAction && /*#__PURE__*/ (0, ($parcel$interopDefault($80FZR$react))).createElement((0, $183c5173677598aa$exports.ActionButton), {
        isQuiet: true,
        onPress: ()=>onAction?.(),
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($74103ae0a349d695$exports))), 'spectrum-Tags-actionButton')
    }, actionLabel))))));
});
