// @adobe/react-spectrum/dist/private/textfield/TextFieldBase.js
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {createFocusableRef as $c234463e9ef56637$export$79d69eee6ae4b329} from "../utils/useDOMRef.js";
import {Field as $3967792f95357356$export$a455218a85c89869} from "../label/Field.js";
import $id7wL$intlStringsjs from "./intlStrings.js";
import "../textfield_vars.css";
import $id7wL$textfield_vars_cssmjs from "../textfield_vars_css.mjs";
import $id7wL$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $id7wL$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {mergeProps as $id7wL$mergeProps} from "react-aria/mergeProps";
import $id7wL$react, {forwardRef as $id7wL$forwardRef, useRef as $id7wL$useRef, useImperativeHandle as $id7wL$useImperativeHandle, cloneElement as $id7wL$cloneElement} from "react";
import {useFocusRing as $id7wL$useFocusRing} from "react-aria/useFocusRing";
import {useHover as $id7wL$useHover} from "react-aria/useHover";
import {useId as $id7wL$useId} from "react-aria/useId";
import {useLocalizedStringFormatter as $id7wL$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 












const $1f88830e88ee8f61$export$d22444a338b6e3c2 = /*#__PURE__*/ (0, $id7wL$forwardRef)(function TextFieldBase(props, ref) {
    let { validationState: validationState = props.isInvalid ? 'invalid' : null, icon: icon, isQuiet: isQuiet = false, isDisabled: isDisabled, multiLine: multiLine, autoFocus: autoFocus, inputClassName: inputClassName, wrapperChildren: wrapperChildren, labelProps: labelProps, inputProps: inputProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, inputRef: userInputRef, isLoading: isLoading, loadingIndicator: loadingIndicator, validationIconClassName: validationIconClassName, disableFocusRing: disableFocusRing } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $id7wL$useHover)({
        isDisabled: isDisabled
    });
    let domRef = (0, $id7wL$useRef)(null);
    let defaultInputRef = (0, $id7wL$useRef)(null);
    let inputRef = userInputRef || defaultInputRef;
    // Expose imperative interface for ref
    (0, $id7wL$useImperativeHandle)(ref, ()=>({
            ...(0, $c234463e9ef56637$export$79d69eee6ae4b329)(domRef, inputRef),
            select () {
                if (inputRef.current) inputRef.current.select();
            },
            getInputElement () {
                return inputRef.current;
            }
        }));
    let ElementType = multiLine ? 'textarea' : 'input';
    let isInvalid = validationState === 'invalid' && !isDisabled;
    if (icon) {
        let UNSAFE_className = (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($id7wL$textfield_vars_cssmjs))), icon.props && icon.props.UNSAFE_className, 'spectrum-Textfield-icon');
        icon = /*#__PURE__*/ (0, $id7wL$cloneElement)(icon, {
            UNSAFE_className: UNSAFE_className,
            size: 'S'
        });
    }
    let stringFormatter = (0, $id7wL$useLocalizedStringFormatter)((0, ($parcel$interopDefault($id7wL$intlStringsjs))), '@react-spectrum/textfield');
    let validId = (0, $id7wL$useId)();
    let validationIcon = isInvalid ? /*#__PURE__*/ (0, $id7wL$react).createElement((0, $id7wL$spectrumiconsuiAlertMedium), null) : /*#__PURE__*/ (0, $id7wL$react).createElement((0, $id7wL$spectrumiconsuiCheckmarkMedium), {
        id: validId,
        "aria-hidden": true,
        "aria-label": stringFormatter.format('valid')
    });
    let validation = /*#__PURE__*/ (0, $id7wL$cloneElement)(validationIcon, {
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($id7wL$textfield_vars_cssmjs))), 'spectrum-Textfield-validationIcon', validationIconClassName)
    });
    // Add validation icon IDREF to aria-describedby when validationState is valid
    let inputPropsAriaDescribedBy = inputProps['aria-describedby'];
    if (!isInvalid && validationState === 'valid' && !isLoading && !isDisabled && (!inputPropsAriaDescribedBy || !inputPropsAriaDescribedBy.includes(validId))) inputProps['aria-describedby'] = [
        inputPropsAriaDescribedBy,
        validId
    ].join(' ').trim();
    let { focusProps: focusProps, isFocusVisible: isFocusVisible } = (0, $id7wL$useFocusRing)({
        isTextInput: true,
        autoFocus: autoFocus
    });
    let textField = /*#__PURE__*/ (0, $id7wL$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($id7wL$textfield_vars_cssmjs))), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': isInvalid,
            'spectrum-Textfield--valid': validationState === 'valid' && !isDisabled,
            'spectrum-Textfield--loadable': loadingIndicator,
            'spectrum-Textfield--quiet': isQuiet,
            'spectrum-Textfield--multiline': multiLine,
            'focus-ring': !disableFocusRing && isFocusVisible
        })
    }, /*#__PURE__*/ (0, $id7wL$react).createElement(ElementType, {
        ...(0, $id7wL$mergeProps)(inputProps, hoverProps, focusProps),
        ref: inputRef,
        rows: multiLine ? 1 : undefined,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($id7wL$textfield_vars_cssmjs))), 'spectrum-Textfield-input', {
            'spectrum-Textfield-inputIcon': icon,
            'is-hovered': isHovered
        }, inputClassName)
    }), icon, validationState && !isLoading && !isDisabled ? validation : null, isLoading && loadingIndicator, wrapperChildren);
    return /*#__PURE__*/ (0, $id7wL$react).createElement((0, $3967792f95357356$export$a455218a85c89869), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        wrapperClassName: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($id7wL$textfield_vars_cssmjs))), 'spectrum-Textfield-wrapper', {
            'spectrum-Textfield-wrapper--quiet': isQuiet
        }),
        showErrorIcon: false,
        ref: domRef
    }, textField);
});


export {$1f88830e88ee8f61$export$d22444a338b6e3c2 as TextFieldBase};
