// @adobe/react-spectrum/dist/private/textfield/TextFieldBase.cjs
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $b93966d678e0af07$exports = require("../label/Field.cjs");
var $1106ad279bbcb7ca$exports = require("./intlStrings.cjs");
require("../textfield_vars.css");
var $5d389146adc85829$exports = require("../textfield_vars_css.cjs");
var $jxKwg$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $jxKwg$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $jxKwg$reactariamergeProps = require("react-aria/mergeProps");
var $jxKwg$react = require("react");
var $jxKwg$reactariauseFocusRing = require("react-aria/useFocusRing");
var $jxKwg$reactariauseHover = require("react-aria/useHover");
var $jxKwg$reactariauseId = require("react-aria/useId");
var $jxKwg$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "TextFieldBase", function () { return $827dbb466e199966$export$d22444a338b6e3c2; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 












const $827dbb466e199966$export$d22444a338b6e3c2 = /*#__PURE__*/ (0, $jxKwg$react.forwardRef)(function TextFieldBase(props, ref) {
    let { validationState: validationState = props.isInvalid ? 'invalid' : null, icon: icon, isQuiet: isQuiet = false, isDisabled: isDisabled, multiLine: multiLine, autoFocus: autoFocus, inputClassName: inputClassName, wrapperChildren: wrapperChildren, labelProps: labelProps, inputProps: inputProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, inputRef: userInputRef, isLoading: isLoading, loadingIndicator: loadingIndicator, validationIconClassName: validationIconClassName, disableFocusRing: disableFocusRing } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $jxKwg$reactariauseHover.useHover)({
        isDisabled: isDisabled
    });
    let domRef = (0, $jxKwg$react.useRef)(null);
    let defaultInputRef = (0, $jxKwg$react.useRef)(null);
    let inputRef = userInputRef || defaultInputRef;
    // Expose imperative interface for ref
    (0, $jxKwg$react.useImperativeHandle)(ref, ()=>({
            ...(0, $65aea7b37663976b$exports.createFocusableRef)(domRef, inputRef),
            select () {
                if (inputRef.current) inputRef.current.select();
            },
            getInputElement () {
                return inputRef.current;
            }
        }));
    let ElementType = multiLine ? 'textarea' : 'input';
    let isInvalid = validationState === 'invalid' && !isDisabled;
    if (icon) {
        let UNSAFE_className = (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), icon.props && icon.props.UNSAFE_className, 'spectrum-Textfield-icon');
        icon = /*#__PURE__*/ (0, $jxKwg$react.cloneElement)(icon, {
            UNSAFE_className: UNSAFE_className,
            size: 'S'
        });
    }
    let stringFormatter = (0, $jxKwg$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($1106ad279bbcb7ca$exports))), '@react-spectrum/textfield');
    let validId = (0, $jxKwg$reactariauseId.useId)();
    let validationIcon = isInvalid ? /*#__PURE__*/ (0, ($parcel$interopDefault($jxKwg$react))).createElement((0, ($parcel$interopDefault($jxKwg$spectrumiconsuiAlertMedium))), null) : /*#__PURE__*/ (0, ($parcel$interopDefault($jxKwg$react))).createElement((0, ($parcel$interopDefault($jxKwg$spectrumiconsuiCheckmarkMedium))), {
        id: validId,
        "aria-hidden": true,
        "aria-label": stringFormatter.format('valid')
    });
    let validation = /*#__PURE__*/ (0, $jxKwg$react.cloneElement)(validationIcon, {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-validationIcon', validationIconClassName)
    });
    // Add validation icon IDREF to aria-describedby when validationState is valid
    let inputPropsAriaDescribedBy = inputProps['aria-describedby'];
    if (!isInvalid && validationState === 'valid' && !isLoading && !isDisabled && (!inputPropsAriaDescribedBy || !inputPropsAriaDescribedBy.includes(validId))) inputProps['aria-describedby'] = [
        inputPropsAriaDescribedBy,
        validId
    ].join(' ').trim();
    let { focusProps: focusProps, isFocusVisible: isFocusVisible } = (0, $jxKwg$reactariauseFocusRing.useFocusRing)({
        isTextInput: true,
        autoFocus: autoFocus
    });
    let textField = /*#__PURE__*/ (0, ($parcel$interopDefault($jxKwg$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': isInvalid,
            'spectrum-Textfield--valid': validationState === 'valid' && !isDisabled,
            'spectrum-Textfield--loadable': loadingIndicator,
            'spectrum-Textfield--quiet': isQuiet,
            'spectrum-Textfield--multiline': multiLine,
            'focus-ring': !disableFocusRing && isFocusVisible
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jxKwg$react))).createElement(ElementType, {
        ...(0, $jxKwg$reactariamergeProps.mergeProps)(inputProps, hoverProps, focusProps),
        ref: inputRef,
        rows: multiLine ? 1 : undefined,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-input', {
            'spectrum-Textfield-inputIcon': icon,
            'is-hovered': isHovered
        }, inputClassName)
    }), icon, validationState && !isLoading && !isDisabled ? validation : null, isLoading && loadingIndicator, wrapperChildren);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($jxKwg$react))).createElement((0, $b93966d678e0af07$exports.Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        wrapperClassName: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($5d389146adc85829$exports))), 'spectrum-Textfield-wrapper', {
            'spectrum-Textfield-wrapper--quiet': isQuiet
        }),
        showErrorIcon: false,
        ref: domRef
    }, textField);
});
