// @adobe/react-spectrum/dist/private/card/CardBase.js
import {Checkbox as $986e1e93e04146a6$export$48513f6b9f8ce62d} from "../checkbox/Checkbox.js";
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686} from "../utils/Slots.js";
import "../card_vars.css";
import $iMWH0$card_vars_cssmjs from "../card_vars_css.mjs";
import {useCardViewContext as $09157122c9607ee5$export$fea0b38586ec8f13} from "./CardViewContext.js";
import {useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useHasChild as $584638b763a93bff$export$e52e2242b6d0f1d4} from "../utils/useHasChild.js";
import {useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {filterDOMProps as $iMWH0$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusRing as $iMWH0$FocusRing} from "react-aria/FocusRing";
import {getFocusableTreeWalker as $iMWH0$getFocusableTreeWalker} from "react-aria/private/focus/FocusScope";
import {mergeProps as $iMWH0$mergeProps} from "react-aria/mergeProps";
import {nodeContains as $iMWH0$nodeContains} from "react-aria/private/utils/shadowdom/DOMFunctions";
import $iMWH0$react, {useRef as $iMWH0$useRef, useState as $iMWH0$useState, useCallback as $iMWH0$useCallback, useMemo as $iMWH0$useMemo} from "react";
import {useFocusWithin as $iMWH0$useFocusWithin} from "react-aria/useFocusWithin";
import {useHover as $iMWH0$useHover} from "react-aria/useHover";
import {useLayoutEffect as $iMWH0$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useResizeObserver as $iMWH0$useResizeObserver} from "react-aria/private/utils/useResizeObserver";
import {useSlotId as $iMWH0$useSlotId} from "react-aria/private/utils/useId";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
// @ts-nocheck
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



















const $cdc2130588546f60$export$7a6ccaf429ad93a8 = /*#__PURE__*/ (0, $iMWH0$react).forwardRef(function CardBase(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    let context = (0, $09157122c9607ee5$export$fea0b38586ec8f13)() || {}; // we can call again here, won't change from Card.tsx
    let { state: state } = context;
    let manager = state === null || state === void 0 ? void 0 : state.selectionManager;
    let { isQuiet: isQuiet, orientation: orientation = 'vertical', articleProps: articleProps = {
        role: 'article'
    }, item: item, layout: layout, children: children } = props;
    let key = item === null || item === void 0 ? void 0 : item.key;
    let isSelected = manager === null || manager === void 0 ? void 0 : manager.isSelected(key);
    let isDisabled = state === null || state === void 0 ? void 0 : state.disabledKeys.has(key);
    let onChange = ()=>manager === null || manager === void 0 ? void 0 : manager.select(key);
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    let { cardProps: cardProps, titleProps: titleProps, contentProps: contentProps } = $cdc2130588546f60$var$useCard(props);
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let gridRef = (0, $iMWH0$useRef)(undefined);
    let checkboxRef = (0, $iMWH0$useRef)(null);
    // cards are only interactive if there is a selection manager and it allows selection
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $iMWH0$useHover)({
        isDisabled: manager === undefined || (manager === null || manager === void 0 ? void 0 : manager.selectionMode) === 'none' || isDisabled
    });
    let [isFocused, setIsFocused] = (0, $iMWH0$useState)(false);
    let { focusWithinProps: focusWithinProps } = (0, $iMWH0$useFocusWithin)({
        onFocusWithinChange: setIsFocused,
        isDisabled: isDisabled
    });
    // ToDo: see css for comment about avatar under selector .spectrum-Card--noLayout.spectrum-Card--default
    let hasPreviewImage = (0, $584638b763a93bff$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs)))['spectrum-Card-image']}`, gridRef);
    let hasPreviewIllustration = (0, $584638b763a93bff$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs)))['spectrum-Card-illustration']}`, gridRef);
    let hasPreview = hasPreviewImage || hasPreviewIllustration;
    // this is for horizontal cards
    let [height, setHeight] = (0, $iMWH0$useState)(NaN);
    let updateHeight = (0, $iMWH0$useCallback)(()=>{
        if (orientation !== 'horizontal') return;
        let cardHeight = gridRef.current.getBoundingClientRect().height;
        setHeight(cardHeight);
    }, [
        orientation,
        gridRef,
        setHeight
    ]);
    (0, $iMWH0$useResizeObserver)({
        ref: gridRef,
        onResize: updateHeight
    });
    let aspectRatioEnforce = undefined;
    if (orientation === 'horizontal' && !isNaN(height)) aspectRatioEnforce = {
        height: `${height}px`,
        width: `${height}px`
    };
    let slots = (0, $iMWH0$useMemo)(()=>({
            image: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-image'),
                objectFit: orientation === 'horizontal' ? 'cover' : 'contain',
                alt: '',
                ...aspectRatioEnforce
            },
            illustration: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-illustration'),
                ...aspectRatioEnforce
            },
            avatar: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-avatar'),
                size: 'avatar-size-400'
            },
            heading: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-heading'),
                ...titleProps
            },
            content: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-content'),
                ...contentProps
            },
            detail: {
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-detail')
            }
        }), // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        titleProps,
        contentProps,
        height,
        isQuiet,
        orientation
    ]);
    (0, $iMWH0$useLayoutEffect)(()=>{
        if (gridRef === null || gridRef === void 0 ? void 0 : gridRef.current) {
            let walker = (0, $iMWH0$getFocusableTreeWalker)(gridRef.current);
            let nextNode = walker.nextNode();
            while(nextNode != null){
                if (checkboxRef.current && !(0, $iMWH0$nodeContains)(checkboxRef.current.UNSAFE_getDOMNode(), nextNode)) {
                    console.warn('Card does not support focusable elements, please contact the team regarding your use case.');
                    break;
                }
                nextNode = walker.nextNode();
            }
        }
    }, [
        children
    ]);
    return /*#__PURE__*/ (0, $iMWH0$react).createElement((0, $iMWH0$FocusRing), {
        focusRingClass: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'focus-ring')
    }, /*#__PURE__*/ (0, $iMWH0$react).createElement("div", {
        ...styleProps,
        ...(0, $iMWH0$mergeProps)(cardProps, focusWithinProps, hoverProps, (0, $iMWH0$filterDOMProps)(props), articleProps),
        ref: domRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card', {
            'spectrum-Card--default': !isQuiet && orientation !== 'horizontal',
            'spectrum-Card--isQuiet': isQuiet && orientation !== 'horizontal',
            'spectrum-Card--horizontal': orientation === 'horizontal',
            'spectrum-Card--noPreview': !hasPreview,
            'is-hovered': isHovered,
            'is-focused': isFocused,
            'is-selected': isSelected,
            'spectrum-Card--waterfall': layout === 'waterfall',
            'spectrum-Card--gallery': layout === 'gallery',
            'spectrum-Card--grid': layout === 'grid',
            'spectrum-Card--noLayout': layout !== 'waterfall' && layout !== 'gallery' && layout !== 'grid'
        }, styleProps.className)
    }, /*#__PURE__*/ (0, $iMWH0$react).createElement("div", {
        ref: gridRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-grid')
    }, manager && manager.selectionMode !== 'none' && /*#__PURE__*/ (0, $iMWH0$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-checkboxWrapper')
    }, /*#__PURE__*/ (0, $iMWH0$react).createElement((0, $986e1e93e04146a6$export$48513f6b9f8ce62d), {
        ref: checkboxRef,
        isDisabled: isDisabled,
        excludeFromTabOrder: true,
        isSelected: isSelected,
        onChange: onChange,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-checkbox'),
        isEmphasized: true,
        "aria-label": "select"
    })), /*#__PURE__*/ (0, $iMWH0$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: slots
    }, children), /*#__PURE__*/ (0, $iMWH0$react).createElement("div", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($iMWH0$card_vars_cssmjs))), 'spectrum-Card-decoration')
    }))));
});
function $cdc2130588546f60$var$useCard(props) {
    let titleId = (0, $iMWH0$useSlotId)();
    let descriptionId = (0, $iMWH0$useSlotId)();
    let titleProps = (0, $iMWH0$useMemo)(()=>({
            id: titleId
        }), [
        titleId
    ]);
    let contentProps = (0, $iMWH0$useMemo)(()=>({
            id: descriptionId
        }), [
        descriptionId
    ]);
    return {
        cardProps: {
            ...(0, $iMWH0$filterDOMProps)(props),
            'aria-labelledby': titleId,
            'aria-describedby': descriptionId,
            tabIndex: 0
        },
        titleProps: titleProps,
        contentProps: contentProps
    };
}


export {$cdc2130588546f60$export$7a6ccaf429ad93a8 as CardBase};
