// @adobe/react-spectrum/dist/private/card/CardBase.cjs
var $9bc060484abc63af$exports = require("../checkbox/Checkbox.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
require("../card_vars.css");
var $59e87deeac09a752$exports = require("../card_vars_css.cjs");
var $c5cd545e21c17a4a$exports = require("./CardViewContext.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $dd6348d4a1a51ff9$exports = require("../utils/useHasChild.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $bLAEJ$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $bLAEJ$reactariaFocusRing = require("react-aria/FocusRing");
var $bLAEJ$reactariaprivatefocusFocusScope = require("react-aria/private/focus/FocusScope");
var $bLAEJ$reactariamergeProps = require("react-aria/mergeProps");
var $bLAEJ$reactariaprivateutilsshadowdomDOMFunctions = require("react-aria/private/utils/shadowdom/DOMFunctions");
var $bLAEJ$react = require("react");
var $bLAEJ$reactariauseFocusWithin = require("react-aria/useFocusWithin");
var $bLAEJ$reactariauseHover = require("react-aria/useHover");
var $bLAEJ$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $bLAEJ$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");
var $bLAEJ$reactariaprivateutilsuseId = require("react-aria/private/utils/useId");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "CardBase", function () { return $93b4a57699139677$export$7a6ccaf429ad93a8; });
// @ts-nocheck
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



















const $93b4a57699139677$export$7a6ccaf429ad93a8 = /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).forwardRef(function CardBase(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    let context = (0, $c5cd545e21c17a4a$exports.useCardViewContext)() || {}; // we can call again here, won't change from Card.tsx
    let { state: state } = context;
    let manager = state?.selectionManager;
    let { isQuiet: isQuiet, orientation: orientation = 'vertical', articleProps: articleProps = {
        role: 'article'
    }, item: item, layout: layout, children: children } = props;
    let key = item?.key;
    let isSelected = manager?.isSelected(key);
    let isDisabled = state?.disabledKeys.has(key);
    let onChange = ()=>manager?.select(key);
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    let { cardProps: cardProps, titleProps: titleProps, contentProps: contentProps } = $93b4a57699139677$var$useCard(props);
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let gridRef = (0, $bLAEJ$react.useRef)(undefined);
    let checkboxRef = (0, $bLAEJ$react.useRef)(null);
    // cards are only interactive if there is a selection manager and it allows selection
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $bLAEJ$reactariauseHover.useHover)({
        isDisabled: manager === undefined || manager?.selectionMode === 'none' || isDisabled
    });
    let [isFocused, setIsFocused] = (0, $bLAEJ$react.useState)(false);
    let { focusWithinProps: focusWithinProps } = (0, $bLAEJ$reactariauseFocusWithin.useFocusWithin)({
        onFocusWithinChange: setIsFocused,
        isDisabled: isDisabled
    });
    // ToDo: see css for comment about avatar under selector .spectrum-Card--noLayout.spectrum-Card--default
    let hasPreviewImage = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($59e87deeac09a752$exports)))['spectrum-Card-image']}`, gridRef);
    let hasPreviewIllustration = (0, $dd6348d4a1a51ff9$exports.useHasChild)(`.${(0, ($parcel$interopDefault($59e87deeac09a752$exports)))['spectrum-Card-illustration']}`, gridRef);
    let hasPreview = hasPreviewImage || hasPreviewIllustration;
    // this is for horizontal cards
    let [height, setHeight] = (0, $bLAEJ$react.useState)(NaN);
    let updateHeight = (0, $bLAEJ$react.useCallback)(()=>{
        if (orientation !== 'horizontal') return;
        let cardHeight = gridRef.current.getBoundingClientRect().height;
        setHeight(cardHeight);
    }, [
        orientation,
        gridRef,
        setHeight
    ]);
    (0, $bLAEJ$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: gridRef,
        onResize: updateHeight
    });
    let aspectRatioEnforce = undefined;
    if (orientation === 'horizontal' && !isNaN(height)) aspectRatioEnforce = {
        height: `${height}px`,
        width: `${height}px`
    };
    let slots = (0, $bLAEJ$react.useMemo)(()=>({
            image: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-image'),
                objectFit: orientation === 'horizontal' ? 'cover' : 'contain',
                alt: '',
                ...aspectRatioEnforce
            },
            illustration: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-illustration'),
                ...aspectRatioEnforce
            },
            avatar: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-avatar'),
                size: 'avatar-size-400'
            },
            heading: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-heading'),
                ...titleProps
            },
            content: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-content'),
                ...contentProps
            },
            detail: {
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-detail')
            }
        }), // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        titleProps,
        contentProps,
        height,
        isQuiet,
        orientation
    ]);
    (0, $bLAEJ$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        if (gridRef?.current) {
            let walker = (0, $bLAEJ$reactariaprivatefocusFocusScope.getFocusableTreeWalker)(gridRef.current);
            let nextNode = walker.nextNode();
            while(nextNode != null){
                if (checkboxRef.current && !(0, $bLAEJ$reactariaprivateutilsshadowdomDOMFunctions.nodeContains)(checkboxRef.current.UNSAFE_getDOMNode(), nextNode)) {
                    console.warn('Card does not support focusable elements, please contact the team regarding your use case.');
                    break;
                }
                nextNode = walker.nextNode();
            }
        }
    }, [
        children
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).createElement((0, $bLAEJ$reactariaFocusRing.FocusRing), {
        focusRingClass: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'focus-ring')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).createElement("div", {
        ...styleProps,
        ...(0, $bLAEJ$reactariamergeProps.mergeProps)(cardProps, focusWithinProps, hoverProps, (0, $bLAEJ$reactariafilterDOMProps.filterDOMProps)(props), articleProps),
        ref: domRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card', {
            'spectrum-Card--default': !isQuiet && orientation !== 'horizontal',
            'spectrum-Card--isQuiet': isQuiet && orientation !== 'horizontal',
            'spectrum-Card--horizontal': orientation === 'horizontal',
            'spectrum-Card--noPreview': !hasPreview,
            'is-hovered': isHovered,
            'is-focused': isFocused,
            'is-selected': isSelected,
            'spectrum-Card--waterfall': layout === 'waterfall',
            'spectrum-Card--gallery': layout === 'gallery',
            'spectrum-Card--grid': layout === 'grid',
            'spectrum-Card--noLayout': layout !== 'waterfall' && layout !== 'gallery' && layout !== 'grid'
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).createElement("div", {
        ref: gridRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-grid')
    }, manager && manager.selectionMode !== 'none' && /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-checkboxWrapper')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).createElement((0, $9bc060484abc63af$exports.Checkbox), {
        ref: checkboxRef,
        isDisabled: isDisabled,
        excludeFromTabOrder: true,
        isSelected: isSelected,
        onChange: onChange,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-checkbox'),
        isEmphasized: true,
        "aria-label": "select"
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: slots
    }, children), /*#__PURE__*/ (0, ($parcel$interopDefault($bLAEJ$react))).createElement("div", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-Card-decoration')
    }))));
});
function $93b4a57699139677$var$useCard(props) {
    let titleId = (0, $bLAEJ$reactariaprivateutilsuseId.useSlotId)();
    let descriptionId = (0, $bLAEJ$reactariaprivateutilsuseId.useSlotId)();
    let titleProps = (0, $bLAEJ$react.useMemo)(()=>({
            id: titleId
        }), [
        titleId
    ]);
    let contentProps = (0, $bLAEJ$react.useMemo)(()=>({
            id: descriptionId
        }), [
        descriptionId
    ]);
    return {
        cardProps: {
            ...(0, $bLAEJ$reactariafilterDOMProps.filterDOMProps)(props),
            'aria-labelledby': titleId,
            'aria-describedby': descriptionId,
            tabIndex: 0
        },
        titleProps: titleProps,
        contentProps: contentProps
    };
}
