// @adobe/react-spectrum/dist/private/card/CardView.js
import {CardBase as $cdc2130588546f60$export$7a6ccaf429ad93a8} from "./CardBase.js";
import {CardViewContext as $09157122c9607ee5$export$64992ac69f286e5c, useCardViewContext as $09157122c9607ee5$export$fea0b38586ec8f13} from "./CardViewContext.js";
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import $kLk5i$intlStringsjs from "./intlStrings.js";
import {ProgressCircle as $277696409c391eff$export$c79b9d6b4cc92af7} from "../progress/ProgressCircle.js";
import "../card_vars.css";
import $kLk5i$card_vars_cssmjs from "../card_vars_css.mjs";
import {useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8, useUnwrapDOMRef as $c234463e9ef56637$export$1d5cc31d9d8df817} from "../utils/useDOMRef.js";
import {useProvider as $089943c7a219141c$export$693cdb10cec23617} from "../provider/Provider.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {GridCollection as $kLk5i$GridCollection} from "react-stately/private/grid/GridCollection";
import {mergeProps as $kLk5i$mergeProps} from "react-aria/mergeProps";
import $kLk5i$react, {useMemo as $kLk5i$useMemo, useCallback as $kLk5i$useCallback, useRef as $kLk5i$useRef} from "react";
import {useCollator as $kLk5i$useCollator} from "react-aria/useCollator";
import {useGrid as $kLk5i$useGrid} from "react-aria/private/grid/useGrid";
import {useGridCell as $kLk5i$useGridCell} from "react-aria/private/grid/useGridCell";
import {useGridRow as $kLk5i$useGridRow} from "react-aria/private/grid/useGridRow";
import {useGridState as $kLk5i$useGridState} from "react-stately/private/grid/useGridState";
import {useListState as $kLk5i$useListState} from "react-stately/useListState";
import {useLocale as $kLk5i$useLocale} from "react-aria/I18nProvider";
import {useLocalizedStringFormatter as $kLk5i$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {Virtualizer as $kLk5i$Virtualizer} from "react-aria/private/virtualizer/Virtualizer";
import {VirtualizerItem as $kLk5i$VirtualizerItem} from "react-aria/private/virtualizer/VirtualizerItem";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
// @ts-nocheck
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





















const $91cd21cb8400ff0f$export$7e52c821f7b6f422 = /*#__PURE__*/ (0, $kLk5i$react).forwardRef(function CardView(props, ref) {
    let { scale: scale } = (0, $089943c7a219141c$export$693cdb10cec23617)();
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let { isQuiet: isQuiet, renderEmptyState: renderEmptyState, layout: layout, loadingState: loadingState, onLoadMore: onLoadMore, cardOrientation: cardOrientation = 'vertical' } = props;
    let collator = (0, $kLk5i$useCollator)({
        usage: 'search',
        sensitivity: 'base'
    });
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let cardViewLayout = (0, $kLk5i$useMemo)(()=>typeof layout === 'function' ? new layout({
            collator: collator,
            cardOrientation: cardOrientation,
            scale: scale
        }) : layout, [
        layout,
        collator,
        cardOrientation,
        scale
    ]);
    let layoutType = cardViewLayout.layoutType;
    let { direction: direction } = (0, $kLk5i$useLocale)();
    let { collection: collection } = (0, $kLk5i$useListState)(props);
    let gridCollection = (0, $kLk5i$useMemo)(()=>new (0, $kLk5i$GridCollection)({
            columnCount: 1,
            items: [
                ...collection
            ].map((item)=>({
                    // Makes the Grid row use the keys the user provides to the cards so that selection change via interactions returns the card keys
                    ...item,
                    hasChildNodes: true,
                    childNodes: [
                        {
                            key: `cell-${item.key}`,
                            type: 'cell',
                            value: null,
                            level: 0,
                            rendered: null,
                            textValue: item.textValue,
                            hasChildNodes: false,
                            childNodes: []
                        }
                    ]
                }))
        }), [
        collection
    ]);
    let state = (0, $kLk5i$useGridState)({
        ...props,
        selectionMode: cardOrientation === 'horizontal' && layoutType === 'grid' ? 'none' : props.selectionMode,
        collection: gridCollection,
        focusMode: 'cell'
    });
    cardViewLayout.collection = gridCollection;
    cardViewLayout.disabledKeys = state.disabledKeys;
    let { gridProps: gridProps } = (0, $kLk5i$useGrid)({
        ...props,
        isVirtualized: true,
        keyboardDelegate: cardViewLayout
    }, state, domRef);
    let renderWrapper = (0, $kLk5i$useCallback)((parent, reusableView)=>/*#__PURE__*/ (0, $kLk5i$react).createElement((0, $kLk5i$VirtualizerItem), {
            key: reusableView.key,
            layoutInfo: reusableView.layoutInfo,
            virtualizer: reusableView.virtualizer,
            parent: parent === null || parent === void 0 ? void 0 : parent.layoutInfo
        }, reusableView.rendered), []);
    let focusedKey = state.selectionManager.focusedKey;
    let focusedItem = gridCollection.getItem(state.selectionManager.focusedKey);
    if ((focusedItem === null || focusedItem === void 0 ? void 0 : focusedItem.parentKey) != null) focusedKey = focusedItem.parentKey;
    let persistedKeys = (0, $kLk5i$useMemo)(()=>focusedKey != null ? new Set([
            focusedKey
        ]) : null, [
        focusedKey
    ]);
    // TODO: does aria-row count and aria-col count need to be modified? Perhaps aria-col count needs to be omitted
    return /*#__PURE__*/ (0, $kLk5i$react).createElement((0, $09157122c9607ee5$export$64992ac69f286e5c).Provider, {
        value: {
            state: state,
            isQuiet: isQuiet,
            layout: cardViewLayout,
            cardOrientation: cardOrientation,
            renderEmptyState: renderEmptyState
        }
    }, /*#__PURE__*/ (0, $kLk5i$react).createElement((0, $kLk5i$Virtualizer), {
        ...gridProps,
        ...styleProps,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($kLk5i$card_vars_cssmjs))), 'spectrum-CardView'),
        ref: domRef,
        persistedKeys: persistedKeys,
        scrollDirection: "vertical",
        layout: cardViewLayout,
        collection: gridCollection,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        layoutOptions: (0, $kLk5i$useMemo)(()=>({
                isLoading: isLoading,
                direction: direction
            }), [
            isLoading,
            direction
        ]),
        renderWrapper: renderWrapper,
        style: {
            ...styleProps.style,
            scrollPaddingTop: cardViewLayout.margin || 0
        }
    }, (0, $kLk5i$useCallback)((type, item)=>{
        if (type === 'item') return /*#__PURE__*/ (0, $kLk5i$react).createElement($91cd21cb8400ff0f$var$InternalCard, {
            item: item
        });
        else if (type === 'loader') return /*#__PURE__*/ (0, $kLk5i$react).createElement($91cd21cb8400ff0f$var$LoadingState, null);
        else if (type === 'placeholder') return /*#__PURE__*/ (0, $kLk5i$react).createElement($91cd21cb8400ff0f$var$EmptyState, null);
    }, [])));
});
function $91cd21cb8400ff0f$var$LoadingState() {
    let { state: state } = (0, $09157122c9607ee5$export$fea0b38586ec8f13)();
    let stringFormatter = (0, $kLk5i$useLocalizedStringFormatter)((0, ($parcel$interopDefault($kLk5i$intlStringsjs))), '@react-spectrum/card');
    return /*#__PURE__*/ (0, $kLk5i$react).createElement($91cd21cb8400ff0f$var$CenteredWrapper, null, /*#__PURE__*/ (0, $kLk5i$react).createElement((0, $277696409c391eff$export$c79b9d6b4cc92af7), {
        isIndeterminate: true,
        "aria-label": state.collection.size > 0 ? stringFormatter.format('loadingMore') : stringFormatter.format('loading')
    }));
}
function $91cd21cb8400ff0f$var$EmptyState() {
    let { renderEmptyState: renderEmptyState } = (0, $09157122c9607ee5$export$fea0b38586ec8f13)();
    let emptyState = renderEmptyState ? renderEmptyState() : null;
    if (emptyState == null) return null;
    return /*#__PURE__*/ (0, $kLk5i$react).createElement($91cd21cb8400ff0f$var$CenteredWrapper, null, emptyState);
}
function $91cd21cb8400ff0f$var$CenteredWrapper({ children: children }) {
    let { state: state } = (0, $09157122c9607ee5$export$fea0b38586ec8f13)();
    return /*#__PURE__*/ (0, $kLk5i$react).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($kLk5i$card_vars_cssmjs))), 'spectrum-CardView-centeredWrapper')
    }, /*#__PURE__*/ (0, $kLk5i$react).createElement("div", {
        role: "gridcell"
    }, children));
}
function $91cd21cb8400ff0f$var$InternalCard(props) {
    let { item: item } = props;
    let cellNode = [
        ...item.childNodes
    ][0];
    let { state: state, cardOrientation: cardOrientation, isQuiet: isQuiet, layout: layout } = (0, $09157122c9607ee5$export$fea0b38586ec8f13)();
    let layoutType = layout.layoutType;
    let rowRef = (0, $kLk5i$useRef)(undefined);
    let cellRef = (0, $kLk5i$useRef)(undefined);
    let unwrappedRef = (0, $c234463e9ef56637$export$1d5cc31d9d8df817)(cellRef);
    let { rowProps: gridRowProps } = (0, $kLk5i$useGridRow)({
        node: item,
        isVirtualized: true
    }, state, rowRef);
    let { gridCellProps: gridCellProps } = (0, $kLk5i$useGridCell)({
        node: cellNode,
        focusMode: 'cell'
    }, state, unwrappedRef);
    // Prevent space key from scrolling the CardView if triggered on a disabled item or on a Card in a selectionMode="none" CardView.
    let allowsInteraction = state.selectionManager.selectionMode !== 'none';
    let isDisabled = !allowsInteraction || state.disabledKeys.has(item.key);
    let onKeyDown = (e)=>{
        if (e.key === ' ' && isDisabled) e.preventDefault();
    };
    let rowProps = (0, $kLk5i$mergeProps)(gridRowProps, {
        onKeyDown: onKeyDown
    });
    if (layoutType === 'grid' || layoutType === 'gallery') isQuiet = true;
    if (layoutType !== 'grid') cardOrientation = 'vertical';
    // We don't want to focus the checkbox (or any other focusable elements) within the Card
    // when pressing the arrow keys so we delete the key down handler here. Arrow key navigation between
    // the cards in the CardView is handled by useGrid => useSelectableCollection instead.
    delete gridCellProps.onKeyDownCapture;
    return /*#__PURE__*/ (0, $kLk5i$react).createElement("div", {
        ...rowProps,
        ref: rowRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($kLk5i$card_vars_cssmjs))), 'spectrum-CardView-row')
    }, /*#__PURE__*/ (0, $kLk5i$react).createElement((0, $cdc2130588546f60$export$7a6ccaf429ad93a8), {
        ref: cellRef,
        articleProps: gridCellProps,
        isQuiet: isQuiet,
        orientation: cardOrientation,
        item: item,
        layout: layoutType
    }, item.rendered));
}


export {$91cd21cb8400ff0f$export$7e52c821f7b6f422 as CardView};
