// @adobe/react-spectrum/dist/private/card/CardView.mjs
import {CardBase as $957034e7d12d8044$export$7a6ccaf429ad93a8} from "./CardBase.mjs";
import {CardViewContext as $161b6b8b1c8e2230$export$64992ac69f286e5c, useCardViewContext as $161b6b8b1c8e2230$export$fea0b38586ec8f13} from "./CardViewContext.mjs";
import {classNames as $6e6392558d48dfec$export$ce4ab0c55987d1ff} from "../utils/classNames.mjs";
import $eEzdj$intlStringsmjs from "./intlStrings.mjs";
import {ProgressCircle as $1cfe37e7feefa23d$export$c79b9d6b4cc92af7} from "../progress/ProgressCircle.mjs";
import "../card_vars.css";
import $eEzdj$card_vars_cssmjs from "../card_vars_css.mjs";
import {useDOMRef as $3c2c983d5210446c$export$c2c55ef9111cafd8, useUnwrapDOMRef as $3c2c983d5210446c$export$1d5cc31d9d8df817} from "../utils/useDOMRef.mjs";
import {useProvider as $71dfb0e0358a12de$export$693cdb10cec23617} from "../provider/Provider.mjs";
import {useStyleProps as $63d03c54ca5e4b88$export$b8e6fb9d2dff3f41} from "../utils/styleProps.mjs";
import {GridCollection as $eEzdj$GridCollection} from "react-stately/private/grid/GridCollection";
import {mergeProps as $eEzdj$mergeProps} from "react-aria/mergeProps";
import $eEzdj$react, {useMemo as $eEzdj$useMemo, useCallback as $eEzdj$useCallback, useRef as $eEzdj$useRef} from "react";
import {useCollator as $eEzdj$useCollator} from "react-aria/useCollator";
import {useGrid as $eEzdj$useGrid} from "react-aria/private/grid/useGrid";
import {useGridCell as $eEzdj$useGridCell} from "react-aria/private/grid/useGridCell";
import {useGridRow as $eEzdj$useGridRow} from "react-aria/private/grid/useGridRow";
import {useGridState as $eEzdj$useGridState} from "react-stately/private/grid/useGridState";
import {useListState as $eEzdj$useListState} from "react-stately/useListState";
import {useLocale as $eEzdj$useLocale} from "react-aria/I18nProvider";
import {useLocalizedStringFormatter as $eEzdj$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {Virtualizer as $eEzdj$Virtualizer} from "react-aria/private/virtualizer/Virtualizer";
import {VirtualizerItem as $eEzdj$VirtualizerItem} from "react-aria/private/virtualizer/VirtualizerItem";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
// @ts-nocheck
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





















const $f30e78dbf4cce5fb$export$7e52c821f7b6f422 = /*#__PURE__*/ (0, $eEzdj$react).forwardRef(function CardView(props, ref) {
    let { scale: scale } = (0, $71dfb0e0358a12de$export$693cdb10cec23617)();
    let { styleProps: styleProps } = (0, $63d03c54ca5e4b88$export$b8e6fb9d2dff3f41)(props);
    let domRef = (0, $3c2c983d5210446c$export$c2c55ef9111cafd8)(ref);
    let { isQuiet: isQuiet, renderEmptyState: renderEmptyState, layout: layout, loadingState: loadingState, onLoadMore: onLoadMore, cardOrientation: cardOrientation = 'vertical' } = props;
    let collator = (0, $eEzdj$useCollator)({
        usage: 'search',
        sensitivity: 'base'
    });
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let cardViewLayout = (0, $eEzdj$useMemo)(()=>typeof layout === 'function' ? new layout({
            collator: collator,
            cardOrientation: cardOrientation,
            scale: scale
        }) : layout, [
        layout,
        collator,
        cardOrientation,
        scale
    ]);
    let layoutType = cardViewLayout.layoutType;
    let { direction: direction } = (0, $eEzdj$useLocale)();
    let { collection: collection } = (0, $eEzdj$useListState)(props);
    let gridCollection = (0, $eEzdj$useMemo)(()=>new (0, $eEzdj$GridCollection)({
            columnCount: 1,
            items: [
                ...collection
            ].map((item)=>({
                    // Makes the Grid row use the keys the user provides to the cards so that selection change via interactions returns the card keys
                    ...item,
                    hasChildNodes: true,
                    childNodes: [
                        {
                            key: `cell-${item.key}`,
                            type: 'cell',
                            value: null,
                            level: 0,
                            rendered: null,
                            textValue: item.textValue,
                            hasChildNodes: false,
                            childNodes: []
                        }
                    ]
                }))
        }), [
        collection
    ]);
    let state = (0, $eEzdj$useGridState)({
        ...props,
        selectionMode: cardOrientation === 'horizontal' && layoutType === 'grid' ? 'none' : props.selectionMode,
        collection: gridCollection,
        focusMode: 'cell'
    });
    cardViewLayout.collection = gridCollection;
    cardViewLayout.disabledKeys = state.disabledKeys;
    let { gridProps: gridProps } = (0, $eEzdj$useGrid)({
        ...props,
        isVirtualized: true,
        keyboardDelegate: cardViewLayout
    }, state, domRef);
    let renderWrapper = (0, $eEzdj$useCallback)((parent, reusableView)=>/*#__PURE__*/ (0, $eEzdj$react).createElement((0, $eEzdj$VirtualizerItem), {
            key: reusableView.key,
            layoutInfo: reusableView.layoutInfo,
            virtualizer: reusableView.virtualizer,
            parent: parent?.layoutInfo
        }, reusableView.rendered), []);
    let focusedKey = state.selectionManager.focusedKey;
    let focusedItem = gridCollection.getItem(state.selectionManager.focusedKey);
    if (focusedItem?.parentKey != null) focusedKey = focusedItem.parentKey;
    let persistedKeys = (0, $eEzdj$useMemo)(()=>focusedKey != null ? new Set([
            focusedKey
        ]) : null, [
        focusedKey
    ]);
    // TODO: does aria-row count and aria-col count need to be modified? Perhaps aria-col count needs to be omitted
    return /*#__PURE__*/ (0, $eEzdj$react).createElement((0, $161b6b8b1c8e2230$export$64992ac69f286e5c).Provider, {
        value: {
            state: state,
            isQuiet: isQuiet,
            layout: cardViewLayout,
            cardOrientation: cardOrientation,
            renderEmptyState: renderEmptyState
        }
    }, /*#__PURE__*/ (0, $eEzdj$react).createElement((0, $eEzdj$Virtualizer), {
        ...gridProps,
        ...styleProps,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($eEzdj$card_vars_cssmjs))), 'spectrum-CardView'),
        ref: domRef,
        persistedKeys: persistedKeys,
        scrollDirection: "vertical",
        layout: cardViewLayout,
        collection: gridCollection,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        layoutOptions: (0, $eEzdj$useMemo)(()=>({
                isLoading: isLoading,
                direction: direction
            }), [
            isLoading,
            direction
        ]),
        renderWrapper: renderWrapper,
        style: {
            ...styleProps.style,
            scrollPaddingTop: cardViewLayout.margin || 0
        }
    }, (0, $eEzdj$useCallback)((type, item)=>{
        if (type === 'item') return /*#__PURE__*/ (0, $eEzdj$react).createElement($f30e78dbf4cce5fb$var$InternalCard, {
            item: item
        });
        else if (type === 'loader') return /*#__PURE__*/ (0, $eEzdj$react).createElement($f30e78dbf4cce5fb$var$LoadingState, null);
        else if (type === 'placeholder') return /*#__PURE__*/ (0, $eEzdj$react).createElement($f30e78dbf4cce5fb$var$EmptyState, null);
    }, [])));
});
function $f30e78dbf4cce5fb$var$LoadingState() {
    let { state: state } = (0, $161b6b8b1c8e2230$export$fea0b38586ec8f13)();
    let stringFormatter = (0, $eEzdj$useLocalizedStringFormatter)((0, ($parcel$interopDefault($eEzdj$intlStringsmjs))), '@react-spectrum/card');
    return /*#__PURE__*/ (0, $eEzdj$react).createElement($f30e78dbf4cce5fb$var$CenteredWrapper, null, /*#__PURE__*/ (0, $eEzdj$react).createElement((0, $1cfe37e7feefa23d$export$c79b9d6b4cc92af7), {
        isIndeterminate: true,
        "aria-label": state.collection.size > 0 ? stringFormatter.format('loadingMore') : stringFormatter.format('loading')
    }));
}
function $f30e78dbf4cce5fb$var$EmptyState() {
    let { renderEmptyState: renderEmptyState } = (0, $161b6b8b1c8e2230$export$fea0b38586ec8f13)();
    let emptyState = renderEmptyState ? renderEmptyState() : null;
    if (emptyState == null) return null;
    return /*#__PURE__*/ (0, $eEzdj$react).createElement($f30e78dbf4cce5fb$var$CenteredWrapper, null, emptyState);
}
function $f30e78dbf4cce5fb$var$CenteredWrapper({ children: children }) {
    let { state: state } = (0, $161b6b8b1c8e2230$export$fea0b38586ec8f13)();
    return /*#__PURE__*/ (0, $eEzdj$react).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($eEzdj$card_vars_cssmjs))), 'spectrum-CardView-centeredWrapper')
    }, /*#__PURE__*/ (0, $eEzdj$react).createElement("div", {
        role: "gridcell"
    }, children));
}
function $f30e78dbf4cce5fb$var$InternalCard(props) {
    let { item: item } = props;
    let cellNode = [
        ...item.childNodes
    ][0];
    let { state: state, cardOrientation: cardOrientation, isQuiet: isQuiet, layout: layout } = (0, $161b6b8b1c8e2230$export$fea0b38586ec8f13)();
    let layoutType = layout.layoutType;
    let rowRef = (0, $eEzdj$useRef)(undefined);
    let cellRef = (0, $eEzdj$useRef)(undefined);
    let unwrappedRef = (0, $3c2c983d5210446c$export$1d5cc31d9d8df817)(cellRef);
    let { rowProps: gridRowProps } = (0, $eEzdj$useGridRow)({
        node: item,
        isVirtualized: true
    }, state, rowRef);
    let { gridCellProps: gridCellProps } = (0, $eEzdj$useGridCell)({
        node: cellNode,
        focusMode: 'cell'
    }, state, unwrappedRef);
    // Prevent space key from scrolling the CardView if triggered on a disabled item or on a Card in a selectionMode="none" CardView.
    let allowsInteraction = state.selectionManager.selectionMode !== 'none';
    let isDisabled = !allowsInteraction || state.disabledKeys.has(item.key);
    let onKeyDown = (e)=>{
        if (e.key === ' ' && isDisabled) e.preventDefault();
    };
    let rowProps = (0, $eEzdj$mergeProps)(gridRowProps, {
        onKeyDown: onKeyDown
    });
    if (layoutType === 'grid' || layoutType === 'gallery') isQuiet = true;
    if (layoutType !== 'grid') cardOrientation = 'vertical';
    // We don't want to focus the checkbox (or any other focusable elements) within the Card
    // when pressing the arrow keys so we delete the key down handler here. Arrow key navigation between
    // the cards in the CardView is handled by useGrid => useSelectableCollection instead.
    delete gridCellProps.onKeyDownCapture;
    return /*#__PURE__*/ (0, $eEzdj$react).createElement("div", {
        ...rowProps,
        ref: rowRef,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($eEzdj$card_vars_cssmjs))), 'spectrum-CardView-row')
    }, /*#__PURE__*/ (0, $eEzdj$react).createElement((0, $957034e7d12d8044$export$7a6ccaf429ad93a8), {
        ref: cellRef,
        articleProps: gridCellProps,
        isQuiet: isQuiet,
        orientation: cardOrientation,
        item: item,
        layout: layoutType
    }, item.rendered));
}


export {$f30e78dbf4cce5fb$export$7e52c821f7b6f422 as CardView};
