// @adobe/react-spectrum/dist/private/card/CardBase.mjs
import {Checkbox as $b50e47f9c64ebdde$export$48513f6b9f8ce62d} from "../checkbox/Checkbox.mjs";
import {classNames as $6e6392558d48dfec$export$ce4ab0c55987d1ff} from "../utils/classNames.mjs";
import {SlotProvider as $62024859ff9f1f8a$export$8107b24b91795686} from "../utils/Slots.mjs";
import "../card_vars.css";
import $5WJjN$card_vars_cssmjs from "../card_vars_css.mjs";
import {useCardViewContext as $161b6b8b1c8e2230$export$fea0b38586ec8f13} from "./CardViewContext.mjs";
import {useDOMRef as $3c2c983d5210446c$export$c2c55ef9111cafd8} from "../utils/useDOMRef.mjs";
import {useHasChild as $f57c7d8d50bdc255$export$e52e2242b6d0f1d4} from "../utils/useHasChild.mjs";
import {useProviderProps as $71dfb0e0358a12de$export$521c373ccc32c300} from "../provider/Provider.mjs";
import {useStyleProps as $63d03c54ca5e4b88$export$b8e6fb9d2dff3f41} from "../utils/styleProps.mjs";
import {filterDOMProps as $5WJjN$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusRing as $5WJjN$FocusRing} from "react-aria/FocusRing";
import {getFocusableTreeWalker as $5WJjN$getFocusableTreeWalker} from "react-aria/private/focus/FocusScope";
import {mergeProps as $5WJjN$mergeProps} from "react-aria/mergeProps";
import {nodeContains as $5WJjN$nodeContains} from "react-aria/private/utils/shadowdom/DOMFunctions";
import $5WJjN$react, {useRef as $5WJjN$useRef, useState as $5WJjN$useState, useCallback as $5WJjN$useCallback, useMemo as $5WJjN$useMemo} from "react";
import {useFocusWithin as $5WJjN$useFocusWithin} from "react-aria/useFocusWithin";
import {useHover as $5WJjN$useHover} from "react-aria/useHover";
import {useLayoutEffect as $5WJjN$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useResizeObserver as $5WJjN$useResizeObserver} from "react-aria/private/utils/useResizeObserver";
import {useSlotId as $5WJjN$useSlotId} from "react-aria/private/utils/useId";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
// @ts-nocheck
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



















const $957034e7d12d8044$export$7a6ccaf429ad93a8 = /*#__PURE__*/ (0, $5WJjN$react).forwardRef(function CardBase(props, ref) {
    props = (0, $71dfb0e0358a12de$export$521c373ccc32c300)(props);
    let context = (0, $161b6b8b1c8e2230$export$fea0b38586ec8f13)() || {}; // we can call again here, won't change from Card.tsx
    let { state: state } = context;
    let manager = state?.selectionManager;
    let { isQuiet: isQuiet, orientation: orientation = 'vertical', articleProps: articleProps = {
        role: 'article'
    }, item: item, layout: layout, children: children } = props;
    let key = item?.key;
    let isSelected = manager?.isSelected(key);
    let isDisabled = state?.disabledKeys.has(key);
    let onChange = ()=>manager?.select(key);
    let { styleProps: styleProps } = (0, $63d03c54ca5e4b88$export$b8e6fb9d2dff3f41)(props);
    let { cardProps: cardProps, titleProps: titleProps, contentProps: contentProps } = $957034e7d12d8044$var$useCard(props);
    let domRef = (0, $3c2c983d5210446c$export$c2c55ef9111cafd8)(ref);
    let gridRef = (0, $5WJjN$useRef)(undefined);
    let checkboxRef = (0, $5WJjN$useRef)(null);
    // cards are only interactive if there is a selection manager and it allows selection
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $5WJjN$useHover)({
        isDisabled: manager === undefined || manager?.selectionMode === 'none' || isDisabled
    });
    let [isFocused, setIsFocused] = (0, $5WJjN$useState)(false);
    let { focusWithinProps: focusWithinProps } = (0, $5WJjN$useFocusWithin)({
        onFocusWithinChange: setIsFocused,
        isDisabled: isDisabled
    });
    // ToDo: see css for comment about avatar under selector .spectrum-Card--noLayout.spectrum-Card--default
    let hasPreviewImage = (0, $f57c7d8d50bdc255$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs)))['spectrum-Card-image']}`, gridRef);
    let hasPreviewIllustration = (0, $f57c7d8d50bdc255$export$e52e2242b6d0f1d4)(`.${(0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs)))['spectrum-Card-illustration']}`, gridRef);
    let hasPreview = hasPreviewImage || hasPreviewIllustration;
    // this is for horizontal cards
    let [height, setHeight] = (0, $5WJjN$useState)(NaN);
    let updateHeight = (0, $5WJjN$useCallback)(()=>{
        if (orientation !== 'horizontal') return;
        let cardHeight = gridRef.current.getBoundingClientRect().height;
        setHeight(cardHeight);
    }, [
        orientation,
        gridRef,
        setHeight
    ]);
    (0, $5WJjN$useResizeObserver)({
        ref: gridRef,
        onResize: updateHeight
    });
    let aspectRatioEnforce = undefined;
    if (orientation === 'horizontal' && !isNaN(height)) aspectRatioEnforce = {
        height: `${height}px`,
        width: `${height}px`
    };
    let slots = (0, $5WJjN$useMemo)(()=>({
            image: {
                UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-image'),
                objectFit: orientation === 'horizontal' ? 'cover' : 'contain',
                alt: '',
                ...aspectRatioEnforce
            },
            illustration: {
                UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-illustration'),
                ...aspectRatioEnforce
            },
            avatar: {
                UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-avatar'),
                size: 'avatar-size-400'
            },
            heading: {
                UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-heading'),
                ...titleProps
            },
            content: {
                UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-content'),
                ...contentProps
            },
            detail: {
                UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-detail')
            }
        }), // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        titleProps,
        contentProps,
        height,
        isQuiet,
        orientation
    ]);
    (0, $5WJjN$useLayoutEffect)(()=>{
        if (gridRef?.current) {
            let walker = (0, $5WJjN$getFocusableTreeWalker)(gridRef.current);
            let nextNode = walker.nextNode();
            while(nextNode != null){
                if (checkboxRef.current && !(0, $5WJjN$nodeContains)(checkboxRef.current.UNSAFE_getDOMNode(), nextNode)) {
                    console.warn('Card does not support focusable elements, please contact the team regarding your use case.');
                    break;
                }
                nextNode = walker.nextNode();
            }
        }
    }, [
        children
    ]);
    return /*#__PURE__*/ (0, $5WJjN$react).createElement((0, $5WJjN$FocusRing), {
        focusRingClass: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'focus-ring')
    }, /*#__PURE__*/ (0, $5WJjN$react).createElement("div", {
        ...styleProps,
        ...(0, $5WJjN$mergeProps)(cardProps, focusWithinProps, hoverProps, (0, $5WJjN$filterDOMProps)(props), articleProps),
        ref: domRef,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card', {
            'spectrum-Card--default': !isQuiet && orientation !== 'horizontal',
            'spectrum-Card--isQuiet': isQuiet && orientation !== 'horizontal',
            'spectrum-Card--horizontal': orientation === 'horizontal',
            'spectrum-Card--noPreview': !hasPreview,
            'is-hovered': isHovered,
            'is-focused': isFocused,
            'is-selected': isSelected,
            'spectrum-Card--waterfall': layout === 'waterfall',
            'spectrum-Card--gallery': layout === 'gallery',
            'spectrum-Card--grid': layout === 'grid',
            'spectrum-Card--noLayout': layout !== 'waterfall' && layout !== 'gallery' && layout !== 'grid'
        }, styleProps.className)
    }, /*#__PURE__*/ (0, $5WJjN$react).createElement("div", {
        ref: gridRef,
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-grid')
    }, manager && manager.selectionMode !== 'none' && /*#__PURE__*/ (0, $5WJjN$react).createElement("div", {
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-checkboxWrapper')
    }, /*#__PURE__*/ (0, $5WJjN$react).createElement((0, $b50e47f9c64ebdde$export$48513f6b9f8ce62d), {
        ref: checkboxRef,
        isDisabled: isDisabled,
        excludeFromTabOrder: true,
        isSelected: isSelected,
        onChange: onChange,
        UNSAFE_className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-checkbox'),
        isEmphasized: true,
        "aria-label": "select"
    })), /*#__PURE__*/ (0, $5WJjN$react).createElement((0, $62024859ff9f1f8a$export$8107b24b91795686), {
        slots: slots
    }, children), /*#__PURE__*/ (0, $5WJjN$react).createElement("div", {
        className: (0, $6e6392558d48dfec$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($5WJjN$card_vars_cssmjs))), 'spectrum-Card-decoration')
    }))));
});
function $957034e7d12d8044$var$useCard(props) {
    let titleId = (0, $5WJjN$useSlotId)();
    let descriptionId = (0, $5WJjN$useSlotId)();
    let titleProps = (0, $5WJjN$useMemo)(()=>({
            id: titleId
        }), [
        titleId
    ]);
    let contentProps = (0, $5WJjN$useMemo)(()=>({
            id: descriptionId
        }), [
        descriptionId
    ]);
    return {
        cardProps: {
            ...(0, $5WJjN$filterDOMProps)(props),
            'aria-labelledby': titleId,
            'aria-describedby': descriptionId,
            tabIndex: 0
        },
        titleProps: titleProps,
        contentProps: contentProps
    };
}


export {$957034e7d12d8044$export$7a6ccaf429ad93a8 as CardBase};
