// @adobe/react-spectrum/dist/private/card/CardView.cjs
var $93b4a57699139677$exports = require("./CardBase.cjs");
var $c5cd545e21c17a4a$exports = require("./CardViewContext.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $9cca57d92973a563$exports = require("./intlStrings.cjs");
var $948c2416aa3a9507$exports = require("../progress/ProgressCircle.cjs");
require("../card_vars.css");
var $59e87deeac09a752$exports = require("../card_vars_css.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $ohoUA$reactstatelyprivategridGridCollection = require("react-stately/private/grid/GridCollection");
var $ohoUA$reactariamergeProps = require("react-aria/mergeProps");
var $ohoUA$react = require("react");
var $ohoUA$reactariauseCollator = require("react-aria/useCollator");
var $ohoUA$reactariaprivategriduseGrid = require("react-aria/private/grid/useGrid");
var $ohoUA$reactariaprivategriduseGridCell = require("react-aria/private/grid/useGridCell");
var $ohoUA$reactariaprivategriduseGridRow = require("react-aria/private/grid/useGridRow");
var $ohoUA$reactstatelyprivategriduseGridState = require("react-stately/private/grid/useGridState");
var $ohoUA$reactstatelyuseListState = require("react-stately/useListState");
var $ohoUA$reactariaI18nProvider = require("react-aria/I18nProvider");
var $ohoUA$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $ohoUA$reactariaprivatevirtualizerVirtualizer = require("react-aria/private/virtualizer/Virtualizer");
var $ohoUA$reactariaprivatevirtualizerVirtualizerItem = require("react-aria/private/virtualizer/VirtualizerItem");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "CardView", function () { return $017dba522854c0f2$export$7e52c821f7b6f422; });
// @ts-nocheck
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





















const $017dba522854c0f2$export$7e52c821f7b6f422 = /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).forwardRef(function CardView(props, ref) {
    let { scale: scale } = (0, $544fc82701fc93e9$exports.useProvider)();
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let { isQuiet: isQuiet, renderEmptyState: renderEmptyState, layout: layout, loadingState: loadingState, onLoadMore: onLoadMore, cardOrientation: cardOrientation = 'vertical' } = props;
    let collator = (0, $ohoUA$reactariauseCollator.useCollator)({
        usage: 'search',
        sensitivity: 'base'
    });
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let cardViewLayout = (0, $ohoUA$react.useMemo)(()=>typeof layout === 'function' ? new layout({
            collator: collator,
            cardOrientation: cardOrientation,
            scale: scale
        }) : layout, [
        layout,
        collator,
        cardOrientation,
        scale
    ]);
    let layoutType = cardViewLayout.layoutType;
    let { direction: direction } = (0, $ohoUA$reactariaI18nProvider.useLocale)();
    let { collection: collection } = (0, $ohoUA$reactstatelyuseListState.useListState)(props);
    let gridCollection = (0, $ohoUA$react.useMemo)(()=>new (0, $ohoUA$reactstatelyprivategridGridCollection.GridCollection)({
            columnCount: 1,
            items: [
                ...collection
            ].map((item)=>({
                    // Makes the Grid row use the keys the user provides to the cards so that selection change via interactions returns the card keys
                    ...item,
                    hasChildNodes: true,
                    childNodes: [
                        {
                            key: `cell-${item.key}`,
                            type: 'cell',
                            value: null,
                            level: 0,
                            rendered: null,
                            textValue: item.textValue,
                            hasChildNodes: false,
                            childNodes: []
                        }
                    ]
                }))
        }), [
        collection
    ]);
    let state = (0, $ohoUA$reactstatelyprivategriduseGridState.useGridState)({
        ...props,
        selectionMode: cardOrientation === 'horizontal' && layoutType === 'grid' ? 'none' : props.selectionMode,
        collection: gridCollection,
        focusMode: 'cell'
    });
    cardViewLayout.collection = gridCollection;
    cardViewLayout.disabledKeys = state.disabledKeys;
    let { gridProps: gridProps } = (0, $ohoUA$reactariaprivategriduseGrid.useGrid)({
        ...props,
        isVirtualized: true,
        keyboardDelegate: cardViewLayout
    }, state, domRef);
    let renderWrapper = (0, $ohoUA$react.useCallback)((parent, reusableView)=>/*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement((0, $ohoUA$reactariaprivatevirtualizerVirtualizerItem.VirtualizerItem), {
            key: reusableView.key,
            layoutInfo: reusableView.layoutInfo,
            virtualizer: reusableView.virtualizer,
            parent: parent?.layoutInfo
        }, reusableView.rendered), []);
    let focusedKey = state.selectionManager.focusedKey;
    let focusedItem = gridCollection.getItem(state.selectionManager.focusedKey);
    if (focusedItem?.parentKey != null) focusedKey = focusedItem.parentKey;
    let persistedKeys = (0, $ohoUA$react.useMemo)(()=>focusedKey != null ? new Set([
            focusedKey
        ]) : null, [
        focusedKey
    ]);
    // TODO: does aria-row count and aria-col count need to be modified? Perhaps aria-col count needs to be omitted
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement((0, $c5cd545e21c17a4a$exports.CardViewContext).Provider, {
        value: {
            state: state,
            isQuiet: isQuiet,
            layout: cardViewLayout,
            cardOrientation: cardOrientation,
            renderEmptyState: renderEmptyState
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement((0, $ohoUA$reactariaprivatevirtualizerVirtualizer.Virtualizer), {
        ...gridProps,
        ...styleProps,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-CardView'),
        ref: domRef,
        persistedKeys: persistedKeys,
        scrollDirection: "vertical",
        layout: cardViewLayout,
        collection: gridCollection,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        layoutOptions: (0, $ohoUA$react.useMemo)(()=>({
                isLoading: isLoading,
                direction: direction
            }), [
            isLoading,
            direction
        ]),
        renderWrapper: renderWrapper,
        style: {
            ...styleProps.style,
            scrollPaddingTop: cardViewLayout.margin || 0
        }
    }, (0, $ohoUA$react.useCallback)((type, item)=>{
        if (type === 'item') return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement($017dba522854c0f2$var$InternalCard, {
            item: item
        });
        else if (type === 'loader') return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement($017dba522854c0f2$var$LoadingState, null);
        else if (type === 'placeholder') return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement($017dba522854c0f2$var$EmptyState, null);
    }, [])));
});
function $017dba522854c0f2$var$LoadingState() {
    let { state: state } = (0, $c5cd545e21c17a4a$exports.useCardViewContext)();
    let stringFormatter = (0, $ohoUA$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($9cca57d92973a563$exports))), '@react-spectrum/card');
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement($017dba522854c0f2$var$CenteredWrapper, null, /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement((0, $948c2416aa3a9507$exports.ProgressCircle), {
        isIndeterminate: true,
        "aria-label": state.collection.size > 0 ? stringFormatter.format('loadingMore') : stringFormatter.format('loading')
    }));
}
function $017dba522854c0f2$var$EmptyState() {
    let { renderEmptyState: renderEmptyState } = (0, $c5cd545e21c17a4a$exports.useCardViewContext)();
    let emptyState = renderEmptyState ? renderEmptyState() : null;
    if (emptyState == null) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement($017dba522854c0f2$var$CenteredWrapper, null, emptyState);
}
function $017dba522854c0f2$var$CenteredWrapper({ children: children }) {
    let { state: state } = (0, $c5cd545e21c17a4a$exports.useCardViewContext)();
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-CardView-centeredWrapper')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement("div", {
        role: "gridcell"
    }, children));
}
function $017dba522854c0f2$var$InternalCard(props) {
    let { item: item } = props;
    let cellNode = [
        ...item.childNodes
    ][0];
    let { state: state, cardOrientation: cardOrientation, isQuiet: isQuiet, layout: layout } = (0, $c5cd545e21c17a4a$exports.useCardViewContext)();
    let layoutType = layout.layoutType;
    let rowRef = (0, $ohoUA$react.useRef)(undefined);
    let cellRef = (0, $ohoUA$react.useRef)(undefined);
    let unwrappedRef = (0, $65aea7b37663976b$exports.useUnwrapDOMRef)(cellRef);
    let { rowProps: gridRowProps } = (0, $ohoUA$reactariaprivategriduseGridRow.useGridRow)({
        node: item,
        isVirtualized: true
    }, state, rowRef);
    let { gridCellProps: gridCellProps } = (0, $ohoUA$reactariaprivategriduseGridCell.useGridCell)({
        node: cellNode,
        focusMode: 'cell'
    }, state, unwrappedRef);
    // Prevent space key from scrolling the CardView if triggered on a disabled item or on a Card in a selectionMode="none" CardView.
    let allowsInteraction = state.selectionManager.selectionMode !== 'none';
    let isDisabled = !allowsInteraction || state.disabledKeys.has(item.key);
    let onKeyDown = (e)=>{
        if (e.key === ' ' && isDisabled) e.preventDefault();
    };
    let rowProps = (0, $ohoUA$reactariamergeProps.mergeProps)(gridRowProps, {
        onKeyDown: onKeyDown
    });
    if (layoutType === 'grid' || layoutType === 'gallery') isQuiet = true;
    if (layoutType !== 'grid') cardOrientation = 'vertical';
    // We don't want to focus the checkbox (or any other focusable elements) within the Card
    // when pressing the arrow keys so we delete the key down handler here. Arrow key navigation between
    // the cards in the CardView is handled by useGrid => useSelectableCollection instead.
    delete gridCellProps.onKeyDownCapture;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement("div", {
        ...rowProps,
        ref: rowRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($59e87deeac09a752$exports))), 'spectrum-CardView-row')
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ohoUA$react))).createElement((0, $93b4a57699139677$exports.CardBase), {
        ref: cellRef,
        articleProps: gridCellProps,
        isQuiet: isQuiet,
        orientation: cardOrientation,
        item: item,
        layout: layoutType
    }, item.rendered));
}
