// @adobe/react-spectrum/dist/private/actiongroup/ActionGroup.js
import {ActionButton as $c265dbb41bfd0210$export$cfc7921d29ef7b80} from "../button/ActionButton.js";
import "../button_vars.css";
import $h7VIV$button_vars_cssmjs from "../button_vars_css.mjs";
import {classNames as $512ba93e663f149c$export$ce4ab0c55987d1ff} from "../utils/classNames.js";
import {ClearSlots as $68f4bc2c1abc5618$export$ceb145244332b7a2, SlotProvider as $68f4bc2c1abc5618$export$8107b24b91795686, useSlotProps as $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3} from "../utils/Slots.js";
import $h7VIV$intlStringsjs from "./intlStrings.js";
import {Menu as $79ddee63a726ea3d$export$d9b273488cd8ce6f} from "../menu/Menu.js";
import {MenuTrigger as $9f6ebde23392f425$export$27d2ad3c5815583e} from "../menu/MenuTrigger.js";
import {Provider as $089943c7a219141c$export$2881499e37b75b9a, useProviderProps as $089943c7a219141c$export$521c373ccc32c300} from "../provider/Provider.js";
import "../actiongroup_vars.css";
import $h7VIV$actiongroup_vars_cssmjs from "../actiongroup_vars_css.mjs";
import {Text as $42dd7396e689e4e6$export$5f1af8db9871e1d6} from "../text/Text.js";
import {Tooltip as $3f07dad5c19b53b6$export$28c660c63b792dea} from "../tooltip/Tooltip.js";
import {TooltipTrigger as $3e76a1633f5aa2b7$export$8c610744efcf8a1d} from "../tooltip/TooltipTrigger.js";
import {useDOMRef as $c234463e9ef56637$export$c2c55ef9111cafd8} from "../utils/useDOMRef.js";
import {useStyleProps as $120fbea2d95e11ed$export$b8e6fb9d2dff3f41} from "../utils/styleProps.js";
import {useActionGroup as $h7VIV$useActionGroup} from "react-aria/private/actiongroup/useActionGroup";
import $h7VIV$spectrumiconsuiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import {filterDOMProps as $h7VIV$filterDOMProps} from "react-aria/filterDOMProps";
import {FocusScope as $h7VIV$FocusScope} from "react-aria/FocusScope";
import {Item as $h7VIV$Item} from "react-stately/Item";
import {useListState as $h7VIV$useListState} from "react-stately/useListState";
import {mergeProps as $h7VIV$mergeProps} from "react-aria/mergeProps";
import $h7VIV$spectrumiconsworkflowMore from "@spectrum-icons/workflow/More";
import {PressResponder as $h7VIV$PressResponder} from "react-aria/private/interactions/PressResponder";
import $h7VIV$react, {forwardRef as $h7VIV$forwardRef, useRef as $h7VIV$useRef, useMemo as $h7VIV$useMemo, useCallback as $h7VIV$useCallback, useState as $h7VIV$useState} from "react";
import {useActionGroupItem as $h7VIV$useActionGroupItem} from "react-aria/private/actiongroup/useActionGroupItem";
import {useHover as $h7VIV$useHover} from "react-aria/useHover";
import {useId as $h7VIV$useId} from "react-aria/useId";
import {useLayoutEffect as $h7VIV$useLayoutEffect} from "react-aria/private/utils/useLayoutEffect";
import {useLocalizedStringFormatter as $h7VIV$useLocalizedStringFormatter} from "react-aria/useLocalizedStringFormatter";
import {useResizeObserver as $h7VIV$useResizeObserver} from "react-aria/private/utils/useResizeObserver";
import {useValueEffect as $h7VIV$useValueEffect} from "react-aria/private/utils/useValueEffect";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






























const $78c8311cc10fd6f1$export$c21a5597f732a168 = /*#__PURE__*/ (0, $h7VIV$forwardRef)(function ActionGroup(props, ref) {
    props = (0, $089943c7a219141c$export$521c373ccc32c300)(props);
    props = (0, $68f4bc2c1abc5618$export$1e5c9e6e4e15efe3)(props, 'actionGroup');
    let { isEmphasized: isEmphasized, density: density, isJustified: isJustified, isDisabled: isDisabled, orientation: orientation = 'horizontal', isQuiet: isQuiet, staticColor: staticColor, overflowMode: overflowMode = 'wrap', onAction: onAction, buttonLabelBehavior: buttonLabelBehavior, summaryIcon: summaryIcon, ...otherProps } = props;
    let domRef = (0, $c234463e9ef56637$export$c2c55ef9111cafd8)(ref);
    let wrapperRef = (0, $h7VIV$useRef)(null);
    let state = (0, $h7VIV$useListState)({
        ...props,
        suppressTextValueWarning: true
    });
    let { actionGroupProps: actionGroupProps } = (0, $h7VIV$useActionGroup)(props, state, domRef);
    let isVertical = orientation === 'vertical';
    let providerProps = {
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isQuiet: isQuiet
    };
    let { styleProps: styleProps } = (0, $120fbea2d95e11ed$export$b8e6fb9d2dff3f41)(props);
    // Only hide button text if every item contains more than just plain text (we assume an icon).
    let isIconCollapsible = (0, $h7VIV$useMemo)(()=>[
            ...state.collection
        ].every((item)=>typeof item.rendered !== 'string'), [
        state.collection
    ]);
    let [{ visibleItems: visibleItems, hideButtonText: hideButtonText, isMeasuring: isMeasuring }, setVisibleItems] = (0, $h7VIV$useValueEffect)({
        visibleItems: state.collection.size,
        hideButtonText: buttonLabelBehavior === 'hide' && isIconCollapsible,
        isMeasuring: false
    });
    let selectionMode = state.selectionManager.selectionMode;
    let updateOverflow = (0, $h7VIV$useCallback)(()=>{
        if (overflowMode === 'wrap') return;
        if (orientation === 'vertical' && selectionMode !== 'none') // Collapsing vertical action groups with selection is currently unsupported by Spectrum.
        return;
        let computeVisibleItems = (visibleItems)=>{
            if (domRef.current && wrapperRef.current) {
                let listItems = Array.from(domRef.current.children);
                let containerSize = orientation === 'horizontal' ? wrapperRef.current.getBoundingClientRect().width : wrapperRef.current.getBoundingClientRect().height;
                let isShowingMenu = visibleItems < state.collection.size;
                let calculatedSize = 0;
                let newVisibleItems = 0;
                if (isShowingMenu) {
                    let item = listItems.pop();
                    if (item) calculatedSize += orientation === 'horizontal' ? $78c8311cc10fd6f1$var$outerWidth(item, false, true) : $78c8311cc10fd6f1$var$outerHeight(item, false, true);
                }
                for (let [i, item] of listItems.entries()){
                    calculatedSize += orientation === 'horizontal' ? $78c8311cc10fd6f1$var$outerWidth(item, i === 0, i === listItems.length - 1) : $78c8311cc10fd6f1$var$outerHeight(item, i === 0, i === listItems.length - 1);
                    if (Math.round(calculatedSize) <= Math.round(containerSize)) newVisibleItems++;
                    else break;
                }
                // If selection is enabled, and not all of the items fit, collapse all of them into a dropdown
                // immediately rather than having some visible and some not.
                if (selectionMode !== 'none' && newVisibleItems < state.collection.size) return 0;
                return newVisibleItems;
            }
            return visibleItems;
        };
        setVisibleItems(function*() {
            let hideButtonText = buttonLabelBehavior === 'hide' && isIconCollapsible;
            // Update to show all items.
            yield {
                visibleItems: state.collection.size,
                hideButtonText: hideButtonText,
                isMeasuring: true
            };
            // Measure, and update to show the items that fit.
            let newVisibleItems = computeVisibleItems(state.collection.size);
            let isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
            // If not all of the buttons fit, and buttonLabelBehavior is 'collapse', then first try hiding
            // the button text and only showing icons. Only if that still doesn't fit collapse into a menu.
            if (newVisibleItems < state.collection.size && buttonLabelBehavior === 'collapse' && isIconCollapsible) {
                yield {
                    visibleItems: state.collection.size,
                    hideButtonText: true,
                    isMeasuring: true
                };
                newVisibleItems = computeVisibleItems(state.collection.size);
                isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
                hideButtonText = true;
            }
            yield {
                visibleItems: newVisibleItems,
                hideButtonText: hideButtonText,
                isMeasuring: isMeasuring
            };
            // If the number of items is less than the number of children,
            // then update again to ensure that the menu fits.
            if (isMeasuring) yield {
                visibleItems: computeVisibleItems(newVisibleItems),
                hideButtonText: hideButtonText,
                isMeasuring: false
            };
        });
    }, [
        domRef,
        state.collection,
        setVisibleItems,
        overflowMode,
        selectionMode,
        buttonLabelBehavior,
        isIconCollapsible,
        orientation
    ]);
    // Watch the parent element for size changes. Watching only the action group itself may not work
    // in all scenarios because it may not shrink when available space is reduced.
    let parentRef = (0, $h7VIV$useMemo)(()=>({
            get current () {
                var _wrapperRef_current;
                return (_wrapperRef_current = wrapperRef.current) === null || _wrapperRef_current === void 0 ? void 0 : _wrapperRef_current.parentElement;
            }
        }), [
        wrapperRef
    ]);
    (0, $h7VIV$useResizeObserver)({
        ref: overflowMode !== 'wrap' ? parentRef : undefined,
        onResize: updateOverflow
    });
    (0, $h7VIV$useLayoutEffect)(updateOverflow, [
        updateOverflow,
        state.collection
    ]);
    let children = [
        ...state.collection
    ];
    let menuItem = null;
    let menuProps = {};
    // If there are no visible items, don't apply any props to the action group container
    // and pass all aria labeling props through to the menu button.
    if (overflowMode === 'collapse' && visibleItems === 0) {
        menuProps = (0, $h7VIV$filterDOMProps)(props, {
            labelable: true
        });
        actionGroupProps = {};
    }
    if (overflowMode === 'collapse' && visibleItems < state.collection.size) {
        let menuChildren = children.slice(visibleItems);
        children = children.slice(0, visibleItems);
        menuItem = /*#__PURE__*/ (0, $h7VIV$react).createElement($78c8311cc10fd6f1$var$ActionGroupMenu, {
            ...menuProps,
            items: menuChildren,
            onAction: (key)=>onAction === null || onAction === void 0 ? void 0 : onAction(key),
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            state: state,
            summaryIcon: summaryIcon,
            hideButtonText: hideButtonText,
            isOnlyItem: visibleItems === 0,
            orientation: orientation
        });
    }
    let style = {
        ...styleProps.style,
        // While measuring, take up as much space as possible.
        flexBasis: isMeasuring ? '100%' : undefined
    };
    return /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $h7VIV$FocusScope), null, /*#__PURE__*/ (0, $h7VIV$react).createElement("div", {
        ...styleProps,
        style: style,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$actiongroup_vars_cssmjs))), 'flex-container', styleProps.className),
        ref: wrapperRef
    }, /*#__PURE__*/ (0, $h7VIV$react).createElement("div", {
        ...actionGroupProps,
        ref: domRef,
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$actiongroup_vars_cssmjs))), 'flex-gap', 'spectrum-ActionGroup', {
            'spectrum-ActionGroup--quiet': isQuiet,
            'spectrum-ActionGroup--vertical': isVertical,
            'spectrum-ActionGroup--compact': density === 'compact',
            'spectrum-ActionGroup--justified': isJustified && !isMeasuring,
            'spectrum-ActionGroup--overflowCollapse': overflowMode === 'collapse'
        }, otherProps.UNSAFE_className)
    }, /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $089943c7a219141c$export$2881499e37b75b9a), providerProps, children.map((item)=>/*#__PURE__*/ (0, $h7VIV$react).createElement($78c8311cc10fd6f1$var$ActionGroupItem, {
            key: item.key,
            onAction: onAction,
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            item: item,
            state: state,
            hideButtonText: hideButtonText,
            orientation: orientation
        })), menuItem))));
});
function $78c8311cc10fd6f1$var$ActionGroupItem({ item: item, state: state, isDisabled: isDisabled, isEmphasized: isEmphasized, staticColor: staticColor, onAction: onAction, hideButtonText: hideButtonText, orientation: orientation }) {
    let ref = (0, $h7VIV$useRef)(null);
    let { buttonProps: buttonProps } = (0, $h7VIV$useActionGroupItem)({
        key: item.key
    }, state);
    isDisabled = isDisabled || state.disabledKeys.has(item.key);
    let isSelected = state.selectionManager.isSelected(item.key);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $h7VIV$useHover)({
        isDisabled: isDisabled
    });
    let domProps = (0, $h7VIV$filterDOMProps)(item.props);
    if (onAction && !isDisabled) buttonProps = (0, $h7VIV$mergeProps)(buttonProps, {
        onPress: ()=>onAction(item.key)
    });
    // If button text is hidden, we need to show it as a tooltip instead, so
    // go find the text element in the DOM after rendering.
    let textId = (0, $h7VIV$useId)();
    let [textContent, setTextContent] = (0, $h7VIV$useState)('');
    (0, $h7VIV$useLayoutEffect)(()=>{
        var _document_getElementById;
        if (hideButtonText) setTextContent((_document_getElementById = document.getElementById(textId)) === null || _document_getElementById === void 0 ? void 0 : _document_getElementById.textContent);
    }, [
        hideButtonText,
        item.rendered,
        textId
    ]);
    let button = // Use a PressResponder to send DOM props through.
    // ActionButton doesn't allow overriding the role by default.
    /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $h7VIV$PressResponder), (0, $h7VIV$mergeProps)(buttonProps, hoverProps, domProps), /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $68f4bc2c1abc5618$export$ceb145244332b7a2), null, /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText
            }
        }
    }, /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $c265dbb41bfd0210$export$cfc7921d29ef7b80), {
        ref: ref,
        // @ts-ignore (private)
        hideButtonText: hideButtonText,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$actiongroup_vars_cssmjs))), 'spectrum-ActionGroup-item', {
            'is-selected': isSelected,
            'is-hovered': isHovered,
            'spectrum-ActionGroup-item--iconOnly': hideButtonText,
            'spectrum-ActionGroup-item--isDisabled': isDisabled
        }, (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$button_vars_cssmjs))), {
            'spectrum-ActionButton--emphasized': isEmphasized,
            'is-selected': isSelected
        })),
        isDisabled: isDisabled,
        staticColor: staticColor,
        "aria-label": item['aria-label'],
        "aria-labelledby": item['aria-label'] == null && hideButtonText ? textId : undefined
    }, item.rendered))));
    if (hideButtonText && textContent) button = /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $3e76a1633f5aa2b7$export$8c610744efcf8a1d), {
        placement: orientation === 'vertical' ? 'end' : 'top'
    }, button, /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $3f07dad5c19b53b6$export$28c660c63b792dea), null, textContent));
    if (item.wrapper) button = item.wrapper(button);
    return button;
}
function $78c8311cc10fd6f1$var$ActionGroupMenu({ state: state, isDisabled: isDisabled, isEmphasized: isEmphasized, staticColor: staticColor, items: items, onAction: onAction, summaryIcon: summaryIcon, hideButtonText: hideButtonText, isOnlyItem: isOnlyItem, orientation: orientation, ...otherProps }) {
    // Use the key of the first item within the menu as the key of the button.
    // The key must actually exist in the collection for focus to work correctly.
    let key = items[0].key;
    let { buttonProps: buttonProps } = (0, $h7VIV$useActionGroupItem)({
        key: key
    }, state);
    let stringFormatter = (0, $h7VIV$useLocalizedStringFormatter)((0, ($parcel$interopDefault($h7VIV$intlStringsjs))), '@react-spectrum/actiongroup');
    // The menu button shouldn't act like an actual action group item.
    delete buttonProps.onPress;
    delete buttonProps.role;
    delete buttonProps['aria-checked'];
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $h7VIV$useHover)({
        isDisabled: isDisabled
    });
    // If no aria-label or aria-labelledby is given, provide a default one.
    let ariaLabel = otherProps['aria-label'] || (otherProps['aria-labelledby'] ? undefined : stringFormatter.format('more'));
    let ariaLabelledby = otherProps['aria-labelledby'];
    let textId = (0, $h7VIV$useId)();
    let id = (0, $h7VIV$useId)();
    // Summary icon only applies when selection is enabled.
    if (state.selectionManager.selectionMode === 'none') summaryIcon = null;
    let iconOnly = false;
    // If there is a selection, show the selected state on the menu button.
    let isSelected = state.selectionManager.selectionMode !== 'none' && !state.selectionManager.isEmpty;
    // If single selection and empty selection is not allowed, swap the contents of the button to the selected item (like a Picker).
    if (!summaryIcon && state.selectionManager.selectionMode === 'single' && state.selectionManager.disallowEmptySelection && state.selectionManager.firstSelectedKey != null) {
        let selectedItem = state.collection.getItem(state.selectionManager.firstSelectedKey);
        if (selectedItem) {
            summaryIcon = selectedItem.rendered;
            if (typeof summaryIcon === 'string') summaryIcon = /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $42dd7396e689e4e6$export$5f1af8db9871e1d6), null, summaryIcon);
            iconOnly = !!hideButtonText;
            ariaLabelledby = `${ariaLabelledby !== null && ariaLabelledby !== void 0 ? ariaLabelledby : id} ${textId}`;
        }
    }
    if (summaryIcon) // If there's a custom summary icon, also add a chevron.
    summaryIcon = /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $h7VIV$react).Fragment, null, /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $h7VIV$spectrumiconsuiChevronDownMedium), {
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$actiongroup_vars_cssmjs))), 'spectrum-ActionGroup-menu-chevron')
    }), /*#__PURE__*/ (0, $h7VIV$react).createElement("span", {
        className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$actiongroup_vars_cssmjs))), 'spectrum-ActionGroup-menu-contents', {
            'spectrum-ActionGroup-item--iconOnly': iconOnly
        })
    }, summaryIcon));
    return(// Use a PressResponder to send DOM props through.
    /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $9f6ebde23392f425$export$27d2ad3c5815583e), {
        align: isOnlyItem ? 'start' : 'end',
        direction: orientation === 'vertical' ? 'end' : 'bottom'
    }, /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $68f4bc2c1abc5618$export$8107b24b91795686), {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText,
                UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$actiongroup_vars_cssmjs))), 'spectrum-ActionGroup-menu-text')
            }
        }
    }, /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $h7VIV$PressResponder), (0, $h7VIV$mergeProps)(buttonProps, hoverProps), /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $c265dbb41bfd0210$export$cfc7921d29ef7b80), {
        ...otherProps,
        id: id,
        "aria-label": ariaLabel,
        "aria-labelledby": ariaLabelledby,
        UNSAFE_className: (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$actiongroup_vars_cssmjs))), 'spectrum-ActionGroup-item', 'spectrum-ActionGroup-menu', {
            'is-hovered': isHovered,
            'is-selected': isSelected
        }, (0, $512ba93e663f149c$export$ce4ab0c55987d1ff)((0, ($parcel$interopDefault($h7VIV$button_vars_cssmjs))), {
            'is-selected': isSelected,
            'spectrum-ActionButton--emphasized': isEmphasized
        })),
        isDisabled: isDisabled,
        staticColor: staticColor
    }, summaryIcon || /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $h7VIV$spectrumiconsworkflowMore), null)))), /*#__PURE__*/ (0, $h7VIV$react).createElement((0, $79ddee63a726ea3d$export$d9b273488cd8ce6f), {
        items: items,
        disabledKeys: state.disabledKeys,
        selectionMode: state.selectionManager.selectionMode,
        selectedKeys: state.selectionManager.selectedKeys,
        disallowEmptySelection: state.selectionManager.disallowEmptySelection,
        onSelectionChange: (keys)=>state.selectionManager.setSelectedKeys(keys),
        onAction: onAction
    }, (node)=>/*#__PURE__*/ (0, $h7VIV$react).createElement((0, $h7VIV$Item), {
            textValue: node.textValue,
            ...(0, $h7VIV$filterDOMProps)(node.props)
        }, node.rendered))));
}
function $78c8311cc10fd6f1$var$outerWidth(element, ignoreLeftMargin, ignoreRightMargin) {
    let style = window.getComputedStyle(element);
    return element.getBoundingClientRect().width + (ignoreLeftMargin ? 0 : $78c8311cc10fd6f1$var$toNumber(style.marginLeft)) + (ignoreRightMargin ? 0 : $78c8311cc10fd6f1$var$toNumber(style.marginRight));
}
function $78c8311cc10fd6f1$var$outerHeight(element, ignoreTopMargin, ignoreBottomMargin) {
    let style = window.getComputedStyle(element);
    return element.getBoundingClientRect().height + (ignoreTopMargin ? 0 : $78c8311cc10fd6f1$var$toNumber(style.marginTop)) + (ignoreBottomMargin ? 0 : $78c8311cc10fd6f1$var$toNumber(style.marginBottom));
}
function $78c8311cc10fd6f1$var$toNumber(value) {
    let parsed = parseInt(value, 10);
    return isNaN(parsed) ? 0 : parsed;
}


export {$78c8311cc10fd6f1$export$c21a5597f732a168 as ActionGroup};
