// @adobe/react-spectrum/dist/private/actiongroup/ActionGroup.cjs
var $183c5173677598aa$exports = require("../button/ActionButton.cjs");
require("../button_vars.css");
var $869138cbe3b599dc$exports = require("../button_vars_css.cjs");
var $69fd630bd812ba47$exports = require("../utils/classNames.cjs");
var $feede71cddc0c5f3$exports = require("../utils/Slots.cjs");
var $b9a8a32d848cb509$exports = require("./intlStrings.cjs");
var $802fb5441f76e7b0$exports = require("../menu/Menu.cjs");
var $98227f5fd590c993$exports = require("../menu/MenuTrigger.cjs");
var $544fc82701fc93e9$exports = require("../provider/Provider.cjs");
require("../actiongroup_vars.css");
var $80eda55ee0c557c4$exports = require("../actiongroup_vars_css.cjs");
var $15e3b68ec42125a9$exports = require("../text/Text.cjs");
var $f1974881be69ddc4$exports = require("../tooltip/Tooltip.cjs");
var $ff31b6c981164b8a$exports = require("../tooltip/TooltipTrigger.cjs");
var $65aea7b37663976b$exports = require("../utils/useDOMRef.cjs");
var $b8f90d51c4908137$exports = require("../utils/styleProps.cjs");
var $gC6oF$reactariaprivateactiongroupuseActionGroup = require("react-aria/private/actiongroup/useActionGroup");
var $gC6oF$spectrumiconsuiChevronDownMedium = require("@spectrum-icons/ui/ChevronDownMedium");
var $gC6oF$reactariafilterDOMProps = require("react-aria/filterDOMProps");
var $gC6oF$reactariaFocusScope = require("react-aria/FocusScope");
var $gC6oF$reactstatelyItem = require("react-stately/Item");
var $gC6oF$reactstatelyuseListState = require("react-stately/useListState");
var $gC6oF$reactariamergeProps = require("react-aria/mergeProps");
var $gC6oF$spectrumiconsworkflowMore = require("@spectrum-icons/workflow/More");
var $gC6oF$reactariaprivateinteractionsPressResponder = require("react-aria/private/interactions/PressResponder");
var $gC6oF$react = require("react");
var $gC6oF$reactariaprivateactiongroupuseActionGroupItem = require("react-aria/private/actiongroup/useActionGroupItem");
var $gC6oF$reactariauseHover = require("react-aria/useHover");
var $gC6oF$reactariauseId = require("react-aria/useId");
var $gC6oF$reactariaprivateutilsuseLayoutEffect = require("react-aria/private/utils/useLayoutEffect");
var $gC6oF$reactariauseLocalizedStringFormatter = require("react-aria/useLocalizedStringFormatter");
var $gC6oF$reactariaprivateutilsuseResizeObserver = require("react-aria/private/utils/useResizeObserver");
var $gC6oF$reactariaprivateutilsuseValueEffect = require("react-aria/private/utils/useValueEffect");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ActionGroup", function () { return $1f2a1f451a6aa23a$export$c21a5597f732a168; });
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






























const $1f2a1f451a6aa23a$export$c21a5597f732a168 = /*#__PURE__*/ (0, $gC6oF$react.forwardRef)(function ActionGroup(props, ref) {
    props = (0, $544fc82701fc93e9$exports.useProviderProps)(props);
    props = (0, $feede71cddc0c5f3$exports.useSlotProps)(props, 'actionGroup');
    let { isEmphasized: isEmphasized, density: density, isJustified: isJustified, isDisabled: isDisabled, orientation: orientation = 'horizontal', isQuiet: isQuiet, staticColor: staticColor, overflowMode: overflowMode = 'wrap', onAction: onAction, buttonLabelBehavior: buttonLabelBehavior, summaryIcon: summaryIcon, ...otherProps } = props;
    let domRef = (0, $65aea7b37663976b$exports.useDOMRef)(ref);
    let wrapperRef = (0, $gC6oF$react.useRef)(null);
    let state = (0, $gC6oF$reactstatelyuseListState.useListState)({
        ...props,
        suppressTextValueWarning: true
    });
    let { actionGroupProps: actionGroupProps } = (0, $gC6oF$reactariaprivateactiongroupuseActionGroup.useActionGroup)(props, state, domRef);
    let isVertical = orientation === 'vertical';
    let providerProps = {
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isQuiet: isQuiet
    };
    let { styleProps: styleProps } = (0, $b8f90d51c4908137$exports.useStyleProps)(props);
    // Only hide button text if every item contains more than just plain text (we assume an icon).
    let isIconCollapsible = (0, $gC6oF$react.useMemo)(()=>[
            ...state.collection
        ].every((item)=>typeof item.rendered !== 'string'), [
        state.collection
    ]);
    let [{ visibleItems: visibleItems, hideButtonText: hideButtonText, isMeasuring: isMeasuring }, setVisibleItems] = (0, $gC6oF$reactariaprivateutilsuseValueEffect.useValueEffect)({
        visibleItems: state.collection.size,
        hideButtonText: buttonLabelBehavior === 'hide' && isIconCollapsible,
        isMeasuring: false
    });
    let selectionMode = state.selectionManager.selectionMode;
    let updateOverflow = (0, $gC6oF$react.useCallback)(()=>{
        if (overflowMode === 'wrap') return;
        if (orientation === 'vertical' && selectionMode !== 'none') // Collapsing vertical action groups with selection is currently unsupported by Spectrum.
        return;
        let computeVisibleItems = (visibleItems)=>{
            if (domRef.current && wrapperRef.current) {
                let listItems = Array.from(domRef.current.children);
                let containerSize = orientation === 'horizontal' ? wrapperRef.current.getBoundingClientRect().width : wrapperRef.current.getBoundingClientRect().height;
                let isShowingMenu = visibleItems < state.collection.size;
                let calculatedSize = 0;
                let newVisibleItems = 0;
                if (isShowingMenu) {
                    let item = listItems.pop();
                    if (item) calculatedSize += orientation === 'horizontal' ? $1f2a1f451a6aa23a$var$outerWidth(item, false, true) : $1f2a1f451a6aa23a$var$outerHeight(item, false, true);
                }
                for (let [i, item] of listItems.entries()){
                    calculatedSize += orientation === 'horizontal' ? $1f2a1f451a6aa23a$var$outerWidth(item, i === 0, i === listItems.length - 1) : $1f2a1f451a6aa23a$var$outerHeight(item, i === 0, i === listItems.length - 1);
                    if (Math.round(calculatedSize) <= Math.round(containerSize)) newVisibleItems++;
                    else break;
                }
                // If selection is enabled, and not all of the items fit, collapse all of them into a dropdown
                // immediately rather than having some visible and some not.
                if (selectionMode !== 'none' && newVisibleItems < state.collection.size) return 0;
                return newVisibleItems;
            }
            return visibleItems;
        };
        setVisibleItems(function*() {
            let hideButtonText = buttonLabelBehavior === 'hide' && isIconCollapsible;
            // Update to show all items.
            yield {
                visibleItems: state.collection.size,
                hideButtonText: hideButtonText,
                isMeasuring: true
            };
            // Measure, and update to show the items that fit.
            let newVisibleItems = computeVisibleItems(state.collection.size);
            let isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
            // If not all of the buttons fit, and buttonLabelBehavior is 'collapse', then first try hiding
            // the button text and only showing icons. Only if that still doesn't fit collapse into a menu.
            if (newVisibleItems < state.collection.size && buttonLabelBehavior === 'collapse' && isIconCollapsible) {
                yield {
                    visibleItems: state.collection.size,
                    hideButtonText: true,
                    isMeasuring: true
                };
                newVisibleItems = computeVisibleItems(state.collection.size);
                isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
                hideButtonText = true;
            }
            yield {
                visibleItems: newVisibleItems,
                hideButtonText: hideButtonText,
                isMeasuring: isMeasuring
            };
            // If the number of items is less than the number of children,
            // then update again to ensure that the menu fits.
            if (isMeasuring) yield {
                visibleItems: computeVisibleItems(newVisibleItems),
                hideButtonText: hideButtonText,
                isMeasuring: false
            };
        });
    }, [
        domRef,
        state.collection,
        setVisibleItems,
        overflowMode,
        selectionMode,
        buttonLabelBehavior,
        isIconCollapsible,
        orientation
    ]);
    // Watch the parent element for size changes. Watching only the action group itself may not work
    // in all scenarios because it may not shrink when available space is reduced.
    let parentRef = (0, $gC6oF$react.useMemo)(()=>({
            get current () {
                return wrapperRef.current?.parentElement;
            }
        }), [
        wrapperRef
    ]);
    (0, $gC6oF$reactariaprivateutilsuseResizeObserver.useResizeObserver)({
        ref: overflowMode !== 'wrap' ? parentRef : undefined,
        onResize: updateOverflow
    });
    (0, $gC6oF$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(updateOverflow, [
        updateOverflow,
        state.collection
    ]);
    let children = [
        ...state.collection
    ];
    let menuItem = null;
    let menuProps = {};
    // If there are no visible items, don't apply any props to the action group container
    // and pass all aria labeling props through to the menu button.
    if (overflowMode === 'collapse' && visibleItems === 0) {
        menuProps = (0, $gC6oF$reactariafilterDOMProps.filterDOMProps)(props, {
            labelable: true
        });
        actionGroupProps = {};
    }
    if (overflowMode === 'collapse' && visibleItems < state.collection.size) {
        let menuChildren = children.slice(visibleItems);
        children = children.slice(0, visibleItems);
        menuItem = /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement($1f2a1f451a6aa23a$var$ActionGroupMenu, {
            ...menuProps,
            items: menuChildren,
            onAction: (key)=>onAction?.(key),
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            state: state,
            summaryIcon: summaryIcon,
            hideButtonText: hideButtonText,
            isOnlyItem: visibleItems === 0,
            orientation: orientation
        });
    }
    let style = {
        ...styleProps.style,
        // While measuring, take up as much space as possible.
        flexBasis: isMeasuring ? '100%' : undefined
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $gC6oF$reactariaFocusScope.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement("div", {
        ...styleProps,
        style: style,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($80eda55ee0c557c4$exports))), 'flex-container', styleProps.className),
        ref: wrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement("div", {
        ...actionGroupProps,
        ref: domRef,
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($80eda55ee0c557c4$exports))), 'flex-gap', 'spectrum-ActionGroup', {
            'spectrum-ActionGroup--quiet': isQuiet,
            'spectrum-ActionGroup--vertical': isVertical,
            'spectrum-ActionGroup--compact': density === 'compact',
            'spectrum-ActionGroup--justified': isJustified && !isMeasuring,
            'spectrum-ActionGroup--overflowCollapse': overflowMode === 'collapse'
        }, otherProps.UNSAFE_className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $544fc82701fc93e9$exports.Provider), providerProps, children.map((item)=>/*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement($1f2a1f451a6aa23a$var$ActionGroupItem, {
            key: item.key,
            onAction: onAction,
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            item: item,
            state: state,
            hideButtonText: hideButtonText,
            orientation: orientation
        })), menuItem))));
});
function $1f2a1f451a6aa23a$var$ActionGroupItem({ item: item, state: state, isDisabled: isDisabled, isEmphasized: isEmphasized, staticColor: staticColor, onAction: onAction, hideButtonText: hideButtonText, orientation: orientation }) {
    let ref = (0, $gC6oF$react.useRef)(null);
    let { buttonProps: buttonProps } = (0, $gC6oF$reactariaprivateactiongroupuseActionGroupItem.useActionGroupItem)({
        key: item.key
    }, state);
    isDisabled = isDisabled || state.disabledKeys.has(item.key);
    let isSelected = state.selectionManager.isSelected(item.key);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gC6oF$reactariauseHover.useHover)({
        isDisabled: isDisabled
    });
    let domProps = (0, $gC6oF$reactariafilterDOMProps.filterDOMProps)(item.props);
    if (onAction && !isDisabled) buttonProps = (0, $gC6oF$reactariamergeProps.mergeProps)(buttonProps, {
        onPress: ()=>onAction(item.key)
    });
    // If button text is hidden, we need to show it as a tooltip instead, so
    // go find the text element in the DOM after rendering.
    let textId = (0, $gC6oF$reactariauseId.useId)();
    let [textContent, setTextContent] = (0, $gC6oF$react.useState)('');
    (0, $gC6oF$reactariaprivateutilsuseLayoutEffect.useLayoutEffect)(()=>{
        if (hideButtonText) setTextContent(document.getElementById(textId)?.textContent);
    }, [
        hideButtonText,
        item.rendered,
        textId
    ]);
    let button = // Use a PressResponder to send DOM props through.
    // ActionButton doesn't allow overriding the role by default.
    /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $gC6oF$reactariaprivateinteractionsPressResponder.PressResponder), (0, $gC6oF$reactariamergeProps.mergeProps)(buttonProps, hoverProps, domProps), /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $feede71cddc0c5f3$exports.ClearSlots), null, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText
            }
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $183c5173677598aa$exports.ActionButton), {
        ref: ref,
        // @ts-ignore (private)
        hideButtonText: hideButtonText,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($80eda55ee0c557c4$exports))), 'spectrum-ActionGroup-item', {
            'is-selected': isSelected,
            'is-hovered': isHovered,
            'spectrum-ActionGroup-item--iconOnly': hideButtonText,
            'spectrum-ActionGroup-item--isDisabled': isDisabled
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), {
            'spectrum-ActionButton--emphasized': isEmphasized,
            'is-selected': isSelected
        })),
        isDisabled: isDisabled,
        staticColor: staticColor,
        "aria-label": item['aria-label'],
        "aria-labelledby": item['aria-label'] == null && hideButtonText ? textId : undefined
    }, item.rendered))));
    if (hideButtonText && textContent) button = /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $ff31b6c981164b8a$exports.TooltipTrigger), {
        placement: orientation === 'vertical' ? 'end' : 'top'
    }, button, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $f1974881be69ddc4$exports.Tooltip), null, textContent));
    if (item.wrapper) button = item.wrapper(button);
    return button;
}
function $1f2a1f451a6aa23a$var$ActionGroupMenu({ state: state, isDisabled: isDisabled, isEmphasized: isEmphasized, staticColor: staticColor, items: items, onAction: onAction, summaryIcon: summaryIcon, hideButtonText: hideButtonText, isOnlyItem: isOnlyItem, orientation: orientation, ...otherProps }) {
    // Use the key of the first item within the menu as the key of the button.
    // The key must actually exist in the collection for focus to work correctly.
    let key = items[0].key;
    let { buttonProps: buttonProps } = (0, $gC6oF$reactariaprivateactiongroupuseActionGroupItem.useActionGroupItem)({
        key: key
    }, state);
    let stringFormatter = (0, $gC6oF$reactariauseLocalizedStringFormatter.useLocalizedStringFormatter)((0, ($parcel$interopDefault($b9a8a32d848cb509$exports))), '@react-spectrum/actiongroup');
    // The menu button shouldn't act like an actual action group item.
    delete buttonProps.onPress;
    delete buttonProps.role;
    delete buttonProps['aria-checked'];
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gC6oF$reactariauseHover.useHover)({
        isDisabled: isDisabled
    });
    // If no aria-label or aria-labelledby is given, provide a default one.
    let ariaLabel = otherProps['aria-label'] || (otherProps['aria-labelledby'] ? undefined : stringFormatter.format('more'));
    let ariaLabelledby = otherProps['aria-labelledby'];
    let textId = (0, $gC6oF$reactariauseId.useId)();
    let id = (0, $gC6oF$reactariauseId.useId)();
    // Summary icon only applies when selection is enabled.
    if (state.selectionManager.selectionMode === 'none') summaryIcon = null;
    let iconOnly = false;
    // If there is a selection, show the selected state on the menu button.
    let isSelected = state.selectionManager.selectionMode !== 'none' && !state.selectionManager.isEmpty;
    // If single selection and empty selection is not allowed, swap the contents of the button to the selected item (like a Picker).
    if (!summaryIcon && state.selectionManager.selectionMode === 'single' && state.selectionManager.disallowEmptySelection && state.selectionManager.firstSelectedKey != null) {
        let selectedItem = state.collection.getItem(state.selectionManager.firstSelectedKey);
        if (selectedItem) {
            summaryIcon = selectedItem.rendered;
            if (typeof summaryIcon === 'string') summaryIcon = /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $15e3b68ec42125a9$exports.Text), null, summaryIcon);
            iconOnly = !!hideButtonText;
            ariaLabelledby = `${ariaLabelledby ?? id} ${textId}`;
        }
    }
    if (summaryIcon) // If there's a custom summary icon, also add a chevron.
    summaryIcon = /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, ($parcel$interopDefault($gC6oF$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, ($parcel$interopDefault($gC6oF$spectrumiconsuiChevronDownMedium))), {
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($80eda55ee0c557c4$exports))), 'spectrum-ActionGroup-menu-chevron')
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement("span", {
        className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($80eda55ee0c557c4$exports))), 'spectrum-ActionGroup-menu-contents', {
            'spectrum-ActionGroup-item--iconOnly': iconOnly
        })
    }, summaryIcon));
    return(// Use a PressResponder to send DOM props through.
    /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $98227f5fd590c993$exports.MenuTrigger), {
        align: isOnlyItem ? 'start' : 'end',
        direction: orientation === 'vertical' ? 'end' : 'bottom'
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $feede71cddc0c5f3$exports.SlotProvider), {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText,
                UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($80eda55ee0c557c4$exports))), 'spectrum-ActionGroup-menu-text')
            }
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $gC6oF$reactariaprivateinteractionsPressResponder.PressResponder), (0, $gC6oF$reactariamergeProps.mergeProps)(buttonProps, hoverProps), /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $183c5173677598aa$exports.ActionButton), {
        ...otherProps,
        id: id,
        "aria-label": ariaLabel,
        "aria-labelledby": ariaLabelledby,
        UNSAFE_className: (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($80eda55ee0c557c4$exports))), 'spectrum-ActionGroup-item', 'spectrum-ActionGroup-menu', {
            'is-hovered': isHovered,
            'is-selected': isSelected
        }, (0, $69fd630bd812ba47$exports.classNames)((0, ($parcel$interopDefault($869138cbe3b599dc$exports))), {
            'is-selected': isSelected,
            'spectrum-ActionButton--emphasized': isEmphasized
        })),
        isDisabled: isDisabled,
        staticColor: staticColor
    }, summaryIcon || /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, ($parcel$interopDefault($gC6oF$spectrumiconsworkflowMore))), null)))), /*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $802fb5441f76e7b0$exports.Menu), {
        items: items,
        disabledKeys: state.disabledKeys,
        selectionMode: state.selectionManager.selectionMode,
        selectedKeys: state.selectionManager.selectedKeys,
        disallowEmptySelection: state.selectionManager.disallowEmptySelection,
        onSelectionChange: (keys)=>state.selectionManager.setSelectedKeys(keys),
        onAction: onAction
    }, (node)=>/*#__PURE__*/ (0, ($parcel$interopDefault($gC6oF$react))).createElement((0, $gC6oF$reactstatelyItem.Item), {
            textValue: node.textValue,
            ...(0, $gC6oF$reactariafilterDOMProps.filterDOMProps)(node.props)
        }, node.rendered))));
}
function $1f2a1f451a6aa23a$var$outerWidth(element, ignoreLeftMargin, ignoreRightMargin) {
    let style = window.getComputedStyle(element);
    return element.getBoundingClientRect().width + (ignoreLeftMargin ? 0 : $1f2a1f451a6aa23a$var$toNumber(style.marginLeft)) + (ignoreRightMargin ? 0 : $1f2a1f451a6aa23a$var$toNumber(style.marginRight));
}
function $1f2a1f451a6aa23a$var$outerHeight(element, ignoreTopMargin, ignoreBottomMargin) {
    let style = window.getComputedStyle(element);
    return element.getBoundingClientRect().height + (ignoreTopMargin ? 0 : $1f2a1f451a6aa23a$var$toNumber(style.marginTop)) + (ignoreBottomMargin ? 0 : $1f2a1f451a6aa23a$var$toNumber(style.marginBottom));
}
function $1f2a1f451a6aa23a$var$toNumber(value) {
    let parsed = parseInt(value, 10);
    return isNaN(parsed) ? 0 : parsed;
}
